<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 82 Integral Population Models | Applied Statistics for the Natural Sciences</title>
  <meta name="description" content="Chapter 82 Integral Population Models | Applied Statistics for the Natural Sciences" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 82 Integral Population Models | Applied Statistics for the Natural Sciences" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="sarasouther/DataAnalysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 82 Integral Population Models | Applied Statistics for the Natural Sciences" />
  
  
  

<meta name="author" content="Sara Souther" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="population-sensation.html"/>
<link rel="next" href="stochastic-population-dynamics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<link href="libs/htmltools-fill-0.5.9/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.11/grViz.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Applied Statistics for the Natural Sciences</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#learning-objectives"><i class="fa fa-check"></i><b>1.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#using-r-in-this-course"><i class="fa fa-check"></i><b>1.2</b> Using R in This Course</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#what-you-need-to-use-r"><i class="fa fa-check"></i><b>1.2.1</b> What you need to use R</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#create-an-r-file"><i class="fa fa-check"></i><b>1.2.2</b> Create an R file</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#using-working-directories-and-opening-files"><i class="fa fa-check"></i><b>1.2.3</b> Using working directories and opening files</a></li>
<li class="chapter" data-level="1.2.4" data-path="index.html"><a href="index.html#r-packages"><i class="fa fa-check"></i><b>1.2.4</b> R Packages</a></li>
<li class="chapter" data-level="1.2.5" data-path="index.html"><a href="index.html#other-information-about-r-code"><i class="fa fa-check"></i><b>1.2.5</b> Other information about R code</a></li>
<li class="chapter" data-level="1.2.6" data-path="index.html"><a href="index.html#getting-started-with-r"><i class="fa fa-check"></i><b>1.2.6</b> Getting started with R</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#how-to-use-this-online-manual"><i class="fa fa-check"></i><b>1.3</b> How to use this online manual</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#how-this-manual-fits-into-the-course"><i class="fa fa-check"></i><b>1.3.1</b> How this manual fits into the course</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#tips-for-getting-the-most-out-of-this-book"><i class="fa fa-check"></i><b>1.3.2</b> Tips for getting the most out of this book</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#a-note-on-r-code-in-this-book"><i class="fa fa-check"></i><b>1.3.3</b> A note on R code in this book</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#course-structure"><i class="fa fa-check"></i><b>1.4</b> Course structure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="back-to-the-basics.html"><a href="back-to-the-basics.html"><i class="fa fa-check"></i><b>2</b> Back to the basics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="back-to-the-basics.html"><a href="back-to-the-basics.html#watch-the-video"><i class="fa fa-check"></i><b>2.1</b> Watch the video</a></li>
<li class="chapter" data-level="2.2" data-path="back-to-the-basics.html"><a href="back-to-the-basics.html#what-can-statistics-tell-us"><i class="fa fa-check"></i><b>2.2</b> What can statistics tell us?</a></li>
<li class="chapter" data-level="2.3" data-path="back-to-the-basics.html"><a href="back-to-the-basics.html#the-frequentist-framework"><i class="fa fa-check"></i><b>2.3</b> The Frequentist Framework</a></li>
<li class="chapter" data-level="2.4" data-path="back-to-the-basics.html"><a href="back-to-the-basics.html#bayesian-statistics"><i class="fa fa-check"></i><b>2.4</b> Bayesian statistics</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="back-to-the-basics.html"><a href="back-to-the-basics.html#when-might-you-choose-bayesian-statistics"><i class="fa fa-check"></i><b>2.4.1</b> When might you choose Bayesian statistics?</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="back-to-the-basics.html"><a href="back-to-the-basics.html#fundaments-of-frequentist-statistics"><i class="fa fa-check"></i><b>2.5</b> Fundaments of frequentist statistics</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="back-to-the-basics.html"><a href="back-to-the-basics.html#statistical-error"><i class="fa fa-check"></i><b>2.5.1</b> Statistical error</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="back-to-the-basics.html"><a href="back-to-the-basics.html#what-is-a-model"><i class="fa fa-check"></i><b>2.6</b> What is a model?</a></li>
<li class="chapter" data-level="2.7" data-path="back-to-the-basics.html"><a href="back-to-the-basics.html#assignment"><i class="fa fa-check"></i><b>2.7</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html"><i class="fa fa-check"></i><b>3</b> Probability: The Language of Uncertainty</a>
<ul>
<li class="chapter" data-level="3.1" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#watch-the-video-1"><i class="fa fa-check"></i><b>3.1</b> Watch the video</a></li>
<li class="chapter" data-level="3.2" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#probability-is-everywhere-and-we-often-get-it-wrong"><i class="fa fa-check"></i><b>3.2</b> Probability is everywhere, and we often get it wrong</a></li>
<li class="chapter" data-level="3.3" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#from-uncertainty-to-inference"><i class="fa fa-check"></i><b>3.3</b> From uncertainty to inference</a></li>
<li class="chapter" data-level="3.4" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#samplespace"><i class="fa fa-check"></i><b>3.4</b> Defining the sample space</a></li>
<li class="chapter" data-level="3.5" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#longrun"><i class="fa fa-check"></i><b>3.5</b> Probability as long-run frequency</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#simulation-watching-probability-stabilize"><i class="fa fa-check"></i><b>3.5.1</b> Simulation: watching probability stabilize</a></li>
<li class="chapter" data-level="3.5.2" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#what-long-run-frequency-means-for-your-research"><i class="fa fa-check"></i><b>3.5.2</b> What long-run frequency means for your research</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#conditional"><i class="fa fa-check"></i><b>3.6</b> Conditional probability and dependence</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#how-knowing-one-thing-changes-what-we-expect"><i class="fa fa-check"></i><b>3.6.1</b> How knowing one thing changes what we expect</a></li>
<li class="chapter" data-level="3.6.2" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#independence-when-knowing-one-thing-changes-nothing"><i class="fa fa-check"></i><b>3.6.2</b> Independence: when knowing one thing changes nothing</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#nullmodels"><i class="fa fa-check"></i><b>3.7</b> From probability to statistical thinking</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#probability-distributions-the-full-picture-of-uncertainty"><i class="fa fa-check"></i><b>3.7.1</b> Probability distributions: the full picture of uncertainty</a></li>
<li class="chapter" data-level="3.7.2" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#nullmodel"><i class="fa fa-check"></i><b>3.7.2</b> Null models: what does chance alone look like?</a></li>
<li class="chapter" data-level="3.7.3" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#permtest"><i class="fa fa-check"></i><b>3.7.3</b> A permutation test: building a null distribution</a></li>
<li class="chapter" data-level="3.7.4" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#pvalue"><i class="fa fa-check"></i><b>3.7.4</b> The p-value: measuring surprise</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#roadmap"><i class="fa fa-check"></i><b>3.8</b> Where probability appears in ecology and statistics</a></li>
<li class="chapter" data-level="3.9" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#probsummary"><i class="fa fa-check"></i><b>3.9</b> Summary</a></li>
<li class="chapter" data-level="3.10" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#probassignment"><i class="fa fa-check"></i><b>3.10</b> Assignment: Probability thinking in your research system</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#part-1-defining-the-sample-space"><i class="fa fa-check"></i><b>3.10.1</b> Part 1: Defining the sample space</a></li>
<li class="chapter" data-level="3.10.2" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#part-2-probability-as-long-run-frequency"><i class="fa fa-check"></i><b>3.10.2</b> Part 2: Probability as long-run frequency</a></li>
<li class="chapter" data-level="3.10.3" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#part-3-conditional-probability"><i class="fa fa-check"></i><b>3.10.3</b> Part 3: Conditional probability</a></li>
<li class="chapter" data-level="3.10.4" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#part-4-independence-and-dependence"><i class="fa fa-check"></i><b>3.10.4</b> Part 4: Independence and dependence</a></li>
<li class="chapter" data-level="3.10.5" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#part-5-null-model-thinking"><i class="fa fa-check"></i><b>3.10.5</b> Part 5: Null model thinking</a></li>
<li class="chapter" data-level="3.10.6" data-path="probability-the-language-of-uncertainty.html"><a href="probability-the-language-of-uncertainty.html#part-6-connecting-to-your-analyses"><i class="fa fa-check"></i><b>3.10.6</b> Part 6: Connecting to your analyses</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html"><i class="fa fa-check"></i><b>4</b> Data Types and Why They Matter</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#vartypes"><i class="fa fa-check"></i><b>4.1</b> Types of variables</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#categorical-variables"><i class="fa fa-check"></i><b>4.1.1</b> Categorical variables</a></li>
<li class="chapter" data-level="4.1.2" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#numerical-variables"><i class="fa fa-check"></i><b>4.1.2</b> Numerical variables</a></li>
<li class="chapter" data-level="4.1.3" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#ordinal-variables"><i class="fa fa-check"></i><b>4.1.3</b> Ordinal variables</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#residuals"><i class="fa fa-check"></i><b>4.2</b> From data to residuals</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#what-is-a-residual"><i class="fa fa-check"></i><b>4.2.1</b> What is a residual?</a></li>
<li class="chapter" data-level="4.2.2" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#why-the-shape-of-residuals-matters"><i class="fa fa-check"></i><b>4.2.2</b> Why the shape of residuals matters</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#errorstructure"><i class="fa fa-check"></i><b>4.3</b> Normal vs. non-normal error</a></li>
<li class="chapter" data-level="4.4" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#distribution-table"><i class="fa fa-check"></i><b>4.4</b> Linking data types to distributions and models</a></li>
<li class="chapter" data-level="4.5" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#distributions"><i class="fa fa-check"></i><b>4.5</b> Common statistical distributions</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#the-normal-distribution"><i class="fa fa-check"></i><b>4.5.1</b> The normal distribution</a></li>
<li class="chapter" data-level="4.5.2" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#the-binomial-distribution"><i class="fa fa-check"></i><b>4.5.2</b> The binomial distribution</a></li>
<li class="chapter" data-level="4.5.3" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#the-poisson-distribution"><i class="fa fa-check"></i><b>4.5.3</b> The Poisson distribution</a></li>
<li class="chapter" data-level="4.5.4" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#the-negative-binomial-distribution"><i class="fa fa-check"></i><b>4.5.4</b> The negative binomial distribution</a></li>
<li class="chapter" data-level="4.5.5" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#the-uniform-distribution"><i class="fa fa-check"></i><b>4.5.5</b> The uniform distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#datatypes-summary"><i class="fa fa-check"></i><b>4.6</b> Summary</a></li>
<li class="chapter" data-level="4.7" data-path="data-types-and-why-they-matter.html"><a href="data-types-and-why-they-matter.html#datatypes-assignment"><i class="fa fa-check"></i><b>4.7</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html"><i class="fa fa-check"></i><b>5</b> Data Exploration and Assumption Checking</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#why-explore-before-you-analyze"><i class="fa fa-check"></i><b>5.1</b> Why explore before you analyze?</a></li>
<li class="chapter" data-level="5.2" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#the-dataset-forest-seedling-survival"><i class="fa fa-check"></i><b>5.2</b> The dataset: Forest seedling survival</a></li>
<li class="chapter" data-level="5.3" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#getting-to-know-your-data"><i class="fa fa-check"></i><b>5.3</b> Getting to know your data</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#first-look-structure-and-dimensions"><i class="fa fa-check"></i><b>5.3.1</b> First look: Structure and dimensions</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#summary-statistics-what-are-the-ranges"><i class="fa fa-check"></i><b>5.3.2</b> Summary statistics: What are the ranges?</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#checking-for-missing-values"><i class="fa fa-check"></i><b>5.3.3</b> Checking for missing values</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#identifying-impossible-values"><i class="fa fa-check"></i><b>5.3.4</b> Identifying impossible values</a></li>
<li class="chapter" data-level="5.3.5" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#quick-reference-common-data-problems"><i class="fa fa-check"></i><b>5.3.5</b> Quick reference: Common data problems</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#visualizing-distributions"><i class="fa fa-check"></i><b>5.4</b> Visualizing distributions</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#continuous-variables-histograms"><i class="fa fa-check"></i><b>5.4.1</b> Continuous variables: Histograms</a></li>
<li class="chapter" data-level="5.4.2" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#continuous-variables-density-plots"><i class="fa fa-check"></i><b>5.4.2</b> Continuous variables: Density plots</a></li>
<li class="chapter" data-level="5.4.3" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#boxplots-comparing-groups-and-spotting-outliers"><i class="fa fa-check"></i><b>5.4.3</b> Boxplots: Comparing groups and spotting outliers</a></li>
<li class="chapter" data-level="5.4.4" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#qq-plots-checking-normality-visually"><i class="fa fa-check"></i><b>5.4.4</b> QQ plots: Checking normality visually</a></li>
<li class="chapter" data-level="5.4.5" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#scatterplots-exploring-relationships"><i class="fa fa-check"></i><b>5.4.5</b> Scatterplots: Exploring relationships</a></li>
<li class="chapter" data-level="5.4.6" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#scatterplot-matrices-the-big-picture"><i class="fa fa-check"></i><b>5.4.6</b> Scatterplot matrices: The big picture</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#key-questions-to-ask-during-exploration"><i class="fa fa-check"></i><b>5.5</b> Key questions to ask during exploration</a></li>
<li class="chapter" data-level="5.6" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#assumptions-of-parametric-tests"><i class="fa fa-check"></i><b>5.6</b> Assumptions of parametric tests</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#why-assumptions-matter"><i class="fa fa-check"></i><b>5.6.1</b> Why assumptions matter</a></li>
<li class="chapter" data-level="5.6.2" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#the-big-four-assumptions-for-linear-models"><i class="fa fa-check"></i><b>5.6.2</b> The Big Four assumptions for linear models</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#checking-assumptions-after-fitting-a-model"><i class="fa fa-check"></i><b>5.7</b> Checking assumptions after fitting a model</a>
<ul>
<li class="chapter" data-level="5.7.1" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#fitting-a-preliminary-model"><i class="fa fa-check"></i><b>5.7.1</b> Fitting a preliminary model</a></li>
<li class="chapter" data-level="5.7.2" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#base-r-diagnostic-plots"><i class="fa fa-check"></i><b>5.7.2</b> Base R diagnostic plots</a></li>
<li class="chapter" data-level="5.7.3" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#using-the-performance-package"><i class="fa fa-check"></i><b>5.7.3</b> Using the performance package</a></li>
<li class="chapter" data-level="5.7.4" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#interpreting-diagnostic-output-a-worked-example"><i class="fa fa-check"></i><b>5.7.4</b> Interpreting diagnostic output: A worked example</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#summary-the-eda-workflow"><i class="fa fa-check"></i><b>5.8</b> Summary: The EDA workflow</a>
<ul>
<li class="chapter" data-level="5.8.1" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#before-fitting-a-model"><i class="fa fa-check"></i><b>5.8.1</b> Before fitting a model:</a></li>
<li class="chapter" data-level="5.8.2" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#after-fitting-a-model"><i class="fa fa-check"></i><b>5.8.2</b> After fitting a model:</a></li>
</ul></li>
<li class="chapter" data-level="5.9" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#key-takeaways"><i class="fa fa-check"></i><b>5.9</b> Key takeaways</a></li>
<li class="chapter" data-level="5.10" data-path="data-exploration-and-assumption-checking.html"><a href="data-exploration-and-assumption-checking.html#assignment-1"><i class="fa fa-check"></i><b>5.10</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="describing-data.html"><a href="describing-data.html"><i class="fa fa-check"></i><b>6</b> Describing Data</a>
<ul>
<li class="chapter" data-level="6.1" data-path="describing-data.html"><a href="describing-data.html#the-dataset-milkweed-seedling-heights"><i class="fa fa-check"></i><b>6.1</b> The dataset: Milkweed seedling heights</a></li>
<li class="chapter" data-level="6.2" data-path="describing-data.html"><a href="describing-data.html#location-where-is-the-center"><i class="fa fa-check"></i><b>6.2</b> Location: Where is the center?</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="describing-data.html"><a href="describing-data.html#the-mean"><i class="fa fa-check"></i><b>6.2.1</b> The mean</a></li>
<li class="chapter" data-level="6.2.2" data-path="describing-data.html"><a href="describing-data.html#the-median"><i class="fa fa-check"></i><b>6.2.2</b> The median</a></li>
<li class="chapter" data-level="6.2.3" data-path="describing-data.html"><a href="describing-data.html#mean-vs.-median-which-should-you-report"><i class="fa fa-check"></i><b>6.2.3</b> Mean vs. median: Which should you report?</a></li>
<li class="chapter" data-level="6.2.4" data-path="describing-data.html"><a href="describing-data.html#the-mode"><i class="fa fa-check"></i><b>6.2.4</b> The mode</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="describing-data.html"><a href="describing-data.html#spread-how-variable-are-the-observations"><i class="fa fa-check"></i><b>6.3</b> Spread: How variable are the observations?</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="describing-data.html"><a href="describing-data.html#range"><i class="fa fa-check"></i><b>6.3.1</b> Range</a></li>
<li class="chapter" data-level="6.3.2" data-path="describing-data.html"><a href="describing-data.html#variance"><i class="fa fa-check"></i><b>6.3.2</b> Variance</a></li>
<li class="chapter" data-level="6.3.3" data-path="describing-data.html"><a href="describing-data.html#standard-deviation"><i class="fa fa-check"></i><b>6.3.3</b> Standard deviation</a></li>
<li class="chapter" data-level="6.3.4" data-path="describing-data.html"><a href="describing-data.html#standard-error"><i class="fa fa-check"></i><b>6.3.4</b> Standard error</a></li>
<li class="chapter" data-level="6.3.5" data-path="describing-data.html"><a href="describing-data.html#interquartile-range-iqr"><i class="fa fa-check"></i><b>6.3.5</b> Interquartile range (IQR)</a></li>
<li class="chapter" data-level="6.3.6" data-path="describing-data.html"><a href="describing-data.html#summary-table-measures-of-spread"><i class="fa fa-check"></i><b>6.3.6</b> Summary table: Measures of spread</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="describing-data.html"><a href="describing-data.html#shape-is-the-distribution-symmetric"><i class="fa fa-check"></i><b>6.4</b> Shape: Is the distribution symmetric?</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="describing-data.html"><a href="describing-data.html#comparing-mean-and-median"><i class="fa fa-check"></i><b>6.4.1</b> Comparing mean and median</a></li>
<li class="chapter" data-level="6.4.2" data-path="describing-data.html"><a href="describing-data.html#visualizing-shape-with-histograms"><i class="fa fa-check"></i><b>6.4.2</b> Visualizing shape with histograms</a></li>
<li class="chapter" data-level="6.4.3" data-path="describing-data.html"><a href="describing-data.html#the-normal-gaussian-distribution"><i class="fa fa-check"></i><b>6.4.3</b> The normal (Gaussian) distribution</a></li>
<li class="chapter" data-level="6.4.4" data-path="describing-data.html"><a href="describing-data.html#skewed-distributions"><i class="fa fa-check"></i><b>6.4.4</b> Skewed distributions</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="describing-data.html"><a href="describing-data.html#outliers-unusual-observations"><i class="fa fa-check"></i><b>6.5</b> Outliers: Unusual observations</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="describing-data.html"><a href="describing-data.html#detecting-outliers-with-boxplots"><i class="fa fa-check"></i><b>6.5.1</b> Detecting outliers with boxplots</a></li>
<li class="chapter" data-level="6.5.2" data-path="describing-data.html"><a href="describing-data.html#boxplots-for-comparing-groups"><i class="fa fa-check"></i><b>6.5.2</b> Boxplots for comparing groups</a></li>
<li class="chapter" data-level="6.5.3" data-path="describing-data.html"><a href="describing-data.html#what-to-do-with-outliers"><i class="fa fa-check"></i><b>6.5.3</b> What to do with outliers</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="describing-data.html"><a href="describing-data.html#describing-categorical-data"><i class="fa fa-check"></i><b>6.6</b> Describing categorical data</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="describing-data.html"><a href="describing-data.html#frequencies-and-proportions"><i class="fa fa-check"></i><b>6.6.1</b> Frequencies and proportions</a></li>
<li class="chapter" data-level="6.6.2" data-path="describing-data.html"><a href="describing-data.html#visualizing-categorical-data-with-bar-plots"><i class="fa fa-check"></i><b>6.6.2</b> Visualizing categorical data with bar plots</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="describing-data.html"><a href="describing-data.html#confidence-intervals"><i class="fa fa-check"></i><b>6.7</b> Confidence intervals</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="describing-data.html"><a href="describing-data.html#calculating-a-95-ci-for-the-mean"><i class="fa fa-check"></i><b>6.7.1</b> Calculating a 95% CI for the mean</a></li>
<li class="chapter" data-level="6.7.2" data-path="describing-data.html"><a href="describing-data.html#using-t.test-for-quick-cis"><i class="fa fa-check"></i><b>6.7.2</b> Using t.test() for quick CIs</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="describing-data.html"><a href="describing-data.html#putting-it-all-together-complete-description"><i class="fa fa-check"></i><b>6.8</b> Putting it all together: Complete description</a></li>
<li class="chapter" data-level="6.9" data-path="describing-data.html"><a href="describing-data.html#reporting-in-publications"><i class="fa fa-check"></i><b>6.9</b> Reporting in publications</a></li>
<li class="chapter" data-level="6.10" data-path="describing-data.html"><a href="describing-data.html#key-takeaways-1"><i class="fa fa-check"></i><b>6.10</b> Key takeaways</a></li>
<li class="chapter" data-level="6.11" data-path="describing-data.html"><a href="describing-data.html#looking-ahead"><i class="fa fa-check"></i><b>6.11</b> Looking ahead</a></li>
<li class="chapter" data-level="6.12" data-path="describing-data.html"><a href="describing-data.html#assignment-2"><i class="fa fa-check"></i><b>6.12</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="ecological-sampling.html"><a href="ecological-sampling.html"><i class="fa fa-check"></i><b>7</b> Ecological sampling</a>
<ul>
<li class="chapter" data-level="7.1" data-path="ecological-sampling.html"><a href="ecological-sampling.html#background"><i class="fa fa-check"></i><b>7.1</b> Background</a></li>
<li class="chapter" data-level="7.2" data-path="ecological-sampling.html"><a href="ecological-sampling.html#what-kind-of-study-are-you-designing"><i class="fa fa-check"></i><b>7.2</b> What kind of study are you designing?</a></li>
<li class="chapter" data-level="7.3" data-path="ecological-sampling.html"><a href="ecological-sampling.html#sampling-approaches"><i class="fa fa-check"></i><b>7.3</b> Sampling approaches</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="ecological-sampling.html"><a href="ecological-sampling.html#random-sampling"><i class="fa fa-check"></i><b>7.3.1</b> Random sampling</a></li>
<li class="chapter" data-level="7.3.2" data-path="ecological-sampling.html"><a href="ecological-sampling.html#stratified-random-sampling"><i class="fa fa-check"></i><b>7.3.2</b> Stratified random sampling</a></li>
<li class="chapter" data-level="7.3.3" data-path="ecological-sampling.html"><a href="ecological-sampling.html#gridded-random-sampling"><i class="fa fa-check"></i><b>7.3.3</b> Gridded random sampling</a></li>
<li class="chapter" data-level="7.3.4" data-path="ecological-sampling.html"><a href="ecological-sampling.html#random-cluster-sampling"><i class="fa fa-check"></i><b>7.3.4</b> Random cluster sampling</a></li>
<li class="chapter" data-level="7.3.5" data-path="ecological-sampling.html"><a href="ecological-sampling.html#grts-sampling-generalized-random-tessellation-stratified-sampling"><i class="fa fa-check"></i><b>7.3.5</b> GRTS Sampling (Generalized Random-Tessellation Stratified Sampling)</a></li>
<li class="chapter" data-level="7.3.6" data-path="ecological-sampling.html"><a href="ecological-sampling.html#understanding-design-weights"><i class="fa fa-check"></i><b>7.3.6</b> Understanding Design Weights</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="ecological-sampling.html"><a href="ecological-sampling.html#systematic-sampling"><i class="fa fa-check"></i><b>7.4</b> Systematic Sampling</a></li>
<li class="chapter" data-level="7.5" data-path="ecological-sampling.html"><a href="ecological-sampling.html#adaptive-sampling-sampling-rare-or-patchy-species"><i class="fa fa-check"></i><b>7.5</b> Adaptive Sampling (Sampling Rare or Patchy Species)</a></li>
<li class="chapter" data-level="7.6" data-path="ecological-sampling.html"><a href="ecological-sampling.html#sampling-along-gradients"><i class="fa fa-check"></i><b>7.6</b> Sampling Along Gradients</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="ecological-sampling.html"><a href="ecological-sampling.html#what-is-a-gradient-study"><i class="fa fa-check"></i><b>7.6.1</b> What is a gradient study?</a></li>
<li class="chapter" data-level="7.6.2" data-path="ecological-sampling.html"><a href="ecological-sampling.html#designing-a-gradient-study"><i class="fa fa-check"></i><b>7.6.2</b> Designing a gradient study</a></li>
<li class="chapter" data-level="7.6.3" data-path="ecological-sampling.html"><a href="ecological-sampling.html#gradient-sampling-in-r"><i class="fa fa-check"></i><b>7.6.3</b> Gradient sampling in R</a></li>
<li class="chapter" data-level="7.6.4" data-path="ecological-sampling.html"><a href="ecological-sampling.html#common-pitfalls-in-gradient-studies"><i class="fa fa-check"></i><b>7.6.4</b> Common pitfalls in gradient studies</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="ecological-sampling.html"><a href="ecological-sampling.html#imperfect-detection-in-ecological-sampling"><i class="fa fa-check"></i><b>7.7</b> Imperfect Detection in Ecological Sampling</a></li>
<li class="chapter" data-level="7.8" data-path="ecological-sampling.html"><a href="ecological-sampling.html#temporal-sampling-and-study-designs-through-time"><i class="fa fa-check"></i><b>7.8</b> Temporal Sampling and Study Designs Through Time</a>
<ul>
<li class="chapter" data-level="7.8.1" data-path="ecological-sampling.html"><a href="ecological-sampling.html#sampling-panels"><i class="fa fa-check"></i><b>7.8.1</b> Sampling Panels</a></li>
<li class="chapter" data-level="7.8.2" data-path="ecological-sampling.html"><a href="ecological-sampling.html#baci-and-other-experimental-designs"><i class="fa fa-check"></i><b>7.8.2</b> BACI and Other Experimental Designs</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="ecological-sampling.html"><a href="ecological-sampling.html#plot-shapes-sizes-and-transects"><i class="fa fa-check"></i><b>7.9</b> Plot shapes, sizes, and transects</a></li>
<li class="chapter" data-level="7.10" data-path="ecological-sampling.html"><a href="ecological-sampling.html#distance-sampling-and-point-counts"><i class="fa fa-check"></i><b>7.10</b> Distance Sampling and Point Counts</a></li>
<li class="chapter" data-level="7.11" data-path="ecological-sampling.html"><a href="ecological-sampling.html#reducing-sampling-error"><i class="fa fa-check"></i><b>7.11</b> Reducing sampling error</a></li>
<li class="chapter" data-level="7.12" data-path="ecological-sampling.html"><a href="ecological-sampling.html#sample-size-determination-in-ecology-effect-sizes-and-precision"><i class="fa fa-check"></i><b>7.12</b> Sample Size Determination in Ecology: Effect Sizes and Precision</a></li>
<li class="chapter" data-level="7.13" data-path="ecological-sampling.html"><a href="ecological-sampling.html#variables-of-interest"><i class="fa fa-check"></i><b>7.13</b> Variables of interest</a></li>
<li class="chapter" data-level="7.14" data-path="ecological-sampling.html"><a href="ecological-sampling.html#test-your-knowledge"><i class="fa fa-check"></i><b>7.14</b> Test your knowledge</a></li>
<li class="chapter" data-level="7.15" data-path="ecological-sampling.html"><a href="ecological-sampling.html#assignment-3"><i class="fa fa-check"></i><b>7.15</b> Assignment</a>
<ul>
<li class="chapter" data-level="7.15.1" data-path="ecological-sampling.html"><a href="ecological-sampling.html#example-methods-sampling-design"><i class="fa fa-check"></i><b>7.15.1</b> Example Methods: Sampling Design</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html"><i class="fa fa-check"></i><b>8</b> Experimental Design: From Question to Study Structure</a>
<ul>
<li class="chapter" data-level="8.1" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#studytypes"><i class="fa fa-check"></i><b>8.1</b> Observational vs. manipulative studies</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#manipulative-experiments"><i class="fa fa-check"></i><b>8.1.1</b> Manipulative experiments</a></li>
<li class="chapter" data-level="8.1.2" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#observational-studies"><i class="fa fa-check"></i><b>8.1.2</b> Observational studies</a></li>
<li class="chapter" data-level="8.1.3" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#natural-experiments-and-quasi-experiments"><i class="fa fa-check"></i><b>8.1.3</b> Natural experiments and quasi-experiments</a></li>
<li class="chapter" data-level="8.1.4" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#space-for-time-substitution"><i class="fa fa-check"></i><b>8.1.4</b> Space-for-time substitution</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#controls"><i class="fa fa-check"></i><b>8.2</b> The logic of controls and comparison</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#what-makes-a-good-control"><i class="fa fa-check"></i><b>8.2.1</b> What makes a good control?</a></li>
<li class="chapter" data-level="8.2.2" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#procedural-controls"><i class="fa fa-check"></i><b>8.2.2</b> Procedural controls</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#randomization"><i class="fa fa-check"></i><b>8.3</b> Randomization: why and how</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#what-randomization-does"><i class="fa fa-check"></i><b>8.3.1</b> What randomization does</a></li>
<li class="chapter" data-level="8.3.2" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#how-to-randomize"><i class="fa fa-check"></i><b>8.3.2</b> How to randomize</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#designs"><i class="fa fa-check"></i><b>8.4</b> Common experimental designs</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#completely-randomized-design-crd"><i class="fa fa-check"></i><b>8.4.1</b> Completely Randomized Design (CRD)</a></li>
<li class="chapter" data-level="8.4.2" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#blocking"><i class="fa fa-check"></i><b>8.4.2</b> Randomized Complete Block Design (RCBD)</a></li>
<li class="chapter" data-level="8.4.3" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#factorial"><i class="fa fa-check"></i><b>8.4.3</b> Factorial designs</a></li>
<li class="chapter" data-level="8.4.4" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#nested"><i class="fa fa-check"></i><b>8.4.4</b> Nested designs</a></li>
<li class="chapter" data-level="8.4.5" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#splitplot"><i class="fa fa-check"></i><b>8.4.5</b> Split-plot designs</a></li>
<li class="chapter" data-level="8.4.6" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#repeatedmeasures"><i class="fa fa-check"></i><b>8.4.6</b> Repeated measures and longitudinal designs</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#baci"><i class="fa fa-check"></i><b>8.5</b> Before-After-Control-Impact (BACI) designs</a>
<ul>
<li class="chapter" data-level="8.5.1" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#the-problem-baci-solves"><i class="fa fa-check"></i><b>8.5.1</b> The problem BACI solves</a></li>
<li class="chapter" data-level="8.5.2" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#the-four-standard-baci-designs"><i class="fa fa-check"></i><b>8.5.2</b> The four standard BACI designs</a></li>
<li class="chapter" data-level="8.5.3" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#the-baci-interaction"><i class="fa fa-check"></i><b>8.5.3</b> The BACI interaction</a></li>
<li class="chapter" data-level="8.5.4" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#statistical-model-for-baci"><i class="fa fa-check"></i><b>8.5.4</b> Statistical model for BACI</a></li>
<li class="chapter" data-level="8.5.5" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#baci-pitfalls"><i class="fa fa-check"></i><b>8.5.5</b> BACI pitfalls</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#replication"><i class="fa fa-check"></i><b>8.6</b> Identifying your unit of replication</a>
<ul>
<li class="chapter" data-level="8.6.1" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#the-rule"><i class="fa fa-check"></i><b>8.6.1</b> The rule</a></li>
<li class="chapter" data-level="8.6.2" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#what-to-do-with-subsamples"><i class="fa fa-check"></i><b>8.6.2</b> What to do with subsamples</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#design-to-analysis"><i class="fa fa-check"></i><b>8.7</b> Connecting design to analysis</a></li>
<li class="chapter" data-level="8.8" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#designsummary"><i class="fa fa-check"></i><b>8.8</b> Summary</a></li>
<li class="chapter" data-level="8.9" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#designassignment"><i class="fa fa-check"></i><b>8.9</b> Assignment</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#part-1-study-type"><i class="fa fa-check"></i><b>8.9.1</b> Part 1: Study type</a></li>
<li class="chapter" data-level="8.9.2" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#part-2-design-structure"><i class="fa fa-check"></i><b>8.9.2</b> Part 2: Design structure</a></li>
<li class="chapter" data-level="8.9.3" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#part-3-controls-and-confounds"><i class="fa fa-check"></i><b>8.9.3</b> Part 3: Controls and confounds</a></li>
<li class="chapter" data-level="8.9.4" data-path="experimental-design-from-question-to-study-structure.html"><a href="experimental-design-from-question-to-study-structure.html#part-4-design-to-model-connection"><i class="fa fa-check"></i><b>8.9.4</b> Part 4: Design-to-model connection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="power-analysis.html"><a href="power-analysis.html"><i class="fa fa-check"></i><b>9</b> Power analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="power-analysis.html"><a href="power-analysis.html#prework"><i class="fa fa-check"></i><b>9.1</b> Prework</a></li>
<li class="chapter" data-level="9.2" data-path="power-analysis.html"><a href="power-analysis.html#statistical-power"><i class="fa fa-check"></i><b>9.2</b> Statistical power</a></li>
<li class="chapter" data-level="9.3" data-path="power-analysis.html"><a href="power-analysis.html#introduction-why-power-matters"><i class="fa fa-check"></i><b>9.3</b> Introduction: Why power matters</a></li>
<li class="chapter" data-level="9.4" data-path="power-analysis.html"><a href="power-analysis.html#setup"><i class="fa fa-check"></i><b>9.4</b> Setup</a></li>
<li class="chapter" data-level="9.5" data-path="power-analysis.html"><a href="power-analysis.html#the-four-ingredients-of-power"><i class="fa fa-check"></i><b>9.5</b> The four ingredients of power</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="power-analysis.html"><a href="power-analysis.html#why-0.80-for-power"><i class="fa fa-check"></i><b>9.5.1</b> Why 0.80 for power?</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="power-analysis.html"><a href="power-analysis.html#effect-size-the-key-to-power-analysis"><i class="fa fa-check"></i><b>9.6</b> Effect size: The key to power analysis</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="power-analysis.html"><a href="power-analysis.html#cohens-d-effect-size-for-comparing-two-means"><i class="fa fa-check"></i><b>9.6.1</b> Cohen’s d: Effect size for comparing two means</a></li>
<li class="chapter" data-level="9.6.2" data-path="power-analysis.html"><a href="power-analysis.html#calculating-effect-size-from-pilot-data"><i class="fa fa-check"></i><b>9.6.2</b> Calculating effect size from pilot data</a></li>
<li class="chapter" data-level="9.6.3" data-path="power-analysis.html"><a href="power-analysis.html#cohens-f-effect-size-for-anova"><i class="fa fa-check"></i><b>9.6.3</b> Cohen’s f: Effect size for ANOVA</a></li>
<li class="chapter" data-level="9.6.4" data-path="power-analysis.html"><a href="power-analysis.html#other-effect-size-measures"><i class="fa fa-check"></i><b>9.6.4</b> Other effect size measures</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="power-analysis.html"><a href="power-analysis.html#power-analysis-for-common-tests"><i class="fa fa-check"></i><b>9.7</b> Power analysis for common tests</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="power-analysis.html"><a href="power-analysis.html#two-sample-t-test"><i class="fa fa-check"></i><b>9.7.1</b> Two-sample t-test</a></li>
<li class="chapter" data-level="9.7.2" data-path="power-analysis.html"><a href="power-analysis.html#paired-t-test"><i class="fa fa-check"></i><b>9.7.2</b> Paired t-test</a></li>
<li class="chapter" data-level="9.7.3" data-path="power-analysis.html"><a href="power-analysis.html#one-way-anova"><i class="fa fa-check"></i><b>9.7.3</b> One-way ANOVA</a></li>
<li class="chapter" data-level="9.7.4" data-path="power-analysis.html"><a href="power-analysis.html#correlation"><i class="fa fa-check"></i><b>9.7.4</b> Correlation</a></li>
<li class="chapter" data-level="9.7.5" data-path="power-analysis.html"><a href="power-analysis.html#chi-square-test"><i class="fa fa-check"></i><b>9.7.5</b> Chi-square test</a></li>
<li class="chapter" data-level="9.7.6" data-path="power-analysis.html"><a href="power-analysis.html#linear-regression"><i class="fa fa-check"></i><b>9.7.6</b> Linear regression</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="power-analysis.html"><a href="power-analysis.html#visualizing-power-power-curves"><i class="fa fa-check"></i><b>9.8</b> Visualizing power: Power curves</a></li>
<li class="chapter" data-level="9.9" data-path="power-analysis.html"><a href="power-analysis.html#what-if-you-cant-achieve-adequate-power"><i class="fa fa-check"></i><b>9.9</b> What if you can’t achieve adequate power?</a>
<ul>
<li class="chapter" data-level="9.9.1" data-path="power-analysis.html"><a href="power-analysis.html#increase-effect-size-through-design"><i class="fa fa-check"></i><b>9.9.1</b> 1. Increase effect size (through design)</a></li>
<li class="chapter" data-level="9.9.2" data-path="power-analysis.html"><a href="power-analysis.html#increase-%CE%B1-with-caution"><i class="fa fa-check"></i><b>9.9.2</b> 2. Increase α (with caution)</a></li>
<li class="chapter" data-level="9.9.3" data-path="power-analysis.html"><a href="power-analysis.html#accept-lower-power-and-be-honest-about-it"><i class="fa fa-check"></i><b>9.9.3</b> 3. Accept lower power (and be honest about it)</a></li>
<li class="chapter" data-level="9.9.4" data-path="power-analysis.html"><a href="power-analysis.html#change-the-question"><i class="fa fa-check"></i><b>9.9.4</b> 4. Change the question</a></li>
<li class="chapter" data-level="9.9.5" data-path="power-analysis.html"><a href="power-analysis.html#collaborate"><i class="fa fa-check"></i><b>9.9.5</b> 5. Collaborate</a></li>
</ul></li>
<li class="chapter" data-level="9.10" data-path="power-analysis.html"><a href="power-analysis.html#post-hoc-power-dont-do-it"><i class="fa fa-check"></i><b>9.10</b> Post-hoc power: Don’t do it</a></li>
<li class="chapter" data-level="9.11" data-path="power-analysis.html"><a href="power-analysis.html#power-for-complex-designs-simulation"><i class="fa fa-check"></i><b>9.11</b> Power for complex designs: Simulation</a>
<ul>
<li class="chapter" data-level="9.11.1" data-path="power-analysis.html"><a href="power-analysis.html#example-power-for-a-mixed-model-using-simr"><i class="fa fa-check"></i><b>9.11.1</b> Example: Power for a mixed model using simr</a></li>
<li class="chapter" data-level="9.11.2" data-path="power-analysis.html"><a href="power-analysis.html#example-power-for-factorial-designs-using-superpower"><i class="fa fa-check"></i><b>9.11.2</b> Example: Power for factorial designs using Superpower</a></li>
</ul></li>
<li class="chapter" data-level="9.12" data-path="power-analysis.html"><a href="power-analysis.html#practical-workflow-for-power-analysis"><i class="fa fa-check"></i><b>9.12</b> Practical workflow for power analysis</a>
<ul>
<li class="chapter" data-level="9.12.1" data-path="power-analysis.html"><a href="power-analysis.html#step-1-define-your-primary-hypothesis"><i class="fa fa-check"></i><b>9.12.1</b> Step 1: Define your primary hypothesis</a></li>
<li class="chapter" data-level="9.12.2" data-path="power-analysis.html"><a href="power-analysis.html#step-2-determine-the-minimum-meaningful-effect"><i class="fa fa-check"></i><b>9.12.2</b> Step 2: Determine the minimum meaningful effect</a></li>
<li class="chapter" data-level="9.12.3" data-path="power-analysis.html"><a href="power-analysis.html#step-3-estimate-variability"><i class="fa fa-check"></i><b>9.12.3</b> Step 3: Estimate variability</a></li>
<li class="chapter" data-level="9.12.4" data-path="power-analysis.html"><a href="power-analysis.html#step-4-calculate-required-sample-size"><i class="fa fa-check"></i><b>9.12.4</b> Step 4: Calculate required sample size</a></li>
<li class="chapter" data-level="9.12.5" data-path="power-analysis.html"><a href="power-analysis.html#step-5-compare-to-feasibility"><i class="fa fa-check"></i><b>9.12.5</b> Step 5: Compare to feasibility</a></li>
<li class="chapter" data-level="9.12.6" data-path="power-analysis.html"><a href="power-analysis.html#step-6-document-everything"><i class="fa fa-check"></i><b>9.12.6</b> Step 6: Document everything</a></li>
</ul></li>
<li class="chapter" data-level="9.13" data-path="power-analysis.html"><a href="power-analysis.html#common-mistakes-in-power-analysis"><i class="fa fa-check"></i><b>9.13</b> Common mistakes in power analysis</a></li>
<li class="chapter" data-level="9.14" data-path="power-analysis.html"><a href="power-analysis.html#summary"><i class="fa fa-check"></i><b>9.14</b> Summary</a></li>
<li class="chapter" data-level="9.15" data-path="power-analysis.html"><a href="power-analysis.html#key-takeaways-2"><i class="fa fa-check"></i><b>9.15</b> Key takeaways</a></li>
<li class="chapter" data-level="9.16" data-path="power-analysis.html"><a href="power-analysis.html#assignment-4"><i class="fa fa-check"></i><b>9.16</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html"><i class="fa fa-check"></i><b>10</b> Selecting Statistical Tests</a>
<ul>
<li class="chapter" data-level="10.1" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#response-and-explanatory-variables"><i class="fa fa-check"></i><b>10.1</b> Response and explanatory variables</a></li>
<li class="chapter" data-level="10.2" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#variable-types-refresher"><i class="fa fa-check"></i><b>10.2</b> Variable types refresher</a></li>
<li class="chapter" data-level="10.3" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#the-decision-matrix"><i class="fa fa-check"></i><b>10.3</b> The decision matrix</a></li>
<li class="chapter" data-level="10.4" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#setup-1"><i class="fa fa-check"></i><b>10.4</b> Setup</a></li>
<li class="chapter" data-level="10.5" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#chi-square-and-g-tests"><i class="fa fa-check"></i><b>10.5</b> Chi-square and G-tests</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#the-scenario"><i class="fa fa-check"></i><b>10.5.1</b> The scenario</a></li>
<li class="chapter" data-level="10.5.2" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#create-the-data"><i class="fa fa-check"></i><b>10.5.2</b> Create the data</a></li>
<li class="chapter" data-level="10.5.3" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#run-the-test"><i class="fa fa-check"></i><b>10.5.3</b> Run the test</a></li>
<li class="chapter" data-level="10.5.4" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#interpret-the-output"><i class="fa fa-check"></i><b>10.5.4</b> Interpret the output</a></li>
<li class="chapter" data-level="10.5.5" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#calculate-effect-size"><i class="fa fa-check"></i><b>10.5.5</b> Calculate effect size</a></li>
<li class="chapter" data-level="10.5.6" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#visualize"><i class="fa fa-check"></i><b>10.5.6</b> Visualize</a></li>
<li class="chapter" data-level="10.5.7" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#write-the-results"><i class="fa fa-check"></i><b>10.5.7</b> Write the results</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#t-test"><i class="fa fa-check"></i><b>10.6</b> t-test</a>
<ul>
<li class="chapter" data-level="10.6.1" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#the-scenario-1"><i class="fa fa-check"></i><b>10.6.1</b> The scenario</a></li>
<li class="chapter" data-level="10.6.2" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#create-the-data-1"><i class="fa fa-check"></i><b>10.6.2</b> Create the data</a></li>
<li class="chapter" data-level="10.6.3" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#check-assumptions"><i class="fa fa-check"></i><b>10.6.3</b> Check assumptions</a></li>
<li class="chapter" data-level="10.6.4" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#run-the-test-1"><i class="fa fa-check"></i><b>10.6.4</b> Run the test</a></li>
<li class="chapter" data-level="10.6.5" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#interpret-the-output-1"><i class="fa fa-check"></i><b>10.6.5</b> Interpret the output</a></li>
<li class="chapter" data-level="10.6.6" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#calculate-effect-size-1"><i class="fa fa-check"></i><b>10.6.6</b> Calculate effect size</a></li>
<li class="chapter" data-level="10.6.7" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#visualize-1"><i class="fa fa-check"></i><b>10.6.7</b> Visualize</a></li>
<li class="chapter" data-level="10.6.8" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#write-the-results-1"><i class="fa fa-check"></i><b>10.6.8</b> Write the results</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#anova"><i class="fa fa-check"></i><b>10.7</b> ANOVA</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#the-scenario-2"><i class="fa fa-check"></i><b>10.7.1</b> The scenario</a></li>
<li class="chapter" data-level="10.7.2" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#create-the-data-2"><i class="fa fa-check"></i><b>10.7.2</b> Create the data</a></li>
<li class="chapter" data-level="10.7.3" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#check-assumptions-1"><i class="fa fa-check"></i><b>10.7.3</b> Check assumptions</a></li>
<li class="chapter" data-level="10.7.4" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#run-the-test-2"><i class="fa fa-check"></i><b>10.7.4</b> Run the test</a></li>
<li class="chapter" data-level="10.7.5" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#interpret-the-output-2"><i class="fa fa-check"></i><b>10.7.5</b> Interpret the output</a></li>
<li class="chapter" data-level="10.7.6" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#post-hoc-comparisons"><i class="fa fa-check"></i><b>10.7.6</b> Post-hoc comparisons</a></li>
<li class="chapter" data-level="10.7.7" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#calculate-effect-size-2"><i class="fa fa-check"></i><b>10.7.7</b> Calculate effect size</a></li>
<li class="chapter" data-level="10.7.8" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#visualize-2"><i class="fa fa-check"></i><b>10.7.8</b> Visualize</a></li>
<li class="chapter" data-level="10.7.9" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#write-the-results-2"><i class="fa fa-check"></i><b>10.7.9</b> Write the results</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#linear-regression-1"><i class="fa fa-check"></i><b>10.8</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#the-scenario-3"><i class="fa fa-check"></i><b>10.8.1</b> The scenario</a></li>
<li class="chapter" data-level="10.8.2" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#create-the-data-3"><i class="fa fa-check"></i><b>10.8.2</b> Create the data</a></li>
<li class="chapter" data-level="10.8.3" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#explore-the-relationship-first"><i class="fa fa-check"></i><b>10.8.3</b> Explore the relationship first</a></li>
<li class="chapter" data-level="10.8.4" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#run-the-test-3"><i class="fa fa-check"></i><b>10.8.4</b> Run the test</a></li>
<li class="chapter" data-level="10.8.5" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#interpret-the-output-3"><i class="fa fa-check"></i><b>10.8.5</b> Interpret the output</a></li>
<li class="chapter" data-level="10.8.6" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#check-assumptions-with-diagnostic-plots"><i class="fa fa-check"></i><b>10.8.6</b> Check assumptions with diagnostic plots</a></li>
<li class="chapter" data-level="10.8.7" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#the-solution-polynomial-regression"><i class="fa fa-check"></i><b>10.8.7</b> The solution: Polynomial regression</a></li>
<li class="chapter" data-level="10.8.8" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#visualize-3"><i class="fa fa-check"></i><b>10.8.8</b> Visualize</a></li>
<li class="chapter" data-level="10.8.9" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#write-the-results-3"><i class="fa fa-check"></i><b>10.8.9</b> Write the results</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#logistic-regression"><i class="fa fa-check"></i><b>10.9</b> Logistic Regression</a>
<ul>
<li class="chapter" data-level="10.9.1" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#the-scenario-4"><i class="fa fa-check"></i><b>10.9.1</b> The scenario</a></li>
<li class="chapter" data-level="10.9.2" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#create-the-data-4"><i class="fa fa-check"></i><b>10.9.2</b> Create the data</a></li>
<li class="chapter" data-level="10.9.3" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#visualize-raw-data"><i class="fa fa-check"></i><b>10.9.3</b> Visualize raw data</a></li>
<li class="chapter" data-level="10.9.4" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#run-the-test-4"><i class="fa fa-check"></i><b>10.9.4</b> Run the test</a></li>
<li class="chapter" data-level="10.9.5" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#interpret-the-output-4"><i class="fa fa-check"></i><b>10.9.5</b> Interpret the output</a></li>
<li class="chapter" data-level="10.9.6" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#test-significance"><i class="fa fa-check"></i><b>10.9.6</b> Test significance</a></li>
<li class="chapter" data-level="10.9.7" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#visualize-with-fitted-probability-curve"><i class="fa fa-check"></i><b>10.9.7</b> Visualize with fitted probability curve</a></li>
<li class="chapter" data-level="10.9.8" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#write-the-results-4"><i class="fa fa-check"></i><b>10.9.8</b> Write the results</a></li>
</ul></li>
<li class="chapter" data-level="10.10" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#summary-choosing-the-right-test"><i class="fa fa-check"></i><b>10.10</b> Summary: Choosing the right test</a></li>
<li class="chapter" data-level="10.11" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#what-if-assumptions-arent-met"><i class="fa fa-check"></i><b>10.11</b> What if assumptions aren’t met?</a></li>
<li class="chapter" data-level="10.12" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#looking-ahead-generalized-linear-models"><i class="fa fa-check"></i><b>10.12</b> Looking ahead: Generalized Linear Models</a></li>
<li class="chapter" data-level="10.13" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#assignment-5"><i class="fa fa-check"></i><b>10.13</b> Assignment</a>
<ul>
<li class="chapter" data-level="10.13.1" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#part-1-identify-the-test"><i class="fa fa-check"></i><b>10.13.1</b> Part 1: Identify the test</a></li>
<li class="chapter" data-level="10.13.2" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#part-2-run-and-interpret"><i class="fa fa-check"></i><b>10.13.2</b> Part 2: Run and interpret</a></li>
<li class="chapter" data-level="10.13.3" data-path="selecting-statistical-tests.html"><a href="selecting-statistical-tests.html#deliverables"><i class="fa fa-check"></i><b>10.13.3</b> Deliverables</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="basic-statistical-testing.html"><a href="basic-statistical-testing.html"><i class="fa fa-check"></i><b>11</b> Basic statistical testing</a>
<ul>
<li class="chapter" data-level="11.1" data-path="basic-statistical-testing.html"><a href="basic-statistical-testing.html#a-short-statistical-review"><i class="fa fa-check"></i><b>11.1</b> A short statistical review</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html"><i class="fa fa-check"></i><b>12</b> Hypothesis Testing Fundamentals</a>
<ul>
<li class="chapter" data-level="12.1" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html#the-core-question"><i class="fa fa-check"></i><b>12.1</b> The core question</a></li>
<li class="chapter" data-level="12.2" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html#a-worked-example-seedling-heights-after-fire"><i class="fa fa-check"></i><b>12.2</b> A worked example: Seedling heights after fire</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html#the-scenario-5"><i class="fa fa-check"></i><b>12.2.1</b> The scenario</a></li>
<li class="chapter" data-level="12.2.2" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html#the-data"><i class="fa fa-check"></i><b>12.2.2</b> The data</a></li>
<li class="chapter" data-level="12.2.3" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html#visualize-first"><i class="fa fa-check"></i><b>12.2.3</b> Visualize first</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html#step-1-state-the-hypotheses"><i class="fa fa-check"></i><b>12.3</b> Step 1: State the hypotheses</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html#the-null-hypothesis-h%E2%82%80"><i class="fa fa-check"></i><b>12.3.1</b> The null hypothesis (H₀)</a></li>
<li class="chapter" data-level="12.3.2" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html#the-alternative-hypothesis-h_a"><i class="fa fa-check"></i><b>12.3.2</b> The alternative hypothesis (H_A)</a></li>
<li class="chapter" data-level="12.3.3" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html#one-tailed-vs.-two-tailed-tests"><i class="fa fa-check"></i><b>12.3.3</b> One-tailed vs. two-tailed tests</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="hypothesis-testing-fundamentals.html"><a href="hypothesis-testing-fundamentals.html#step-2-calculate-a-test-statistic"><i class="fa fa-check"></i><b>12.4</b> Step 2: Calculate a test statistic</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html"><i class="fa fa-check"></i><b>13</b> Comparing Groups: t-tests and ANOVA</a>
<ul>
<li class="chapter" data-level="13.1" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#the-big-picture-partitioning-variance"><i class="fa fa-check"></i><b>13.1</b> The big picture: Partitioning variance</a></li>
<li class="chapter" data-level="13.2" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#setup-2"><i class="fa fa-check"></i><b>13.2</b> Setup</a></li>
<li class="chapter" data-level="13.3" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#part-1-t-tests"><i class="fa fa-check"></i><b>13.3</b> Part 1: t-tests</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#one-sample-t-test"><i class="fa fa-check"></i><b>13.3.1</b> One-sample t-test</a></li>
<li class="chapter" data-level="13.3.2" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#the-question"><i class="fa fa-check"></i><b>13.3.2</b> The question</a></li>
<li class="chapter" data-level="13.3.3" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#ecological-example"><i class="fa fa-check"></i><b>13.3.3</b> Ecological example</a></li>
<li class="chapter" data-level="13.3.4" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#hypotheses"><i class="fa fa-check"></i><b>13.3.4</b> Hypotheses</a></li>
<li class="chapter" data-level="13.3.5" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#check-assumptions-2"><i class="fa fa-check"></i><b>13.3.5</b> Check assumptions</a></li>
<li class="chapter" data-level="13.3.6" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#run-the-test-5"><i class="fa fa-check"></i><b>13.3.6</b> Run the test</a></li>
<li class="chapter" data-level="13.3.7" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#interpret-the-output-5"><i class="fa fa-check"></i><b>13.3.7</b> Interpret the output</a></li>
<li class="chapter" data-level="13.3.8" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#effect-size"><i class="fa fa-check"></i><b>13.3.8</b> Effect size</a></li>
<li class="chapter" data-level="13.3.9" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#results-statement"><i class="fa fa-check"></i><b>13.3.9</b> Results statement</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#two-sample-t-test-1"><i class="fa fa-check"></i><b>13.4</b> Two-sample t-test</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#the-question-1"><i class="fa fa-check"></i><b>13.4.1</b> The question</a></li>
<li class="chapter" data-level="13.4.2" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#ecological-example-1"><i class="fa fa-check"></i><b>13.4.2</b> Ecological example</a></li>
<li class="chapter" data-level="13.4.3" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#hypotheses-1"><i class="fa fa-check"></i><b>13.4.3</b> Hypotheses</a></li>
<li class="chapter" data-level="13.4.4" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#check-assumptions-3"><i class="fa fa-check"></i><b>13.4.4</b> Check assumptions</a></li>
<li class="chapter" data-level="13.4.5" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#run-the-test-6"><i class="fa fa-check"></i><b>13.4.5</b> Run the test</a></li>
<li class="chapter" data-level="13.4.6" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#interpret-the-output-6"><i class="fa fa-check"></i><b>13.4.6</b> Interpret the output</a></li>
<li class="chapter" data-level="13.4.7" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#effect-size-1"><i class="fa fa-check"></i><b>13.4.7</b> Effect size</a></li>
<li class="chapter" data-level="13.4.8" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#visualize-4"><i class="fa fa-check"></i><b>13.4.8</b> Visualize</a></li>
<li class="chapter" data-level="13.4.9" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#sample-methods-and-results"><i class="fa fa-check"></i><b>13.4.9</b> Sample methods and results</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#paired-t-test-1"><i class="fa fa-check"></i><b>13.5</b> Paired t-test</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#the-question-2"><i class="fa fa-check"></i><b>13.5.1</b> The question</a></li>
<li class="chapter" data-level="13.5.2" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#ecological-example-2"><i class="fa fa-check"></i><b>13.5.2</b> Ecological example</a></li>
<li class="chapter" data-level="13.5.3" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#why-paired-tests-are-more-powerful"><i class="fa fa-check"></i><b>13.5.3</b> Why paired tests are more powerful</a></li>
<li class="chapter" data-level="13.5.4" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#hypotheses-2"><i class="fa fa-check"></i><b>13.5.4</b> Hypotheses</a></li>
<li class="chapter" data-level="13.5.5" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#check-assumptions-4"><i class="fa fa-check"></i><b>13.5.5</b> Check assumptions</a></li>
<li class="chapter" data-level="13.5.6" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#run-the-test-7"><i class="fa fa-check"></i><b>13.5.6</b> Run the test</a></li>
<li class="chapter" data-level="13.5.7" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#effect-size-2"><i class="fa fa-check"></i><b>13.5.7</b> Effect size</a></li>
<li class="chapter" data-level="13.5.8" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#results-statement-1"><i class="fa fa-check"></i><b>13.5.8</b> Results statement</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#when-assumptions-are-violated"><i class="fa fa-check"></i><b>13.6</b> When assumptions are violated</a></li>
<li class="chapter" data-level="13.7" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#part-2-anova"><i class="fa fa-check"></i><b>13.7</b> Part 2: ANOVA</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#from-t-test-to-anova"><i class="fa fa-check"></i><b>13.7.1</b> From t-test to ANOVA</a></li>
<li class="chapter" data-level="13.7.2" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#the-logic-of-anova"><i class="fa fa-check"></i><b>13.7.2</b> The logic of ANOVA</a></li>
<li class="chapter" data-level="13.7.3" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#one-way-anova-1"><i class="fa fa-check"></i><b>13.7.3</b> One-way ANOVA</a></li>
<li class="chapter" data-level="13.7.4" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#the-question-3"><i class="fa fa-check"></i><b>13.7.4</b> The question</a></li>
<li class="chapter" data-level="13.7.5" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#ecological-example-3"><i class="fa fa-check"></i><b>13.7.5</b> Ecological example</a></li>
<li class="chapter" data-level="13.7.6" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#hypotheses-3"><i class="fa fa-check"></i><b>13.7.6</b> Hypotheses</a></li>
<li class="chapter" data-level="13.7.7" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#check-assumptions-5"><i class="fa fa-check"></i><b>13.7.7</b> Check assumptions</a></li>
<li class="chapter" data-level="13.7.8" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#run-the-test-8"><i class="fa fa-check"></i><b>13.7.8</b> Run the test</a></li>
<li class="chapter" data-level="13.7.9" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#interpret-the-anova-table"><i class="fa fa-check"></i><b>13.7.9</b> Interpret the ANOVA table</a></li>
<li class="chapter" data-level="13.7.10" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#effect-size-3"><i class="fa fa-check"></i><b>13.7.10</b> Effect size</a></li>
<li class="chapter" data-level="13.7.11" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#post-hoc-tests-which-groups-differ"><i class="fa fa-check"></i><b>13.7.11</b> Post-hoc tests: Which groups differ?</a></li>
<li class="chapter" data-level="13.7.12" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#visualize-5"><i class="fa fa-check"></i><b>13.7.12</b> Visualize</a></li>
<li class="chapter" data-level="13.7.13" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#sample-methods-and-results-1"><i class="fa fa-check"></i><b>13.7.13</b> Sample methods and results</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#beyond-one-way-anova"><i class="fa fa-check"></i><b>13.8</b> Beyond one-way ANOVA</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#two-way-anova-factorial-designs"><i class="fa fa-check"></i><b>13.8.1</b> Two-way ANOVA (Factorial designs)</a></li>
<li class="chapter" data-level="13.8.2" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#ancova-continuous-covariate"><i class="fa fa-check"></i><b>13.8.2</b> ANCOVA (Continuous covariate)</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#the-connection-to-linear-models"><i class="fa fa-check"></i><b>13.9</b> The connection to linear models</a></li>
<li class="chapter" data-level="13.10" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#summary-t-tests-vs.-anova"><i class="fa fa-check"></i><b>13.10</b> Summary: t-tests vs. ANOVA</a></li>
<li class="chapter" data-level="13.11" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#key-assumptions-and-alternatives"><i class="fa fa-check"></i><b>13.11</b> Key assumptions and alternatives</a></li>
<li class="chapter" data-level="13.12" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#key-takeaways-3"><i class="fa fa-check"></i><b>13.12</b> Key takeaways</a></li>
<li class="chapter" data-level="13.13" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#assignment-6"><i class="fa fa-check"></i><b>13.13</b> Assignment</a>
<ul>
<li class="chapter" data-level="13.13.1" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#part-1-conceptual-questions"><i class="fa fa-check"></i><b>13.13.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="13.13.2" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#part-2-t-test-analysis"><i class="fa fa-check"></i><b>13.13.2</b> Part 2: t-test analysis</a></li>
<li class="chapter" data-level="13.13.3" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#part-3-anova-analysis"><i class="fa fa-check"></i><b>13.13.3</b> Part 3: ANOVA analysis</a></li>
<li class="chapter" data-level="13.13.4" data-path="comparing-groups-t-tests-and-anova.html"><a href="comparing-groups-t-tests-and-anova.html#part-4-reflection"><i class="fa fa-check"></i><b>13.13.4</b> Part 4: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html"><i class="fa fa-check"></i><b>14</b> Testing Relationships: Regression</a>
<ul>
<li class="chapter" data-level="14.1" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#the-core-idea"><i class="fa fa-check"></i><b>14.1</b> The core idea</a></li>
<li class="chapter" data-level="14.2" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#the-regression-equation"><i class="fa fa-check"></i><b>14.2</b> The regression equation</a></li>
<li class="chapter" data-level="14.3" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#setup-3"><i class="fa fa-check"></i><b>14.3</b> Setup</a></li>
<li class="chapter" data-level="14.4" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#part-1-simple-linear-regression"><i class="fa fa-check"></i><b>14.4</b> Part 1: Simple Linear Regression</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#ecological-example-tree-growth-and-light-availability"><i class="fa fa-check"></i><b>14.4.1</b> Ecological example: Tree growth and light availability</a></li>
<li class="chapter" data-level="14.4.2" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#visualize-first-always-plot-your-data"><i class="fa fa-check"></i><b>14.4.2</b> Visualize first: Always plot your data</a></li>
<li class="chapter" data-level="14.4.3" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#hypotheses-4"><i class="fa fa-check"></i><b>14.4.3</b> Hypotheses</a></li>
<li class="chapter" data-level="14.4.4" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#fit-the-model"><i class="fa fa-check"></i><b>14.4.4</b> Fit the model</a></li>
<li class="chapter" data-level="14.4.5" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#interpret-the-output-7"><i class="fa fa-check"></i><b>14.4.5</b> Interpret the output</a></li>
<li class="chapter" data-level="14.4.6" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#visualize-with-fitted-line"><i class="fa fa-check"></i><b>14.4.6</b> Visualize with fitted line</a></li>
<li class="chapter" data-level="14.4.7" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#check-assumptions-6"><i class="fa fa-check"></i><b>14.4.7</b> Check assumptions</a></li>
<li class="chapter" data-level="14.4.8" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#confidence-and-prediction-intervals"><i class="fa fa-check"></i><b>14.4.8</b> Confidence and prediction intervals</a></li>
<li class="chapter" data-level="14.4.9" data-path="testing-relationships-regression.html"><a href="testing-relationships-regression.html#sample-methods-and-results-2"><i class="fa fa-check"></i><b>14.4.9</b> Sample methods and results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="part-2-when-things-go-wrong.html"><a href="part-2-when-things-go-wrong.html"><i class="fa fa-check"></i><b>15</b> Part 2: When things go wrong</a>
<ul>
<li class="chapter" data-level="15.1" data-path="part-2-when-things-go-wrong.html"><a href="part-2-when-things-go-wrong.html#problem-1-non-linear-relationships"><i class="fa fa-check"></i><b>15.1</b> Problem 1: Non-linear relationships</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="part-2-when-things-go-wrong.html"><a href="part-2-when-things-go-wrong.html#detection"><i class="fa fa-check"></i><b>15.1.1</b> Detection</a></li>
<li class="chapter" data-level="15.1.2" data-path="part-2-when-things-go-wrong.html"><a href="part-2-when-things-go-wrong.html#solution-polynomial-regression"><i class="fa fa-check"></i><b>15.1.2</b> Solution: Polynomial regression</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="part-2-when-things-go-wrong.html"><a href="part-2-when-things-go-wrong.html#problem-2-heteroscedasticity-unequal-variance"><i class="fa fa-check"></i><b>15.2</b> Problem 2: Heteroscedasticity (unequal variance)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="part-2-when-things-go-wrong.html"><a href="part-2-when-things-go-wrong.html#detection-1"><i class="fa fa-check"></i><b>15.2.1</b> Detection</a></li>
<li class="chapter" data-level="15.2.2" data-path="part-2-when-things-go-wrong.html"><a href="part-2-when-things-go-wrong.html#solutions"><i class="fa fa-check"></i><b>15.2.2</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="part-2-when-things-go-wrong.html"><a href="part-2-when-things-go-wrong.html#problem-3-influential-points"><i class="fa fa-check"></i><b>15.3</b> Problem 3: Influential points</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="part-2-when-things-go-wrong.html"><a href="part-2-when-things-go-wrong.html#detection-cooks-distance"><i class="fa fa-check"></i><b>15.3.1</b> Detection: Cook’s Distance</a></li>
<li class="chapter" data-level="15.3.2" data-path="part-2-when-things-go-wrong.html"><a href="part-2-when-things-go-wrong.html#what-to-do"><i class="fa fa-check"></i><b>15.3.2</b> What to do</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="part-3-multiple-regression.html"><a href="part-3-multiple-regression.html"><i class="fa fa-check"></i><b>16</b> Part 3: Multiple Regression</a>
<ul>
<li class="chapter" data-level="16.1" data-path="part-3-multiple-regression.html"><a href="part-3-multiple-regression.html#ecological-example-predicting-tree-growth"><i class="fa fa-check"></i><b>16.1</b> Ecological example: Predicting tree growth</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="part-3-multiple-regression.html"><a href="part-3-multiple-regression.html#interpret-the-coefficients"><i class="fa fa-check"></i><b>16.1.1</b> Interpret the coefficients</a></li>
<li class="chapter" data-level="16.1.2" data-path="part-3-multiple-regression.html"><a href="part-3-multiple-regression.html#visualizing-multiple-regression"><i class="fa fa-check"></i><b>16.1.2</b> Visualizing multiple regression</a></li>
<li class="chapter" data-level="16.1.3" data-path="part-3-multiple-regression.html"><a href="part-3-multiple-regression.html#comparing-models-which-predictors-matter"><i class="fa fa-check"></i><b>16.1.3</b> Comparing models: Which predictors matter?</a></li>
<li class="chapter" data-level="16.1.4" data-path="part-3-multiple-regression.html"><a href="part-3-multiple-regression.html#multicollinearity-when-predictors-are-correlated"><i class="fa fa-check"></i><b>16.1.4</b> Multicollinearity: When predictors are correlated</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html"><i class="fa fa-check"></i><b>17</b> Part 4: Reporting regression results</a>
<ul>
<li class="chapter" data-level="17.1" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#what-to-include"><i class="fa fa-check"></i><b>17.1</b> What to include</a></li>
<li class="chapter" data-level="17.2" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#results-statement-templates"><i class="fa fa-check"></i><b>17.2</b> Results statement templates</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#simple-regression"><i class="fa fa-check"></i><b>17.2.1</b> Simple regression</a></li>
<li class="chapter" data-level="17.2.2" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#multiple-regression"><i class="fa fa-check"></i><b>17.2.2</b> Multiple regression</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#complete-example"><i class="fa fa-check"></i><b>17.3</b> Complete example</a></li>
<li class="chapter" data-level="17.4" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#connection-to-linear-models-and-glms"><i class="fa fa-check"></i><b>17.4</b> Connection to linear models and GLMs</a></li>
<li class="chapter" data-level="17.5" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#summary-table"><i class="fa fa-check"></i><b>17.5</b> Summary table</a></li>
<li class="chapter" data-level="17.6" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#key-assumptions-and-checks"><i class="fa fa-check"></i><b>17.6</b> Key assumptions and checks</a></li>
<li class="chapter" data-level="17.7" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#key-takeaways-4"><i class="fa fa-check"></i><b>17.7</b> Key takeaways</a></li>
<li class="chapter" data-level="17.8" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#assignment-7"><i class="fa fa-check"></i><b>17.8</b> Assignment</a>
<ul>
<li class="chapter" data-level="17.8.1" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#part-1-conceptual-questions-1"><i class="fa fa-check"></i><b>17.8.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="17.8.2" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#part-2-simple-regression-analysis"><i class="fa fa-check"></i><b>17.8.2</b> Part 2: Simple regression analysis</a></li>
<li class="chapter" data-level="17.8.3" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#part-3-multiple-regression-optional-challenge"><i class="fa fa-check"></i><b>17.8.3</b> Part 3: Multiple regression (optional challenge)</a></li>
<li class="chapter" data-level="17.8.4" data-path="part-4-reporting-regression-results.html"><a href="part-4-reporting-regression-results.html#part-4-reflection-1"><i class="fa fa-check"></i><b>17.8.4</b> Part 4: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="analyzing-categorical-data-chi-square-and-related-tests.html"><a href="analyzing-categorical-data-chi-square-and-related-tests.html"><i class="fa fa-check"></i><b>18</b> Analyzing Categorical Data: Chi-square and Related Tests</a>
<ul>
<li class="chapter" data-level="18.1" data-path="analyzing-categorical-data-chi-square-and-related-tests.html"><a href="analyzing-categorical-data-chi-square-and-related-tests.html#the-core-question-1"><i class="fa fa-check"></i><b>18.1</b> The core question</a></li>
<li class="chapter" data-level="18.2" data-path="analyzing-categorical-data-chi-square-and-related-tests.html"><a href="analyzing-categorical-data-chi-square-and-related-tests.html#types-of-chi-square-tests"><i class="fa fa-check"></i><b>18.2</b> Types of chi-square tests</a></li>
<li class="chapter" data-level="18.3" data-path="analyzing-categorical-data-chi-square-and-related-tests.html"><a href="analyzing-categorical-data-chi-square-and-related-tests.html#setup-4"><i class="fa fa-check"></i><b>18.3</b> Setup</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html"><i class="fa fa-check"></i><b>19</b> Part 1: Chi-square Test of Independence</a>
<ul>
<li class="chapter" data-level="19.1" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#the-question-4"><i class="fa fa-check"></i><b>19.1</b> The question</a></li>
<li class="chapter" data-level="19.2" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#ecological-example-nest-success-and-habitat-type"><i class="fa fa-check"></i><b>19.2</b> Ecological example: Nest success and habitat type</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#hypotheses-5"><i class="fa fa-check"></i><b>19.2.1</b> Hypotheses</a></li>
<li class="chapter" data-level="19.2.2" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#the-logic-observed-vs.-expected"><i class="fa fa-check"></i><b>19.2.2</b> The logic: Observed vs. expected</a></li>
<li class="chapter" data-level="19.2.3" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#the-chi-square-statistic"><i class="fa fa-check"></i><b>19.2.3</b> The chi-square statistic</a></li>
<li class="chapter" data-level="19.2.4" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#run-the-test-9"><i class="fa fa-check"></i><b>19.2.4</b> Run the test</a></li>
<li class="chapter" data-level="19.2.5" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#interpret-the-output-8"><i class="fa fa-check"></i><b>19.2.5</b> Interpret the output</a></li>
<li class="chapter" data-level="19.2.6" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#examine-the-pattern-which-cells-drive-the-result"><i class="fa fa-check"></i><b>19.2.6</b> Examine the pattern: Which cells drive the result?</a></li>
<li class="chapter" data-level="19.2.7" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#check-assumptions-7"><i class="fa fa-check"></i><b>19.2.7</b> Check assumptions</a></li>
<li class="chapter" data-level="19.2.8" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#effect-size-cram%C3%A9rs-v"><i class="fa fa-check"></i><b>19.2.8</b> Effect size: Cramér’s V</a></li>
<li class="chapter" data-level="19.2.9" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#visualize-6"><i class="fa fa-check"></i><b>19.2.9</b> Visualize</a></li>
<li class="chapter" data-level="19.2.10" data-path="part-1-chi-square-test-of-independence.html"><a href="part-1-chi-square-test-of-independence.html#sample-methods-and-results-3"><i class="fa fa-check"></i><b>19.2.10</b> Sample methods and results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="part-2-chi-square-goodness-of-fit.html"><a href="part-2-chi-square-goodness-of-fit.html"><i class="fa fa-check"></i><b>20</b> Part 2: Chi-square Goodness of Fit</a>
<ul>
<li class="chapter" data-level="20.1" data-path="part-2-chi-square-goodness-of-fit.html"><a href="part-2-chi-square-goodness-of-fit.html#the-question-5"><i class="fa fa-check"></i><b>20.1</b> The question</a></li>
<li class="chapter" data-level="20.2" data-path="part-2-chi-square-goodness-of-fit.html"><a href="part-2-chi-square-goodness-of-fit.html#ecological-example-seed-dispersal-distances"><i class="fa fa-check"></i><b>20.2</b> Ecological example: Seed dispersal distances</a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="part-2-chi-square-goodness-of-fit.html"><a href="part-2-chi-square-goodness-of-fit.html#hypotheses-6"><i class="fa fa-check"></i><b>20.2.1</b> Hypotheses</a></li>
<li class="chapter" data-level="20.2.2" data-path="part-2-chi-square-goodness-of-fit.html"><a href="part-2-chi-square-goodness-of-fit.html#run-the-test-10"><i class="fa fa-check"></i><b>20.2.2</b> Run the test</a></li>
<li class="chapter" data-level="20.2.3" data-path="part-2-chi-square-goodness-of-fit.html"><a href="part-2-chi-square-goodness-of-fit.html#interpret"><i class="fa fa-check"></i><b>20.2.3</b> Interpret</a></li>
<li class="chapter" data-level="20.2.4" data-path="part-2-chi-square-goodness-of-fit.html"><a href="part-2-chi-square-goodness-of-fit.html#visualize-7"><i class="fa fa-check"></i><b>20.2.4</b> Visualize</a></li>
<li class="chapter" data-level="20.2.5" data-path="part-2-chi-square-goodness-of-fit.html"><a href="part-2-chi-square-goodness-of-fit.html#results-statement-2"><i class="fa fa-check"></i><b>20.2.5</b> Results statement</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="part-3-alternative-tests.html"><a href="part-3-alternative-tests.html"><i class="fa fa-check"></i><b>21</b> Part 3: Alternative Tests</a>
<ul>
<li class="chapter" data-level="21.1" data-path="part-3-alternative-tests.html"><a href="part-3-alternative-tests.html#g-test-likelihood-ratio-test"><i class="fa fa-check"></i><b>21.1</b> G-test (Likelihood Ratio Test)</a></li>
<li class="chapter" data-level="21.2" data-path="part-3-alternative-tests.html"><a href="part-3-alternative-tests.html#fishers-exact-test"><i class="fa fa-check"></i><b>21.2</b> Fisher’s Exact Test</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="part-3-alternative-tests.html"><a href="part-3-alternative-tests.html#example-rare-species-occurrence"><i class="fa fa-check"></i><b>21.2.1</b> Example: Rare species occurrence</a></li>
<li class="chapter" data-level="21.2.2" data-path="part-3-alternative-tests.html"><a href="part-3-alternative-tests.html#interpret-fishers-output"><i class="fa fa-check"></i><b>21.2.2</b> Interpret Fisher’s output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="part-4-22-tables-and-odds-ratios.html"><a href="part-4-22-tables-and-odds-ratios.html"><i class="fa fa-check"></i><b>22</b> Part 4: 2×2 Tables and Odds Ratios</a>
<ul>
<li class="chapter" data-level="22.1" data-path="part-4-22-tables-and-odds-ratios.html"><a href="part-4-22-tables-and-odds-ratios.html#understanding-odds-and-odds-ratios"><i class="fa fa-check"></i><b>22.1</b> Understanding odds and odds ratios</a>
<ul>
<li class="chapter" data-level="22.1.1" data-path="part-4-22-tables-and-odds-ratios.html"><a href="part-4-22-tables-and-odds-ratios.html#visualize-22-tables"><i class="fa fa-check"></i><b>22.1.1</b> Visualize 2×2 tables</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="part-5-common-pitfalls-and-solutions.html"><a href="part-5-common-pitfalls-and-solutions.html"><i class="fa fa-check"></i><b>23</b> Part 5: Common Pitfalls and Solutions</a>
<ul>
<li class="chapter" data-level="23.1" data-path="part-5-common-pitfalls-and-solutions.html"><a href="part-5-common-pitfalls-and-solutions.html#pitfall-1-low-expected-counts"><i class="fa fa-check"></i><b>23.1</b> Pitfall 1: Low expected counts</a></li>
<li class="chapter" data-level="23.2" data-path="part-5-common-pitfalls-and-solutions.html"><a href="part-5-common-pitfalls-and-solutions.html#pitfall-2-non-independent-observations"><i class="fa fa-check"></i><b>23.2</b> Pitfall 2: Non-independent observations</a></li>
<li class="chapter" data-level="23.3" data-path="part-5-common-pitfalls-and-solutions.html"><a href="part-5-common-pitfalls-and-solutions.html#pitfall-3-confusing-proportions-with-counts"><i class="fa fa-check"></i><b>23.3</b> Pitfall 3: Confusing proportions with counts</a></li>
<li class="chapter" data-level="23.4" data-path="part-5-common-pitfalls-and-solutions.html"><a href="part-5-common-pitfalls-and-solutions.html#pitfall-4-testing-the-wrong-hypothesis"><i class="fa fa-check"></i><b>23.4</b> Pitfall 4: Testing the wrong hypothesis</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="part-6-connection-to-glms.html"><a href="part-6-connection-to-glms.html"><i class="fa fa-check"></i><b>24</b> Part 6: Connection to GLMs</a></li>
<li class="chapter" data-level="25" data-path="part-7-summary.html"><a href="part-7-summary.html"><i class="fa fa-check"></i><b>25</b> Part 7: Summary</a>
<ul>
<li class="chapter" data-level="25.1" data-path="part-7-summary.html"><a href="part-7-summary.html#which-test-to-use"><i class="fa fa-check"></i><b>25.1</b> Which test to use?</a></li>
<li class="chapter" data-level="25.2" data-path="part-7-summary.html"><a href="part-7-summary.html#assumptions-checklist"><i class="fa fa-check"></i><b>25.2</b> Assumptions checklist</a></li>
<li class="chapter" data-level="25.3" data-path="part-7-summary.html"><a href="part-7-summary.html#reporting-checklist"><i class="fa fa-check"></i><b>25.3</b> Reporting checklist</a></li>
<li class="chapter" data-level="25.4" data-path="part-7-summary.html"><a href="part-7-summary.html#sample-methods-and-results-templates"><i class="fa fa-check"></i><b>25.4</b> Sample methods and results templates</a>
<ul>
<li class="chapter" data-level="25.4.1" data-path="part-7-summary.html"><a href="part-7-summary.html#chi-square-test-of-independence"><i class="fa fa-check"></i><b>25.4.1</b> Chi-square test of independence</a></li>
<li class="chapter" data-level="25.4.2" data-path="part-7-summary.html"><a href="part-7-summary.html#goodness-of-fit"><i class="fa fa-check"></i><b>25.4.2</b> Goodness of fit</a></li>
<li class="chapter" data-level="25.4.3" data-path="part-7-summary.html"><a href="part-7-summary.html#fishers-exact-test-with-odds-ratio"><i class="fa fa-check"></i><b>25.4.3</b> Fisher’s exact test with odds ratio</a></li>
</ul></li>
<li class="chapter" data-level="25.5" data-path="part-7-summary.html"><a href="part-7-summary.html#key-takeaways-5"><i class="fa fa-check"></i><b>25.5</b> Key takeaways</a></li>
<li class="chapter" data-level="25.6" data-path="part-7-summary.html"><a href="part-7-summary.html#assignment-8"><i class="fa fa-check"></i><b>25.6</b> Assignment</a>
<ul>
<li class="chapter" data-level="25.6.1" data-path="part-7-summary.html"><a href="part-7-summary.html#part-1-conceptual-questions-2"><i class="fa fa-check"></i><b>25.6.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="25.6.2" data-path="part-7-summary.html"><a href="part-7-summary.html#part-2-test-of-independence"><i class="fa fa-check"></i><b>25.6.2</b> Part 2: Test of independence</a></li>
<li class="chapter" data-level="25.6.3" data-path="part-7-summary.html"><a href="part-7-summary.html#part-3-goodness-of-fit"><i class="fa fa-check"></i><b>25.6.3</b> Part 3: Goodness of fit</a></li>
<li class="chapter" data-level="25.6.4" data-path="part-7-summary.html"><a href="part-7-summary.html#part-4-small-sample-analysis"><i class="fa fa-check"></i><b>25.6.4</b> Part 4: Small sample analysis</a></li>
<li class="chapter" data-level="25.6.5" data-path="part-7-summary.html"><a href="part-7-summary.html#part-5-reflection"><i class="fa fa-check"></i><b>25.6.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="generalized-linear-models-glms.html"><a href="generalized-linear-models-glms.html"><i class="fa fa-check"></i><b>26</b> Generalized Linear Models (GLMs)</a>
<ul>
<li class="chapter" data-level="26.1" data-path="generalized-linear-models-glms.html"><a href="generalized-linear-models-glms.html#the-problem-with-classic-methods"><i class="fa fa-check"></i><b>26.1</b> The problem with “classic” methods</a></li>
<li class="chapter" data-level="26.2" data-path="generalized-linear-models-glms.html"><a href="generalized-linear-models-glms.html#the-glm-framework"><i class="fa fa-check"></i><b>26.2</b> The GLM framework</a>
<ul>
<li class="chapter" data-level="26.2.1" data-path="generalized-linear-models-glms.html"><a href="generalized-linear-models-glms.html#the-linear-predictor-%CE%B7"><i class="fa fa-check"></i><b>26.2.1</b> 1. The linear predictor (η)</a></li>
<li class="chapter" data-level="26.2.2" data-path="generalized-linear-models-glms.html"><a href="generalized-linear-models-glms.html#the-error-distribution-family"><i class="fa fa-check"></i><b>26.2.2</b> 2. The error distribution (family)</a></li>
<li class="chapter" data-level="26.2.3" data-path="generalized-linear-models-glms.html"><a href="generalized-linear-models-glms.html#the-link-function-g"><i class="fa fa-check"></i><b>26.2.3</b> 3. The link function (g)</a></li>
</ul></li>
<li class="chapter" data-level="26.3" data-path="generalized-linear-models-glms.html"><a href="generalized-linear-models-glms.html#setup-5"><i class="fa fa-check"></i><b>26.3</b> Setup</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="part-1-everything-you-know-is-a-glm.html"><a href="part-1-everything-you-know-is-a-glm.html"><i class="fa fa-check"></i><b>27</b> Part 1: Everything you know is a GLM</a>
<ul>
<li class="chapter" data-level="27.1" data-path="part-1-everything-you-know-is-a-glm.html"><a href="part-1-everything-you-know-is-a-glm.html#anova-as-a-glm"><i class="fa fa-check"></i><b>27.1</b> ANOVA as a GLM</a></li>
<li class="chapter" data-level="27.2" data-path="part-1-everything-you-know-is-a-glm.html"><a href="part-1-everything-you-know-is-a-glm.html#regression-as-a-glm"><i class="fa fa-check"></i><b>27.2</b> Regression as a GLM</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="part-2-understanding-link-functions.html"><a href="part-2-understanding-link-functions.html"><i class="fa fa-check"></i><b>28</b> Part 2: Understanding Link Functions</a>
<ul>
<li class="chapter" data-level="28.1" data-path="part-2-understanding-link-functions.html"><a href="part-2-understanding-link-functions.html#the-identity-link-gaussian-glm"><i class="fa fa-check"></i><b>28.1</b> The identity link (Gaussian GLM)</a></li>
<li class="chapter" data-level="28.2" data-path="part-2-understanding-link-functions.html"><a href="part-2-understanding-link-functions.html#the-log-link-poisson-glm-for-counts"><i class="fa fa-check"></i><b>28.2</b> The log link (Poisson GLM for counts)</a></li>
<li class="chapter" data-level="28.3" data-path="part-2-understanding-link-functions.html"><a href="part-2-understanding-link-functions.html#the-logit-link-binomial-glm-for-proportionsbinary"><i class="fa fa-check"></i><b>28.3</b> The logit link (Binomial GLM for proportions/binary)</a>
<ul>
<li class="chapter" data-level="28.3.1" data-path="part-2-understanding-link-functions.html"><a href="part-2-understanding-link-functions.html#demonstrating-the-logit-transformation"><i class="fa fa-check"></i><b>28.3.1</b> Demonstrating the logit transformation</a></li>
<li class="chapter" data-level="28.3.2" data-path="part-2-understanding-link-functions.html"><a href="part-2-understanding-link-functions.html#creating-a-logistic-curve-from-coefficients"><i class="fa fa-check"></i><b>28.3.2</b> Creating a logistic curve from coefficients</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="29" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html"><i class="fa fa-check"></i><b>29</b> Part 3: Logistic Regression (Binomial GLM)</a>
<ul>
<li class="chapter" data-level="29.1" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html#when-to-use-it"><i class="fa fa-check"></i><b>29.1</b> When to use it</a></li>
<li class="chapter" data-level="29.2" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html#ecological-example-plant-survival-along-a-moisture-gradient"><i class="fa fa-check"></i><b>29.2</b> Ecological example: Plant survival along a moisture gradient</a>
<ul>
<li class="chapter" data-level="29.2.1" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html#visualize-the-raw-data"><i class="fa fa-check"></i><b>29.2.1</b> Visualize the raw data</a></li>
<li class="chapter" data-level="29.2.2" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html#fit-the-model-1"><i class="fa fa-check"></i><b>29.2.2</b> Fit the model</a></li>
<li class="chapter" data-level="29.2.3" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html#interpret-the-output-9"><i class="fa fa-check"></i><b>29.2.3</b> Interpret the output</a></li>
<li class="chapter" data-level="29.2.4" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html#test-significance-1"><i class="fa fa-check"></i><b>29.2.4</b> Test significance</a></li>
<li class="chapter" data-level="29.2.5" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html#generate-predictions"><i class="fa fa-check"></i><b>29.2.5</b> Generate predictions</a></li>
<li class="chapter" data-level="29.2.6" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html#visualize-with-fitted-curve"><i class="fa fa-check"></i><b>29.2.6</b> Visualize with fitted curve</a></li>
<li class="chapter" data-level="29.2.7" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html#calculate-the-ld50-ed50"><i class="fa fa-check"></i><b>29.2.7</b> Calculate the LD50 / ED50</a></li>
<li class="chapter" data-level="29.2.8" data-path="part-3-logistic-regression-binomial-glm.html"><a href="part-3-logistic-regression-binomial-glm.html#sample-methods-and-results-4"><i class="fa fa-check"></i><b>29.2.8</b> Sample methods and results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="30" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html"><i class="fa fa-check"></i><b>30</b> Part 4: Poisson Regression (Count Data)</a>
<ul>
<li class="chapter" data-level="30.1" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html#when-to-use-it-1"><i class="fa fa-check"></i><b>30.1</b> When to use it</a></li>
<li class="chapter" data-level="30.2" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html#ecological-example-pollinator-visits-and-flower-display"><i class="fa fa-check"></i><b>30.2</b> Ecological example: Pollinator visits and flower display</a>
<ul>
<li class="chapter" data-level="30.2.1" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html#visualize-8"><i class="fa fa-check"></i><b>30.2.1</b> Visualize</a></li>
<li class="chapter" data-level="30.2.2" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html#fit-the-model-2"><i class="fa fa-check"></i><b>30.2.2</b> Fit the model</a></li>
<li class="chapter" data-level="30.2.3" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html#interpret-coefficients"><i class="fa fa-check"></i><b>30.2.3</b> Interpret coefficients</a></li>
<li class="chapter" data-level="30.2.4" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html#check-for-overdispersion"><i class="fa fa-check"></i><b>30.2.4</b> Check for overdispersion</a></li>
<li class="chapter" data-level="30.2.5" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html#handling-overdispersion"><i class="fa fa-check"></i><b>30.2.5</b> Handling overdispersion</a></li>
<li class="chapter" data-level="30.2.6" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html#generate-predictions-1"><i class="fa fa-check"></i><b>30.2.6</b> Generate predictions</a></li>
<li class="chapter" data-level="30.2.7" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html#visualize-9"><i class="fa fa-check"></i><b>30.2.7</b> Visualize</a></li>
<li class="chapter" data-level="30.2.8" data-path="part-4-poisson-regression-count-data.html"><a href="part-4-poisson-regression-count-data.html#sample-results"><i class="fa fa-check"></i><b>30.2.8</b> Sample results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="31" data-path="part-5-glm-diagnostics.html"><a href="part-5-glm-diagnostics.html"><i class="fa fa-check"></i><b>31</b> Part 5: GLM Diagnostics</a>
<ul>
<li class="chapter" data-level="31.1" data-path="part-5-glm-diagnostics.html"><a href="part-5-glm-diagnostics.html#using-dharma-for-glm-diagnostics"><i class="fa fa-check"></i><b>31.1</b> Using DHARMa for GLM diagnostics</a></li>
<li class="chapter" data-level="31.2" data-path="part-5-glm-diagnostics.html"><a href="part-5-glm-diagnostics.html#traditional-glm-diagnostics"><i class="fa fa-check"></i><b>31.2</b> Traditional GLM diagnostics</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="part-6-model-comparison.html"><a href="part-6-model-comparison.html"><i class="fa fa-check"></i><b>32</b> Part 6: Model Comparison</a>
<ul>
<li class="chapter" data-level="32.1" data-path="part-6-model-comparison.html"><a href="part-6-model-comparison.html#comparing-models-with-aic"><i class="fa fa-check"></i><b>32.1</b> Comparing models with AIC</a></li>
<li class="chapter" data-level="32.2" data-path="part-6-model-comparison.html"><a href="part-6-model-comparison.html#likelihood-ratio-tests"><i class="fa fa-check"></i><b>32.2</b> Likelihood ratio tests</a></li>
</ul></li>
<li class="chapter" data-level="33" data-path="part-7-summary-tables.html"><a href="part-7-summary-tables.html"><i class="fa fa-check"></i><b>33</b> Part 7: Summary Tables</a>
<ul>
<li class="chapter" data-level="33.1" data-path="part-7-summary-tables.html"><a href="part-7-summary-tables.html#choosing-the-right-glm"><i class="fa fa-check"></i><b>33.1</b> Choosing the right GLM</a></li>
<li class="chapter" data-level="33.2" data-path="part-7-summary-tables.html"><a href="part-7-summary-tables.html#quick-reference-for-interpretation"><i class="fa fa-check"></i><b>33.2</b> Quick reference for interpretation</a></li>
<li class="chapter" data-level="33.3" data-path="part-7-summary-tables.html"><a href="part-7-summary-tables.html#assumptions-to-check"><i class="fa fa-check"></i><b>33.3</b> Assumptions to check</a></li>
<li class="chapter" data-level="33.4" data-path="part-7-summary-tables.html"><a href="part-7-summary-tables.html#key-takeaways-6"><i class="fa fa-check"></i><b>33.4</b> Key takeaways</a></li>
<li class="chapter" data-level="33.5" data-path="part-7-summary-tables.html"><a href="part-7-summary-tables.html#assignment-9"><i class="fa fa-check"></i><b>33.5</b> Assignment</a>
<ul>
<li class="chapter" data-level="33.5.1" data-path="part-7-summary-tables.html"><a href="part-7-summary-tables.html#part-1-conceptual-questions-3"><i class="fa fa-check"></i><b>33.5.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="33.5.2" data-path="part-7-summary-tables.html"><a href="part-7-summary-tables.html#part-2-logistic-regression"><i class="fa fa-check"></i><b>33.5.2</b> Part 2: Logistic regression</a></li>
<li class="chapter" data-level="33.5.3" data-path="part-7-summary-tables.html"><a href="part-7-summary-tables.html#part-3-poisson-regression"><i class="fa fa-check"></i><b>33.5.3</b> Part 3: Poisson regression</a></li>
<li class="chapter" data-level="33.5.4" data-path="part-7-summary-tables.html"><a href="part-7-summary-tables.html#part-4-reflection-2"><i class="fa fa-check"></i><b>33.5.4</b> Part 4: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="zero-inflated-and-hurdle-models.html"><a href="zero-inflated-and-hurdle-models.html"><i class="fa fa-check"></i><b>34</b> Zero-Inflated and Hurdle Models</a>
<ul>
<li class="chapter" data-level="34.1" data-path="zero-inflated-and-hurdle-models.html"><a href="zero-inflated-and-hurdle-models.html#why-so-many-zeros"><i class="fa fa-check"></i><b>34.1</b> Why so many zeros?</a></li>
<li class="chapter" data-level="34.2" data-path="zero-inflated-and-hurdle-models.html"><a href="zero-inflated-and-hurdle-models.html#setup-6"><i class="fa fa-check"></i><b>34.2</b> Setup</a></li>
</ul></li>
<li class="chapter" data-level="35" data-path="part-1-recognizing-zero-inflation.html"><a href="part-1-recognizing-zero-inflation.html"><i class="fa fa-check"></i><b>35</b> Part 1: Recognizing Zero-Inflation</a>
<ul>
<li class="chapter" data-level="35.1" data-path="part-1-recognizing-zero-inflation.html"><a href="part-1-recognizing-zero-inflation.html#what-does-zero-inflation-look-like"><i class="fa fa-check"></i><b>35.1</b> What does zero-inflation look like?</a></li>
<li class="chapter" data-level="35.2" data-path="part-1-recognizing-zero-inflation.html"><a href="part-1-recognizing-zero-inflation.html#testing-for-zero-inflation"><i class="fa fa-check"></i><b>35.2</b> Testing for zero-inflation</a></li>
<li class="chapter" data-level="35.3" data-path="part-1-recognizing-zero-inflation.html"><a href="part-1-recognizing-zero-inflation.html#formal-test-with-dharma"><i class="fa fa-check"></i><b>35.3</b> Formal test with DHARMa</a></li>
</ul></li>
<li class="chapter" data-level="36" data-path="part-2-the-conceptual-framework.html"><a href="part-2-the-conceptual-framework.html"><i class="fa fa-check"></i><b>36</b> Part 2: The Conceptual Framework</a>
<ul>
<li class="chapter" data-level="36.1" data-path="part-2-the-conceptual-framework.html"><a href="part-2-the-conceptual-framework.html#two-processes-one-dataset"><i class="fa fa-check"></i><b>36.1</b> Two processes, one dataset</a>
<ul>
<li class="chapter" data-level="36.1.1" data-path="part-2-the-conceptual-framework.html"><a href="part-2-the-conceptual-framework.html#ecological-interpretation"><i class="fa fa-check"></i><b>36.1.1</b> Ecological interpretation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="37" data-path="part-3-zero-inflated-models.html"><a href="part-3-zero-inflated-models.html"><i class="fa fa-check"></i><b>37</b> Part 3: Zero-Inflated Models</a>
<ul>
<li class="chapter" data-level="37.1" data-path="part-3-zero-inflated-models.html"><a href="part-3-zero-inflated-models.html#example-pollinator-visits-to-flowers"><i class="fa fa-check"></i><b>37.1</b> Example: Pollinator visits to flowers</a></li>
<li class="chapter" data-level="37.2" data-path="part-3-zero-inflated-models.html"><a href="part-3-zero-inflated-models.html#fitting-a-zero-inflated-poisson-zip"><i class="fa fa-check"></i><b>37.2</b> Fitting a zero-inflated Poisson (ZIP)</a>
<ul>
<li class="chapter" data-level="37.2.1" data-path="part-3-zero-inflated-models.html"><a href="part-3-zero-inflated-models.html#interpreting-zip-output"><i class="fa fa-check"></i><b>37.2.1</b> Interpreting ZIP output</a></li>
</ul></li>
<li class="chapter" data-level="37.3" data-path="part-3-zero-inflated-models.html"><a href="part-3-zero-inflated-models.html#zero-inflated-negative-binomial-zinb"><i class="fa fa-check"></i><b>37.3</b> Zero-inflated negative binomial (ZINB)</a></li>
<li class="chapter" data-level="37.4" data-path="part-3-zero-inflated-models.html"><a href="part-3-zero-inflated-models.html#comparing-models"><i class="fa fa-check"></i><b>37.4</b> Comparing models</a></li>
</ul></li>
<li class="chapter" data-level="38" data-path="part-4-hurdle-models.html"><a href="part-4-hurdle-models.html"><i class="fa fa-check"></i><b>38</b> Part 4: Hurdle Models</a>
<ul>
<li class="chapter" data-level="38.1" data-path="part-4-hurdle-models.html"><a href="part-4-hurdle-models.html#how-hurdle-models-differ"><i class="fa fa-check"></i><b>38.1</b> How hurdle models differ</a>
<ul>
<li class="chapter" data-level="38.1.1" data-path="part-4-hurdle-models.html"><a href="part-4-hurdle-models.html#when-to-use-each"><i class="fa fa-check"></i><b>38.1.1</b> When to use each</a></li>
</ul></li>
<li class="chapter" data-level="38.2" data-path="part-4-hurdle-models.html"><a href="part-4-hurdle-models.html#fitting-a-hurdle-model"><i class="fa fa-check"></i><b>38.2</b> Fitting a hurdle model</a>
<ul>
<li class="chapter" data-level="38.2.1" data-path="part-4-hurdle-models.html"><a href="part-4-hurdle-models.html#interpreting-hurdle-output"><i class="fa fa-check"></i><b>38.2.1</b> Interpreting hurdle output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="39" data-path="part-5-choosing-between-approaches.html"><a href="part-5-choosing-between-approaches.html"><i class="fa fa-check"></i><b>39</b> Part 5: Choosing Between Approaches</a>
<ul>
<li class="chapter" data-level="39.1" data-path="part-5-choosing-between-approaches.html"><a href="part-5-choosing-between-approaches.html#decision-flowchart"><i class="fa fa-check"></i><b>39.1</b> Decision flowchart</a></li>
<li class="chapter" data-level="39.2" data-path="part-5-choosing-between-approaches.html"><a href="part-5-choosing-between-approaches.html#practical-comparison"><i class="fa fa-check"></i><b>39.2</b> Practical comparison</a></li>
</ul></li>
<li class="chapter" data-level="40" data-path="part-6-diagnostics.html"><a href="part-6-diagnostics.html"><i class="fa fa-check"></i><b>40</b> Part 6: Diagnostics</a>
<ul>
<li class="chapter" data-level="40.1" data-path="part-6-diagnostics.html"><a href="part-6-diagnostics.html#dharma-for-zero-inflated-models"><i class="fa fa-check"></i><b>40.1</b> DHARMa for zero-inflated models</a></li>
<li class="chapter" data-level="40.2" data-path="part-6-diagnostics.html"><a href="part-6-diagnostics.html#check-zero-inflation"><i class="fa fa-check"></i><b>40.2</b> Check zero-inflation</a></li>
<li class="chapter" data-level="40.3" data-path="part-6-diagnostics.html"><a href="part-6-diagnostics.html#check-dispersion"><i class="fa fa-check"></i><b>40.3</b> Check dispersion</a></li>
<li class="chapter" data-level="40.4" data-path="part-6-diagnostics.html"><a href="part-6-diagnostics.html#compare-observed-vs-predicted"><i class="fa fa-check"></i><b>40.4</b> Compare observed vs predicted</a></li>
</ul></li>
<li class="chapter" data-level="41" data-path="part-7-complete-example-with-mixed-effects.html"><a href="part-7-complete-example-with-mixed-effects.html"><i class="fa fa-check"></i><b>41</b> Part 7: Complete Example with Mixed Effects</a>
<ul>
<li class="chapter" data-level="41.1" data-path="part-7-complete-example-with-mixed-effects.html"><a href="part-7-complete-example-with-mixed-effects.html#extract-variance-components"><i class="fa fa-check"></i><b>41.1</b> Extract variance components</a></li>
<li class="chapter" data-level="41.2" data-path="part-7-complete-example-with-mixed-effects.html"><a href="part-7-complete-example-with-mixed-effects.html#estimated-marginal-means"><i class="fa fa-check"></i><b>41.2</b> Estimated marginal means</a></li>
</ul></li>
<li class="chapter" data-level="42" data-path="part-8-reporting.html"><a href="part-8-reporting.html"><i class="fa fa-check"></i><b>42</b> Part 8: Reporting</a>
<ul>
<li class="chapter" data-level="42.1" data-path="part-8-reporting.html"><a href="part-8-reporting.html#what-to-report"><i class="fa fa-check"></i><b>42.1</b> What to report</a></li>
<li class="chapter" data-level="42.2" data-path="part-8-reporting.html"><a href="part-8-reporting.html#sample-methods-and-results-5"><i class="fa fa-check"></i><b>42.2</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="42.2.1" data-path="part-8-reporting.html"><a href="part-8-reporting.html#methods"><i class="fa fa-check"></i><b>42.2.1</b> Methods</a></li>
<li class="chapter" data-level="42.2.2" data-path="part-8-reporting.html"><a href="part-8-reporting.html#results"><i class="fa fa-check"></i><b>42.2.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="42.3" data-path="part-8-reporting.html"><a href="part-8-reporting.html#common-mistakes-to-avoid"><i class="fa fa-check"></i><b>42.3</b> Common mistakes to avoid</a></li>
<li class="chapter" data-level="42.4" data-path="part-8-reporting.html"><a href="part-8-reporting.html#key-takeaways-7"><i class="fa fa-check"></i><b>42.4</b> Key takeaways</a></li>
<li class="chapter" data-level="42.5" data-path="part-8-reporting.html"><a href="part-8-reporting.html#assignment-10"><i class="fa fa-check"></i><b>42.5</b> Assignment</a>
<ul>
<li class="chapter" data-level="42.5.1" data-path="part-8-reporting.html"><a href="part-8-reporting.html#part-1-conceptual-questions-4"><i class="fa fa-check"></i><b>42.5.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="42.5.2" data-path="part-8-reporting.html"><a href="part-8-reporting.html#part-2-recognizing-zero-inflation"><i class="fa fa-check"></i><b>42.5.2</b> Part 2: Recognizing zero-inflation</a></li>
<li class="chapter" data-level="42.5.3" data-path="part-8-reporting.html"><a href="part-8-reporting.html#part-3-model-fitting"><i class="fa fa-check"></i><b>42.5.3</b> Part 3: Model fitting</a></li>
<li class="chapter" data-level="42.5.4" data-path="part-8-reporting.html"><a href="part-8-reporting.html#part-4-model-comparison"><i class="fa fa-check"></i><b>42.5.4</b> Part 4: Model comparison</a></li>
<li class="chapter" data-level="42.5.5" data-path="part-8-reporting.html"><a href="part-8-reporting.html#part-5-reflection-1"><i class="fa fa-check"></i><b>42.5.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="43" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html"><i class="fa fa-check"></i><b>43</b> Mixed Effects Models</a>
<ul>
<li class="chapter" data-level="43.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#why-mixed-models-matter"><i class="fa fa-check"></i><b>43.1</b> Why mixed models matter</a></li>
<li class="chapter" data-level="43.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#setup-7"><i class="fa fa-check"></i><b>43.2</b> Setup</a></li>
<li class="chapter" data-level="43.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-1-fixed-vs.-random-effects"><i class="fa fa-check"></i><b>43.3</b> Part 1: Fixed vs. Random Effects</a>
<ul>
<li class="chapter" data-level="43.3.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#fixed-effects"><i class="fa fa-check"></i><b>43.3.1</b> Fixed effects</a></li>
<li class="chapter" data-level="43.3.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#random-effects"><i class="fa fa-check"></i><b>43.3.2</b> Random effects</a></li>
<li class="chapter" data-level="43.3.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-gray-area"><i class="fa fa-check"></i><b>43.3.3</b> The gray area</a></li>
</ul></li>
<li class="chapter" data-level="43.4" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-2-design-model-structure"><i class="fa fa-check"></i><b>43.4</b> Part 2: Design → Model Structure</a></li>
<li class="chapter" data-level="43.5" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-3-randomized-complete-block-design"><i class="fa fa-check"></i><b>43.5</b> Part 3: Randomized Complete Block Design</a>
<ul>
<li class="chapter" data-level="43.5.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-design"><i class="fa fa-check"></i><b>43.5.1</b> The design</a></li>
<li class="chapter" data-level="43.5.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-model"><i class="fa fa-check"></i><b>43.5.2</b> The model</a></li>
<li class="chapter" data-level="43.5.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#interpret-the-output-10"><i class="fa fa-check"></i><b>43.5.3</b> Interpret the output</a></li>
<li class="chapter" data-level="43.5.4" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#post-hoc-comparisons-1"><i class="fa fa-check"></i><b>43.5.4</b> Post-hoc comparisons</a></li>
<li class="chapter" data-level="43.5.5" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#what-if-we-ignored-the-blocking"><i class="fa fa-check"></i><b>43.5.5</b> What if we ignored the blocking?</a></li>
<li class="chapter" data-level="43.5.6" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#sample-results-1"><i class="fa fa-check"></i><b>43.5.6</b> Sample results</a></li>
</ul></li>
<li class="chapter" data-level="43.6" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-4-nested-hierarchical-designs"><i class="fa fa-check"></i><b>43.6</b> Part 4: Nested (Hierarchical) Designs</a>
<ul>
<li class="chapter" data-level="43.6.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-design-1"><i class="fa fa-check"></i><b>43.6.1</b> The design</a></li>
<li class="chapter" data-level="43.6.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-model-1"><i class="fa fa-check"></i><b>43.6.2</b> The model</a></li>
<li class="chapter" data-level="43.6.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#interpret-the-variance-components"><i class="fa fa-check"></i><b>43.6.3</b> Interpret the variance components</a></li>
<li class="chapter" data-level="43.6.4" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#calculate-icc-intraclass-correlation"><i class="fa fa-check"></i><b>43.6.4</b> Calculate ICC (Intraclass Correlation)</a></li>
<li class="chapter" data-level="43.6.5" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#test-fixed-effects"><i class="fa fa-check"></i><b>43.6.5</b> Test fixed effects</a></li>
<li class="chapter" data-level="43.6.6" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#important-note-on-degrees-of-freedom"><i class="fa fa-check"></i><b>43.6.6</b> Important note on degrees of freedom</a></li>
<li class="chapter" data-level="43.6.7" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#sample-methods-and-results-6"><i class="fa fa-check"></i><b>43.6.7</b> Sample methods and results</a></li>
</ul></li>
<li class="chapter" data-level="43.7" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-5-crossed-random-effects"><i class="fa fa-check"></i><b>43.7</b> Part 5: Crossed Random Effects</a>
<ul>
<li class="chapter" data-level="43.7.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-design-2"><i class="fa fa-check"></i><b>43.7.1</b> The design</a></li>
<li class="chapter" data-level="43.7.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-model-2"><i class="fa fa-check"></i><b>43.7.2</b> The model</a></li>
<li class="chapter" data-level="43.7.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#interpret-1"><i class="fa fa-check"></i><b>43.7.3</b> Interpret</a></li>
<li class="chapter" data-level="43.7.4" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#when-to-use-crossed-vs.-nested"><i class="fa fa-check"></i><b>43.7.4</b> When to use crossed vs. nested</a></li>
</ul></li>
<li class="chapter" data-level="43.8" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-6-repeated-measures-and-longitudinal-data"><i class="fa fa-check"></i><b>43.8</b> Part 6: Repeated Measures and Longitudinal Data</a>
<ul>
<li class="chapter" data-level="43.8.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-design-3"><i class="fa fa-check"></i><b>43.8.1</b> The design</a></li>
<li class="chapter" data-level="43.8.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#random-intercept-model"><i class="fa fa-check"></i><b>43.8.2</b> Random intercept model</a></li>
<li class="chapter" data-level="43.8.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#random-slope-model"><i class="fa fa-check"></i><b>43.8.3</b> Random slope model</a></li>
<li class="chapter" data-level="43.8.4" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#interpret-random-slope-output"><i class="fa fa-check"></i><b>43.8.4</b> Interpret random slope output</a></li>
<li class="chapter" data-level="43.8.5" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#compare-models"><i class="fa fa-check"></i><b>43.8.5</b> Compare models</a></li>
<li class="chapter" data-level="43.8.6" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#when-to-use-random-slopes"><i class="fa fa-check"></i><b>43.8.6</b> When to use random slopes</a></li>
</ul></li>
<li class="chapter" data-level="43.9" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-7-model-diagnostics"><i class="fa fa-check"></i><b>43.9</b> Part 7: Model Diagnostics</a>
<ul>
<li class="chapter" data-level="43.9.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#check-residuals"><i class="fa fa-check"></i><b>43.9.1</b> Check residuals</a></li>
<li class="chapter" data-level="43.9.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#check-random-effects-normality"><i class="fa fa-check"></i><b>43.9.2</b> Check random effects normality</a></li>
<li class="chapter" data-level="43.9.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#check-for-influential-groups"><i class="fa fa-check"></i><b>43.9.3</b> Check for influential groups</a></li>
</ul></li>
<li class="chapter" data-level="43.10" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-8-reporting-mixed-models"><i class="fa fa-check"></i><b>43.10</b> Part 8: Reporting Mixed Models</a>
<ul>
<li class="chapter" data-level="43.10.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#what-to-report-1"><i class="fa fa-check"></i><b>43.10.1</b> What to report</a></li>
<li class="chapter" data-level="43.10.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#r-squared-for-mixed-models"><i class="fa fa-check"></i><b>43.10.2</b> R-squared for mixed models</a></li>
<li class="chapter" data-level="43.10.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#sample-methods-and-results-7"><i class="fa fa-check"></i><b>43.10.3</b> Sample methods and results</a></li>
</ul></li>
<li class="chapter" data-level="43.11" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-9-common-issues-and-solutions"><i class="fa fa-check"></i><b>43.11</b> Part 9: Common Issues and Solutions</a>
<ul>
<li class="chapter" data-level="43.11.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#singular-fit-convergence-warnings"><i class="fa fa-check"></i><b>43.11.1</b> Singular fit / convergence warnings</a></li>
<li class="chapter" data-level="43.11.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#how-many-levels-do-you-need-for-random-effects"><i class="fa fa-check"></i><b>43.11.2</b> How many levels do you need for random effects?</a></li>
<li class="chapter" data-level="43.11.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#nested-factors-with-same-labels"><i class="fa fa-check"></i><b>43.11.3</b> Nested factors with same labels</a></li>
</ul></li>
<li class="chapter" data-level="43.12" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-8-baci-analysis-a-complete-worked-example"><i class="fa fa-check"></i><b>43.12</b> Part 8: BACI Analysis — A Complete Worked Example</a>
<ul>
<li class="chapter" data-level="43.12.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-scenario-6"><i class="fa fa-check"></i><b>43.12.1</b> The scenario</a></li>
<li class="chapter" data-level="43.12.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#visualize-the-baci-pattern"><i class="fa fa-check"></i><b>43.12.2</b> Visualize the BACI pattern</a></li>
<li class="chapter" data-level="43.12.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-baci-model"><i class="fa fa-check"></i><b>43.12.3</b> The BACI model</a></li>
<li class="chapter" data-level="43.12.4" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#interpret-the-baci-output"><i class="fa fa-check"></i><b>43.12.4</b> Interpret the BACI output</a></li>
<li class="chapter" data-level="43.12.5" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#understanding-the-coefficients"><i class="fa fa-check"></i><b>43.12.5</b> Understanding the coefficients</a></li>
<li class="chapter" data-level="43.12.6" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#estimated-marginal-means-1"><i class="fa fa-check"></i><b>43.12.6</b> Estimated marginal means</a></li>
<li class="chapter" data-level="43.12.7" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#baci-effect-size-and-confidence-interval"><i class="fa fa-check"></i><b>43.12.7</b> BACI effect size and confidence interval</a></li>
<li class="chapter" data-level="43.12.8" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#sample-methods-and-results-for-baci"><i class="fa fa-check"></i><b>43.12.8</b> Sample methods and results for BACI</a></li>
</ul></li>
<li class="chapter" data-level="43.13" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-9-temporal-autocorrelation"><i class="fa fa-check"></i><b>43.13</b> Part 9: Temporal Autocorrelation</a>
<ul>
<li class="chapter" data-level="43.13.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#when-does-it-matter"><i class="fa fa-check"></i><b>43.13.1</b> When does it matter?</a></li>
<li class="chapter" data-level="43.13.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#detecting-temporal-autocorrelation"><i class="fa fa-check"></i><b>43.13.2</b> Detecting temporal autocorrelation</a></li>
<li class="chapter" data-level="43.13.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#what-autocorrelation-does-to-your-analysis"><i class="fa fa-check"></i><b>43.13.3</b> What autocorrelation does to your analysis</a></li>
</ul></li>
<li class="chapter" data-level="43.14" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#solutions-for-temporal-autocorrelation"><i class="fa fa-check"></i><b>43.14</b> Solutions for temporal autocorrelation</a>
<ul>
<li class="chapter" data-level="43.14.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#solution-1-random-slopes-often-sufficient"><i class="fa fa-check"></i><b>43.14.1</b> Solution 1: Random slopes (often sufficient)</a></li>
<li class="chapter" data-level="43.14.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#solution-2-explicit-autocorrelation-structure-nlme"><i class="fa fa-check"></i><b>43.14.2</b> Solution 2: Explicit autocorrelation structure (nlme)</a></li>
<li class="chapter" data-level="43.14.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#comparing-approaches"><i class="fa fa-check"></i><b>43.14.3</b> Comparing approaches</a></li>
</ul></li>
<li class="chapter" data-level="43.15" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#when-to-worry-vs.-when-to-relax"><i class="fa fa-check"></i><b>43.15</b> When to worry vs. when to relax</a></li>
<li class="chapter" data-level="43.16" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#the-bridge-to-time-series"><i class="fa fa-check"></i><b>43.16</b> The bridge to time series</a></li>
<li class="chapter" data-level="43.17" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-10-summary-tables"><i class="fa fa-check"></i><b>43.17</b> Part 10: Summary Tables</a>
<ul>
<li class="chapter" data-level="43.17.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#design-to-model-structure"><i class="fa fa-check"></i><b>43.17.1</b> Design to model structure</a></li>
</ul></li>
<li class="chapter" data-level="43.18" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#syntax-quick-reference"><i class="fa fa-check"></i><b>43.18</b> Syntax quick reference</a></li>
<li class="chapter" data-level="43.19" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#key-takeaways-8"><i class="fa fa-check"></i><b>43.19</b> Key takeaways</a></li>
<li class="chapter" data-level="43.20" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#assignment-11"><i class="fa fa-check"></i><b>43.20</b> Assignment</a>
<ul>
<li class="chapter" data-level="43.20.1" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-1-identify-the-structure"><i class="fa fa-check"></i><b>43.20.1</b> Part 1: Identify the structure</a></li>
<li class="chapter" data-level="43.20.2" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-2-blocked-design-analysis"><i class="fa fa-check"></i><b>43.20.2</b> Part 2: Blocked design analysis</a></li>
<li class="chapter" data-level="43.20.3" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-3-baci-analysis"><i class="fa fa-check"></i><b>43.20.3</b> Part 3: BACI analysis</a></li>
<li class="chapter" data-level="43.20.4" data-path="mixed-effects-models.html"><a href="mixed-effects-models.html#part-4-reflection-3"><i class="fa fa-check"></i><b>43.20.4</b> Part 4: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="44" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html"><i class="fa fa-check"></i><b>44</b> Generalized Additive Models (GAMs)</a>
<ul>
<li class="chapter" data-level="44.1" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#what-is-a-gam"><i class="fa fa-check"></i><b>44.1</b> What is a GAM?</a></li>
<li class="chapter" data-level="44.2" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#when-to-use-gams"><i class="fa fa-check"></i><b>44.2</b> When to use GAMs</a></li>
<li class="chapter" data-level="44.3" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#setup-8"><i class="fa fa-check"></i><b>44.3</b> Setup</a></li>
<li class="chapter" data-level="44.4" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-1-the-basics-of-smoothing"><i class="fa fa-check"></i><b>44.4</b> Part 1: The Basics of Smoothing</a></li>
<li class="chapter" data-level="44.5" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#what-is-a-smoother"><i class="fa fa-check"></i><b>44.5</b> What is a smoother?</a></li>
<li class="chapter" data-level="44.6" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#how-smoothers-work-conceptually"><i class="fa fa-check"></i><b>44.6</b> How smoothers work (conceptually)</a></li>
<li class="chapter" data-level="44.7" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#the-wigglyness-penalty"><i class="fa fa-check"></i><b>44.7</b> The wigglyness penalty</a></li>
<li class="chapter" data-level="44.8" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-2-fitting-gams-with-mgcv"><i class="fa fa-check"></i><b>44.8</b> Part 2: Fitting GAMs with mgcv</a></li>
<li class="chapter" data-level="44.9" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#basic-syntax"><i class="fa fa-check"></i><b>44.9</b> Basic syntax</a></li>
<li class="chapter" data-level="44.10" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#understanding-the-summary"><i class="fa fa-check"></i><b>44.10</b> Understanding the summary</a></li>
<li class="chapter" data-level="44.11" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#what-is-edf-effective-degrees-of-freedom"><i class="fa fa-check"></i><b>44.11</b> What is EDF (Effective Degrees of Freedom)?</a></li>
<li class="chapter" data-level="44.12" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-3-types-of-smooth-terms"><i class="fa fa-check"></i><b>44.12</b> Part 3: Types of Smooth Terms</a></li>
<li class="chapter" data-level="44.13" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#basic-smooth-s"><i class="fa fa-check"></i><b>44.13</b> Basic smooth: s()</a></li>
<li class="chapter" data-level="44.14" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#cyclic-smooths-for-temporal-data"><i class="fa fa-check"></i><b>44.14</b> Cyclic smooths for temporal data</a></li>
<li class="chapter" data-level="44.15" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#tensor-product-smooths-te-and-ti"><i class="fa fa-check"></i><b>44.15</b> Tensor product smooths: te() and ti()</a></li>
<li class="chapter" data-level="44.16" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#factor-smooth-interactions-by"><i class="fa fa-check"></i><b>44.16</b> Factor-smooth interactions: by =</a></li>
<li class="chapter" data-level="44.17" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-4-gams-with-different-distributions"><i class="fa fa-check"></i><b>44.17</b> Part 4: GAMs with Different Distributions</a></li>
<li class="chapter" data-level="44.18" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#common-families"><i class="fa fa-check"></i><b>44.18</b> Common families</a></li>
<li class="chapter" data-level="44.19" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#example-species-distribution-modeling"><i class="fa fa-check"></i><b>44.19</b> Example: Species distribution modeling</a></li>
<li class="chapter" data-level="44.20" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#example-count-data"><i class="fa fa-check"></i><b>44.20</b> Example: Count data</a></li>
<li class="chapter" data-level="44.21" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-5-model-checking-and-diagnostics"><i class="fa fa-check"></i><b>44.21</b> Part 5: Model Checking and Diagnostics</a></li>
<li class="chapter" data-level="44.22" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#basic-diagnostics"><i class="fa fa-check"></i><b>44.22</b> Basic diagnostics</a></li>
<li class="chapter" data-level="44.23" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#checking-basis-dimension-k"><i class="fa fa-check"></i><b>44.23</b> Checking basis dimension (k)</a></li>
<li class="chapter" data-level="44.24" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#dharma-diagnostics"><i class="fa fa-check"></i><b>44.24</b> DHARMa diagnostics</a></li>
<li class="chapter" data-level="44.25" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#concurvity"><i class="fa fa-check"></i><b>44.25</b> Concurvity</a></li>
<li class="chapter" data-level="44.26" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-6-visualization"><i class="fa fa-check"></i><b>44.26</b> Part 6: Visualization</a></li>
<li class="chapter" data-level="44.27" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#using-gratia-for-modern-plots"><i class="fa fa-check"></i><b>44.27</b> Using gratia for modern plots</a></li>
<li class="chapter" data-level="44.28" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#visualizing-interactions"><i class="fa fa-check"></i><b>44.28</b> Visualizing interactions</a></li>
<li class="chapter" data-level="44.29" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#plotting-on-response-scale"><i class="fa fa-check"></i><b>44.29</b> Plotting on response scale</a></li>
<li class="chapter" data-level="44.30" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-7-model-selection"><i class="fa fa-check"></i><b>44.30</b> Part 7: Model Selection</a></li>
<li class="chapter" data-level="44.31" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#comparing-gam-to-glm"><i class="fa fa-check"></i><b>44.31</b> Comparing GAM to GLM</a></li>
<li class="chapter" data-level="44.32" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#comparing-different-smooths"><i class="fa fa-check"></i><b>44.32</b> Comparing different smooths</a></li>
<li class="chapter" data-level="44.33" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#should-i-include-this-smooth"><i class="fa fa-check"></i><b>44.33</b> Should I include this smooth?</a></li>
<li class="chapter" data-level="44.34" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-8-gams-with-random-effects-gamms"><i class="fa fa-check"></i><b>44.34</b> Part 8: GAMs with Random Effects (GAMMs)</a></li>
<li class="chapter" data-level="44.35" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#alternative-gamm-function"><i class="fa fa-check"></i><b>44.35</b> Alternative: gamm() function</a></li>
<li class="chapter" data-level="44.36" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#using-bam-for-large-datasets"><i class="fa fa-check"></i><b>44.36</b> Using bam() for large datasets</a></li>
<li class="chapter" data-level="44.37" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-9-complete-example"><i class="fa fa-check"></i><b>44.37</b> Part 9: Complete Example</a></li>
<li class="chapter" data-level="44.38" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-10-reporting"><i class="fa fa-check"></i><b>44.38</b> Part 10: Reporting</a></li>
<li class="chapter" data-level="44.39" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#what-to-report-2"><i class="fa fa-check"></i><b>44.39</b> What to report</a></li>
<li class="chapter" data-level="44.40" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#sample-methods-and-results-8"><i class="fa fa-check"></i><b>44.40</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="44.40.1" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#methods-1"><i class="fa fa-check"></i><b>44.40.1</b> Methods</a></li>
<li class="chapter" data-level="44.40.2" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#results-1"><i class="fa fa-check"></i><b>44.40.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="44.41" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#key-takeaways-9"><i class="fa fa-check"></i><b>44.41</b> Key takeaways</a></li>
<li class="chapter" data-level="44.42" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#assignment-12"><i class="fa fa-check"></i><b>44.42</b> Assignment</a>
<ul>
<li class="chapter" data-level="44.42.1" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-1-conceptual-questions-5"><i class="fa fa-check"></i><b>44.42.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="44.42.2" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-2-basic-gam-fitting"><i class="fa fa-check"></i><b>44.42.2</b> Part 2: Basic GAM fitting</a></li>
<li class="chapter" data-level="44.42.3" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-3-gam-vs-glm-comparison"><i class="fa fa-check"></i><b>44.42.3</b> Part 3: GAM vs GLM comparison</a></li>
<li class="chapter" data-level="44.42.4" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-4-multiple-smooths"><i class="fa fa-check"></i><b>44.42.4</b> Part 4: Multiple smooths</a></li>
<li class="chapter" data-level="44.42.5" data-path="generalized-additive-models-gams.html"><a href="generalized-additive-models-gams.html#part-5-reflection-2"><i class="fa fa-check"></i><b>44.42.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="45" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html"><i class="fa fa-check"></i><b>45</b> Bootstrapping and Resampling Methods</a>
<ul>
<li class="chapter" data-level="45.1" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#the-core-insight"><i class="fa fa-check"></i><b>45.1</b> The core insight</a></li>
<li class="chapter" data-level="45.2" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#when-to-use-bootstrapping"><i class="fa fa-check"></i><b>45.2</b> When to use bootstrapping</a></li>
<li class="chapter" data-level="45.3" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#setup-9"><i class="fa fa-check"></i><b>45.3</b> Setup</a></li>
<li class="chapter" data-level="45.4" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-1-the-bootstrap-principle"><i class="fa fa-check"></i><b>45.4</b> Part 1: The Bootstrap Principle</a></li>
<li class="chapter" data-level="45.5" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#a-simple-example"><i class="fa fa-check"></i><b>45.5</b> A simple example</a>
<ul>
<li class="chapter" data-level="45.5.1" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#the-standard-approach-parametric"><i class="fa fa-check"></i><b>45.5.1</b> The standard approach (parametric)</a></li>
<li class="chapter" data-level="45.5.2" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#the-bootstrap-approach-non-parametric"><i class="fa fa-check"></i><b>45.5.2</b> The bootstrap approach (non-parametric)</a></li>
</ul></li>
<li class="chapter" data-level="45.6" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#why-with-replacement"><i class="fa fa-check"></i><b>45.6</b> Why “with replacement”?</a></li>
<li class="chapter" data-level="45.7" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-2-bootstrap-confidence-intervals"><i class="fa fa-check"></i><b>45.7</b> Part 2: Bootstrap Confidence Intervals</a></li>
<li class="chapter" data-level="45.8" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#method-1-percentile-interval"><i class="fa fa-check"></i><b>45.8</b> Method 1: Percentile interval</a></li>
<li class="chapter" data-level="45.9" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#method-2-basic-reverse-percentile-interval"><i class="fa fa-check"></i><b>45.9</b> Method 2: Basic (reverse percentile) interval</a></li>
<li class="chapter" data-level="45.10" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#method-3-bca-bias-corrected-and-accelerated"><i class="fa fa-check"></i><b>45.10</b> Method 3: BCa (Bias-Corrected and Accelerated)</a></li>
<li class="chapter" data-level="45.11" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#comparing-ci-methods"><i class="fa fa-check"></i><b>45.11</b> Comparing CI methods</a></li>
<li class="chapter" data-level="45.12" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-3-bootstrap-standard-errors"><i class="fa fa-check"></i><b>45.12</b> Part 3: Bootstrap Standard Errors</a></li>
<li class="chapter" data-level="45.13" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-4-when-to-bootstrap"><i class="fa fa-check"></i><b>45.13</b> Part 4: When to Bootstrap</a></li>
<li class="chapter" data-level="45.14" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#situations-where-bootstrapping-shines"><i class="fa fa-check"></i><b>45.14</b> Situations where bootstrapping shines</a>
<ul>
<li class="chapter" data-level="45.14.1" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#statistics-without-standard-formulas"><i class="fa fa-check"></i><b>45.14.1</b> 1. Statistics without standard formulas</a></li>
<li class="chapter" data-level="45.14.2" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#ratios"><i class="fa fa-check"></i><b>45.14.2</b> 2. Ratios</a></li>
<li class="chapter" data-level="45.14.3" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#non-normal-data"><i class="fa fa-check"></i><b>45.14.3</b> 3. Non-normal data</a></li>
<li class="chapter" data-level="45.14.4" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#small-samples"><i class="fa fa-check"></i><b>45.14.4</b> 4. Small samples</a></li>
</ul></li>
<li class="chapter" data-level="45.15" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#when-not-to-bootstrap"><i class="fa fa-check"></i><b>45.15</b> When NOT to bootstrap</a></li>
<li class="chapter" data-level="45.16" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-5-the-jackknife"><i class="fa fa-check"></i><b>45.16</b> Part 5: The Jackknife</a></li>
<li class="chapter" data-level="45.17" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#how-it-works"><i class="fa fa-check"></i><b>45.17</b> How it works</a></li>
<li class="chapter" data-level="45.18" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#identifying-influential-observations"><i class="fa fa-check"></i><b>45.18</b> Identifying influential observations</a></li>
<li class="chapter" data-level="45.19" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-6-practical-applications"><i class="fa fa-check"></i><b>45.19</b> Part 6: Practical Applications</a></li>
<li class="chapter" data-level="45.20" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#application-1-correlation-coefficients"><i class="fa fa-check"></i><b>45.20</b> Application 1: Correlation coefficients</a></li>
<li class="chapter" data-level="45.21" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#application-2-regression-coefficients"><i class="fa fa-check"></i><b>45.21</b> Application 2: Regression coefficients</a></li>
<li class="chapter" data-level="45.22" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#application-3-difference-between-groups"><i class="fa fa-check"></i><b>45.22</b> Application 3: Difference between groups</a></li>
<li class="chapter" data-level="45.23" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#application-4-custom-ecological-metrics"><i class="fa fa-check"></i><b>45.23</b> Application 4: Custom ecological metrics</a></li>
<li class="chapter" data-level="45.24" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-7-using-the-boot-package"><i class="fa fa-check"></i><b>45.24</b> Part 7: Using the <code>boot</code> Package</a></li>
<li class="chapter" data-level="45.25" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#basic-workflow"><i class="fa fa-check"></i><b>45.25</b> Basic workflow</a></li>
<li class="chapter" data-level="45.26" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#extracting-confidence-intervals"><i class="fa fa-check"></i><b>45.26</b> Extracting confidence intervals</a></li>
<li class="chapter" data-level="45.27" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#visualizing-bootstrap-results"><i class="fa fa-check"></i><b>45.27</b> Visualizing bootstrap results</a></li>
<li class="chapter" data-level="45.28" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#example-correlation-with-boot-package"><i class="fa fa-check"></i><b>45.28</b> Example: Correlation with boot package</a></li>
<li class="chapter" data-level="45.29" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#example-regression-slope-with-boot-package"><i class="fa fa-check"></i><b>45.29</b> Example: Regression slope with boot package</a></li>
<li class="chapter" data-level="45.30" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#example-multiple-statistics-at-once"><i class="fa fa-check"></i><b>45.30</b> Example: Multiple statistics at once</a></li>
<li class="chapter" data-level="45.31" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-8-reporting-bootstrap-results"><i class="fa fa-check"></i><b>45.31</b> Part 8: Reporting Bootstrap Results</a></li>
<li class="chapter" data-level="45.32" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#what-to-report-3"><i class="fa fa-check"></i><b>45.32</b> What to report</a></li>
<li class="chapter" data-level="45.33" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#sample-methods-and-results-9"><i class="fa fa-check"></i><b>45.33</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="45.33.1" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#methods-brief-version"><i class="fa fa-check"></i><b>45.33.1</b> Methods (brief version)</a></li>
<li class="chapter" data-level="45.33.2" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#methods-detailed-version"><i class="fa fa-check"></i><b>45.33.2</b> Methods (detailed version)</a></li>
<li class="chapter" data-level="45.33.3" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#results-2"><i class="fa fa-check"></i><b>45.33.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="45.34" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#key-takeaways-10"><i class="fa fa-check"></i><b>45.34</b> Key takeaways</a></li>
<li class="chapter" data-level="45.35" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#assignment-13"><i class="fa fa-check"></i><b>45.35</b> Assignment</a>
<ul>
<li class="chapter" data-level="45.35.1" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-1-conceptual-questions-6"><i class="fa fa-check"></i><b>45.35.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="45.35.2" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-2-manual-bootstrap"><i class="fa fa-check"></i><b>45.35.2</b> Part 2: Manual bootstrap</a></li>
<li class="chapter" data-level="45.35.3" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-3-using-the-boot-package"><i class="fa fa-check"></i><b>45.35.3</b> Part 3: Using the boot package</a></li>
<li class="chapter" data-level="45.35.4" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-4-applied-problem"><i class="fa fa-check"></i><b>45.35.4</b> Part 4: Applied problem</a></li>
<li class="chapter" data-level="45.35.5" data-path="bootstrapping-and-resampling-methods.html"><a href="bootstrapping-and-resampling-methods.html#part-5-reflection-3"><i class="fa fa-check"></i><b>45.35.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="46" data-path="model-selection-and-multi-model-inference.html"><a href="model-selection-and-multi-model-inference.html"><i class="fa fa-check"></i><b>46</b> Model Selection and Multi-Model Inference</a>
<ul>
<li class="chapter" data-level="46.1" data-path="model-selection-and-multi-model-inference.html"><a href="model-selection-and-multi-model-inference.html#when-to-use-model-selection"><i class="fa fa-check"></i><b>46.1</b> When to use model selection</a></li>
<li class="chapter" data-level="46.2" data-path="model-selection-and-multi-model-inference.html"><a href="model-selection-and-multi-model-inference.html#a-guiding-example-drought-resilience-in-emory-oak-woodlands"><i class="fa fa-check"></i><b>46.2</b> A guiding example: Drought resilience in Emory oak woodlands</a></li>
<li class="chapter" data-level="46.3" data-path="model-selection-and-multi-model-inference.html"><a href="model-selection-and-multi-model-inference.html#setup-10"><i class="fa fa-check"></i><b>46.3</b> Setup</a></li>
</ul></li>
<li class="chapter" data-level="47" data-path="part-1-dealing-with-collinearity.html"><a href="part-1-dealing-with-collinearity.html"><i class="fa fa-check"></i><b>47</b> Part 1: Dealing with Collinearity</a>
<ul>
<li class="chapter" data-level="47.1" data-path="part-1-dealing-with-collinearity.html"><a href="part-1-dealing-with-collinearity.html#why-collinearity-is-a-problem"><i class="fa fa-check"></i><b>47.1</b> Why collinearity is a problem</a></li>
<li class="chapter" data-level="47.2" data-path="part-1-dealing-with-collinearity.html"><a href="part-1-dealing-with-collinearity.html#create-example-data"><i class="fa fa-check"></i><b>47.2</b> Create example data</a></li>
<li class="chapter" data-level="47.3" data-path="part-1-dealing-with-collinearity.html"><a href="part-1-dealing-with-collinearity.html#step-1-visualize-correlations"><i class="fa fa-check"></i><b>47.3</b> Step 1: Visualize correlations</a></li>
<li class="chapter" data-level="47.4" data-path="part-1-dealing-with-collinearity.html"><a href="part-1-dealing-with-collinearity.html#step-2-identify-highly-correlated-pairs"><i class="fa fa-check"></i><b>47.4</b> Step 2: Identify highly correlated pairs</a></li>
<li class="chapter" data-level="47.5" data-path="part-1-dealing-with-collinearity.html"><a href="part-1-dealing-with-collinearity.html#step-3-calculate-variance-inflation-factors-vif"><i class="fa fa-check"></i><b>47.5</b> Step 3: Calculate Variance Inflation Factors (VIF)</a></li>
<li class="chapter" data-level="47.6" data-path="part-1-dealing-with-collinearity.html"><a href="part-1-dealing-with-collinearity.html#step-4-iterative-vif-reduction"><i class="fa fa-check"></i><b>47.6</b> Step 4: Iterative VIF reduction</a></li>
<li class="chapter" data-level="47.7" data-path="part-1-dealing-with-collinearity.html"><a href="part-1-dealing-with-collinearity.html#step-5-verify-final-vifs"><i class="fa fa-check"></i><b>47.7</b> Step 5: Verify final VIFs</a></li>
<li class="chapter" data-level="47.8" data-path="part-1-dealing-with-collinearity.html"><a href="part-1-dealing-with-collinearity.html#blended-approach-expert-statistical"><i class="fa fa-check"></i><b>47.8</b> Blended approach: Expert + Statistical</a></li>
</ul></li>
<li class="chapter" data-level="48" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html"><i class="fa fa-check"></i><b>48</b> Part 2: Information-Theoretic Model Selection (AIC)</a>
<ul>
<li class="chapter" data-level="48.1" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html#the-philosophy"><i class="fa fa-check"></i><b>48.1</b> The philosophy</a></li>
<li class="chapter" data-level="48.2" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html#aicc-for-small-samples"><i class="fa fa-check"></i><b>48.2</b> AICc for small samples</a></li>
<li class="chapter" data-level="48.3" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html#key-concepts"><i class="fa fa-check"></i><b>48.3</b> Key concepts</a>
<ul>
<li class="chapter" data-level="48.3.1" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html#delta-aic-%CE%B4aic"><i class="fa fa-check"></i><b>48.3.1</b> Delta AIC (ΔAIC)</a></li>
<li class="chapter" data-level="48.3.2" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html#akaike-weights-w%E1%B5%A2"><i class="fa fa-check"></i><b>48.3.2</b> Akaike weights (wᵢ)</a></li>
</ul></li>
<li class="chapter" data-level="48.4" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html#model-selection-with-mumin"><i class="fa fa-check"></i><b>48.4</b> Model selection with MuMIn</a>
<ul>
<li class="chapter" data-level="48.4.1" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html#generate-all-subsets"><i class="fa fa-check"></i><b>48.4.1</b> Generate all subsets</a></li>
<li class="chapter" data-level="48.4.2" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html#interpret-the-output-11"><i class="fa fa-check"></i><b>48.4.2</b> Interpret the output</a></li>
<li class="chapter" data-level="48.4.3" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html#identify-important-variables"><i class="fa fa-check"></i><b>48.4.3</b> Identify important variables</a></li>
<li class="chapter" data-level="48.4.4" data-path="part-2-information-theoretic-model-selection-aic.html"><a href="part-2-information-theoretic-model-selection-aic.html#visualize-model-selection"><i class="fa fa-check"></i><b>48.4.4</b> Visualize model selection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="49" data-path="part-3-model-averaging.html"><a href="part-3-model-averaging.html"><i class="fa fa-check"></i><b>49</b> Part 3: Model Averaging</a>
<ul>
<li class="chapter" data-level="49.1" data-path="part-3-model-averaging.html"><a href="part-3-model-averaging.html#when-to-average"><i class="fa fa-check"></i><b>49.1</b> When to average</a></li>
<li class="chapter" data-level="49.2" data-path="part-3-model-averaging.html"><a href="part-3-model-averaging.html#full-vs.-conditional-averaging"><i class="fa fa-check"></i><b>49.2</b> Full vs. conditional averaging</a></li>
<li class="chapter" data-level="49.3" data-path="part-3-model-averaging.html"><a href="part-3-model-averaging.html#perform-model-averaging"><i class="fa fa-check"></i><b>49.3</b> Perform model averaging</a>
<ul>
<li class="chapter" data-level="49.3.1" data-path="part-3-model-averaging.html"><a href="part-3-model-averaging.html#extract-averaged-coefficients"><i class="fa fa-check"></i><b>49.3.1</b> Extract averaged coefficients</a></li>
<li class="chapter" data-level="49.3.2" data-path="part-3-model-averaging.html"><a href="part-3-model-averaging.html#variable-importance-from-averaged-model"><i class="fa fa-check"></i><b>49.3.2</b> Variable importance from averaged model</a></li>
</ul></li>
<li class="chapter" data-level="49.4" data-path="part-3-model-averaging.html"><a href="part-3-model-averaging.html#predictions-from-averaged-model"><i class="fa fa-check"></i><b>49.4</b> Predictions from averaged model</a></li>
</ul></li>
<li class="chapter" data-level="50" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html"><i class="fa fa-check"></i><b>50</b> Part 4: Comparing Model Selection Approaches</a>
<ul>
<li class="chapter" data-level="50.1" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#overview-of-alternatives"><i class="fa fa-check"></i><b>50.1</b> Overview of alternatives</a></li>
<li class="chapter" data-level="50.2" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#when-to-use-each-1"><i class="fa fa-check"></i><b>50.2</b> When to use each</a></li>
<li class="chapter" data-level="50.3" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#brief-example-lasso"><i class="fa fa-check"></i><b>50.3</b> Brief example: LASSO</a></li>
<li class="chapter" data-level="50.4" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#part-5-reporting-model-selection-results"><i class="fa fa-check"></i><b>50.4</b> Part 5: Reporting Model Selection Results</a></li>
<li class="chapter" data-level="50.5" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#what-to-include-1"><i class="fa fa-check"></i><b>50.5</b> What to include</a></li>
<li class="chapter" data-level="50.6" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#creating-a-publication-ready-model-selection-table"><i class="fa fa-check"></i><b>50.6</b> Creating a publication-ready model selection table</a></li>
<li class="chapter" data-level="50.7" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#sample-methods-and-results-10"><i class="fa fa-check"></i><b>50.7</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="50.7.1" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#methods-2"><i class="fa fa-check"></i><b>50.7.1</b> Methods</a></li>
<li class="chapter" data-level="50.7.2" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#results-3"><i class="fa fa-check"></i><b>50.7.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="50.8" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#key-takeaways-11"><i class="fa fa-check"></i><b>50.8</b> Key takeaways</a></li>
<li class="chapter" data-level="50.9" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#assignment-14"><i class="fa fa-check"></i><b>50.9</b> Assignment</a>
<ul>
<li class="chapter" data-level="50.9.1" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#part-1-conceptual-questions-7"><i class="fa fa-check"></i><b>50.9.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="50.9.2" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#part-2-collinearity-reduction"><i class="fa fa-check"></i><b>50.9.2</b> Part 2: Collinearity reduction</a></li>
<li class="chapter" data-level="50.9.3" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#part-3-model-selection"><i class="fa fa-check"></i><b>50.9.3</b> Part 3: Model selection</a></li>
<li class="chapter" data-level="50.9.4" data-path="part-4-comparing-model-selection-approaches.html"><a href="part-4-comparing-model-selection-approaches.html#part-4-reflection-4"><i class="fa fa-check"></i><b>50.9.4</b> Part 4: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="51" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html"><i class="fa fa-check"></i><b>51</b> Time Series Analysis and Forecasting</a>
<ul>
<li class="chapter" data-level="51.1" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#what-is-a-time-series"><i class="fa fa-check"></i><b>51.1</b> What is a time series?</a></li>
<li class="chapter" data-level="51.2" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#why-time-series-analysis-matters-in-ecology"><i class="fa fa-check"></i><b>51.2</b> Why time series analysis matters in ecology</a></li>
<li class="chapter" data-level="51.3" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#setup-11"><i class="fa fa-check"></i><b>51.3</b> Setup</a></li>
<li class="chapter" data-level="51.4" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-1-time-series-structure"><i class="fa fa-check"></i><b>51.4</b> Part 1: Time Series Structure</a></li>
<li class="chapter" data-level="51.5" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#components-of-a-time-series"><i class="fa fa-check"></i><b>51.5</b> Components of a time series</a></li>
<li class="chapter" data-level="51.6" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#example-simulating-a-time-series"><i class="fa fa-check"></i><b>51.6</b> Example: Simulating a time series</a></li>
<li class="chapter" data-level="51.7" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#creating-time-series-objects-in-r"><i class="fa fa-check"></i><b>51.7</b> Creating time series objects in R</a></li>
<li class="chapter" data-level="51.8" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-2-exploratory-data-analysis"><i class="fa fa-check"></i><b>51.8</b> Part 2: Exploratory Data Analysis</a></li>
<li class="chapter" data-level="51.9" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#visualizing-time-series"><i class="fa fa-check"></i><b>51.9</b> Visualizing time series</a></li>
<li class="chapter" data-level="51.10" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#autocorrelation"><i class="fa fa-check"></i><b>51.10</b> Autocorrelation</a></li>
<li class="chapter" data-level="51.11" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#partial-autocorrelation-pacf"><i class="fa fa-check"></i><b>51.11</b> Partial autocorrelation (PACF)</a></li>
<li class="chapter" data-level="51.12" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-3-stationarity"><i class="fa fa-check"></i><b>51.12</b> Part 3: Stationarity</a></li>
<li class="chapter" data-level="51.13" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#what-is-stationarity"><i class="fa fa-check"></i><b>51.13</b> What is stationarity?</a></li>
<li class="chapter" data-level="51.14" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#testing-for-stationarity"><i class="fa fa-check"></i><b>51.14</b> Testing for stationarity</a></li>
<li class="chapter" data-level="51.15" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#making-a-series-stationary"><i class="fa fa-check"></i><b>51.15</b> Making a series stationary</a>
<ul>
<li class="chapter" data-level="51.15.1" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#differencing"><i class="fa fa-check"></i><b>51.15.1</b> Differencing</a></li>
<li class="chapter" data-level="51.15.2" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#seasonal-differencing"><i class="fa fa-check"></i><b>51.15.2</b> Seasonal differencing</a></li>
<li class="chapter" data-level="51.15.3" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#log-transformation"><i class="fa fa-check"></i><b>51.15.3</b> Log transformation</a></li>
</ul></li>
<li class="chapter" data-level="51.16" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-4-decomposition"><i class="fa fa-check"></i><b>51.16</b> Part 4: Decomposition</a></li>
<li class="chapter" data-level="51.17" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#classical-decomposition"><i class="fa fa-check"></i><b>51.17</b> Classical decomposition</a></li>
<li class="chapter" data-level="51.18" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#stl-decomposition"><i class="fa fa-check"></i><b>51.18</b> STL decomposition</a></li>
<li class="chapter" data-level="51.19" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#extracting-components"><i class="fa fa-check"></i><b>51.19</b> Extracting components</a></li>
<li class="chapter" data-level="51.20" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-5-arima-models"><i class="fa fa-check"></i><b>51.20</b> Part 5: ARIMA Models</a></li>
<li class="chapter" data-level="51.21" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#the-arima-framework"><i class="fa fa-check"></i><b>51.21</b> The ARIMA framework</a></li>
<li class="chapter" data-level="51.22" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#understanding-ar-and-ma"><i class="fa fa-check"></i><b>51.22</b> Understanding AR and MA</a></li>
<li class="chapter" data-level="51.23" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#seasonal-arima-sarima"><i class="fa fa-check"></i><b>51.23</b> Seasonal ARIMA (SARIMA)</a></li>
<li class="chapter" data-level="51.24" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#fitting-arima-with-auto.arima"><i class="fa fa-check"></i><b>51.24</b> Fitting ARIMA with auto.arima()</a>
<ul>
<li class="chapter" data-level="51.24.1" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#interpreting-the-output"><i class="fa fa-check"></i><b>51.24.1</b> Interpreting the output</a></li>
</ul></li>
<li class="chapter" data-level="51.25" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#checking-residuals"><i class="fa fa-check"></i><b>51.25</b> Checking residuals</a></li>
<li class="chapter" data-level="51.26" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-6-forecasting"><i class="fa fa-check"></i><b>51.26</b> Part 6: Forecasting</a></li>
<li class="chapter" data-level="51.27" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#generating-forecasts"><i class="fa fa-check"></i><b>51.27</b> Generating forecasts</a></li>
<li class="chapter" data-level="51.28" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#understanding-prediction-intervals"><i class="fa fa-check"></i><b>51.28</b> Understanding prediction intervals</a></li>
<li class="chapter" data-level="51.29" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#evaluating-forecast-accuracy"><i class="fa fa-check"></i><b>51.29</b> Evaluating forecast accuracy</a>
<ul>
<li class="chapter" data-level="51.29.1" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#traintest-split"><i class="fa fa-check"></i><b>51.29.1</b> Train/test split</a></li>
<li class="chapter" data-level="51.29.2" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#error-metrics"><i class="fa fa-check"></i><b>51.29.2</b> Error metrics</a></li>
</ul></li>
<li class="chapter" data-level="51.30" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-7-ecological-case-studies"><i class="fa fa-check"></i><b>51.30</b> Part 7: Ecological Case Studies</a></li>
<li class="chapter" data-level="51.31" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#case-study-1-population-trends"><i class="fa fa-check"></i><b>51.31</b> Case Study 1: Population trends</a></li>
<li class="chapter" data-level="51.32" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#case-study-2-seasonal-climate-data"><i class="fa fa-check"></i><b>51.32</b> Case Study 2: Seasonal climate data</a></li>
<li class="chapter" data-level="51.33" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#case-study-3-phenological-timing"><i class="fa fa-check"></i><b>51.33</b> Case Study 3: Phenological timing</a></li>
<li class="chapter" data-level="51.34" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-8-advanced-topics-brief-introduction"><i class="fa fa-check"></i><b>51.34</b> Part 8: Advanced Topics (Brief Introduction)</a></li>
<li class="chapter" data-level="51.35" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#state-space-models"><i class="fa fa-check"></i><b>51.35</b> State-space models</a></li>
<li class="chapter" data-level="51.36" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#exponential-smoothing-ets"><i class="fa fa-check"></i><b>51.36</b> Exponential smoothing (ETS)</a></li>
<li class="chapter" data-level="51.37" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#dynamic-regression"><i class="fa fa-check"></i><b>51.37</b> Dynamic regression</a></li>
<li class="chapter" data-level="51.38" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-9-practical-tips"><i class="fa fa-check"></i><b>51.38</b> Part 9: Practical Tips</a></li>
<li class="chapter" data-level="51.39" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#common-pitfalls"><i class="fa fa-check"></i><b>51.39</b> Common pitfalls</a></li>
<li class="chapter" data-level="51.40" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#when-to-use-time-series-vs-regression"><i class="fa fa-check"></i><b>51.40</b> When to use time series vs regression</a></li>
<li class="chapter" data-level="51.41" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#dealing-with-missing-data"><i class="fa fa-check"></i><b>51.41</b> Dealing with missing data</a></li>
<li class="chapter" data-level="51.42" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-10-reporting-1"><i class="fa fa-check"></i><b>51.42</b> Part 10: Reporting</a></li>
<li class="chapter" data-level="51.43" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#what-to-report-4"><i class="fa fa-check"></i><b>51.43</b> What to report</a></li>
<li class="chapter" data-level="51.44" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#sample-methods-and-results-11"><i class="fa fa-check"></i><b>51.44</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="51.44.1" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#methods-3"><i class="fa fa-check"></i><b>51.44.1</b> Methods</a></li>
<li class="chapter" data-level="51.44.2" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#results-4"><i class="fa fa-check"></i><b>51.44.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="51.45" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#key-takeaways-12"><i class="fa fa-check"></i><b>51.45</b> Key takeaways</a></li>
<li class="chapter" data-level="51.46" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#assignment-15"><i class="fa fa-check"></i><b>51.46</b> Assignment</a>
<ul>
<li class="chapter" data-level="51.46.1" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-1-conceptual-questions-8"><i class="fa fa-check"></i><b>51.46.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="51.46.2" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-2-exploratory-analysis"><i class="fa fa-check"></i><b>51.46.2</b> Part 2: Exploratory analysis</a></li>
<li class="chapter" data-level="51.46.3" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-3-arima-modeling"><i class="fa fa-check"></i><b>51.46.3</b> Part 3: ARIMA modeling</a></li>
<li class="chapter" data-level="51.46.4" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-4-forecasting"><i class="fa fa-check"></i><b>51.46.4</b> Part 4: Forecasting</a></li>
<li class="chapter" data-level="51.46.5" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-5-ecological-application"><i class="fa fa-check"></i><b>51.46.5</b> Part 5: Ecological application</a></li>
<li class="chapter" data-level="51.46.6" data-path="time-series-analysis-and-forecasting.html"><a href="time-series-analysis-and-forecasting.html#part-6-reflection"><i class="fa fa-check"></i><b>51.46.6</b> Part 6: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="52" data-path="ordination-methods.html"><a href="ordination-methods.html"><i class="fa fa-check"></i><b>52</b> Ordination Methods</a>
<ul>
<li class="chapter" data-level="52.1" data-path="ordination-methods.html"><a href="ordination-methods.html#the-core-idea-1"><i class="fa fa-check"></i><b>52.1</b> The core idea</a></li>
<li class="chapter" data-level="52.2" data-path="ordination-methods.html"><a href="ordination-methods.html#setup-12"><i class="fa fa-check"></i><b>52.2</b> Setup</a></li>
<li class="chapter" data-level="52.3" data-path="ordination-methods.html"><a href="ordination-methods.html#part-1-visualizing-ordination-logic"><i class="fa fa-check"></i><b>52.3</b> Part 1: Visualizing Ordination Logic</a></li>
<li class="chapter" data-level="52.4" data-path="ordination-methods.html"><a href="ordination-methods.html#one-species"><i class="fa fa-check"></i><b>52.4</b> One species</a></li>
<li class="chapter" data-level="52.5" data-path="ordination-methods.html"><a href="ordination-methods.html#two-species"><i class="fa fa-check"></i><b>52.5</b> Two species</a></li>
<li class="chapter" data-level="52.6" data-path="ordination-methods.html"><a href="ordination-methods.html#three-species-and-beyond"><i class="fa fa-check"></i><b>52.6</b> Three species… and beyond</a></li>
<li class="chapter" data-level="52.7" data-path="ordination-methods.html"><a href="ordination-methods.html#part-2-nmds-non-metric-multidimensional-scaling"><i class="fa fa-check"></i><b>52.7</b> Part 2: NMDS (Non-metric Multidimensional Scaling)</a></li>
<li class="chapter" data-level="52.8" data-path="ordination-methods.html"><a href="ordination-methods.html#what-makes-nmds-different"><i class="fa fa-check"></i><b>52.8</b> What makes NMDS different?</a></li>
<li class="chapter" data-level="52.9" data-path="ordination-methods.html"><a href="ordination-methods.html#how-nmds-works"><i class="fa fa-check"></i><b>52.9</b> How NMDS works</a></li>
<li class="chapter" data-level="52.10" data-path="ordination-methods.html"><a href="ordination-methods.html#nmds-in-practice"><i class="fa fa-check"></i><b>52.10</b> NMDS in practice</a>
<ul>
<li class="chapter" data-level="52.10.1" data-path="ordination-methods.html"><a href="ordination-methods.html#data-transformation"><i class="fa fa-check"></i><b>52.10.1</b> Data transformation</a></li>
<li class="chapter" data-level="52.10.2" data-path="ordination-methods.html"><a href="ordination-methods.html#run-nmds"><i class="fa fa-check"></i><b>52.10.2</b> Run NMDS</a></li>
<li class="chapter" data-level="52.10.3" data-path="ordination-methods.html"><a href="ordination-methods.html#interpret-stress"><i class="fa fa-check"></i><b>52.10.3</b> Interpret stress</a></li>
<li class="chapter" data-level="52.10.4" data-path="ordination-methods.html"><a href="ordination-methods.html#basic-nmds-plot"><i class="fa fa-check"></i><b>52.10.4</b> Basic NMDS plot</a></li>
<li class="chapter" data-level="52.10.5" data-path="ordination-methods.html"><a href="ordination-methods.html#better-visualization-with-ggplot"><i class="fa fa-check"></i><b>52.10.5</b> Better visualization with ggplot</a></li>
</ul></li>
<li class="chapter" data-level="52.11" data-path="ordination-methods.html"><a href="ordination-methods.html#interpreting-nmds-axes"><i class="fa fa-check"></i><b>52.11</b> Interpreting NMDS axes</a></li>
<li class="chapter" data-level="52.12" data-path="ordination-methods.html"><a href="ordination-methods.html#part-3-other-ordination-methods"><i class="fa fa-check"></i><b>52.12</b> Part 3: Other Ordination Methods</a></li>
<li class="chapter" data-level="52.13" data-path="ordination-methods.html"><a href="ordination-methods.html#pca-principal-components-analysis"><i class="fa fa-check"></i><b>52.13</b> PCA (Principal Components Analysis)</a></li>
<li class="chapter" data-level="52.14" data-path="ordination-methods.html"><a href="ordination-methods.html#ca-and-dca"><i class="fa fa-check"></i><b>52.14</b> CA and DCA</a></li>
<li class="chapter" data-level="52.15" data-path="ordination-methods.html"><a href="ordination-methods.html#part-4-practical-considerations"><i class="fa fa-check"></i><b>52.15</b> Part 4: Practical Considerations</a></li>
<li class="chapter" data-level="52.16" data-path="ordination-methods.html"><a href="ordination-methods.html#choosing-the-right-dissimilarity"><i class="fa fa-check"></i><b>52.16</b> Choosing the right dissimilarity</a></li>
<li class="chapter" data-level="52.17" data-path="ordination-methods.html"><a href="ordination-methods.html#handling-stress"><i class="fa fa-check"></i><b>52.17</b> Handling stress</a></li>
<li class="chapter" data-level="52.18" data-path="ordination-methods.html"><a href="ordination-methods.html#visualizing-groups"><i class="fa fa-check"></i><b>52.18</b> Visualizing groups</a></li>
<li class="chapter" data-level="52.19" data-path="ordination-methods.html"><a href="ordination-methods.html#creating-publication-quality-figures"><i class="fa fa-check"></i><b>52.19</b> Creating publication-quality figures</a></li>
<li class="chapter" data-level="52.20" data-path="ordination-methods.html"><a href="ordination-methods.html#part-5-reporting-ordination-results"><i class="fa fa-check"></i><b>52.20</b> Part 5: Reporting Ordination Results</a></li>
<li class="chapter" data-level="52.21" data-path="ordination-methods.html"><a href="ordination-methods.html#what-to-report-5"><i class="fa fa-check"></i><b>52.21</b> What to report</a></li>
<li class="chapter" data-level="52.22" data-path="ordination-methods.html"><a href="ordination-methods.html#sample-methods-and-results-12"><i class="fa fa-check"></i><b>52.22</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="52.22.1" data-path="ordination-methods.html"><a href="ordination-methods.html#methods-4"><i class="fa fa-check"></i><b>52.22.1</b> Methods</a></li>
<li class="chapter" data-level="52.22.2" data-path="ordination-methods.html"><a href="ordination-methods.html#results-5"><i class="fa fa-check"></i><b>52.22.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="52.23" data-path="ordination-methods.html"><a href="ordination-methods.html#key-takeaways-13"><i class="fa fa-check"></i><b>52.23</b> Key takeaways</a></li>
<li class="chapter" data-level="52.24" data-path="ordination-methods.html"><a href="ordination-methods.html#assignment-16"><i class="fa fa-check"></i><b>52.24</b> Assignment</a>
<ul>
<li class="chapter" data-level="52.24.1" data-path="ordination-methods.html"><a href="ordination-methods.html#part-1-conceptual-questions-9"><i class="fa fa-check"></i><b>52.24.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="52.24.2" data-path="ordination-methods.html"><a href="ordination-methods.html#part-2-nmds-analysis"><i class="fa fa-check"></i><b>52.24.2</b> Part 2: NMDS analysis</a></li>
<li class="chapter" data-level="52.24.3" data-path="ordination-methods.html"><a href="ordination-methods.html#part-3-environmental-interpretation"><i class="fa fa-check"></i><b>52.24.3</b> Part 3: Environmental interpretation</a></li>
<li class="chapter" data-level="52.24.4" data-path="ordination-methods.html"><a href="ordination-methods.html#part-4-reflection-5"><i class="fa fa-check"></i><b>52.24.4</b> Part 4: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="53" data-path="permutation-based-methods-for-community-data.html"><a href="permutation-based-methods-for-community-data.html"><i class="fa fa-check"></i><b>53</b> Permutation-Based Methods for Community Data</a>
<ul>
<li class="chapter" data-level="53.1" data-path="permutation-based-methods-for-community-data.html"><a href="permutation-based-methods-for-community-data.html#the-permutation-logic"><i class="fa fa-check"></i><b>53.1</b> The permutation logic</a></li>
<li class="chapter" data-level="53.2" data-path="permutation-based-methods-for-community-data.html"><a href="permutation-based-methods-for-community-data.html#setup-13"><i class="fa fa-check"></i><b>53.2</b> Setup</a></li>
</ul></li>
<li class="chapter" data-level="54" data-path="part-1-permanova.html"><a href="part-1-permanova.html"><i class="fa fa-check"></i><b>54</b> Part 1: PERMANOVA</a>
<ul>
<li class="chapter" data-level="54.1" data-path="part-1-permanova.html"><a href="part-1-permanova.html#what-is-permanova"><i class="fa fa-check"></i><b>54.1</b> What is PERMANOVA?</a></li>
<li class="chapter" data-level="54.2" data-path="part-1-permanova.html"><a href="part-1-permanova.html#how-permanova-works"><i class="fa fa-check"></i><b>54.2</b> How PERMANOVA works</a></li>
<li class="chapter" data-level="54.3" data-path="part-1-permanova.html"><a href="part-1-permanova.html#permanova-in-practice"><i class="fa fa-check"></i><b>54.3</b> PERMANOVA in practice</a>
<ul>
<li class="chapter" data-level="54.3.1" data-path="part-1-permanova.html"><a href="part-1-permanova.html#interpret-the-output-12"><i class="fa fa-check"></i><b>54.3.1</b> Interpret the output</a></li>
</ul></li>
<li class="chapter" data-level="54.4" data-path="part-1-permanova.html"><a href="part-1-permanova.html#using-formula-interface"><i class="fa fa-check"></i><b>54.4</b> Using formula interface</a></li>
<li class="chapter" data-level="54.5" data-path="part-1-permanova.html"><a href="part-1-permanova.html#testing-interactions"><i class="fa fa-check"></i><b>54.5</b> Testing interactions</a></li>
<li class="chapter" data-level="54.6" data-path="part-1-permanova.html"><a href="part-1-permanova.html#permanova-assumptions"><i class="fa fa-check"></i><b>54.6</b> PERMANOVA assumptions</a>
<ul>
<li class="chapter" data-level="54.6.1" data-path="part-1-permanova.html"><a href="part-1-permanova.html#checking-dispersion-homogeneity-of-variances"><i class="fa fa-check"></i><b>54.6.1</b> Checking dispersion (homogeneity of variances)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="55" data-path="part-2-pairwise-comparisons.html"><a href="part-2-pairwise-comparisons.html"><i class="fa fa-check"></i><b>55</b> Part 2: Pairwise Comparisons</a></li>
<li class="chapter" data-level="56" data-path="part-3-anosim.html"><a href="part-3-anosim.html"><i class="fa fa-check"></i><b>56</b> Part 3: ANOSIM</a>
<ul>
<li class="chapter" data-level="56.1" data-path="part-3-anosim.html"><a href="part-3-anosim.html#what-is-anosim"><i class="fa fa-check"></i><b>56.1</b> What is ANOSIM?</a>
<ul>
<li class="chapter" data-level="56.1.1" data-path="part-3-anosim.html"><a href="part-3-anosim.html#interpret-anosim-output"><i class="fa fa-check"></i><b>56.1.1</b> Interpret ANOSIM output</a></li>
</ul></li>
<li class="chapter" data-level="56.2" data-path="part-3-anosim.html"><a href="part-3-anosim.html#anosim-vs-permanova"><i class="fa fa-check"></i><b>56.2</b> ANOSIM vs PERMANOVA</a></li>
</ul></li>
<li class="chapter" data-level="57" data-path="part-4-mantel-test.html"><a href="part-4-mantel-test.html"><i class="fa fa-check"></i><b>57</b> Part 4: Mantel Test</a>
<ul>
<li class="chapter" data-level="57.1" data-path="part-4-mantel-test.html"><a href="part-4-mantel-test.html#what-is-the-mantel-test"><i class="fa fa-check"></i><b>57.1</b> What is the Mantel test?</a></li>
<li class="chapter" data-level="57.2" data-path="part-4-mantel-test.html"><a href="part-4-mantel-test.html#mantel-test-in-practice"><i class="fa fa-check"></i><b>57.2</b> Mantel test in practice</a>
<ul>
<li class="chapter" data-level="57.2.1" data-path="part-4-mantel-test.html"><a href="part-4-mantel-test.html#interpret-mantel-output"><i class="fa fa-check"></i><b>57.2.1</b> Interpret Mantel output</a></li>
</ul></li>
<li class="chapter" data-level="57.3" data-path="part-4-mantel-test.html"><a href="part-4-mantel-test.html#partial-mantel-test"><i class="fa fa-check"></i><b>57.3</b> Partial Mantel test</a></li>
</ul></li>
<li class="chapter" data-level="58" data-path="part-5-fitting-environmental-variables-to-ordination.html"><a href="part-5-fitting-environmental-variables-to-ordination.html"><i class="fa fa-check"></i><b>58</b> Part 5: Fitting Environmental Variables to Ordination</a>
<ul>
<li class="chapter" data-level="58.1" data-path="part-5-fitting-environmental-variables-to-ordination.html"><a href="part-5-fitting-environmental-variables-to-ordination.html#the-envfit-function"><i class="fa fa-check"></i><b>58.1</b> The envfit function</a>
<ul>
<li class="chapter" data-level="58.1.1" data-path="part-5-fitting-environmental-variables-to-ordination.html"><a href="part-5-fitting-environmental-variables-to-ordination.html#interpret-envfit"><i class="fa fa-check"></i><b>58.1.1</b> Interpret envfit</a></li>
</ul></li>
<li class="chapter" data-level="58.2" data-path="part-5-fitting-environmental-variables-to-ordination.html"><a href="part-5-fitting-environmental-variables-to-ordination.html#surface-fitting"><i class="fa fa-check"></i><b>58.2</b> Surface fitting</a></li>
</ul></li>
<li class="chapter" data-level="59" data-path="part-6-complete-workflow-example.html"><a href="part-6-complete-workflow-example.html"><i class="fa fa-check"></i><b>59</b> Part 6: Complete Workflow Example</a></li>
<li class="chapter" data-level="60" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html"><i class="fa fa-check"></i><b>60</b> Part 7: Reporting Permutation Results</a>
<ul>
<li class="chapter" data-level="60.1" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#what-to-report-6"><i class="fa fa-check"></i><b>60.1</b> What to report</a></li>
<li class="chapter" data-level="60.2" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#sample-methods-and-results-13"><i class="fa fa-check"></i><b>60.2</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="60.2.1" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#methods-5"><i class="fa fa-check"></i><b>60.2.1</b> Methods</a></li>
<li class="chapter" data-level="60.2.2" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#results-6"><i class="fa fa-check"></i><b>60.2.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="60.3" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#key-takeaways-14"><i class="fa fa-check"></i><b>60.3</b> Key takeaways</a></li>
<li class="chapter" data-level="60.4" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#assignment-17"><i class="fa fa-check"></i><b>60.4</b> Assignment</a>
<ul>
<li class="chapter" data-level="60.4.1" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#part-1-conceptual-questions-10"><i class="fa fa-check"></i><b>60.4.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="60.4.2" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#part-2-permanova-analysis"><i class="fa fa-check"></i><b>60.4.2</b> Part 2: PERMANOVA analysis</a></li>
<li class="chapter" data-level="60.4.3" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#part-3-mantel-test"><i class="fa fa-check"></i><b>60.4.3</b> Part 3: Mantel test</a></li>
<li class="chapter" data-level="60.4.4" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#part-4-complete-workflow"><i class="fa fa-check"></i><b>60.4.4</b> Part 4: Complete workflow</a></li>
<li class="chapter" data-level="60.4.5" data-path="part-7-reporting-permutation-results.html"><a href="part-7-reporting-permutation-results.html#part-5-reflection-4"><i class="fa fa-check"></i><b>60.4.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="61" data-path="diversity-statistics.html"><a href="diversity-statistics.html"><i class="fa fa-check"></i><b>61</b> Diversity Statistics</a>
<ul>
<li class="chapter" data-level="61.1" data-path="diversity-statistics.html"><a href="diversity-statistics.html#the-challenge-of-measuring-diversity"><i class="fa fa-check"></i><b>61.1</b> The challenge of measuring diversity</a></li>
<li class="chapter" data-level="61.2" data-path="diversity-statistics.html"><a href="diversity-statistics.html#setup-14"><i class="fa fa-check"></i><b>61.2</b> Setup</a></li>
<li class="chapter" data-level="61.3" data-path="diversity-statistics.html"><a href="diversity-statistics.html#part-1-types-of-diversity-data"><i class="fa fa-check"></i><b>61.3</b> Part 1: Types of Diversity Data</a></li>
<li class="chapter" data-level="61.4" data-path="diversity-statistics.html"><a href="diversity-statistics.html#abundance-data"><i class="fa fa-check"></i><b>61.4</b> Abundance data</a></li>
<li class="chapter" data-level="61.5" data-path="diversity-statistics.html"><a href="diversity-statistics.html#incidence-presenceabsence-data"><i class="fa fa-check"></i><b>61.5</b> Incidence (presence/absence) data</a></li>
<li class="chapter" data-level="61.6" data-path="diversity-statistics.html"><a href="diversity-statistics.html#incidence-frequency-data"><i class="fa fa-check"></i><b>61.6</b> Incidence frequency data</a></li>
<li class="chapter" data-level="61.7" data-path="diversity-statistics.html"><a href="diversity-statistics.html#choosing-between-data-types"><i class="fa fa-check"></i><b>61.7</b> Choosing between data types</a></li>
<li class="chapter" data-level="61.8" data-path="diversity-statistics.html"><a href="diversity-statistics.html#part-2-diversity-metrics-and-hill-numbers"><i class="fa fa-check"></i><b>61.8</b> Part 2: Diversity Metrics and Hill Numbers</a></li>
<li class="chapter" data-level="61.9" data-path="diversity-statistics.html"><a href="diversity-statistics.html#classic-diversity-indices"><i class="fa fa-check"></i><b>61.9</b> Classic diversity indices</a>
<ul>
<li class="chapter" data-level="61.9.1" data-path="diversity-statistics.html"><a href="diversity-statistics.html#species-richness-s"><i class="fa fa-check"></i><b>61.9.1</b> Species richness (S)</a></li>
<li class="chapter" data-level="61.9.2" data-path="diversity-statistics.html"><a href="diversity-statistics.html#shannon-diversity-index-h"><i class="fa fa-check"></i><b>61.9.2</b> Shannon diversity index (H’)</a></li>
<li class="chapter" data-level="61.9.3" data-path="diversity-statistics.html"><a href="diversity-statistics.html#simpson-diversity-index-d"><i class="fa fa-check"></i><b>61.9.3</b> Simpson diversity index (D)</a></li>
</ul></li>
<li class="chapter" data-level="61.10" data-path="diversity-statistics.html"><a href="diversity-statistics.html#the-problem-with-indices"><i class="fa fa-check"></i><b>61.10</b> The problem with indices</a></li>
<li class="chapter" data-level="61.11" data-path="diversity-statistics.html"><a href="diversity-statistics.html#hill-numbers-the-solution"><i class="fa fa-check"></i><b>61.11</b> Hill numbers: The solution</a></li>
<li class="chapter" data-level="61.12" data-path="diversity-statistics.html"><a href="diversity-statistics.html#interpreting-hill-numbers"><i class="fa fa-check"></i><b>61.12</b> Interpreting Hill numbers</a></li>
<li class="chapter" data-level="61.13" data-path="diversity-statistics.html"><a href="diversity-statistics.html#part-3-the-sampling-problem"><i class="fa fa-check"></i><b>61.13</b> Part 3: The Sampling Problem</a></li>
<li class="chapter" data-level="61.14" data-path="diversity-statistics.html"><a href="diversity-statistics.html#you-never-find-all-the-species"><i class="fa fa-check"></i><b>61.14</b> You never find all the species</a></li>
<li class="chapter" data-level="61.15" data-path="diversity-statistics.html"><a href="diversity-statistics.html#rarefaction-standardizing-by-sample-size"><i class="fa fa-check"></i><b>61.15</b> Rarefaction: Standardizing by sample size</a></li>
<li class="chapter" data-level="61.16" data-path="diversity-statistics.html"><a href="diversity-statistics.html#extrapolation-predicting-unseen-diversity"><i class="fa fa-check"></i><b>61.16</b> Extrapolation: Predicting unseen diversity</a></li>
<li class="chapter" data-level="61.17" data-path="diversity-statistics.html"><a href="diversity-statistics.html#sample-coverage-a-better-standardization"><i class="fa fa-check"></i><b>61.17</b> Sample coverage: A better standardization</a></li>
<li class="chapter" data-level="61.18" data-path="diversity-statistics.html"><a href="diversity-statistics.html#part-4-the-inext-framework"><i class="fa fa-check"></i><b>61.18</b> Part 4: The iNEXT Framework</a></li>
<li class="chapter" data-level="61.19" data-path="diversity-statistics.html"><a href="diversity-statistics.html#basic-workflow-1"><i class="fa fa-check"></i><b>61.19</b> Basic workflow</a>
<ul>
<li class="chapter" data-level="61.19.1" data-path="diversity-statistics.html"><a href="diversity-statistics.html#understanding-the-output"><i class="fa fa-check"></i><b>61.19.1</b> Understanding the output</a></li>
</ul></li>
<li class="chapter" data-level="61.20" data-path="diversity-statistics.html"><a href="diversity-statistics.html#visualizing-rarefactionextrapolation-curves"><i class="fa fa-check"></i><b>61.20</b> Visualizing rarefaction/extrapolation curves</a></li>
<li class="chapter" data-level="61.21" data-path="diversity-statistics.html"><a href="diversity-statistics.html#sample-size-based-vs.-coverage-based-comparison"><i class="fa fa-check"></i><b>61.21</b> Sample size-based vs. coverage-based comparison</a></li>
<li class="chapter" data-level="61.22" data-path="diversity-statistics.html"><a href="diversity-statistics.html#comparing-at-standardized-coverage"><i class="fa fa-check"></i><b>61.22</b> Comparing at standardized coverage</a></li>
<li class="chapter" data-level="61.23" data-path="diversity-statistics.html"><a href="diversity-statistics.html#working-with-incidence-data"><i class="fa fa-check"></i><b>61.23</b> Working with incidence data</a></li>
<li class="chapter" data-level="61.24" data-path="diversity-statistics.html"><a href="diversity-statistics.html#part-5-data-preparation-for-inext"><i class="fa fa-check"></i><b>61.24</b> Part 5: Data Preparation for iNEXT</a></li>
<li class="chapter" data-level="61.25" data-path="diversity-statistics.html"><a href="diversity-statistics.html#from-a-site-species-matrix-abundance"><i class="fa fa-check"></i><b>61.25</b> From a site × species matrix (abundance)</a></li>
<li class="chapter" data-level="61.26" data-path="diversity-statistics.html"><a href="diversity-statistics.html#from-a-site-species-matrix-incidence"><i class="fa fa-check"></i><b>61.26</b> From a site × species matrix (incidence)</a></li>
<li class="chapter" data-level="61.27" data-path="diversity-statistics.html"><a href="diversity-statistics.html#from-long-format-data"><i class="fa fa-check"></i><b>61.27</b> From long-format data</a></li>
</ul></li>
<li class="chapter" data-level="62" data-path="part-6-experimental-design-for-diversity-studies.html"><a href="part-6-experimental-design-for-diversity-studies.html"><i class="fa fa-check"></i><b>62</b> Part 6: Experimental Design for Diversity Studies</a>
<ul>
<li class="chapter" data-level="62.1" data-path="part-6-experimental-design-for-diversity-studies.html"><a href="part-6-experimental-design-for-diversity-studies.html#key-design-considerations"><i class="fa fa-check"></i><b>62.1</b> Key design considerations</a>
<ul>
<li class="chapter" data-level="62.1.1" data-path="part-6-experimental-design-for-diversity-studies.html"><a href="part-6-experimental-design-for-diversity-studies.html#what-is-your-sampling-unit"><i class="fa fa-check"></i><b>62.1.1</b> 1. What is your sampling unit?</a></li>
<li class="chapter" data-level="62.1.2" data-path="part-6-experimental-design-for-diversity-studies.html"><a href="part-6-experimental-design-for-diversity-studies.html#how-many-sampling-units-do-you-need"><i class="fa fa-check"></i><b>62.1.2</b> 2. How many sampling units do you need?</a></li>
<li class="chapter" data-level="62.1.3" data-path="part-6-experimental-design-for-diversity-studies.html"><a href="part-6-experimental-design-for-diversity-studies.html#replication-for-statistical-comparison"><i class="fa fa-check"></i><b>62.1.3</b> 3. Replication for statistical comparison</a></li>
</ul></li>
<li class="chapter" data-level="62.2" data-path="part-6-experimental-design-for-diversity-studies.html"><a href="part-6-experimental-design-for-diversity-studies.html#example-comparing-diversity-between-treatments"><i class="fa fa-check"></i><b>62.2</b> Example: Comparing diversity between treatments</a></li>
<li class="chapter" data-level="62.3" data-path="part-6-experimental-design-for-diversity-studies.html"><a href="part-6-experimental-design-for-diversity-studies.html#workflow-for-comparing-treatments"><i class="fa fa-check"></i><b>62.3</b> Workflow for comparing treatments</a></li>
</ul></li>
<li class="chapter" data-level="63" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html"><i class="fa fa-check"></i><b>63</b> Part 7: Reporting Diversity Results</a>
<ul>
<li class="chapter" data-level="63.1" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html#what-to-report-7"><i class="fa fa-check"></i><b>63.1</b> What to report</a></li>
<li class="chapter" data-level="63.2" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html#sample-methods-and-results-14"><i class="fa fa-check"></i><b>63.2</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="63.2.1" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html#methods-6"><i class="fa fa-check"></i><b>63.2.1</b> Methods</a></li>
<li class="chapter" data-level="63.2.2" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html#results-7"><i class="fa fa-check"></i><b>63.2.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="63.3" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html#key-takeaways-15"><i class="fa fa-check"></i><b>63.3</b> Key takeaways</a></li>
<li class="chapter" data-level="63.4" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html#assignment-18"><i class="fa fa-check"></i><b>63.4</b> Assignment</a>
<ul>
<li class="chapter" data-level="63.4.1" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html#part-1-conceptual-questions-11"><i class="fa fa-check"></i><b>63.4.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="63.4.2" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html#part-2-inext-analysis"><i class="fa fa-check"></i><b>63.4.2</b> Part 2: iNEXT analysis</a></li>
<li class="chapter" data-level="63.4.3" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html#part-3-data-preparation"><i class="fa fa-check"></i><b>63.4.3</b> Part 3: Data preparation</a></li>
<li class="chapter" data-level="63.4.4" data-path="part-7-reporting-diversity-results.html"><a href="part-7-reporting-diversity-results.html#part-4-reflection-6"><i class="fa fa-check"></i><b>63.4.4</b> Part 4: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="64" data-path="functional-trait-analysis.html"><a href="functional-trait-analysis.html"><i class="fa fa-check"></i><b>64</b> Functional Trait Analysis</a>
<ul>
<li class="chapter" data-level="64.1" data-path="functional-trait-analysis.html"><a href="functional-trait-analysis.html#why-traits-matter"><i class="fa fa-check"></i><b>64.1</b> Why traits matter</a></li>
<li class="chapter" data-level="64.2" data-path="functional-trait-analysis.html"><a href="functional-trait-analysis.html#setup-15"><i class="fa fa-check"></i><b>64.2</b> Setup</a></li>
<li class="chapter" data-level="64.3" data-path="functional-trait-analysis.html"><a href="functional-trait-analysis.html#part-1-trait-data-structure"><i class="fa fa-check"></i><b>64.3</b> Part 1: Trait Data Structure</a></li>
<li class="chapter" data-level="64.4" data-path="functional-trait-analysis.html"><a href="functional-trait-analysis.html#the-three-matrices"><i class="fa fa-check"></i><b>64.4</b> The three matrices</a></li>
<li class="chapter" data-level="64.5" data-path="functional-trait-analysis.html"><a href="functional-trait-analysis.html#example-data"><i class="fa fa-check"></i><b>64.5</b> Example data</a></li>
<li class="chapter" data-level="64.6" data-path="functional-trait-analysis.html"><a href="functional-trait-analysis.html#handling-trait-data-types"><i class="fa fa-check"></i><b>64.6</b> Handling trait data types</a></li>
</ul></li>
<li class="chapter" data-level="65" data-path="part-2-community-weighted-means-cwm.html"><a href="part-2-community-weighted-means-cwm.html"><i class="fa fa-check"></i><b>65</b> Part 2: Community-Weighted Means (CWM)</a>
<ul>
<li class="chapter" data-level="65.1" data-path="part-2-community-weighted-means-cwm.html"><a href="part-2-community-weighted-means-cwm.html#what-is-cwm"><i class="fa fa-check"></i><b>65.1</b> What is CWM?</a></li>
<li class="chapter" data-level="65.2" data-path="part-2-community-weighted-means-cwm.html"><a href="part-2-community-weighted-means-cwm.html#calculating-cwm"><i class="fa fa-check"></i><b>65.2</b> Calculating CWM</a></li>
<li class="chapter" data-level="65.3" data-path="part-2-community-weighted-means-cwm.html"><a href="part-2-community-weighted-means-cwm.html#cwm-along-environmental-gradients"><i class="fa fa-check"></i><b>65.3</b> CWM along environmental gradients</a></li>
<li class="chapter" data-level="65.4" data-path="part-2-community-weighted-means-cwm.html"><a href="part-2-community-weighted-means-cwm.html#interpreting-cwm"><i class="fa fa-check"></i><b>65.4</b> Interpreting CWM</a></li>
</ul></li>
<li class="chapter" data-level="66" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html"><i class="fa fa-check"></i><b>66</b> Part 3: Functional Diversity Indices</a>
<ul>
<li class="chapter" data-level="66.1" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#the-main-indices"><i class="fa fa-check"></i><b>66.1</b> The main indices</a></li>
<li class="chapter" data-level="66.2" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#calculating-functional-diversity"><i class="fa fa-check"></i><b>66.2</b> Calculating functional diversity</a></li>
<li class="chapter" data-level="66.3" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#using-fundiversity-tidyverse-friendly"><i class="fa fa-check"></i><b>66.3</b> Using fundiversity (tidyverse-friendly)</a></li>
<li class="chapter" data-level="66.4" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#functional-diversity-along-gradients"><i class="fa fa-check"></i><b>66.4</b> Functional diversity along gradients</a></li>
<li class="chapter" data-level="66.5" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-4-fourth-corner-analysis"><i class="fa fa-check"></i><b>66.5</b> Part 4: Fourth-Corner Analysis</a></li>
<li class="chapter" data-level="66.6" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#the-question-6"><i class="fa fa-check"></i><b>66.6</b> The question</a></li>
<li class="chapter" data-level="66.7" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#interpreting-fourth-corner-results"><i class="fa fa-check"></i><b>66.7</b> Interpreting fourth-corner results</a></li>
<li class="chapter" data-level="66.8" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#adjusting-for-multiple-testing"><i class="fa fa-check"></i><b>66.8</b> Adjusting for multiple testing</a></li>
<li class="chapter" data-level="66.9" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-5-rlq-analysis"><i class="fa fa-check"></i><b>66.9</b> Part 5: RLQ Analysis</a></li>
<li class="chapter" data-level="66.10" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#beyond-fourth-corner"><i class="fa fa-check"></i><b>66.10</b> Beyond fourth-corner</a></li>
<li class="chapter" data-level="66.11" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#visualizing-rlq"><i class="fa fa-check"></i><b>66.11</b> Visualizing RLQ</a></li>
<li class="chapter" data-level="66.12" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#testing-rlq-significance"><i class="fa fa-check"></i><b>66.12</b> Testing RLQ significance</a></li>
<li class="chapter" data-level="66.13" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-6-trait-based-community-assembly"><i class="fa fa-check"></i><b>66.13</b> Part 6: Trait-Based Community Assembly</a></li>
<li class="chapter" data-level="66.14" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#assembly-mechanisms"><i class="fa fa-check"></i><b>66.14</b> Assembly mechanisms</a></li>
<li class="chapter" data-level="66.15" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#null-models"><i class="fa fa-check"></i><b>66.15</b> Null models</a></li>
<li class="chapter" data-level="66.16" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#interpreting-ses"><i class="fa fa-check"></i><b>66.16</b> Interpreting SES</a></li>
<li class="chapter" data-level="66.17" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-7-intraspecific-trait-variation"><i class="fa fa-check"></i><b>66.17</b> Part 7: Intraspecific Trait Variation</a></li>
<li class="chapter" data-level="66.18" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#beyond-species-means"><i class="fa fa-check"></i><b>66.18</b> Beyond species means</a></li>
<li class="chapter" data-level="66.19" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#measuring-itv"><i class="fa fa-check"></i><b>66.19</b> Measuring ITV</a></li>
<li class="chapter" data-level="66.20" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#partitioning-trait-variation"><i class="fa fa-check"></i><b>66.20</b> Partitioning trait variation</a></li>
<li class="chapter" data-level="66.21" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-8-complete-workflow"><i class="fa fa-check"></i><b>66.21</b> Part 8: Complete Workflow</a></li>
<li class="chapter" data-level="66.22" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-9-reporting"><i class="fa fa-check"></i><b>66.22</b> Part 9: Reporting</a></li>
<li class="chapter" data-level="66.23" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#what-to-report-8"><i class="fa fa-check"></i><b>66.23</b> What to report</a></li>
<li class="chapter" data-level="66.24" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#sample-methods-and-results-15"><i class="fa fa-check"></i><b>66.24</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="66.24.1" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#methods-7"><i class="fa fa-check"></i><b>66.24.1</b> Methods</a></li>
<li class="chapter" data-level="66.24.2" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#results-8"><i class="fa fa-check"></i><b>66.24.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="66.25" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#key-takeaways-16"><i class="fa fa-check"></i><b>66.25</b> Key takeaways</a></li>
<li class="chapter" data-level="66.26" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#assignment-19"><i class="fa fa-check"></i><b>66.26</b> Assignment</a>
<ul>
<li class="chapter" data-level="66.26.1" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-1-conceptual-questions-12"><i class="fa fa-check"></i><b>66.26.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="66.26.2" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-2-cwm-analysis"><i class="fa fa-check"></i><b>66.26.2</b> Part 2: CWM analysis</a></li>
<li class="chapter" data-level="66.26.3" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-3-functional-diversity"><i class="fa fa-check"></i><b>66.26.3</b> Part 3: Functional diversity</a></li>
<li class="chapter" data-level="66.26.4" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-4-fourth-corner-analysis-1"><i class="fa fa-check"></i><b>66.26.4</b> Part 4: Fourth-corner analysis</a></li>
<li class="chapter" data-level="66.26.5" data-path="part-3-functional-diversity-indices.html"><a href="part-3-functional-diversity-indices.html#part-5-reflection-5"><i class="fa fa-check"></i><b>66.26.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="67" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html"><i class="fa fa-check"></i><b>67</b> Network Analysis in Ecology</a>
<ul>
<li class="chapter" data-level="67.1" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#what-is-an-ecological-network"><i class="fa fa-check"></i><b>67.1</b> What is an ecological network?</a></li>
<li class="chapter" data-level="67.2" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#types-of-ecological-networks"><i class="fa fa-check"></i><b>67.2</b> Types of ecological networks</a></li>
<li class="chapter" data-level="67.3" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#setup-16"><i class="fa fa-check"></i><b>67.3</b> Setup</a></li>
<li class="chapter" data-level="67.4" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-1-bipartite-networks"><i class="fa fa-check"></i><b>67.4</b> Part 1: Bipartite Networks</a></li>
<li class="chapter" data-level="67.5" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#network-data-structure"><i class="fa fa-check"></i><b>67.5</b> Network data structure</a></li>
<li class="chapter" data-level="67.6" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#visualizing-bipartite-networks"><i class="fa fa-check"></i><b>67.6</b> Visualizing bipartite networks</a></li>
<li class="chapter" data-level="67.7" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#network-level-metrics"><i class="fa fa-check"></i><b>67.7</b> Network-level metrics</a>
<ul>
<li class="chapter" data-level="67.7.1" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#key-metrics-explained"><i class="fa fa-check"></i><b>67.7.1</b> Key metrics explained</a></li>
</ul></li>
<li class="chapter" data-level="67.8" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#species-level-metrics"><i class="fa fa-check"></i><b>67.8</b> Species-level metrics</a>
<ul>
<li class="chapter" data-level="67.8.1" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#species-level-metrics-explained"><i class="fa fa-check"></i><b>67.8.1</b> Species-level metrics explained</a></li>
</ul></li>
<li class="chapter" data-level="67.9" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-2-nestedness"><i class="fa fa-check"></i><b>67.9</b> Part 2: Nestedness</a></li>
<li class="chapter" data-level="67.10" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#what-is-nestedness"><i class="fa fa-check"></i><b>67.10</b> What is nestedness?</a></li>
<li class="chapter" data-level="67.11" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#measuring-nestedness"><i class="fa fa-check"></i><b>67.11</b> Measuring nestedness</a></li>
<li class="chapter" data-level="67.12" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#why-nestedness-matters"><i class="fa fa-check"></i><b>67.12</b> Why nestedness matters</a></li>
<li class="chapter" data-level="67.13" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-3-modularity"><i class="fa fa-check"></i><b>67.13</b> Part 3: Modularity</a></li>
<li class="chapter" data-level="67.14" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#what-is-modularity"><i class="fa fa-check"></i><b>67.14</b> What is modularity?</a></li>
<li class="chapter" data-level="67.15" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#detecting-modules"><i class="fa fa-check"></i><b>67.15</b> Detecting modules</a></li>
<li class="chapter" data-level="67.16" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#visualizing-modules"><i class="fa fa-check"></i><b>67.16</b> Visualizing modules</a></li>
<li class="chapter" data-level="67.17" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#modularity-vs-nestedness"><i class="fa fa-check"></i><b>67.17</b> Modularity vs nestedness</a></li>
<li class="chapter" data-level="67.18" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-4-food-webs-unipartite-networks"><i class="fa fa-check"></i><b>67.18</b> Part 4: Food Webs (Unipartite Networks)</a></li>
<li class="chapter" data-level="67.19" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#food-web-data-structure"><i class="fa fa-check"></i><b>67.19</b> Food web data structure</a></li>
<li class="chapter" data-level="67.20" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#using-igraph-for-food-webs"><i class="fa fa-check"></i><b>67.20</b> Using igraph for food webs</a></li>
<li class="chapter" data-level="67.21" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#visualizing-food-webs"><i class="fa fa-check"></i><b>67.21</b> Visualizing food webs</a></li>
<li class="chapter" data-level="67.22" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#food-web-metrics"><i class="fa fa-check"></i><b>67.22</b> Food web metrics</a></li>
<li class="chapter" data-level="67.23" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-5-network-centrality"><i class="fa fa-check"></i><b>67.23</b> Part 5: Network Centrality</a></li>
<li class="chapter" data-level="67.24" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#why-centrality-matters"><i class="fa fa-check"></i><b>67.24</b> Why centrality matters</a></li>
<li class="chapter" data-level="67.25" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-6-null-models-and-statistical-testing"><i class="fa fa-check"></i><b>67.25</b> Part 6: Null Models and Statistical Testing</a></li>
<li class="chapter" data-level="67.26" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#why-null-models"><i class="fa fa-check"></i><b>67.26</b> Why null models?</a></li>
<li class="chapter" data-level="67.27" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#types-of-null-models"><i class="fa fa-check"></i><b>67.27</b> Types of null models</a></li>
<li class="chapter" data-level="67.28" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#testing-nestedness"><i class="fa fa-check"></i><b>67.28</b> Testing nestedness</a></li>
<li class="chapter" data-level="67.29" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#testing-multiple-metrics"><i class="fa fa-check"></i><b>67.29</b> Testing multiple metrics</a></li>
<li class="chapter" data-level="67.30" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-7-network-robustness"><i class="fa fa-check"></i><b>67.30</b> Part 7: Network Robustness</a></li>
<li class="chapter" data-level="67.31" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#simulating-species-loss"><i class="fa fa-check"></i><b>67.31</b> Simulating species loss</a></li>
<li class="chapter" data-level="67.32" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-8-comparing-networks"><i class="fa fa-check"></i><b>67.32</b> Part 8: Comparing Networks</a></li>
<li class="chapter" data-level="67.33" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#comparing-network-structure"><i class="fa fa-check"></i><b>67.33</b> Comparing network structure</a></li>
<li class="chapter" data-level="67.34" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#network-dissimilarity"><i class="fa fa-check"></i><b>67.34</b> Network dissimilarity</a></li>
<li class="chapter" data-level="67.35" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-9-complete-workflow"><i class="fa fa-check"></i><b>67.35</b> Part 9: Complete Workflow</a></li>
<li class="chapter" data-level="67.36" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-10-reporting-2"><i class="fa fa-check"></i><b>67.36</b> Part 10: Reporting</a></li>
<li class="chapter" data-level="67.37" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#what-to-report-9"><i class="fa fa-check"></i><b>67.37</b> What to report</a></li>
<li class="chapter" data-level="67.38" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#sample-methods-and-results-16"><i class="fa fa-check"></i><b>67.38</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="67.38.1" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#methods-8"><i class="fa fa-check"></i><b>67.38.1</b> Methods</a></li>
<li class="chapter" data-level="67.38.2" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#results-9"><i class="fa fa-check"></i><b>67.38.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="67.39" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#key-takeaways-17"><i class="fa fa-check"></i><b>67.39</b> Key takeaways</a></li>
<li class="chapter" data-level="67.40" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#assignment-20"><i class="fa fa-check"></i><b>67.40</b> Assignment</a>
<ul>
<li class="chapter" data-level="67.40.1" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-1-conceptual-questions-13"><i class="fa fa-check"></i><b>67.40.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="67.40.2" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-2-bipartite-network-analysis"><i class="fa fa-check"></i><b>67.40.2</b> Part 2: Bipartite network analysis</a></li>
<li class="chapter" data-level="67.40.3" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-3-food-web-analysis"><i class="fa fa-check"></i><b>67.40.3</b> Part 3: Food web analysis</a></li>
<li class="chapter" data-level="67.40.4" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-4-robustness-analysis"><i class="fa fa-check"></i><b>67.40.4</b> Part 4: Robustness analysis</a></li>
<li class="chapter" data-level="67.40.5" data-path="network-analysis-in-ecology.html"><a href="network-analysis-in-ecology.html#part-5-reflection-6"><i class="fa fa-check"></i><b>67.40.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="68" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html"><i class="fa fa-check"></i><b>68</b> Null Models and Community Assembly</a>
<ul>
<li class="chapter" data-level="68.1" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#the-fundamental-question"><i class="fa fa-check"></i><b>68.1</b> The fundamental question</a></li>
<li class="chapter" data-level="68.2" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#the-null-model-logic"><i class="fa fa-check"></i><b>68.2</b> The null model logic</a></li>
<li class="chapter" data-level="68.3" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#setup-17"><i class="fa fa-check"></i><b>68.3</b> Setup</a></li>
<li class="chapter" data-level="68.4" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-1-the-philosophy-of-null-models"><i class="fa fa-check"></i><b>68.4</b> Part 1: The Philosophy of Null Models</a></li>
<li class="chapter" data-level="68.5" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#what-makes-a-good-null-model"><i class="fa fa-check"></i><b>68.5</b> What makes a good null model?</a></li>
<li class="chapter" data-level="68.6" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#the-constraint-problem"><i class="fa fa-check"></i><b>68.6</b> The constraint problem</a></li>
<li class="chapter" data-level="68.7" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#type-i-and-type-ii-errors-in-null-models"><i class="fa fa-check"></i><b>68.7</b> Type I and Type II errors in null models</a></li>
<li class="chapter" data-level="68.8" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-2-co-occurrence-analysis"><i class="fa fa-check"></i><b>68.8</b> Part 2: Co-occurrence Analysis</a></li>
<li class="chapter" data-level="68.9" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#the-classic-question"><i class="fa fa-check"></i><b>68.9</b> The classic question</a></li>
<li class="chapter" data-level="68.10" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#creating-a-presence-absence-matrix"><i class="fa fa-check"></i><b>68.10</b> Creating a presence-absence matrix</a></li>
<li class="chapter" data-level="68.11" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#checkerboard-units"><i class="fa fa-check"></i><b>68.11</b> Checkerboard units</a></li>
<li class="chapter" data-level="68.12" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#the-c-score"><i class="fa fa-check"></i><b>68.12</b> The C-score</a></li>
<li class="chapter" data-level="68.13" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#interpreting-co-occurrence-results"><i class="fa fa-check"></i><b>68.13</b> Interpreting co-occurrence results</a></li>
<li class="chapter" data-level="68.14" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#effect-size-ses"><i class="fa fa-check"></i><b>68.14</b> Effect size (SES)</a></li>
<li class="chapter" data-level="68.15" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-3-null-model-algorithms"><i class="fa fa-check"></i><b>68.15</b> Part 3: Null Model Algorithms</a></li>
<li class="chapter" data-level="68.16" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#common-algorithms-for-presence-absence-data"><i class="fa fa-check"></i><b>68.16</b> Common algorithms for presence-absence data</a></li>
<li class="chapter" data-level="68.17" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#the-swap-algorithm-sim9"><i class="fa fa-check"></i><b>68.17</b> The swap algorithm (sim9)</a></li>
<li class="chapter" data-level="68.18" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#using-vegan-for-co-occurrence"><i class="fa fa-check"></i><b>68.18</b> Using vegan for co-occurrence</a></li>
<li class="chapter" data-level="68.19" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-4-phylogenetic-community-structure"><i class="fa fa-check"></i><b>68.19</b> Part 4: Phylogenetic Community Structure</a></li>
<li class="chapter" data-level="68.20" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#the-question-7"><i class="fa fa-check"></i><b>68.20</b> The question</a></li>
<li class="chapter" data-level="68.21" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#the-data-you-need"><i class="fa fa-check"></i><b>68.21</b> The data you need</a></li>
<li class="chapter" data-level="68.22" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#mean-pairwise-distance-mpd"><i class="fa fa-check"></i><b>68.22</b> Mean Pairwise Distance (MPD)</a></li>
<li class="chapter" data-level="68.23" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#mean-nearest-taxon-distance-mntd"><i class="fa fa-check"></i><b>68.23</b> Mean Nearest Taxon Distance (MNTD)</a></li>
<li class="chapter" data-level="68.24" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#standardized-effect-sizes-nri-and-nti"><i class="fa fa-check"></i><b>68.24</b> Standardized effect sizes (NRI and NTI)</a></li>
<li class="chapter" data-level="68.25" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#interpreting-phylogenetic-structure"><i class="fa fa-check"></i><b>68.25</b> Interpreting phylogenetic structure</a></li>
<li class="chapter" data-level="68.26" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-5-null-models-for-phylogenetic-structure"><i class="fa fa-check"></i><b>68.26</b> Part 5: Null Models for Phylogenetic Structure</a></li>
<li class="chapter" data-level="68.27" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#null-model-choices"><i class="fa fa-check"></i><b>68.27</b> Null model choices</a></li>
<li class="chapter" data-level="68.28" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-6-connecting-patterns-to-mechanisms"><i class="fa fa-check"></i><b>68.28</b> Part 6: Connecting Patterns to Mechanisms</a></li>
<li class="chapter" data-level="68.29" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#the-inference-problem"><i class="fa fa-check"></i><b>68.29</b> The inference problem</a></li>
<li class="chapter" data-level="68.30" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#combining-co-occurrence-and-phylogeny"><i class="fa fa-check"></i><b>68.30</b> Combining co-occurrence and phylogeny</a></li>
<li class="chapter" data-level="68.31" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#scale-dependence"><i class="fa fa-check"></i><b>68.31</b> Scale dependence</a></li>
<li class="chapter" data-level="68.32" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-7-functional-trait-null-models"><i class="fa fa-check"></i><b>68.32</b> Part 7: Functional Trait Null Models</a></li>
<li class="chapter" data-level="68.33" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#connecting-to-chapter-23"><i class="fa fa-check"></i><b>68.33</b> Connecting to Chapter 23</a></li>
<li class="chapter" data-level="68.34" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#comparing-phylogenetic-and-functional-structure"><i class="fa fa-check"></i><b>68.34</b> Comparing phylogenetic and functional structure</a></li>
<li class="chapter" data-level="68.35" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-8-complete-workflow-1"><i class="fa fa-check"></i><b>68.35</b> Part 8: Complete Workflow</a></li>
<li class="chapter" data-level="68.36" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-9-reporting-1"><i class="fa fa-check"></i><b>68.36</b> Part 9: Reporting</a></li>
<li class="chapter" data-level="68.37" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#what-to-report-10"><i class="fa fa-check"></i><b>68.37</b> What to report</a></li>
<li class="chapter" data-level="68.38" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#sample-methods-and-results-17"><i class="fa fa-check"></i><b>68.38</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="68.38.1" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#methods-9"><i class="fa fa-check"></i><b>68.38.1</b> Methods</a></li>
<li class="chapter" data-level="68.38.2" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#results-10"><i class="fa fa-check"></i><b>68.38.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="68.39" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#key-takeaways-18"><i class="fa fa-check"></i><b>68.39</b> Key takeaways</a></li>
<li class="chapter" data-level="68.40" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#assignment-21"><i class="fa fa-check"></i><b>68.40</b> Assignment</a>
<ul>
<li class="chapter" data-level="68.40.1" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-1-conceptual-questions-14"><i class="fa fa-check"></i><b>68.40.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="68.40.2" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-2-co-occurrence-analysis-1"><i class="fa fa-check"></i><b>68.40.2</b> Part 2: Co-occurrence analysis</a></li>
<li class="chapter" data-level="68.40.3" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-3-phylogenetic-structure"><i class="fa fa-check"></i><b>68.40.3</b> Part 3: Phylogenetic structure</a></li>
<li class="chapter" data-level="68.40.4" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-4-combining-evidence"><i class="fa fa-check"></i><b>68.40.4</b> Part 4: Combining evidence</a></li>
<li class="chapter" data-level="68.40.5" data-path="null-models-and-community-assembly.html"><a href="null-models-and-community-assembly.html#part-5-reflection-7"><i class="fa fa-check"></i><b>68.40.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="69" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html"><i class="fa fa-check"></i><b>69</b> Introduction: Working with Spatial Data in R</a>
<ul>
<li class="chapter" data-level="69.1" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#setup-18"><i class="fa fa-check"></i><b>69.1</b> Setup</a></li>
<li class="chapter" data-level="69.2" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#part-1-understanding-coordinate-reference-systems-crs"><i class="fa fa-check"></i><b>69.2</b> Part 1: Understanding Coordinate Reference Systems (CRS)</a></li>
<li class="chapter" data-level="69.3" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#the-fundamental-problem"><i class="fa fa-check"></i><b>69.3</b> The fundamental problem</a></li>
<li class="chapter" data-level="69.4" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#geographic-vs.-projected-coordinate-systems"><i class="fa fa-check"></i><b>69.4</b> Geographic vs. Projected Coordinate Systems</a>
<ul>
<li class="chapter" data-level="69.4.1" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#geographic-coordinate-systems-gcs"><i class="fa fa-check"></i><b>69.4.1</b> Geographic Coordinate Systems (GCS)</a></li>
<li class="chapter" data-level="69.4.2" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#projected-coordinate-systems-pcs"><i class="fa fa-check"></i><b>69.4.2</b> Projected Coordinate Systems (PCS)</a></li>
</ul></li>
<li class="chapter" data-level="69.5" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#the-utm-system"><i class="fa fa-check"></i><b>69.5</b> The UTM System</a></li>
<li class="chapter" data-level="69.6" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#epsg-codes"><i class="fa fa-check"></i><b>69.6</b> EPSG Codes</a></li>
<li class="chapter" data-level="69.7" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#crs-in-practice"><i class="fa fa-check"></i><b>69.7</b> CRS in Practice</a></li>
<li class="chapter" data-level="69.8" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#common-crs-problems-and-solutions"><i class="fa fa-check"></i><b>69.8</b> Common CRS Problems and Solutions</a></li>
<li class="chapter" data-level="69.9" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#part-2-types-of-spatial-data"><i class="fa fa-check"></i><b>69.9</b> Part 2: Types of Spatial Data</a></li>
<li class="chapter" data-level="69.10" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#vector-data"><i class="fa fa-check"></i><b>69.10</b> Vector Data</a></li>
<li class="chapter" data-level="69.11" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#raster-data"><i class="fa fa-check"></i><b>69.11</b> Raster Data</a></li>
<li class="chapter" data-level="69.12" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#when-to-use-which"><i class="fa fa-check"></i><b>69.12</b> When to Use Which?</a></li>
<li class="chapter" data-level="69.13" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#part-3-reading-and-writing-spatial-data"><i class="fa fa-check"></i><b>69.13</b> Part 3: Reading and Writing Spatial Data</a></li>
<li class="chapter" data-level="69.14" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#vector-data-with-sf"><i class="fa fa-check"></i><b>69.14</b> Vector Data with sf</a></li>
<li class="chapter" data-level="69.15" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#raster-data-with-terra"><i class="fa fa-check"></i><b>69.15</b> Raster Data with terra</a></li>
<li class="chapter" data-level="69.16" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#part-4-basic-spatial-operations"><i class="fa fa-check"></i><b>69.16</b> Part 4: Basic Spatial Operations</a></li>
<li class="chapter" data-level="69.17" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#vector-operations"><i class="fa fa-check"></i><b>69.17</b> Vector Operations</a></li>
<li class="chapter" data-level="69.18" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#raster-operations"><i class="fa fa-check"></i><b>69.18</b> Raster Operations</a></li>
<li class="chapter" data-level="69.19" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#extracting-raster-values-at-points"><i class="fa fa-check"></i><b>69.19</b> Extracting Raster Values at Points</a></li>
<li class="chapter" data-level="69.20" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#part-5-spatial-autocorrelation"><i class="fa fa-check"></i><b>69.20</b> Part 5: Spatial Autocorrelation</a></li>
<li class="chapter" data-level="69.21" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#what-is-spatial-autocorrelation"><i class="fa fa-check"></i><b>69.21</b> What is Spatial Autocorrelation?</a></li>
<li class="chapter" data-level="69.22" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#why-it-matters"><i class="fa fa-check"></i><b>69.22</b> Why It Matters</a></li>
<li class="chapter" data-level="69.23" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#detecting-spatial-autocorrelation"><i class="fa fa-check"></i><b>69.23</b> Detecting Spatial Autocorrelation</a>
<ul>
<li class="chapter" data-level="69.23.1" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#visual-methods"><i class="fa fa-check"></i><b>69.23.1</b> Visual Methods</a></li>
<li class="chapter" data-level="69.23.2" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#morans-i-test"><i class="fa fa-check"></i><b>69.23.2</b> Moran’s I Test</a></li>
</ul></li>
<li class="chapter" data-level="69.24" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#dealing-with-spatial-autocorrelation"><i class="fa fa-check"></i><b>69.24</b> Dealing with Spatial Autocorrelation</a></li>
<li class="chapter" data-level="69.25" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#part-6-complete-worked-example"><i class="fa fa-check"></i><b>69.25</b> Part 6: Complete Worked Example</a></li>
<li class="chapter" data-level="69.26" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#key-takeaways-19"><i class="fa fa-check"></i><b>69.26</b> Key Takeaways</a></li>
<li class="chapter" data-level="69.27" data-path="introduction-working-with-spatial-data-in-r.html"><a href="introduction-working-with-spatial-data-in-r.html#assignment-22"><i class="fa fa-check"></i><b>69.27</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="70" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html"><i class="fa fa-check"></i><b>70</b> Working with Long-term and Synthetic Data</a>
<ul>
<li class="chapter" data-level="70.1" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#why-this-matters"><i class="fa fa-check"></i><b>70.1</b> Why this matters</a></li>
<li class="chapter" data-level="70.2" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#setup-19"><i class="fa fa-check"></i><b>70.2</b> Setup</a></li>
<li class="chapter" data-level="70.3" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#part-1-challenges-of-data-you-didnt-collect"><i class="fa fa-check"></i><b>70.3</b> Part 1: Challenges of Data You Didn’t Collect</a></li>
<li class="chapter" data-level="70.4" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#the-hidden-assumptions"><i class="fa fa-check"></i><b>70.4</b> The hidden assumptions</a>
<ul>
<li class="chapter" data-level="70.4.1" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#common-problems"><i class="fa fa-check"></i><b>70.4.1</b> Common problems</a></li>
</ul></li>
<li class="chapter" data-level="70.5" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#evaluating-data-quality"><i class="fa fa-check"></i><b>70.5</b> Evaluating data quality</a>
<ul>
<li class="chapter" data-level="70.5.1" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#example-using-prism-climate-data"><i class="fa fa-check"></i><b>70.5.1</b> Example: Using PRISM climate data</a></li>
</ul></li>
<li class="chapter" data-level="70.6" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#part-2-introduction-to-time-series"><i class="fa fa-check"></i><b>70.6</b> Part 2: Introduction to Time Series</a></li>
<li class="chapter" data-level="70.7" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#when-does-repeated-measures-become-time-series"><i class="fa fa-check"></i><b>70.7</b> When does “repeated measures” become “time series”?</a></li>
<li class="chapter" data-level="70.8" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#the-components-of-a-time-series"><i class="fa fa-check"></i><b>70.8</b> The components of a time series</a></li>
<li class="chapter" data-level="70.9" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#stationarity"><i class="fa fa-check"></i><b>70.9</b> Stationarity</a></li>
<li class="chapter" data-level="70.10" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#autocorrelation-in-time-series"><i class="fa fa-check"></i><b>70.10</b> Autocorrelation in time series</a></li>
<li class="chapter" data-level="70.11" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#arima-models-a-brief-introduction"><i class="fa fa-check"></i><b>70.11</b> ARIMA models: A brief introduction</a>
<ul>
<li class="chapter" data-level="70.11.1" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#when-to-use-arima"><i class="fa fa-check"></i><b>70.11.1</b> When to use ARIMA</a></li>
<li class="chapter" data-level="70.11.2" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#connecting-arima-to-what-you-know"><i class="fa fa-check"></i><b>70.11.2</b> Connecting ARIMA to what you know</a></li>
</ul></li>
<li class="chapter" data-level="70.12" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#part-3-combining-datasets"><i class="fa fa-check"></i><b>70.12</b> Part 3: Combining Datasets</a></li>
<li class="chapter" data-level="70.13" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#harmonization-challenges"><i class="fa fa-check"></i><b>70.13</b> Harmonization challenges</a></li>
<li class="chapter" data-level="70.14" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#example-combining-climate-datasets"><i class="fa fa-check"></i><b>70.14</b> Example: Combining climate datasets</a></li>
<li class="chapter" data-level="70.15" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#documenting-your-synthesis"><i class="fa fa-check"></i><b>70.15</b> Documenting your synthesis</a></li>
<li class="chapter" data-level="70.16" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#part-4-introduction-to-meta-analysis"><i class="fa fa-check"></i><b>70.16</b> Part 4: Introduction to Meta-analysis</a></li>
<li class="chapter" data-level="70.17" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#why-meta-analysis"><i class="fa fa-check"></i><b>70.17</b> Why meta-analysis?</a></li>
<li class="chapter" data-level="70.18" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#the-meta-analysis-workflow"><i class="fa fa-check"></i><b>70.18</b> The meta-analysis workflow</a></li>
<li class="chapter" data-level="70.19" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#effect-sizes"><i class="fa fa-check"></i><b>70.19</b> Effect sizes</a></li>
<li class="chapter" data-level="70.20" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#example-meta-analysis-of-grazing-effects"><i class="fa fa-check"></i><b>70.20</b> Example: Meta-analysis of grazing effects</a></li>
<li class="chapter" data-level="70.21" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#calculate-effect-sizes"><i class="fa fa-check"></i><b>70.21</b> Calculate effect sizes</a></li>
<li class="chapter" data-level="70.22" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#fit-the-meta-analysis-model"><i class="fa fa-check"></i><b>70.22</b> Fit the meta-analysis model</a>
<ul>
<li class="chapter" data-level="70.22.1" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#interpret-the-output-13"><i class="fa fa-check"></i><b>70.22.1</b> Interpret the output</a></li>
</ul></li>
<li class="chapter" data-level="70.23" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#visualize-forest-plot"><i class="fa fa-check"></i><b>70.23</b> Visualize: Forest plot</a></li>
<li class="chapter" data-level="70.24" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#test-for-publication-bias"><i class="fa fa-check"></i><b>70.24</b> Test for publication bias</a></li>
<li class="chapter" data-level="70.25" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#moderator-analysis"><i class="fa fa-check"></i><b>70.25</b> Moderator analysis</a></li>
<li class="chapter" data-level="70.26" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#sample-methods-and-results-for-meta-analysis"><i class="fa fa-check"></i><b>70.26</b> Sample methods and results for meta-analysis</a></li>
<li class="chapter" data-level="70.27" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#part-5-practical-guidance"><i class="fa fa-check"></i><b>70.27</b> Part 5: Practical Guidance</a></li>
<li class="chapter" data-level="70.28" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#when-to-use-what"><i class="fa fa-check"></i><b>70.28</b> When to use what</a></li>
<li class="chapter" data-level="70.29" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#know-when-to-get-help"><i class="fa fa-check"></i><b>70.29</b> Know when to get help</a></li>
<li class="chapter" data-level="70.30" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#resources-for-going-deeper"><i class="fa fa-check"></i><b>70.30</b> Resources for going deeper</a></li>
<li class="chapter" data-level="70.31" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#key-takeaways-20"><i class="fa fa-check"></i><b>70.31</b> Key takeaways</a></li>
<li class="chapter" data-level="70.32" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#assignment-23"><i class="fa fa-check"></i><b>70.32</b> Assignment</a>
<ul>
<li class="chapter" data-level="70.32.1" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#part-1-evaluating-external-data"><i class="fa fa-check"></i><b>70.32.1</b> Part 1: Evaluating external data</a></li>
<li class="chapter" data-level="70.32.2" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#part-2-time-series-decomposition"><i class="fa fa-check"></i><b>70.32.2</b> Part 2: Time series decomposition</a></li>
<li class="chapter" data-level="70.32.3" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#part-3-mini-meta-analysis"><i class="fa fa-check"></i><b>70.32.3</b> Part 3: Mini meta-analysis</a></li>
<li class="chapter" data-level="70.32.4" data-path="working-with-long-term-and-synthetic-data.html"><a href="working-with-long-term-and-synthetic-data.html#part-4-reflection-7"><i class="fa fa-check"></i><b>70.32.4</b> Part 4: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="71" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html"><i class="fa fa-check"></i><b>71</b> Spatial Autocorrelation and Spatial Models</a>
<ul>
<li class="chapter" data-level="71.1" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#examples-of-spatial-data-in-ecology"><i class="fa fa-check"></i><b>71.1</b> Examples of spatial data in ecology</a></li>
<li class="chapter" data-level="71.2" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#setup-20"><i class="fa fa-check"></i><b>71.2</b> Setup</a></li>
<li class="chapter" data-level="71.3" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-1-what-is-spatial-autocorrelation"><i class="fa fa-check"></i><b>71.3</b> Part 1: What Is Spatial Autocorrelation?</a></li>
<li class="chapter" data-level="71.4" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#the-intuition"><i class="fa fa-check"></i><b>71.4</b> The intuition</a></li>
<li class="chapter" data-level="71.5" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#simulating-spatial-data"><i class="fa fa-check"></i><b>71.5</b> Simulating spatial data</a></li>
<li class="chapter" data-level="71.6" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-2-quantifying-spatial-autocorrelation"><i class="fa fa-check"></i><b>71.6</b> Part 2: Quantifying Spatial Autocorrelation</a></li>
<li class="chapter" data-level="71.7" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#morans-i"><i class="fa fa-check"></i><b>71.7</b> Moran’s I</a></li>
<li class="chapter" data-level="71.8" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#calculating-morans-i"><i class="fa fa-check"></i><b>71.8</b> Calculating Moran’s I</a></li>
<li class="chapter" data-level="71.9" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#morans-i-scatter-plot"><i class="fa fa-check"></i><b>71.9</b> Moran’s I scatter plot</a></li>
<li class="chapter" data-level="71.10" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#gearys-c"><i class="fa fa-check"></i><b>71.10</b> Geary’s C</a></li>
<li class="chapter" data-level="71.11" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-3-spatial-neighbors-and-weights"><i class="fa fa-check"></i><b>71.11</b> Part 3: Spatial Neighbors and Weights</a></li>
<li class="chapter" data-level="71.12" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#why-define-neighbors"><i class="fa fa-check"></i><b>71.12</b> Why define neighbors?</a></li>
<li class="chapter" data-level="71.13" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#creating-neighbor-lists"><i class="fa fa-check"></i><b>71.13</b> Creating neighbor lists</a></li>
<li class="chapter" data-level="71.14" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#spatial-weights-matrices"><i class="fa fa-check"></i><b>71.14</b> Spatial weights matrices</a></li>
<li class="chapter" data-level="71.15" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#visualizing-neighbors"><i class="fa fa-check"></i><b>71.15</b> Visualizing neighbors</a></li>
<li class="chapter" data-level="71.16" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-4-local-spatial-autocorrelation"><i class="fa fa-check"></i><b>71.16</b> Part 4: Local Spatial Autocorrelation</a></li>
<li class="chapter" data-level="71.17" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#global-vs-local"><i class="fa fa-check"></i><b>71.17</b> Global vs Local</a></li>
<li class="chapter" data-level="71.18" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#local-morans-i"><i class="fa fa-check"></i><b>71.18</b> Local Moran’s I</a></li>
<li class="chapter" data-level="71.19" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#identifying-clusters-and-outliers"><i class="fa fa-check"></i><b>71.19</b> Identifying clusters and outliers</a></li>
<li class="chapter" data-level="71.20" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-5-spatial-correlograms"><i class="fa fa-check"></i><b>71.20</b> Part 5: Spatial Correlograms</a></li>
<li class="chapter" data-level="71.21" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#autocorrelation-across-distances"><i class="fa fa-check"></i><b>71.21</b> Autocorrelation across distances</a></li>
<li class="chapter" data-level="71.22" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-6-spatial-regression-models"><i class="fa fa-check"></i><b>71.22</b> Part 6: Spatial Regression Models</a></li>
<li class="chapter" data-level="71.23" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#the-problem-with-ordinary-regression"><i class="fa fa-check"></i><b>71.23</b> The problem with ordinary regression</a></li>
<li class="chapter" data-level="71.24" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#types-of-spatial-regression"><i class="fa fa-check"></i><b>71.24</b> Types of spatial regression</a></li>
<li class="chapter" data-level="71.25" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#spatial-error-model-sem"><i class="fa fa-check"></i><b>71.25</b> Spatial Error Model (SEM)</a></li>
<li class="chapter" data-level="71.26" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#spatial-lag-model-slm"><i class="fa fa-check"></i><b>71.26</b> Spatial Lag Model (SLM)</a></li>
<li class="chapter" data-level="71.27" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#comparing-models-1"><i class="fa fa-check"></i><b>71.27</b> Comparing models</a></li>
<li class="chapter" data-level="71.28" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#interpreting-spatial-regression"><i class="fa fa-check"></i><b>71.28</b> Interpreting spatial regression</a></li>
<li class="chapter" data-level="71.29" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-7-geostatistics-and-variograms"><i class="fa fa-check"></i><b>71.29</b> Part 7: Geostatistics and Variograms</a></li>
<li class="chapter" data-level="71.30" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#what-is-geostatistics"><i class="fa fa-check"></i><b>71.30</b> What is geostatistics?</a></li>
<li class="chapter" data-level="71.31" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#the-variogram"><i class="fa fa-check"></i><b>71.31</b> The variogram</a></li>
<li class="chapter" data-level="71.32" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#computing-and-plotting-variograms"><i class="fa fa-check"></i><b>71.32</b> Computing and plotting variograms</a></li>
<li class="chapter" data-level="71.33" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#fitting-variogram-models"><i class="fa fa-check"></i><b>71.33</b> Fitting variogram models</a></li>
<li class="chapter" data-level="71.34" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-8-spatial-interpolation-kriging"><i class="fa fa-check"></i><b>71.34</b> Part 8: Spatial Interpolation (Kriging)</a></li>
<li class="chapter" data-level="71.35" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#what-is-kriging"><i class="fa fa-check"></i><b>71.35</b> What is kriging?</a></li>
<li class="chapter" data-level="71.36" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#ordinary-kriging"><i class="fa fa-check"></i><b>71.36</b> Ordinary kriging</a></li>
<li class="chapter" data-level="71.37" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#universal-kriging"><i class="fa fa-check"></i><b>71.37</b> Universal kriging</a></li>
<li class="chapter" data-level="71.38" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-9-ecological-case-studies"><i class="fa fa-check"></i><b>71.38</b> Part 9: Ecological Case Studies</a></li>
<li class="chapter" data-level="71.39" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#case-study-1-soil-nutrients-across-a-landscape"><i class="fa fa-check"></i><b>71.39</b> Case Study 1: Soil nutrients across a landscape</a></li>
<li class="chapter" data-level="71.40" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#case-study-2-species-richness-hotspots"><i class="fa fa-check"></i><b>71.40</b> Case Study 2: Species richness hotspots</a></li>
<li class="chapter" data-level="71.41" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#case-study-3-spatial-regression-comparison"><i class="fa fa-check"></i><b>71.41</b> Case Study 3: Spatial regression comparison</a></li>
<li class="chapter" data-level="71.42" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-10-common-pitfalls-and-tips"><i class="fa fa-check"></i><b>71.42</b> Part 10: Common Pitfalls and Tips</a></li>
<li class="chapter" data-level="71.43" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#pitfalls-to-avoid"><i class="fa fa-check"></i><b>71.43</b> Pitfalls to avoid</a></li>
<li class="chapter" data-level="71.44" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#practical-tips"><i class="fa fa-check"></i><b>71.44</b> Practical tips</a></li>
<li class="chapter" data-level="71.45" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-11-reporting"><i class="fa fa-check"></i><b>71.45</b> Part 11: Reporting</a></li>
<li class="chapter" data-level="71.46" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#what-to-report-11"><i class="fa fa-check"></i><b>71.46</b> What to report</a></li>
<li class="chapter" data-level="71.47" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#sample-methods-and-results-18"><i class="fa fa-check"></i><b>71.47</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="71.47.1" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#methods-10"><i class="fa fa-check"></i><b>71.47.1</b> Methods</a></li>
<li class="chapter" data-level="71.47.2" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#results-11"><i class="fa fa-check"></i><b>71.47.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="71.48" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#key-takeaways-21"><i class="fa fa-check"></i><b>71.48</b> Key takeaways</a></li>
<li class="chapter" data-level="71.49" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#assignment-24"><i class="fa fa-check"></i><b>71.49</b> Assignment</a>
<ul>
<li class="chapter" data-level="71.49.1" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-1-conceptual-questions-15"><i class="fa fa-check"></i><b>71.49.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="71.49.2" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-2-detecting-autocorrelation"><i class="fa fa-check"></i><b>71.49.2</b> Part 2: Detecting autocorrelation</a></li>
<li class="chapter" data-level="71.49.3" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-3-spatial-regression"><i class="fa fa-check"></i><b>71.49.3</b> Part 3: Spatial regression</a></li>
<li class="chapter" data-level="71.49.4" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-4-variogram-and-kriging"><i class="fa fa-check"></i><b>71.49.4</b> Part 4: Variogram and kriging</a></li>
<li class="chapter" data-level="71.49.5" data-path="spatial-autocorrelation-and-spatial-models.html"><a href="spatial-autocorrelation-and-spatial-models.html#part-5-reflection-8"><i class="fa fa-check"></i><b>71.49.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="72" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html"><i class="fa fa-check"></i><b>72</b> Spatial Disease Ecology: Patterns, Spread, and Risk</a>
<ul>
<li class="chapter" data-level="72.1" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#the-core-question-2"><i class="fa fa-check"></i><b>72.1</b> The core question</a></li>
<li class="chapter" data-level="72.2" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#analytical-framework"><i class="fa fa-check"></i><b>72.2</b> Analytical framework</a></li>
<li class="chapter" data-level="72.3" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#setup-21"><i class="fa fa-check"></i><b>72.3</b> Setup</a></li>
<li class="chapter" data-level="72.4" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-1-disease-data-structure"><i class="fa fa-check"></i><b>72.4</b> Part 1: Disease Data Structure</a></li>
<li class="chapter" data-level="72.5" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#types-of-disease-data"><i class="fa fa-check"></i><b>72.5</b> Types of disease data</a></li>
<li class="chapter" data-level="72.6" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#example-simulating-emory-oak-disease-data"><i class="fa fa-check"></i><b>72.6</b> Example: Simulating Emory oak disease data</a></li>
<li class="chapter" data-level="72.7" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-2-mapping-disease-patterns"><i class="fa fa-check"></i><b>72.7</b> Part 2: Mapping Disease Patterns</a></li>
<li class="chapter" data-level="72.8" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#basic-disease-map"><i class="fa fa-check"></i><b>72.8</b> Basic disease map</a></li>
<li class="chapter" data-level="72.9" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#disease-intensity-with-environment"><i class="fa fa-check"></i><b>72.9</b> Disease intensity with environment</a></li>
<li class="chapter" data-level="72.10" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#kernel-density-of-infected-trees"><i class="fa fa-check"></i><b>72.10</b> Kernel density of infected trees</a></li>
<li class="chapter" data-level="72.11" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-3-testing-for-spatial-clustering"><i class="fa fa-check"></i><b>72.11</b> Part 3: Testing for Spatial Clustering</a></li>
<li class="chapter" data-level="72.12" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#is-disease-clustered-beyond-what-tree-distribution-explains"><i class="fa fa-check"></i><b>72.12</b> Is disease clustered beyond what tree distribution explains?</a>
<ul>
<li class="chapter" data-level="72.12.1" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#join-count-test"><i class="fa fa-check"></i><b>72.12.1</b> Join count test</a></li>
<li class="chapter" data-level="72.12.2" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#morans-i-on-residuals-key-test"><i class="fa fa-check"></i><b>72.12.2</b> Moran’s I on residuals (key test!)</a></li>
</ul></li>
<li class="chapter" data-level="72.13" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-4-point-pattern-analysis"><i class="fa fa-check"></i><b>72.13</b> Part 4: Point Pattern Analysis</a></li>
<li class="chapter" data-level="72.14" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#using-spatstat-for-formal-point-pattern-tests"><i class="fa fa-check"></i><b>72.14</b> Using spatstat for formal point pattern tests</a></li>
<li class="chapter" data-level="72.15" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#k-function-testing-for-clustering"><i class="fa fa-check"></i><b>72.15</b> K-function: Testing for clustering</a></li>
<li class="chapter" data-level="72.16" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#cross-type-k-function"><i class="fa fa-check"></i><b>72.16</b> Cross-type K-function</a></li>
<li class="chapter" data-level="72.17" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-5-spatial-disease-regression"><i class="fa fa-check"></i><b>72.17</b> Part 5: Spatial Disease Regression</a></li>
<li class="chapter" data-level="72.18" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#building-a-risk-model"><i class="fa fa-check"></i><b>72.18</b> Building a risk model</a></li>
<li class="chapter" data-level="72.19" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#autologistic-regression"><i class="fa fa-check"></i><b>72.19</b> Autologistic regression</a></li>
<li class="chapter" data-level="72.20" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#interpreting-the-results"><i class="fa fa-check"></i><b>72.20</b> Interpreting the results</a></li>
<li class="chapter" data-level="72.21" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-6-temporal-spread-analysis"><i class="fa fa-check"></i><b>72.21</b> Part 6: Temporal Spread Analysis</a></li>
<li class="chapter" data-level="72.22" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#does-the-spatial-pattern-change-over-time"><i class="fa fa-check"></i><b>72.22</b> Does the spatial pattern change over time?</a></li>
<li class="chapter" data-level="72.23" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#testing-for-directional-spread"><i class="fa fa-check"></i><b>72.23</b> Testing for directional spread</a></li>
<li class="chapter" data-level="72.24" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-7-predicting-at-risk-areas"><i class="fa fa-check"></i><b>72.24</b> Part 7: Predicting At-Risk Areas</a></li>
<li class="chapter" data-level="72.25" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#building-a-risk-surface"><i class="fa fa-check"></i><b>72.25</b> Building a risk surface</a></li>
<li class="chapter" data-level="72.26" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#identifying-high-risk-uninfected-trees"><i class="fa fa-check"></i><b>72.26</b> Identifying high-risk uninfected trees</a></li>
<li class="chapter" data-level="72.27" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-8-distinguishing-hypotheses"><i class="fa fa-check"></i><b>72.27</b> Part 8: Distinguishing Hypotheses</a></li>
<li class="chapter" data-level="72.28" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#summary-of-evidence"><i class="fa fa-check"></i><b>72.28</b> Summary of evidence</a></li>
<li class="chapter" data-level="72.29" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-9-management-recommendations"><i class="fa fa-check"></i><b>72.29</b> Part 9: Management Recommendations</a></li>
<li class="chapter" data-level="72.30" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#actionable-outputs"><i class="fa fa-check"></i><b>72.30</b> Actionable outputs</a></li>
<li class="chapter" data-level="72.31" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-10-reporting-3"><i class="fa fa-check"></i><b>72.31</b> Part 10: Reporting</a></li>
<li class="chapter" data-level="72.32" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#sample-methods-and-results-19"><i class="fa fa-check"></i><b>72.32</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="72.32.1" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#methods-11"><i class="fa fa-check"></i><b>72.32.1</b> Methods</a></li>
<li class="chapter" data-level="72.32.2" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#results-12"><i class="fa fa-check"></i><b>72.32.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="72.33" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#key-takeaways-22"><i class="fa fa-check"></i><b>72.33</b> Key takeaways</a></li>
<li class="chapter" data-level="72.34" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#assignment-25"><i class="fa fa-check"></i><b>72.34</b> Assignment</a>
<ul>
<li class="chapter" data-level="72.34.1" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-1-conceptual-questions-16"><i class="fa fa-check"></i><b>72.34.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="72.34.2" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-2-applied-analysis"><i class="fa fa-check"></i><b>72.34.2</b> Part 2: Applied analysis</a></li>
<li class="chapter" data-level="72.34.3" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-3-temporal-analysis"><i class="fa fa-check"></i><b>72.34.3</b> Part 3: Temporal analysis</a></li>
<li class="chapter" data-level="72.34.4" data-path="spatial-disease-ecology-patterns-spread-and-risk.html"><a href="spatial-disease-ecology-patterns-spread-and-risk.html#part-4-report"><i class="fa fa-check"></i><b>72.34.4</b> Part 4: Report</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="73" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html"><i class="fa fa-check"></i><b>73</b> Machine Learning for Ecology</a>
<ul>
<li class="chapter" data-level="73.1" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#why-machine-learning-in-ecology"><i class="fa fa-check"></i><b>73.1</b> Why machine learning in ecology?</a></li>
<li class="chapter" data-level="73.2" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#setup-22"><i class="fa fa-check"></i><b>73.2</b> Setup</a></li>
<li class="chapter" data-level="73.3" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-1-supervised-vs-unsupervised-learning"><i class="fa fa-check"></i><b>73.3</b> Part 1: Supervised vs Unsupervised Learning</a></li>
<li class="chapter" data-level="73.4" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#supervised-learning"><i class="fa fa-check"></i><b>73.4</b> Supervised learning</a></li>
<li class="chapter" data-level="73.5" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#unsupervised-learning"><i class="fa fa-check"></i><b>73.5</b> Unsupervised learning</a></li>
<li class="chapter" data-level="73.6" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-2-the-traintest-paradigm"><i class="fa fa-check"></i><b>73.6</b> Part 2: The Train/Test Paradigm</a></li>
<li class="chapter" data-level="73.7" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#why-we-split-data"><i class="fa fa-check"></i><b>73.7</b> Why we split data</a></li>
<li class="chapter" data-level="73.8" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#example-species-distribution-data"><i class="fa fa-check"></i><b>73.8</b> Example: Species distribution data</a></li>
<li class="chapter" data-level="73.9" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#creating-traintest-split"><i class="fa fa-check"></i><b>73.9</b> Creating train/test split</a></li>
<li class="chapter" data-level="73.10" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#cross-validation"><i class="fa fa-check"></i><b>73.10</b> Cross-validation</a></li>
<li class="chapter" data-level="73.11" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-3-regularized-regression"><i class="fa fa-check"></i><b>73.11</b> Part 3: Regularized Regression</a></li>
<li class="chapter" data-level="73.12" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#the-problem-with-many-predictors"><i class="fa fa-check"></i><b>73.12</b> The problem with many predictors</a></li>
<li class="chapter" data-level="73.13" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#lasso-ridge-and-elastic-net"><i class="fa fa-check"></i><b>73.13</b> LASSO, Ridge, and Elastic Net</a></li>
<li class="chapter" data-level="73.14" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#lasso-for-variable-selection"><i class="fa fa-check"></i><b>73.14</b> LASSO for variable selection</a></li>
<li class="chapter" data-level="73.15" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#comparing-regularization-approaches"><i class="fa fa-check"></i><b>73.15</b> Comparing regularization approaches</a></li>
<li class="chapter" data-level="73.16" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-4-random-forests"><i class="fa fa-check"></i><b>73.16</b> Part 4: Random Forests</a></li>
<li class="chapter" data-level="73.17" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#the-workhorse-of-ecological-ml"><i class="fa fa-check"></i><b>73.17</b> The workhorse of ecological ML</a></li>
<li class="chapter" data-level="73.18" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#why-random-forests-work-well-for-ecology"><i class="fa fa-check"></i><b>73.18</b> Why random forests work well for ecology</a></li>
<li class="chapter" data-level="73.19" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#fitting-a-random-forest"><i class="fa fa-check"></i><b>73.19</b> Fitting a random forest</a></li>
<li class="chapter" data-level="73.20" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#variable-importance"><i class="fa fa-check"></i><b>73.20</b> Variable importance</a></li>
<li class="chapter" data-level="73.21" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#partial-dependence-plots"><i class="fa fa-check"></i><b>73.21</b> Partial dependence plots</a></li>
<li class="chapter" data-level="73.22" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#evaluating-the-model"><i class="fa fa-check"></i><b>73.22</b> Evaluating the model</a></li>
<li class="chapter" data-level="73.23" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#roc-curve"><i class="fa fa-check"></i><b>73.23</b> ROC curve</a></li>
<li class="chapter" data-level="73.24" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-5-gradient-boosting-xgboost"><i class="fa fa-check"></i><b>73.24</b> Part 5: Gradient Boosting (XGBoost)</a></li>
<li class="chapter" data-level="73.25" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#sequential-learning"><i class="fa fa-check"></i><b>73.25</b> Sequential learning</a></li>
<li class="chapter" data-level="73.26" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#fitting-xgboost"><i class="fa fa-check"></i><b>73.26</b> Fitting XGBoost</a></li>
<li class="chapter" data-level="73.27" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#xgboost-importance"><i class="fa fa-check"></i><b>73.27</b> XGBoost importance</a></li>
<li class="chapter" data-level="73.28" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#comparing-models-2"><i class="fa fa-check"></i><b>73.28</b> Comparing models</a></li>
<li class="chapter" data-level="73.29" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-6-unsupervised-learning"><i class="fa fa-check"></i><b>73.29</b> Part 6: Unsupervised Learning</a></li>
<li class="chapter" data-level="73.30" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#k-means-clustering"><i class="fa fa-check"></i><b>73.30</b> K-means clustering</a></li>
<li class="chapter" data-level="73.31" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#hierarchical-clustering"><i class="fa fa-check"></i><b>73.31</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="73.32" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#dimensionality-reduction"><i class="fa fa-check"></i><b>73.32</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="73.33" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-7-using-tidymodels-for-ml-workflows"><i class="fa fa-check"></i><b>73.33</b> Part 7: Using tidymodels for ML Workflows</a></li>
<li class="chapter" data-level="73.34" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#the-tidymodels-framework"><i class="fa fa-check"></i><b>73.34</b> The tidymodels framework</a></li>
<li class="chapter" data-level="73.35" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#hyperparameter-tuning"><i class="fa fa-check"></i><b>73.35</b> Hyperparameter tuning</a></li>
<li class="chapter" data-level="73.36" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-8-case-study-species-distribution-model"><i class="fa fa-check"></i><b>73.36</b> Part 8: Case Study — Species Distribution Model</a></li>
<li class="chapter" data-level="73.37" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-9-practical-challenges"><i class="fa fa-check"></i><b>73.37</b> Part 9: Practical Challenges</a></li>
<li class="chapter" data-level="73.38" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#small-sample-sizes"><i class="fa fa-check"></i><b>73.38</b> Small sample sizes</a></li>
<li class="chapter" data-level="73.39" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#spatial-autocorrelation"><i class="fa fa-check"></i><b>73.39</b> Spatial autocorrelation</a></li>
<li class="chapter" data-level="73.40" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#class-imbalance"><i class="fa fa-check"></i><b>73.40</b> Class imbalance</a></li>
<li class="chapter" data-level="73.41" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#extrapolation"><i class="fa fa-check"></i><b>73.41</b> Extrapolation</a></li>
<li class="chapter" data-level="73.42" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-10-interpretation-vs-prediction"><i class="fa fa-check"></i><b>73.42</b> Part 10: Interpretation vs Prediction</a></li>
<li class="chapter" data-level="73.43" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#the-black-box-concern"><i class="fa fa-check"></i><b>73.43</b> The black box concern</a></li>
<li class="chapter" data-level="73.44" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#important-caveat"><i class="fa fa-check"></i><b>73.44</b> Important caveat</a></li>
<li class="chapter" data-level="73.45" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-11-reporting-ml-results"><i class="fa fa-check"></i><b>73.45</b> Part 11: Reporting ML Results</a></li>
<li class="chapter" data-level="73.46" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#what-to-report-12"><i class="fa fa-check"></i><b>73.46</b> What to report</a></li>
<li class="chapter" data-level="73.47" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#sample-methods-and-results-20"><i class="fa fa-check"></i><b>73.47</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="73.47.1" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#methods-12"><i class="fa fa-check"></i><b>73.47.1</b> Methods</a></li>
<li class="chapter" data-level="73.47.2" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#results-13"><i class="fa fa-check"></i><b>73.47.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="73.48" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#key-takeaways-23"><i class="fa fa-check"></i><b>73.48</b> Key takeaways</a></li>
<li class="chapter" data-level="73.49" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#assignment-26"><i class="fa fa-check"></i><b>73.49</b> Assignment</a>
<ul>
<li class="chapter" data-level="73.49.1" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-1-conceptual-questions-17"><i class="fa fa-check"></i><b>73.49.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="73.49.2" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-2-model-building"><i class="fa fa-check"></i><b>73.49.2</b> Part 2: Model building</a></li>
<li class="chapter" data-level="73.49.3" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-3-sdm-workflow"><i class="fa fa-check"></i><b>73.49.3</b> Part 3: SDM workflow</a></li>
<li class="chapter" data-level="73.49.4" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-4-cross-validation"><i class="fa fa-check"></i><b>73.49.4</b> Part 4: Cross-validation</a></li>
<li class="chapter" data-level="73.49.5" data-path="machine-learning-for-ecology.html"><a href="machine-learning-for-ecology.html#part-5-reflection-9"><i class="fa fa-check"></i><b>73.49.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="74" data-path="species-distribution-modeling.html"><a href="species-distribution-modeling.html"><i class="fa fa-check"></i><b>74</b> Species Distribution Modeling</a>
<ul>
<li class="chapter" data-level="74.1" data-path="species-distribution-modeling.html"><a href="species-distribution-modeling.html#introduction-1"><i class="fa fa-check"></i><b>74.1</b> Introduction</a></li>
<li class="chapter" data-level="74.2" data-path="species-distribution-modeling.html"><a href="species-distribution-modeling.html#a-brief-review-of-niche-theory"><i class="fa fa-check"></i><b>74.2</b> A brief review of niche theory</a></li>
<li class="chapter" data-level="74.3" data-path="species-distribution-modeling.html"><a href="species-distribution-modeling.html#downloads-for-this-lab"><i class="fa fa-check"></i><b>74.3</b> Downloads for this lab</a></li>
<li class="chapter" data-level="74.4" data-path="species-distribution-modeling.html"><a href="species-distribution-modeling.html#objectives"><i class="fa fa-check"></i><b>74.4</b> Objectives</a></li>
<li class="chapter" data-level="74.5" data-path="species-distribution-modeling.html"><a href="species-distribution-modeling.html#methods-13"><i class="fa fa-check"></i><b>74.5</b> Methods</a>
<ul>
<li class="chapter" data-level="74.5.1" data-path="species-distribution-modeling.html"><a href="species-distribution-modeling.html#lets-get-modeling"><i class="fa fa-check"></i><b>74.5.1</b> Let’s get modeling</a></li>
<li class="chapter" data-level="74.5.2" data-path="species-distribution-modeling.html"><a href="species-distribution-modeling.html#run-the-r-code"><i class="fa fa-check"></i><b>74.5.2</b> Run the R code</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="75" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html"><i class="fa fa-check"></i><b>75</b> Occupancy Models and Detectability</a>
<ul>
<li class="chapter" data-level="75.1" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#the-detection-problem"><i class="fa fa-check"></i><b>75.1</b> The detection problem</a></li>
<li class="chapter" data-level="75.2" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#when-to-use-occupancy-models"><i class="fa fa-check"></i><b>75.2</b> When to use occupancy models</a></li>
<li class="chapter" data-level="75.3" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#setup-23"><i class="fa fa-check"></i><b>75.3</b> Setup</a></li>
<li class="chapter" data-level="75.4" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-1-single-season-occupancy-models"><i class="fa fa-check"></i><b>75.4</b> Part 1: Single-Season Occupancy Models</a></li>
<li class="chapter" data-level="75.5" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#the-basic-model"><i class="fa fa-check"></i><b>75.5</b> The basic model</a></li>
<li class="chapter" data-level="75.6" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#simulating-occupancy-data"><i class="fa fa-check"></i><b>75.6</b> Simulating occupancy data</a></li>
<li class="chapter" data-level="75.7" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#the-detection-history-matrix"><i class="fa fa-check"></i><b>75.7</b> The detection history matrix</a></li>
<li class="chapter" data-level="75.8" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#fitting-occupancy-models-with-unmarked"><i class="fa fa-check"></i><b>75.8</b> Fitting occupancy models with unmarked</a></li>
<li class="chapter" data-level="75.9" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#interpreting-output"><i class="fa fa-check"></i><b>75.9</b> Interpreting output</a></li>
<li class="chapter" data-level="75.10" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-2-covariates-on-detection-and-occupancy"><i class="fa fa-check"></i><b>75.10</b> Part 2: Covariates on Detection and Occupancy</a></li>
<li class="chapter" data-level="75.11" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#why-covariates-matter"><i class="fa fa-check"></i><b>75.11</b> Why covariates matter</a></li>
<li class="chapter" data-level="75.12" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#adding-site-covariates"><i class="fa fa-check"></i><b>75.12</b> Adding site covariates</a></li>
<li class="chapter" data-level="75.13" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#adding-observation-covariates"><i class="fa fa-check"></i><b>75.13</b> Adding observation covariates</a></li>
<li class="chapter" data-level="75.14" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#model-selection-1"><i class="fa fa-check"></i><b>75.14</b> Model selection</a></li>
<li class="chapter" data-level="75.15" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-3-predictions-and-visualization"><i class="fa fa-check"></i><b>75.15</b> Part 3: Predictions and Visualization</a></li>
<li class="chapter" data-level="75.16" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#predicting-occupancy"><i class="fa fa-check"></i><b>75.16</b> Predicting occupancy</a></li>
<li class="chapter" data-level="75.17" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#mapping-occupancy"><i class="fa fa-check"></i><b>75.17</b> Mapping occupancy</a></li>
<li class="chapter" data-level="75.18" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-4-model-goodness-of-fit"><i class="fa fa-check"></i><b>75.18</b> Part 4: Model Goodness-of-Fit</a></li>
<li class="chapter" data-level="75.19" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#checking-model-fit"><i class="fa fa-check"></i><b>75.19</b> Checking model fit</a></li>
<li class="chapter" data-level="75.20" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#assessing-detection-probability"><i class="fa fa-check"></i><b>75.20</b> Assessing detection probability</a></li>
<li class="chapter" data-level="75.21" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-5-multi-season-dynamic-occupancy-models"><i class="fa fa-check"></i><b>75.21</b> Part 5: Multi-Season (Dynamic) Occupancy Models</a></li>
<li class="chapter" data-level="75.22" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#why-multi-season-models"><i class="fa fa-check"></i><b>75.22</b> Why multi-season models?</a></li>
<li class="chapter" data-level="75.23" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#simulating-multi-season-data"><i class="fa fa-check"></i><b>75.23</b> Simulating multi-season data</a></li>
<li class="chapter" data-level="75.24" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#fitting-multi-season-models"><i class="fa fa-check"></i><b>75.24</b> Fitting multi-season models</a></li>
<li class="chapter" data-level="75.25" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#derived-parameters"><i class="fa fa-check"></i><b>75.25</b> Derived parameters</a></li>
<li class="chapter" data-level="75.26" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-6-n-mixture-models-for-abundance"><i class="fa fa-check"></i><b>75.26</b> Part 6: N-Mixture Models for Abundance</a></li>
<li class="chapter" data-level="75.27" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#beyond-presenceabsence"><i class="fa fa-check"></i><b>75.27</b> Beyond presence/absence</a></li>
<li class="chapter" data-level="75.28" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#simulating-count-data"><i class="fa fa-check"></i><b>75.28</b> Simulating count data</a></li>
<li class="chapter" data-level="75.29" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#fitting-n-mixture-models"><i class="fa fa-check"></i><b>75.29</b> Fitting N-mixture models</a></li>
<li class="chapter" data-level="75.30" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#estimating-site-specific-abundance"><i class="fa fa-check"></i><b>75.30</b> Estimating site-specific abundance</a></li>
<li class="chapter" data-level="75.31" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-7-survey-design-considerations"><i class="fa fa-check"></i><b>75.31</b> Part 7: Survey Design Considerations</a></li>
<li class="chapter" data-level="75.32" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#how-many-surveys"><i class="fa fa-check"></i><b>75.32</b> How many surveys?</a></li>
<li class="chapter" data-level="75.33" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#sites-vs-surveys-trade-off"><i class="fa fa-check"></i><b>75.33</b> Sites vs surveys trade-off</a></li>
<li class="chapter" data-level="75.34" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-8-reporting-occupancy-models"><i class="fa fa-check"></i><b>75.34</b> Part 8: Reporting Occupancy Models</a></li>
<li class="chapter" data-level="75.35" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#what-to-report-13"><i class="fa fa-check"></i><b>75.35</b> What to report</a></li>
<li class="chapter" data-level="75.36" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#sample-methods-and-results-21"><i class="fa fa-check"></i><b>75.36</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="75.36.1" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#methods-14"><i class="fa fa-check"></i><b>75.36.1</b> Methods</a></li>
<li class="chapter" data-level="75.36.2" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#results-14"><i class="fa fa-check"></i><b>75.36.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="75.37" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#key-takeaways-24"><i class="fa fa-check"></i><b>75.37</b> Key takeaways</a></li>
<li class="chapter" data-level="75.38" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#assignment-27"><i class="fa fa-check"></i><b>75.38</b> Assignment</a>
<ul>
<li class="chapter" data-level="75.38.1" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-1-conceptual-questions-18"><i class="fa fa-check"></i><b>75.38.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="75.38.2" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-2-single-season-occupancy"><i class="fa fa-check"></i><b>75.38.2</b> Part 2: Single-season occupancy</a></li>
<li class="chapter" data-level="75.38.3" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-3-multi-season-model"><i class="fa fa-check"></i><b>75.38.3</b> Part 3: Multi-season model</a></li>
<li class="chapter" data-level="75.38.4" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-4-n-mixture-model"><i class="fa fa-check"></i><b>75.38.4</b> Part 4: N-mixture model</a></li>
<li class="chapter" data-level="75.38.5" data-path="occupancy-models-and-detectability.html"><a href="occupancy-models-and-detectability.html#part-5-reporting"><i class="fa fa-check"></i><b>75.38.5</b> Part 5: Reporting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="76" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html"><i class="fa fa-check"></i><b>76</b> Structural Equation Modeling I: Foundations and Path Analysis</a>
<ul>
<li class="chapter" data-level="76.1" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#when-should-you-use-sem"><i class="fa fa-check"></i><b>76.1</b> When should you use SEM?</a></li>
<li class="chapter" data-level="76.2" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#how-is-sem-different-from-multiple-regression"><i class="fa fa-check"></i><b>76.2</b> How is SEM different from multiple regression?</a></li>
<li class="chapter" data-level="76.3" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#setup-24"><i class="fa fa-check"></i><b>76.3</b> Setup</a></li>
<li class="chapter" data-level="76.4" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-1-causal-thinking-and-dags"><i class="fa fa-check"></i><b>76.4</b> Part 1: Causal Thinking and DAGs</a></li>
<li class="chapter" data-level="76.5" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#what-is-causality"><i class="fa fa-check"></i><b>76.5</b> What is causality?</a></li>
<li class="chapter" data-level="76.6" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#directed-acyclic-graphs-dags"><i class="fa fa-check"></i><b>76.6</b> Directed Acyclic Graphs (DAGs)</a></li>
<li class="chapter" data-level="76.7" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#three-fundamental-structures"><i class="fa fa-check"></i><b>76.7</b> Three fundamental structures</a>
<ul>
<li class="chapter" data-level="76.7.1" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#chains-mediation"><i class="fa fa-check"></i><b>76.7.1</b> 1. Chains (Mediation)</a></li>
<li class="chapter" data-level="76.7.2" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#forks-confounding"><i class="fa fa-check"></i><b>76.7.2</b> 2. Forks (Confounding)</a></li>
<li class="chapter" data-level="76.7.3" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#colliders"><i class="fa fa-check"></i><b>76.7.3</b> 3. Colliders</a></li>
</ul></li>
<li class="chapter" data-level="76.8" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#summary-what-to-control-for"><i class="fa fa-check"></i><b>76.8</b> Summary: What to control for</a></li>
<li class="chapter" data-level="76.9" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-2-building-a-conceptual-model"><i class="fa fa-check"></i><b>76.9</b> Part 2: Building a Conceptual Model</a></li>
<li class="chapter" data-level="76.10" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#meta-modeling-your-system"><i class="fa fa-check"></i><b>76.10</b> Meta-modeling your system</a>
<ul>
<li class="chapter" data-level="76.10.1" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#define-your-purpose"><i class="fa fa-check"></i><b>76.10.1</b> 1. Define your purpose</a></li>
<li class="chapter" data-level="76.10.2" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#identify-variable-roles"><i class="fa fa-check"></i><b>76.10.2</b> 2. Identify variable roles</a></li>
<li class="chapter" data-level="76.10.3" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#determine-scope"><i class="fa fa-check"></i><b>76.10.3</b> 3. Determine scope</a></li>
</ul></li>
<li class="chapter" data-level="76.11" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#case-study-pollinator-response-to-vegetation-management"><i class="fa fa-check"></i><b>76.11</b> Case study: Pollinator response to vegetation management</a></li>
<li class="chapter" data-level="76.12" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#finding-adjustment-sets"><i class="fa fa-check"></i><b>76.12</b> Finding adjustment sets</a></li>
<li class="chapter" data-level="76.13" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-3-path-analysis-with-lavaan"><i class="fa fa-check"></i><b>76.13</b> Part 3: Path Analysis with lavaan</a></li>
<li class="chapter" data-level="76.14" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#example-data-post-fire-plant-recovery"><i class="fa fa-check"></i><b>76.14</b> Example data: Post-fire plant recovery</a></li>
<li class="chapter" data-level="76.15" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#the-lavaan-package"><i class="fa fa-check"></i><b>76.15</b> The lavaan package</a></li>
<li class="chapter" data-level="76.16" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#step-1-start-simple"><i class="fa fa-check"></i><b>76.16</b> Step 1: Start simple</a></li>
<li class="chapter" data-level="76.17" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#step-2-visualize-the-model"><i class="fa fa-check"></i><b>76.17</b> Step 2: Visualize the model</a></li>
<li class="chapter" data-level="76.18" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#step-3-add-mediation"><i class="fa fa-check"></i><b>76.18</b> Step 3: Add mediation</a></li>
<li class="chapter" data-level="76.19" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#step-4-calculate-indirect-effects"><i class="fa fa-check"></i><b>76.19</b> Step 4: Calculate indirect effects</a></li>
<li class="chapter" data-level="76.20" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-4-model-fit-assessment"><i class="fa fa-check"></i><b>76.20</b> Part 4: Model Fit Assessment</a></li>
<li class="chapter" data-level="76.21" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#why-assess-fit"><i class="fa fa-check"></i><b>76.21</b> Why assess fit?</a></li>
<li class="chapter" data-level="76.22" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#key-fit-indices"><i class="fa fa-check"></i><b>76.22</b> Key fit indices</a></li>
<li class="chapter" data-level="76.23" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#modification-indices"><i class="fa fa-check"></i><b>76.23</b> Modification indices</a></li>
<li class="chapter" data-level="76.24" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-5-a-complete-example"><i class="fa fa-check"></i><b>76.24</b> Part 5: A Complete Example</a></li>
<li class="chapter" data-level="76.25" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#interpreting-the-results-1"><i class="fa fa-check"></i><b>76.25</b> Interpreting the results</a></li>
<li class="chapter" data-level="76.26" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-6-comparing-models"><i class="fa fa-check"></i><b>76.26</b> Part 6: Comparing Models</a></li>
<li class="chapter" data-level="76.27" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#nested-models-likelihood-ratio-test"><i class="fa fa-check"></i><b>76.27</b> Nested models: Likelihood Ratio Test</a></li>
<li class="chapter" data-level="76.28" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#non-nested-models-aicbic"><i class="fa fa-check"></i><b>76.28</b> Non-nested models: AIC/BIC</a></li>
<li class="chapter" data-level="76.29" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-7-assumptions-and-diagnostics"><i class="fa fa-check"></i><b>76.29</b> Part 7: Assumptions and Diagnostics</a></li>
<li class="chapter" data-level="76.30" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#key-assumptions"><i class="fa fa-check"></i><b>76.30</b> Key assumptions</a></li>
<li class="chapter" data-level="76.31" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#checking-multivariate-normality"><i class="fa fa-check"></i><b>76.31</b> Checking multivariate normality</a></li>
<li class="chapter" data-level="76.32" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#sample-size-guidelines"><i class="fa fa-check"></i><b>76.32</b> Sample size guidelines</a></li>
<li class="chapter" data-level="76.33" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#robust-estimation"><i class="fa fa-check"></i><b>76.33</b> Robust estimation</a></li>
<li class="chapter" data-level="76.34" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-8-reporting-sem-results"><i class="fa fa-check"></i><b>76.34</b> Part 8: Reporting SEM Results</a></li>
<li class="chapter" data-level="76.35" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#what-to-report-14"><i class="fa fa-check"></i><b>76.35</b> What to report</a></li>
<li class="chapter" data-level="76.36" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#sample-methods-and-results-22"><i class="fa fa-check"></i><b>76.36</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="76.36.1" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#methods-15"><i class="fa fa-check"></i><b>76.36.1</b> Methods</a></li>
<li class="chapter" data-level="76.36.2" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#results-15"><i class="fa fa-check"></i><b>76.36.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="76.37" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#key-takeaways-25"><i class="fa fa-check"></i><b>76.37</b> Key takeaways</a></li>
<li class="chapter" data-level="76.38" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#assignment-28"><i class="fa fa-check"></i><b>76.38</b> Assignment</a>
<ul>
<li class="chapter" data-level="76.38.1" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-1-conceptual-questions-19"><i class="fa fa-check"></i><b>76.38.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="76.38.2" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-2-build-a-dag"><i class="fa fa-check"></i><b>76.38.2</b> Part 2: Build a DAG</a></li>
<li class="chapter" data-level="76.38.3" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-3-fit-a-path-model"><i class="fa fa-check"></i><b>76.38.3</b> Part 3: Fit a path model</a></li>
<li class="chapter" data-level="76.38.4" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-4-model-comparison-1"><i class="fa fa-check"></i><b>76.38.4</b> Part 4: Model comparison</a></li>
<li class="chapter" data-level="76.38.5" data-path="structural-equation-modeling-i-foundations-and-path-analysis.html"><a href="structural-equation-modeling-i-foundations-and-path-analysis.html#part-5-reporting-1"><i class="fa fa-check"></i><b>76.38.5</b> Part 5: Reporting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="77" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><i class="fa fa-check"></i><b>77</b> Structural Equation Modeling II: Latent Variables and Measurement Models</a>
<ul>
<li class="chapter" data-level="77.1" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#what-is-a-latent-variable"><i class="fa fa-check"></i><b>77.1</b> What is a latent variable?</a></li>
<li class="chapter" data-level="77.2" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#why-use-latent-variables"><i class="fa fa-check"></i><b>77.2</b> Why use latent variables?</a></li>
<li class="chapter" data-level="77.3" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#setup-25"><i class="fa fa-check"></i><b>77.3</b> Setup</a></li>
<li class="chapter" data-level="77.4" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-1-confirmatory-factor-analysis-cfa"><i class="fa fa-check"></i><b>77.4</b> Part 1: Confirmatory Factor Analysis (CFA)</a></li>
<li class="chapter" data-level="77.5" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#what-is-cfa"><i class="fa fa-check"></i><b>77.5</b> What is CFA?</a></li>
<li class="chapter" data-level="77.6" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#cfa-syntax-in-lavaan"><i class="fa fa-check"></i><b>77.6</b> CFA syntax in lavaan</a></li>
<li class="chapter" data-level="77.7" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#example-disturbance-pressure"><i class="fa fa-check"></i><b>77.7</b> Example: Disturbance pressure</a></li>
<li class="chapter" data-level="77.8" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#fitting-a-cfa-model"><i class="fa fa-check"></i><b>77.8</b> Fitting a CFA model</a></li>
<li class="chapter" data-level="77.9" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#interpreting-cfa-output"><i class="fa fa-check"></i><b>77.9</b> Interpreting CFA output</a></li>
<li class="chapter" data-level="77.10" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#identification-in-cfa"><i class="fa fa-check"></i><b>77.10</b> Identification in CFA</a></li>
<li class="chapter" data-level="77.11" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-2-multiple-latent-factors"><i class="fa fa-check"></i><b>77.11</b> Part 2: Multiple Latent Factors</a></li>
<li class="chapter" data-level="77.12" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#when-constructs-are-multidimensional"><i class="fa fa-check"></i><b>77.12</b> When constructs are multidimensional</a></li>
<li class="chapter" data-level="77.13" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#fitting-a-two-factor-cfa"><i class="fa fa-check"></i><b>77.13</b> Fitting a two-factor CFA</a></li>
<li class="chapter" data-level="77.14" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#factor-correlations"><i class="fa fa-check"></i><b>77.14</b> Factor correlations</a></li>
<li class="chapter" data-level="77.15" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-3-full-structural-models"><i class="fa fa-check"></i><b>77.15</b> Part 3: Full Structural Models</a></li>
<li class="chapter" data-level="77.16" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#combining-measurement-and-structural-models"><i class="fa fa-check"></i><b>77.16</b> Combining measurement and structural models</a></li>
<li class="chapter" data-level="77.17" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#example-disturbance-effects-on-biodiversity"><i class="fa fa-check"></i><b>77.17</b> Example: Disturbance effects on biodiversity</a></li>
<li class="chapter" data-level="77.18" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#extracting-results"><i class="fa fa-check"></i><b>77.18</b> Extracting results</a></li>
<li class="chapter" data-level="77.19" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-4-measurement-error-in-predictors"><i class="fa fa-check"></i><b>77.19</b> Part 4: Measurement Error in Predictors</a></li>
<li class="chapter" data-level="77.20" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#why-measurement-error-matters"><i class="fa fa-check"></i><b>77.20</b> Why measurement error matters</a></li>
<li class="chapter" data-level="77.21" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#modeling-measurement-error"><i class="fa fa-check"></i><b>77.21</b> Modeling measurement error</a></li>
<li class="chapter" data-level="77.22" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#comparing-models-with-and-without-error-correction"><i class="fa fa-check"></i><b>77.22</b> Comparing models with and without error correction</a></li>
<li class="chapter" data-level="77.23" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-5-model-identification-deep-dive"><i class="fa fa-check"></i><b>77.23</b> Part 5: Model Identification Deep Dive</a></li>
<li class="chapter" data-level="77.24" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#the-identification-problem"><i class="fa fa-check"></i><b>77.24</b> The identification problem</a></li>
<li class="chapter" data-level="77.25" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#key-identification-rules"><i class="fa fa-check"></i><b>77.25</b> Key identification rules</a></li>
<li class="chapter" data-level="77.26" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#checking-identification"><i class="fa fa-check"></i><b>77.26</b> Checking identification</a></li>
<li class="chapter" data-level="77.27" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-6-advanced-model-assessment"><i class="fa fa-check"></i><b>77.27</b> Part 6: Advanced Model Assessment</a></li>
<li class="chapter" data-level="77.28" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#comparing-nested-cfa-models"><i class="fa fa-check"></i><b>77.28</b> Comparing nested CFA models</a></li>
<li class="chapter" data-level="77.29" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#reliability-measures"><i class="fa fa-check"></i><b>77.29</b> Reliability measures</a></li>
<li class="chapter" data-level="77.30" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#average-variance-extracted-ave"><i class="fa fa-check"></i><b>77.30</b> Average Variance Extracted (AVE)</a></li>
<li class="chapter" data-level="77.31" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-7-reporting-latent-variable-models"><i class="fa fa-check"></i><b>77.31</b> Part 7: Reporting Latent Variable Models</a></li>
<li class="chapter" data-level="77.32" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#what-to-report-for-cfa"><i class="fa fa-check"></i><b>77.32</b> What to report for CFA</a></li>
<li class="chapter" data-level="77.33" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#what-to-report-for-full-sem"><i class="fa fa-check"></i><b>77.33</b> What to report for full SEM</a></li>
<li class="chapter" data-level="77.34" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#sample-methods-and-results-23"><i class="fa fa-check"></i><b>77.34</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="77.34.1" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#methods-16"><i class="fa fa-check"></i><b>77.34.1</b> Methods</a></li>
<li class="chapter" data-level="77.34.2" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#results-16"><i class="fa fa-check"></i><b>77.34.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="77.35" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#key-takeaways-26"><i class="fa fa-check"></i><b>77.35</b> Key takeaways</a></li>
<li class="chapter" data-level="77.36" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#assignment-29"><i class="fa fa-check"></i><b>77.36</b> Assignment</a>
<ul>
<li class="chapter" data-level="77.36.1" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-1-conceptual-questions-20"><i class="fa fa-check"></i><b>77.36.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="77.36.2" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-2-build-a-cfa"><i class="fa fa-check"></i><b>77.36.2</b> Part 2: Build a CFA</a></li>
<li class="chapter" data-level="77.36.3" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-3-two-factor-cfa"><i class="fa fa-check"></i><b>77.36.3</b> Part 3: Two-factor CFA</a></li>
<li class="chapter" data-level="77.36.4" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-4-full-sem"><i class="fa fa-check"></i><b>77.36.4</b> Part 4: Full SEM</a></li>
<li class="chapter" data-level="77.36.5" data-path="structural-equation-modeling-ii-latent-variables-and-measurement-models.html"><a href="structural-equation-modeling-ii-latent-variables-and-measurement-models.html#part-5-reflection-10"><i class="fa fa-check"></i><b>77.36.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="78" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html"><i class="fa fa-check"></i><b>78</b> Structural Equation Modeling III: Piecewise SEM</a>
<ul>
<li class="chapter" data-level="78.1" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#when-to-use-piecewise-vs-covariance-based-sem"><i class="fa fa-check"></i><b>78.1</b> When to use piecewise vs covariance-based SEM</a></li>
<li class="chapter" data-level="78.2" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#setup-26"><i class="fa fa-check"></i><b>78.2</b> Setup</a></li>
<li class="chapter" data-level="78.3" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-1-d-separation-and-model-fit"><i class="fa fa-check"></i><b>78.3</b> Part 1: D-Separation and Model Fit</a></li>
<li class="chapter" data-level="78.4" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#the-logic-of-piecewise-sem"><i class="fa fa-check"></i><b>78.4</b> The logic of piecewise SEM</a></li>
<li class="chapter" data-level="78.5" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#what-is-d-separation"><i class="fa fa-check"></i><b>78.5</b> What is D-separation?</a></li>
<li class="chapter" data-level="78.6" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#example-testing-d-separation"><i class="fa fa-check"></i><b>78.6</b> Example: Testing D-separation</a></li>
<li class="chapter" data-level="78.7" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#fitting-piecewise-sem"><i class="fa fa-check"></i><b>78.7</b> Fitting piecewise SEM</a></li>
<li class="chapter" data-level="78.8" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#understanding-the-output-1"><i class="fa fa-check"></i><b>78.8</b> Understanding the output</a></li>
<li class="chapter" data-level="78.9" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#detecting-missing-paths"><i class="fa fa-check"></i><b>78.9</b> Detecting missing paths</a></li>
<li class="chapter" data-level="78.10" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-2-working-with-real-ecological-data"><i class="fa fa-check"></i><b>78.10</b> Part 2: Working with Real Ecological Data</a></li>
<li class="chapter" data-level="78.11" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#example-post-fire-recovery-keeley-data"><i class="fa fa-check"></i><b>78.11</b> Example: Post-fire recovery (Keeley data)</a></li>
<li class="chapter" data-level="78.12" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#building-the-model"><i class="fa fa-check"></i><b>78.12</b> Building the model</a></li>
<li class="chapter" data-level="78.13" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#visualizing-the-model"><i class="fa fa-check"></i><b>78.13</b> Visualizing the model</a></li>
<li class="chapter" data-level="78.14" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#extracting-results-1"><i class="fa fa-check"></i><b>78.14</b> Extracting results</a></li>
<li class="chapter" data-level="78.15" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-3-glms-in-piecewise-sem"><i class="fa fa-check"></i><b>78.15</b> Part 3: GLMs in Piecewise SEM</a></li>
<li class="chapter" data-level="78.16" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#why-glms-matter"><i class="fa fa-check"></i><b>78.16</b> Why GLMs matter</a></li>
<li class="chapter" data-level="78.17" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#example-binary-response"><i class="fa fa-check"></i><b>78.17</b> Example: Binary response</a></li>
<li class="chapter" data-level="78.18" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#example-count-response"><i class="fa fa-check"></i><b>78.18</b> Example: Count response</a></li>
<li class="chapter" data-level="78.19" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#standardizing-glm-coefficients"><i class="fa fa-check"></i><b>78.19</b> Standardizing GLM coefficients</a></li>
<li class="chapter" data-level="78.20" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-4-mixed-models-in-piecewise-sem"><i class="fa fa-check"></i><b>78.20</b> Part 4: Mixed Models in Piecewise SEM</a></li>
<li class="chapter" data-level="78.21" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#handling-hierarchical-data"><i class="fa fa-check"></i><b>78.21</b> Handling hierarchical data</a></li>
<li class="chapter" data-level="78.22" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#fitting-mixed-models-in-psem"><i class="fa fa-check"></i><b>78.22</b> Fitting mixed models in psem</a></li>
<li class="chapter" data-level="78.23" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#understanding-r%C2%B2-for-mixed-models"><i class="fa fa-check"></i><b>78.23</b> Understanding R² for mixed models</a></li>
<li class="chapter" data-level="78.24" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-5-nonlinearity-and-interactions"><i class="fa fa-check"></i><b>78.24</b> Part 5: Nonlinearity and Interactions</a></li>
<li class="chapter" data-level="78.25" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#polynomial-terms"><i class="fa fa-check"></i><b>78.25</b> Polynomial terms</a></li>
<li class="chapter" data-level="78.26" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#interaction-terms"><i class="fa fa-check"></i><b>78.26</b> Interaction terms</a></li>
<li class="chapter" data-level="78.27" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-6-categorical-predictors"><i class="fa fa-check"></i><b>78.27</b> Part 6: Categorical Predictors</a></li>
<li class="chapter" data-level="78.28" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#treatment-effects-in-sem"><i class="fa fa-check"></i><b>78.28</b> Treatment effects in SEM</a></li>
<li class="chapter" data-level="78.29" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#post-hoc-comparisons-2"><i class="fa fa-check"></i><b>78.29</b> Post-hoc comparisons</a></li>
<li class="chapter" data-level="78.30" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-7-complete-ecological-example"><i class="fa fa-check"></i><b>78.30</b> Part 7: Complete Ecological Example</a></li>
<li class="chapter" data-level="78.31" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#case-study-vegetation-management-and-pollinators"><i class="fa fa-check"></i><b>78.31</b> Case study: Vegetation management and pollinators</a></li>
<li class="chapter" data-level="78.32" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#build-and-fit-the-sem"><i class="fa fa-check"></i><b>78.32</b> Build and fit the SEM</a></li>
<li class="chapter" data-level="78.33" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#extract-key-results"><i class="fa fa-check"></i><b>78.33</b> Extract key results</a></li>
<li class="chapter" data-level="78.34" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-8-reporting-piecewise-sem"><i class="fa fa-check"></i><b>78.34</b> Part 8: Reporting Piecewise SEM</a></li>
<li class="chapter" data-level="78.35" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#what-to-report-15"><i class="fa fa-check"></i><b>78.35</b> What to report</a></li>
<li class="chapter" data-level="78.36" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#sample-methods-and-results-24"><i class="fa fa-check"></i><b>78.36</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="78.36.1" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#methods-17"><i class="fa fa-check"></i><b>78.36.1</b> Methods</a></li>
<li class="chapter" data-level="78.36.2" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#results-17"><i class="fa fa-check"></i><b>78.36.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="78.37" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#key-takeaways-27"><i class="fa fa-check"></i><b>78.37</b> Key takeaways</a></li>
<li class="chapter" data-level="78.38" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#assignment-30"><i class="fa fa-check"></i><b>78.38</b> Assignment</a>
<ul>
<li class="chapter" data-level="78.38.1" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-1-conceptual-questions-21"><i class="fa fa-check"></i><b>78.38.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="78.38.2" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-2-basic-piecewise-sem"><i class="fa fa-check"></i><b>78.38.2</b> Part 2: Basic piecewise SEM</a></li>
<li class="chapter" data-level="78.38.3" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-3-glm-in-piecewise-sem"><i class="fa fa-check"></i><b>78.38.3</b> Part 3: GLM in piecewise SEM</a></li>
<li class="chapter" data-level="78.38.4" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-4-mixed-model-sem"><i class="fa fa-check"></i><b>78.38.4</b> Part 4: Mixed model SEM</a></li>
<li class="chapter" data-level="78.38.5" data-path="structural-equation-modeling-iii-piecewise-sem.html"><a href="structural-equation-modeling-iii-piecewise-sem.html#part-5-reporting-2"><i class="fa fa-check"></i><b>78.38.5</b> Part 5: Reporting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="79" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html"><i class="fa fa-check"></i><b>79</b> Causal Inference and Directed Acyclic Graphs</a>
<ul>
<li class="chapter" data-level="79.1" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#why-causal-inference-matters-in-ecology"><i class="fa fa-check"></i><b>79.1</b> Why causal inference matters in ecology</a></li>
<li class="chapter" data-level="79.2" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#setup-27"><i class="fa fa-check"></i><b>79.2</b> Setup</a></li>
<li class="chapter" data-level="79.3" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-1-from-correlation-to-causation"><i class="fa fa-check"></i><b>79.3</b> Part 1: From Correlation to Causation</a></li>
<li class="chapter" data-level="79.4" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#the-ladder-of-causation"><i class="fa fa-check"></i><b>79.4</b> The ladder of causation</a></li>
<li class="chapter" data-level="79.5" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#why-correlation-causation"><i class="fa fa-check"></i><b>79.5</b> Why correlation ≠ causation</a></li>
<li class="chapter" data-level="79.6" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-2-what-is-a-dag"><i class="fa fa-check"></i><b>79.6</b> Part 2: What is a DAG?</a></li>
<li class="chapter" data-level="79.7" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#directed-acyclic-graphs"><i class="fa fa-check"></i><b>79.7</b> Directed Acyclic Graphs</a></li>
<li class="chapter" data-level="79.8" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#reading-a-dag"><i class="fa fa-check"></i><b>79.8</b> Reading a DAG</a></li>
<li class="chapter" data-level="79.9" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#creating-dags-in-r"><i class="fa fa-check"></i><b>79.9</b> Creating DAGs in R</a></li>
<li class="chapter" data-level="79.10" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-3-the-three-fundamental-structures"><i class="fa fa-check"></i><b>79.10</b> Part 3: The Three Fundamental Structures</a></li>
<li class="chapter" data-level="79.11" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#structure-1-chains-mediation"><i class="fa fa-check"></i><b>79.11</b> Structure 1: Chains (Mediation)</a></li>
<li class="chapter" data-level="79.12" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#structure-2-forks-confounding"><i class="fa fa-check"></i><b>79.12</b> Structure 2: Forks (Confounding)</a></li>
<li class="chapter" data-level="79.13" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#structure-3-colliders"><i class="fa fa-check"></i><b>79.13</b> Structure 3: Colliders</a></li>
<li class="chapter" data-level="79.14" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#summary-the-three-structures"><i class="fa fa-check"></i><b>79.14</b> Summary: The three structures</a></li>
<li class="chapter" data-level="79.15" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-4-paths-and-d-separation"><i class="fa fa-check"></i><b>79.15</b> Part 4: Paths and D-Separation</a></li>
<li class="chapter" data-level="79.16" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#what-is-a-path"><i class="fa fa-check"></i><b>79.16</b> What is a path?</a></li>
<li class="chapter" data-level="79.17" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#when-is-a-path-blocked"><i class="fa fa-check"></i><b>79.17</b> When is a path blocked?</a></li>
<li class="chapter" data-level="79.18" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#example-multiple-paths"><i class="fa fa-check"></i><b>79.18</b> Example: Multiple paths</a></li>
<li class="chapter" data-level="79.19" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-5-the-backdoor-criterion"><i class="fa fa-check"></i><b>79.19</b> Part 5: The Backdoor Criterion</a></li>
<li class="chapter" data-level="79.20" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#causal-vs-non-causal-paths"><i class="fa fa-check"></i><b>79.20</b> Causal vs non-causal paths</a></li>
<li class="chapter" data-level="79.21" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#finding-adjustment-sets-1"><i class="fa fa-check"></i><b>79.21</b> Finding adjustment sets</a></li>
<li class="chapter" data-level="79.22" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#more-complex-example"><i class="fa fa-check"></i><b>79.22</b> More complex example</a></li>
<li class="chapter" data-level="79.23" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-6-common-pitfalls"><i class="fa fa-check"></i><b>79.23</b> Part 6: Common Pitfalls</a></li>
<li class="chapter" data-level="79.24" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#pitfall-1-controlling-for-a-mediator"><i class="fa fa-check"></i><b>79.24</b> Pitfall 1: Controlling for a mediator</a></li>
<li class="chapter" data-level="79.25" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#pitfall-2-controlling-for-a-collider"><i class="fa fa-check"></i><b>79.25</b> Pitfall 2: Controlling for a collider</a></li>
<li class="chapter" data-level="79.26" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#pitfall-3-controlling-for-a-descendant-of-a-collider"><i class="fa fa-check"></i><b>79.26</b> Pitfall 3: Controlling for a descendant of a collider</a></li>
<li class="chapter" data-level="79.27" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#pitfall-4-table-2-fallacy"><i class="fa fa-check"></i><b>79.27</b> Pitfall 4: Table 2 Fallacy</a></li>
<li class="chapter" data-level="79.28" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-7-building-your-own-dag"><i class="fa fa-check"></i><b>79.28</b> Part 7: Building Your Own DAG</a></li>
<li class="chapter" data-level="79.29" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#step-by-step-process"><i class="fa fa-check"></i><b>79.29</b> Step-by-step process</a></li>
<li class="chapter" data-level="79.30" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#example-grazing-effects-on-plant-diversity"><i class="fa fa-check"></i><b>79.30</b> Example: Grazing effects on plant diversity</a></li>
<li class="chapter" data-level="79.31" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-8-when-causal-inference-fails"><i class="fa fa-check"></i><b>79.31</b> Part 8: When Causal Inference Fails</a></li>
<li class="chapter" data-level="79.32" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#unmeasured-confounding"><i class="fa fa-check"></i><b>79.32</b> Unmeasured confounding</a></li>
<li class="chapter" data-level="79.33" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#alternatives-when-dag-assumptions-fail"><i class="fa fa-check"></i><b>79.33</b> Alternatives when DAG assumptions fail</a></li>
<li class="chapter" data-level="79.34" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-9-connecting-dags-to-analysis"><i class="fa fa-check"></i><b>79.34</b> Part 9: Connecting DAGs to Analysis</a></li>
<li class="chapter" data-level="79.35" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#dags-inform-statistical-models"><i class="fa fa-check"></i><b>79.35</b> DAGs inform statistical models</a></li>
<li class="chapter" data-level="79.36" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#example-complete-workflow"><i class="fa fa-check"></i><b>79.36</b> Example: Complete workflow</a></li>
<li class="chapter" data-level="79.37" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-10-reporting-causal-analysis"><i class="fa fa-check"></i><b>79.37</b> Part 10: Reporting Causal Analysis</a></li>
<li class="chapter" data-level="79.38" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#what-to-report-16"><i class="fa fa-check"></i><b>79.38</b> What to report</a></li>
<li class="chapter" data-level="79.39" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#sample-methods-paragraph"><i class="fa fa-check"></i><b>79.39</b> Sample methods paragraph</a></li>
<li class="chapter" data-level="79.40" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#key-takeaways-28"><i class="fa fa-check"></i><b>79.40</b> Key takeaways</a></li>
<li class="chapter" data-level="79.41" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#assignment-31"><i class="fa fa-check"></i><b>79.41</b> Assignment</a>
<ul>
<li class="chapter" data-level="79.41.1" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-1-identify-structures"><i class="fa fa-check"></i><b>79.41.1</b> Part 1: Identify structures</a></li>
<li class="chapter" data-level="79.41.2" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-2-build-a-dag-1"><i class="fa fa-check"></i><b>79.41.2</b> Part 2: Build a DAG</a></li>
<li class="chapter" data-level="79.41.3" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-3-collider-bias-simulation"><i class="fa fa-check"></i><b>79.41.3</b> Part 3: Collider bias simulation</a></li>
<li class="chapter" data-level="79.41.4" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-4-analyze-a-dag"><i class="fa fa-check"></i><b>79.41.4</b> Part 4: Analyze a DAG</a></li>
<li class="chapter" data-level="79.41.5" data-path="causal-inference-and-directed-acyclic-graphs.html"><a href="causal-inference-and-directed-acyclic-graphs.html#part-5-reflection-11"><i class="fa fa-check"></i><b>79.41.5</b> Part 5: Reflection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="80" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html"><i class="fa fa-check"></i><b>80</b> Hierarchical and Bayesian Thinking</a>
<ul>
<li class="chapter" data-level="80.1" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#why-this-matters-for-ecologists"><i class="fa fa-check"></i><b>80.1</b> Why this matters for ecologists</a></li>
<li class="chapter" data-level="80.2" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#setup-28"><i class="fa fa-check"></i><b>80.2</b> Setup</a></li>
<li class="chapter" data-level="80.3" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-1-two-philosophies-of-statistics"><i class="fa fa-check"></i><b>80.3</b> Part 1: Two Philosophies of Statistics</a></li>
<li class="chapter" data-level="80.4" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#frequentist-thinking"><i class="fa fa-check"></i><b>80.4</b> Frequentist thinking</a></li>
<li class="chapter" data-level="80.5" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#bayesian-thinking"><i class="fa fa-check"></i><b>80.5</b> Bayesian thinking</a></li>
<li class="chapter" data-level="80.6" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#a-simple-example-1"><i class="fa fa-check"></i><b>80.6</b> A simple example</a></li>
<li class="chapter" data-level="80.7" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-2-priors"><i class="fa fa-check"></i><b>80.7</b> Part 2: Priors</a></li>
<li class="chapter" data-level="80.8" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#what-are-priors"><i class="fa fa-check"></i><b>80.8</b> What are priors?</a></li>
<li class="chapter" data-level="80.9" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#choosing-priors"><i class="fa fa-check"></i><b>80.9</b> Choosing priors</a></li>
<li class="chapter" data-level="80.10" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#prior-sensitivity"><i class="fa fa-check"></i><b>80.10</b> Prior sensitivity</a></li>
<li class="chapter" data-level="80.11" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-3-hierarchical-models"><i class="fa fa-check"></i><b>80.11</b> Part 3: Hierarchical Models</a></li>
<li class="chapter" data-level="80.12" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#the-partial-pooling-insight"><i class="fa fa-check"></i><b>80.12</b> The partial pooling insight</a></li>
<li class="chapter" data-level="80.13" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#why-hierarchical-models-work"><i class="fa fa-check"></i><b>80.13</b> Why hierarchical models work</a></li>
<li class="chapter" data-level="80.14" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-4-bayesian-regression-with-brms"><i class="fa fa-check"></i><b>80.14</b> Part 4: Bayesian Regression with brms</a></li>
<li class="chapter" data-level="80.15" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#a-simple-bayesian-regression"><i class="fa fa-check"></i><b>80.15</b> A simple Bayesian regression</a></li>
<li class="chapter" data-level="80.16" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#understanding-the-output-2"><i class="fa fa-check"></i><b>80.16</b> Understanding the output</a></li>
<li class="chapter" data-level="80.17" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#visualizing-posteriors"><i class="fa fa-check"></i><b>80.17</b> Visualizing posteriors</a></li>
<li class="chapter" data-level="80.18" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#posterior-predictive-checks"><i class="fa fa-check"></i><b>80.18</b> Posterior predictive checks</a></li>
<li class="chapter" data-level="80.19" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-5-bayesian-hierarchical-models"><i class="fa fa-check"></i><b>80.19</b> Part 5: Bayesian Hierarchical Models</a></li>
<li class="chapter" data-level="80.20" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#fitting-hierarchical-models-with-brms"><i class="fa fa-check"></i><b>80.20</b> Fitting hierarchical models with brms</a></li>
<li class="chapter" data-level="80.21" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#extracting-group-level-estimates"><i class="fa fa-check"></i><b>80.21</b> Extracting group-level estimates</a></li>
<li class="chapter" data-level="80.22" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#comparing-to-frequentist"><i class="fa fa-check"></i><b>80.22</b> Comparing to frequentist</a></li>
<li class="chapter" data-level="80.23" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-6-ecological-example---species-richness"><i class="fa fa-check"></i><b>80.23</b> Part 6: Ecological Example - Species Richness</a></li>
<li class="chapter" data-level="80.24" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#a-more-realistic-model"><i class="fa fa-check"></i><b>80.24</b> A more realistic model</a></li>
<li class="chapter" data-level="80.25" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#setting-priors"><i class="fa fa-check"></i><b>80.25</b> Setting priors</a></li>
<li class="chapter" data-level="80.26" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#visualizing-effects"><i class="fa fa-check"></i><b>80.26</b> Visualizing effects</a></li>
<li class="chapter" data-level="80.27" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#model-diagnostics"><i class="fa fa-check"></i><b>80.27</b> Model diagnostics</a></li>
<li class="chapter" data-level="80.28" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-7-model-comparison"><i class="fa fa-check"></i><b>80.28</b> Part 7: Model Comparison</a></li>
<li class="chapter" data-level="80.29" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#bayesian-model-comparison"><i class="fa fa-check"></i><b>80.29</b> Bayesian model comparison</a></li>
<li class="chapter" data-level="80.30" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-8-predictions-and-uncertainty"><i class="fa fa-check"></i><b>80.30</b> Part 8: Predictions and Uncertainty</a></li>
<li class="chapter" data-level="80.31" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#posterior-predictions"><i class="fa fa-check"></i><b>80.31</b> Posterior predictions</a></li>
<li class="chapter" data-level="80.32" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#predictions-for-new-groups"><i class="fa fa-check"></i><b>80.32</b> Predictions for new groups</a></li>
<li class="chapter" data-level="80.33" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-9-when-to-go-bayesian"><i class="fa fa-check"></i><b>80.33</b> Part 9: When to Go Bayesian</a></li>
<li class="chapter" data-level="80.34" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#advantages-of-bayesian-approach"><i class="fa fa-check"></i><b>80.34</b> Advantages of Bayesian approach</a></li>
<li class="chapter" data-level="80.35" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#when-frequentist-is-fine"><i class="fa fa-check"></i><b>80.35</b> When frequentist is fine</a></li>
<li class="chapter" data-level="80.36" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#practical-considerations"><i class="fa fa-check"></i><b>80.36</b> Practical considerations</a></li>
<li class="chapter" data-level="80.37" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-10-reporting-bayesian-analyses"><i class="fa fa-check"></i><b>80.37</b> Part 10: Reporting Bayesian Analyses</a></li>
<li class="chapter" data-level="80.38" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#what-to-report-17"><i class="fa fa-check"></i><b>80.38</b> What to report</a></li>
<li class="chapter" data-level="80.39" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#sample-methods-and-results-25"><i class="fa fa-check"></i><b>80.39</b> Sample methods and results</a>
<ul>
<li class="chapter" data-level="80.39.1" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#methods-18"><i class="fa fa-check"></i><b>80.39.1</b> Methods</a></li>
<li class="chapter" data-level="80.39.2" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#results-18"><i class="fa fa-check"></i><b>80.39.2</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="80.40" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#key-takeaways-29"><i class="fa fa-check"></i><b>80.40</b> Key takeaways</a></li>
<li class="chapter" data-level="80.41" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#assignment-32"><i class="fa fa-check"></i><b>80.41</b> Assignment</a>
<ul>
<li class="chapter" data-level="80.41.1" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-1-conceptual-questions-22"><i class="fa fa-check"></i><b>80.41.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="80.41.2" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-2-prior-exploration"><i class="fa fa-check"></i><b>80.41.2</b> Part 2: Prior exploration</a></li>
<li class="chapter" data-level="80.41.3" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-3-fit-a-bayesian-hierarchical-model"><i class="fa fa-check"></i><b>80.41.3</b> Part 3: Fit a Bayesian hierarchical model</a></li>
<li class="chapter" data-level="80.41.4" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-4-model-comparison-2"><i class="fa fa-check"></i><b>80.41.4</b> Part 4: Model comparison</a></li>
<li class="chapter" data-level="80.41.5" data-path="hierarchical-and-bayesian-thinking.html"><a href="hierarchical-and-bayesian-thinking.html#part-5-reporting-3"><i class="fa fa-check"></i><b>80.41.5</b> Part 5: Reporting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="81" data-path="population-sensation.html"><a href="population-sensation.html"><i class="fa fa-check"></i><b>81</b> Population sensation</a>
<ul>
<li class="chapter" data-level="81.1" data-path="population-sensation.html"><a href="population-sensation.html#population-characteristics"><i class="fa fa-check"></i><b>81.1</b> Population characteristics</a></li>
<li class="chapter" data-level="81.2" data-path="population-sensation.html"><a href="population-sensation.html#planning-a-census"><i class="fa fa-check"></i><b>81.2</b> Planning a census</a></li>
<li class="chapter" data-level="81.3" data-path="population-sensation.html"><a href="population-sensation.html#life-cycle-diagrams"><i class="fa fa-check"></i><b>81.3</b> Life cycle diagrams</a></li>
<li class="chapter" data-level="81.4" data-path="population-sensation.html"><a href="population-sensation.html#population-demographic-models"><i class="fa fa-check"></i><b>81.4</b> Population demographic models</a>
<ul>
<li class="chapter" data-level="81.4.1" data-path="population-sensation.html"><a href="population-sensation.html#unstructured-models"><i class="fa fa-check"></i><b>81.4.1</b> Unstructured Models</a></li>
<li class="chapter" data-level="81.4.2" data-path="population-sensation.html"><a href="population-sensation.html#structured-models"><i class="fa fa-check"></i><b>81.4.2</b> Structured Models</a></li>
</ul></li>
<li class="chapter" data-level="81.5" data-path="population-sensation.html"><a href="population-sensation.html#integral-projection-models"><i class="fa fa-check"></i><b>81.5</b> Integral projection models</a></li>
<li class="chapter" data-level="81.6" data-path="population-sensation.html"><a href="population-sensation.html#population-dynamics"><i class="fa fa-check"></i><b>81.6</b> Population dynamics</a>
<ul>
<li class="chapter" data-level="81.6.1" data-path="population-sensation.html"><a href="population-sensation.html#population-growth-rate"><i class="fa fa-check"></i><b>81.6.1</b> Population growth rate</a></li>
<li class="chapter" data-level="81.6.2" data-path="population-sensation.html"><a href="population-sensation.html#stable-stage-distribution"><i class="fa fa-check"></i><b>81.6.2</b> Stable stage distribution</a></li>
<li class="chapter" data-level="81.6.3" data-path="population-sensation.html"><a href="population-sensation.html#reproductive-value"><i class="fa fa-check"></i><b>81.6.3</b> Reproductive value</a></li>
<li class="chapter" data-level="81.6.4" data-path="population-sensation.html"><a href="population-sensation.html#sensitivities-absolute-effects"><i class="fa fa-check"></i><b>81.6.4</b> Sensitivities: absolute effects</a></li>
<li class="chapter" data-level="81.6.5" data-path="population-sensation.html"><a href="population-sensation.html#elasticities-proportional-effects"><i class="fa fa-check"></i><b>81.6.5</b> Elasticities: proportional effects</a></li>
<li class="chapter" data-level="81.6.6" data-path="population-sensation.html"><a href="population-sensation.html#beyond-asymptotic-dynamics-transient-behavior"><i class="fa fa-check"></i><b>81.6.6</b> Beyond asymptotic dynamics: transient behavior</a></li>
</ul></li>
<li class="chapter" data-level="81.7" data-path="population-sensation.html"><a href="population-sensation.html#chapter-summary"><i class="fa fa-check"></i><b>81.7</b> Chapter summary</a></li>
<li class="chapter" data-level="81.8" data-path="population-sensation.html"><a href="population-sensation.html#test-your-knowledge-1"><i class="fa fa-check"></i><b>81.8</b> Test your knowledge</a></li>
<li class="chapter" data-level="81.9" data-path="population-sensation.html"><a href="population-sensation.html#assignment-33"><i class="fa fa-check"></i><b>81.9</b> Assignment</a>
<ul>
<li class="chapter" data-level="81.9.1" data-path="population-sensation.html"><a href="population-sensation.html#your-task"><i class="fa fa-check"></i><b>81.9.1</b> Your task</a></li>
<li class="chapter" data-level="81.9.2" data-path="population-sensation.html"><a href="population-sensation.html#guidelines"><i class="fa fa-check"></i><b>81.9.2</b> Guidelines</a></li>
<li class="chapter" data-level="81.9.3" data-path="population-sensation.html"><a href="population-sensation.html#what-to-submit"><i class="fa fa-check"></i><b>81.9.3</b> What to submit</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="82" data-path="integral-population-models.html"><a href="integral-population-models.html"><i class="fa fa-check"></i><b>82</b> Integral Population Models</a>
<ul>
<li class="chapter" data-level="82.1" data-path="integral-population-models.html"><a href="integral-population-models.html#case-study-modeling-an-endangered-hawaiian-plant"><i class="fa fa-check"></i><b>82.1</b> Case Study: Modeling an Endangered Hawaiian Plant</a>
<ul>
<li class="chapter" data-level="82.1.1" data-path="integral-population-models.html"><a href="integral-population-models.html#why-model-s.-lanceolata-populations"><i class="fa fa-check"></i><b>82.1.1</b> Why model S. lanceolata populations?</a></li>
<li class="chapter" data-level="82.1.2" data-path="integral-population-models.html"><a href="integral-population-models.html#the-management-question"><i class="fa fa-check"></i><b>82.1.2</b> The management question:</a></li>
</ul></li>
<li class="chapter" data-level="82.2" data-path="integral-population-models.html"><a href="integral-population-models.html#designing-a-demographic-census-timing-and-measurements"><i class="fa fa-check"></i><b>82.2</b> Designing a Demographic Census: Timing and Measurements</a>
<ul>
<li class="chapter" data-level="82.2.1" data-path="integral-population-models.html"><a href="integral-population-models.html#principle-1-census-timing-must-align-with-the-organisms-life-cycle"><i class="fa fa-check"></i><b>82.2.1</b> Principle 1: Census Timing Must Align with the Organism’s Life Cycle</a></li>
<li class="chapter" data-level="82.2.2" data-path="integral-population-models.html"><a href="integral-population-models.html#principle-2-measurements-must-link-to-vital-rates"><i class="fa fa-check"></i><b>82.2.2</b> Principle 2: Measurements Must Link to Vital Rates</a></li>
<li class="chapter" data-level="82.2.3" data-path="integral-population-models.html"><a href="integral-population-models.html#principle-3-track-individual-fates-through-time"><i class="fa fa-check"></i><b>82.2.3</b> Principle 3: Track Individual Fates Through Time</a></li>
<li class="chapter" data-level="82.2.4" data-path="integral-population-models.html"><a href="integral-population-models.html#principle-4-document-context-and-threats"><i class="fa fa-check"></i><b>82.2.4</b> Principle 4: Document Context and Threats</a></li>
<li class="chapter" data-level="82.2.5" data-path="integral-population-models.html"><a href="integral-population-models.html#understanding-state-variables-in-ipms"><i class="fa fa-check"></i><b>82.2.5</b> Understanding State Variables in IPMs</a></li>
<li class="chapter" data-level="82.2.6" data-path="integral-population-models.html"><a href="integral-population-models.html#explore-potential-state-variables"><i class="fa fa-check"></i><b>82.2.6</b> Explore Potential State Variables</a></li>
<li class="chapter" data-level="82.2.7" data-path="integral-population-models.html"><a href="integral-population-models.html#visualize-distribution-of-candidate-state-variables"><i class="fa fa-check"></i><b>82.2.7</b> Visualize Distribution of Candidate State Variables</a></li>
<li class="chapter" data-level="82.2.8" data-path="integral-population-models.html"><a href="integral-population-models.html#selecting-our-state-variable"><i class="fa fa-check"></i><b>82.2.8</b> Selecting Our State Variable</a></li>
<li class="chapter" data-level="82.2.9" data-path="integral-population-models.html"><a href="integral-population-models.html#the-dimensionality-problem"><i class="fa fa-check"></i><b>82.2.9</b> The dimensionality problem</a></li>
</ul></li>
<li class="chapter" data-level="82.3" data-path="integral-population-models.html"><a href="integral-population-models.html#packages-and-approaches-to-creating-ipms"><i class="fa fa-check"></i><b>82.3</b> Packages and approaches to creating IPMs</a>
<ul>
<li class="chapter" data-level="82.3.1" data-path="integral-population-models.html"><a href="integral-population-models.html#ipmr-a-modern-flexible-ipm-framework"><i class="fa fa-check"></i><b>82.3.1</b> ipmr: a modern, flexible IPM framework</a></li>
<li class="chapter" data-level="82.3.2" data-path="integral-population-models.html"><a href="integral-population-models.html#ipmpack-a-more-automated-ipm-workflow"><i class="fa fa-check"></i><b>82.3.2</b> IPMpack: a more automated IPM workflow</a></li>
<li class="chapter" data-level="82.3.3" data-path="integral-population-models.html"><a href="integral-population-models.html#hand-coded-ipms-maximum-transparency"><i class="fa fa-check"></i><b>82.3.3</b> Hand-coded IPMs: maximum transparency</a></li>
</ul></li>
<li class="chapter" data-level="82.4" data-path="integral-population-models.html"><a href="integral-population-models.html#building-an-ipm-structuring-linear-models-variable-transformations-and-dealing-with-outliers"><i class="fa fa-check"></i><b>82.4</b> Building an IPM: Structuring linear models, variable transformations, and dealing with outliers</a>
<ul>
<li class="chapter" data-level="82.4.1" data-path="integral-population-models.html"><a href="integral-population-models.html#variable-transformations-and-scale"><i class="fa fa-check"></i><b>82.4.1</b> Variable transformations and scale</a></li>
<li class="chapter" data-level="82.4.2" data-path="integral-population-models.html"><a href="integral-population-models.html#dealing-with-zeros-missing-values-and-outliers"><i class="fa fa-check"></i><b>82.4.2</b> Dealing with zeros, missing values, and outliers</a></li>
<li class="chapter" data-level="82.4.3" data-path="integral-population-models.html"><a href="integral-population-models.html#parallel-ipm-workflows"><i class="fa fa-check"></i><b>82.4.3</b> Parallel IPM workflows</a></li>
<li class="chapter" data-level="82.4.4" data-path="integral-population-models.html"><a href="integral-population-models.html#growth-transitions"><i class="fa fa-check"></i><b>82.4.4</b> Growth transitions</a></li>
</ul></li>
<li class="chapter" data-level="82.5" data-path="integral-population-models.html"><a href="integral-population-models.html#constructing-a-transition-matrix"><i class="fa fa-check"></i><b>82.5</b> Constructing a transition matrix</a></li>
<li class="chapter" data-level="82.6" data-path="integral-population-models.html"><a href="integral-population-models.html#building-an-ipm-boundary-points-mesh-points-and-step-size"><i class="fa fa-check"></i><b>82.6</b> Building an IPM: Boundary points, mesh points, and step size</a></li>
<li class="chapter" data-level="82.7" data-path="integral-population-models.html"><a href="integral-population-models.html#model-evaluation-eviction"><i class="fa fa-check"></i><b>82.7</b> Model Evaluation: Eviction</a>
<ul>
<li class="chapter" data-level="82.7.1" data-path="integral-population-models.html"><a href="integral-population-models.html#why-eviction-matters"><i class="fa fa-check"></i><b>82.7.1</b> Why Eviction Matters</a></li>
<li class="chapter" data-level="82.7.2" data-path="integral-population-models.html"><a href="integral-population-models.html#eviction-correction-strategies"><i class="fa fa-check"></i><b>82.7.2</b> Eviction Correction Strategies</a></li>
<li class="chapter" data-level="82.7.3" data-path="integral-population-models.html"><a href="integral-population-models.html#summary-1"><i class="fa fa-check"></i><b>82.7.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="82.8" data-path="integral-population-models.html"><a href="integral-population-models.html#estimating-population-dynamics-parameters"><i class="fa fa-check"></i><b>82.8</b> Estimating population dynamics parameters</a></li>
<li class="chapter" data-level="82.9" data-path="integral-population-models.html"><a href="integral-population-models.html#calculating-confidence-intervals"><i class="fa fa-check"></i><b>82.9</b> Calculating confidence intervals</a></li>
<li class="chapter" data-level="82.10" data-path="integral-population-models.html"><a href="integral-population-models.html#putting-this-all-together-in-code-form"><i class="fa fa-check"></i><b>82.10</b> Putting this all together in code form!</a></li>
<li class="chapter" data-level="82.11" data-path="integral-population-models.html"><a href="integral-population-models.html#test-your-knowledge-2"><i class="fa fa-check"></i><b>82.11</b> Test your knowledge</a></li>
<li class="chapter" data-level="82.12" data-path="integral-population-models.html"><a href="integral-population-models.html#assignment-34"><i class="fa fa-check"></i><b>82.12</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="83" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html"><i class="fa fa-check"></i><b>83</b> Stochastic Population Dynamics</a>
<ul>
<li class="chapter" data-level="83.1" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#why-stochasticity-matters"><i class="fa fa-check"></i><b>83.1</b> Why stochasticity matters</a></li>
<li class="chapter" data-level="83.2" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#types-of-stochasticity"><i class="fa fa-check"></i><b>83.2</b> Types of stochasticity</a></li>
<li class="chapter" data-level="83.3" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#setup-29"><i class="fa fa-check"></i><b>83.3</b> Setup</a></li>
<li class="chapter" data-level="83.4" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-1-stochastic-matrix-models"><i class="fa fa-check"></i><b>83.4</b> Part 1: Stochastic Matrix Models</a></li>
<li class="chapter" data-level="83.5" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#deterministic-baseline"><i class="fa fa-check"></i><b>83.5</b> Deterministic baseline</a></li>
<li class="chapter" data-level="83.6" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#adding-environmental-stochasticity"><i class="fa fa-check"></i><b>83.6</b> Adding environmental stochasticity</a></li>
<li class="chapter" data-level="83.7" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#stochastic-simulation"><i class="fa fa-check"></i><b>83.7</b> Stochastic simulation</a></li>
<li class="chapter" data-level="83.8" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#stochastic-growth-rate-log-%CE%BBs"><i class="fa fa-check"></i><b>83.8</b> Stochastic growth rate (log λs)</a></li>
<li class="chapter" data-level="83.9" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-2-demographic-stochasticity"><i class="fa fa-check"></i><b>83.9</b> Part 2: Demographic Stochasticity</a>
<ul>
<li class="chapter" data-level="83.9.1" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#what-is-demographic-stochasticity"><i class="fa fa-check"></i><b>83.9.1</b> What is demographic stochasticity?</a></li>
<li class="chapter" data-level="83.9.2" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#simulating-demographic-stochasticity"><i class="fa fa-check"></i><b>83.9.2</b> Simulating demographic stochasticity</a></li>
</ul></li>
<li class="chapter" data-level="83.10" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-3-extinction-risk-analysis"><i class="fa fa-check"></i><b>83.10</b> Part 3: Extinction Risk Analysis</a>
<ul>
<li class="chapter" data-level="83.10.1" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#quasi-extinction-thresholds"><i class="fa fa-check"></i><b>83.10.1</b> Quasi-extinction thresholds</a></li>
<li class="chapter" data-level="83.10.2" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#cumulative-extinction-curves"><i class="fa fa-check"></i><b>83.10.2</b> Cumulative extinction curves</a></li>
</ul></li>
<li class="chapter" data-level="83.11" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-4-stochastic-ipms"><i class="fa fa-check"></i><b>83.11</b> Part 4: Stochastic IPMs</a>
<ul>
<li class="chapter" data-level="83.11.1" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#why-ipms-for-stochasticity"><i class="fa fa-check"></i><b>83.11.1</b> Why IPMs for stochasticity?</a></li>
<li class="chapter" data-level="83.11.2" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#building-a-stochastic-ipm-with-ipmr"><i class="fa fa-check"></i><b>83.11.2</b> Building a stochastic IPM with ipmr</a></li>
<li class="chapter" data-level="83.11.3" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#simulating-year-specific-vital-rates"><i class="fa fa-check"></i><b>83.11.3</b> Simulating year-specific vital rates</a></li>
<li class="chapter" data-level="83.11.4" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#building-year-specific-kernels"><i class="fa fa-check"></i><b>83.11.4</b> Building year-specific kernels</a></li>
<li class="chapter" data-level="83.11.5" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#stochastic-ipm-simulation"><i class="fa fa-check"></i><b>83.11.5</b> Stochastic IPM simulation</a></li>
<li class="chapter" data-level="83.11.6" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#visualizing-size-distribution-dynamics"><i class="fa fa-check"></i><b>83.11.6</b> Visualizing size distribution dynamics</a></li>
</ul></li>
<li class="chapter" data-level="83.12" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-5-catastrophes-and-bonanzas"><i class="fa fa-check"></i><b>83.12</b> Part 5: Catastrophes and Bonanzas</a>
<ul>
<li class="chapter" data-level="83.12.1" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#incorporating-rare-events"><i class="fa fa-check"></i><b>83.12.1</b> Incorporating rare events</a></li>
</ul></li>
<li class="chapter" data-level="83.13" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-6-stochastic-sensitivity-analysis"><i class="fa fa-check"></i><b>83.13</b> Part 6: Stochastic Sensitivity Analysis</a>
<ul>
<li class="chapter" data-level="83.13.1" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#which-vital-rates-matter-most-under-stochasticity"><i class="fa fa-check"></i><b>83.13.1</b> Which vital rates matter most under stochasticity?</a></li>
</ul></li>
<li class="chapter" data-level="83.14" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-7-application-to-silene-lanceolata"><i class="fa fa-check"></i><b>83.14</b> Part 7: Application to <em>Silene lanceolata</em></a>
<ul>
<li class="chapter" data-level="83.14.1" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#connecting-to-chapter-18"><i class="fa fa-check"></i><b>83.14.1</b> Connecting to Chapter 18</a></li>
<li class="chapter" data-level="83.14.2" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#minimum-viable-population-analysis"><i class="fa fa-check"></i><b>83.14.2</b> Minimum viable population analysis</a></li>
</ul></li>
<li class="chapter" data-level="83.15" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-8-reporting-stochastic-models"><i class="fa fa-check"></i><b>83.15</b> Part 8: Reporting Stochastic Models</a>
<ul>
<li class="chapter" data-level="83.15.1" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#what-to-report-18"><i class="fa fa-check"></i><b>83.15.1</b> What to report</a></li>
<li class="chapter" data-level="83.15.2" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#sample-methods-and-results-26"><i class="fa fa-check"></i><b>83.15.2</b> Sample methods and results</a></li>
</ul></li>
<li class="chapter" data-level="83.16" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#key-takeaways-30"><i class="fa fa-check"></i><b>83.16</b> Key takeaways</a></li>
<li class="chapter" data-level="83.17" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#assignment-35"><i class="fa fa-check"></i><b>83.17</b> Assignment</a>
<ul>
<li class="chapter" data-level="83.17.1" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-1-conceptual-questions-23"><i class="fa fa-check"></i><b>83.17.1</b> Part 1: Conceptual questions</a></li>
<li class="chapter" data-level="83.17.2" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-2-stochastic-matrix-model"><i class="fa fa-check"></i><b>83.17.2</b> Part 2: Stochastic matrix model</a></li>
<li class="chapter" data-level="83.17.3" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-3-extinction-risk-analysis-1"><i class="fa fa-check"></i><b>83.17.3</b> Part 3: Extinction risk analysis</a></li>
<li class="chapter" data-level="83.17.4" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-4-stochastic-ipm"><i class="fa fa-check"></i><b>83.17.4</b> Part 4: Stochastic IPM</a></li>
<li class="chapter" data-level="83.17.5" data-path="stochastic-population-dynamics.html"><a href="stochastic-population-dynamics.html#part-5-reporting-4"><i class="fa fa-check"></i><b>83.17.5</b> Part 5: Reporting</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/sarasouther/DataAnalysis" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Applied Statistics for the Natural Sciences</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="integral-population-models" class="section level1 hasAnchor" number="82">
<h1><span class="header-section-number">Chapter 82</span> Integral Population Models<a href="integral-population-models.html#integral-population-models" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="case-study-modeling-an-endangered-hawaiian-plant" class="section level2 hasAnchor" number="82.1">
<h2><span class="header-section-number">82.1</span> Case Study: Modeling an Endangered Hawaiian Plant<a href="integral-population-models.html#case-study-modeling-an-endangered-hawaiian-plant" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For our applied example, we’ll build a population model for <em>Silene lanceolata</em> (lanceolate catchfly), a small woody shrub endemic to the Hawaiian Islands. This species is federally listed as threatened/endangered and faces multiple conservation challenges that make it an ideal candidate for demographic modeling.</p>
<div id="why-model-s.-lanceolata-populations" class="section level3 hasAnchor" number="82.1.1">
<h3><span class="header-section-number">82.1.1</span> Why model S. lanceolata populations?<a href="integral-population-models.html#why-model-s.-lanceolata-populations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Hawaiian plant species like S. lanceolata occur in small, isolated populations that are particularly vulnerable to extinction. At Pōhakuloa Training Area (PTA) on Hawaii Island, these native plants face a perfect storm of threats:</p>
<ul>
<li>Invasive plant competition: Fountain grass (<em>Cenchrus setaceus</em>) and fireweed (<em>Senecio madagascariensis</em>) outcompete native species for water, light, and space</li>
<li>Altered fire regimes: Invasive grasses produce fine fuels that dramatically increase fire frequency—a disturbance to which Hawaiian plants have no evolutionary adaptations</li>
<li>Climate change: Increasing drought frequency threatens plants already stressed by competition and fire</li>
<li>Small population sizes: Limited numbers mean that small changes in vital rates (survival, growth, reproduction) can have large consequences for extinction risk</li>
</ul>
</div>
<div id="the-management-question" class="section level3 hasAnchor" number="82.1.2">
<h3><span class="header-section-number">82.1.2</span> The management question:<a href="integral-population-models.html#the-management-question" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Conservation managers need to decide how to allocate limited resources. Should they prioritize invasive plant control? Focus on fire suppression? Protect certain life stages? Population models allow us to project population trajectories under different management scenarios and identify which vital rates have the greatest influence on population growth.
In this example, we’ll use demographic data collected from field studies at PTA to build an Integral Projection Model (IPM) that links individual plant size to survival, growth, and reproduction. This will allow us to evaluate how management actions might influence <em>S. lanceolata</em> persistence.</p>
</div>
</div>
<div id="designing-a-demographic-census-timing-and-measurements" class="section level2 hasAnchor" number="82.2">
<h2><span class="header-section-number">82.2</span> Designing a Demographic Census: Timing and Measurements<a href="integral-population-models.html#designing-a-demographic-census-timing-and-measurements" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before we dive into building the population model, let’s review how demographic data are collected. The design of a census protocol involves several critical decisions that directly impact the quality and interpretability of your population projections.</p>
<div id="principle-1-census-timing-must-align-with-the-organisms-life-cycle" class="section level3 hasAnchor" number="82.2.1">
<h3><span class="header-section-number">82.2.1</span> Principle 1: Census Timing Must Align with the Organism’s Life Cycle<a href="integral-population-models.html#principle-1-census-timing-must-align-with-the-organisms-life-cycle" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For demographic models to be meaningful, censuses must occur at regular intervals that correspond to biologically relevant time steps. Ideally, you want to census:</p>
<p>At the same time each year to capture consistent population states
Before or after major demographic events (reproduction, mortality pulses, germination)
When individuals are most visible/measurable for accurate data collection</p>
<p>For <em>S. lanceolata</em> at Pōhakuloa Training Area, we faced practical constraints. While Hawaii lacks dramatic seasonal variation in temperature, the species does show phenological patterns. We conducted censuses in summer for several pragmatic reasons:</p>
<ul>
<li>Field accessibility: Summer weather allowed consistent access to remote field sites</li>
<li>Plant visibility: Woody shrubs maintain their structure year-round, but summer timing captured both vegetative and reproductive activity</li>
<li>Logistical feasibility: Research team availability and funding cycles</li>
</ul>
<p>The key is that we censused at the same time each year, allowing us to track year-to-year changes in a standardized way. Our annual time step (summer to summer) captures one full cycle of growth, reproduction, and survival.</p>
</div>
<div id="principle-2-measurements-must-link-to-vital-rates" class="section level3 hasAnchor" number="82.2.2">
<h3><span class="header-section-number">82.2.2</span> Principle 2: Measurements Must Link to Vital Rates<a href="integral-population-models.html#principle-2-measurements-must-link-to-vital-rates" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Demographic models require data on survival, growth, and reproduction. The measurements we collect must allow us to estimate these vital rates. For <em>S. lanceolata</em>, we structured our census to capture:</p>
<ul>
<li><p>Size Measurements (for survival and growth): We measured multiple aspects of plant size because size often predicts both survival probability and reproductive output:</p>
<ul>
<li>Foliated length of longest branch (cm): Captures vegetative vigor and photosynthetic capacity</li>
<li>Plant height (cm): Structural size metric, indicates establishment and competitive ability</li>
<li>Plant height with reproductive stalk (cm): Total height including reproductive structures</li>
<li>Stem diameter (cm): Proxy for plant age and woody biomass accumulation</li>
<li>Number of branches (n): Architectural complexity, related to resource acquisition</li>
</ul></li>
</ul>
<p>For our Integral Projection Model (IPM), we’ll select one primary size variable that best predicts survival, growth, and fecundity. Often this is stem diameter or height, but we measured multiple metrics to explore which is most predictive.</p>
<ul>
<li><p>Reproductive Measurements (for fecundity): To estimate how many offspring each individual produces, we counted:</p></li>
<li><p>Number of inflorescences (n): Reproductive effort (number of flowering stalks)</p></li>
<li><p>Number of flowers and seed heads per stalk: Potential and realized seed production</p></li>
<li><p>Seeds per fruit: Obtained from literature since counting tiny seeds in the field is impractical</p></li>
</ul>
<p>These measurements allow us to build a size-fecundity function: larger plants typically produce more inflorescences, more flowers, and ultimately more seeds.</p>
</div>
<div id="principle-3-track-individual-fates-through-time" class="section level3 hasAnchor" number="82.2.3">
<h3><span class="header-section-number">82.2.3</span> Principle 3: Track Individual Fates Through Time<a href="integral-population-models.html#principle-3-track-individual-fates-through-time" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The power of demographic models comes from tracking individuals across census intervals:</p>
<ul>
<li>Did this individual survive from year t to year t+1?</li>
<li>If it survived, how much did it grow (change in size)?</li>
<li>How many offspring did it produce?</li>
</ul>
<p>This requires marking individual plants (tags, GPS coordinates, maps) so we can relocate them in subsequent years. For S. lanceolata, each plant received a unique identifier allowing us to build a longitudinal dataset of individual life histories.</p>
</div>
<div id="principle-4-document-context-and-threats" class="section level3 hasAnchor" number="82.2.4">
<h3><span class="header-section-number">82.2.4</span> Principle 4: Document Context and Threats<a href="integral-population-models.html#principle-4-document-context-and-threats" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Beyond demographic measurements, we recorded factors that might explain variation in vital rates:</p>
<ul>
<li>Browse damage: Evidence of herbivory (ungulates, insects)</li>
<li>Fire history: Burned vs. unburned individuals</li>
<li>Disease/stress: Chlorosis, wilting, pathogen damage</li>
<li>Proximity to invasive plants: Competition effects</li>
</ul>
<p>These covariates allow us to ask: Does fire reduce survival? Do plants near fountain grass grow more slowly? Does herbivory reduce reproduction?</p>
<p>Let’s check out data from our initial collection trip:</p>
<div class="sourceCode" id="cb2185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2185-1"><a href="integral-population-models.html#cb2185-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2185-2"><a href="integral-population-models.html#cb2185-2" tabindex="-1"></a><span class="fu">library</span>(httr)</span>
<span id="cb2185-3"><a href="integral-population-models.html#cb2185-3" tabindex="-1"></a></span>
<span id="cb2185-4"><a href="integral-population-models.html#cb2185-4" tabindex="-1"></a><span class="co"># Create data directory if it doesn&#39;t exist</span></span>
<span id="cb2185-5"><a href="integral-population-models.html#cb2185-5" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;data&quot;</span>)) <span class="fu">dir.create</span>(<span class="st">&quot;data&quot;</span>)</span>
<span id="cb2185-6"><a href="integral-population-models.html#cb2185-6" tabindex="-1"></a></span>
<span id="cb2185-7"><a href="integral-population-models.html#cb2185-7" tabindex="-1"></a><span class="co"># Google Drive file ID</span></span>
<span id="cb2185-8"><a href="integral-population-models.html#cb2185-8" tabindex="-1"></a>file_id <span class="ot">&lt;-</span> <span class="st">&quot;1WCToCDAuSR_Wd2ScRy9yhwiDGijL47OX&quot;</span></span>
<span id="cb2185-9"><a href="integral-population-models.html#cb2185-9" tabindex="-1"></a></span>
<span id="cb2185-10"><a href="integral-population-models.html#cb2185-10" tabindex="-1"></a><span class="fu">GET</span>(</span>
<span id="cb2185-11"><a href="integral-population-models.html#cb2185-11" tabindex="-1"></a>  <span class="at">url =</span> <span class="fu">paste0</span>(</span>
<span id="cb2185-12"><a href="integral-population-models.html#cb2185-12" tabindex="-1"></a>    <span class="st">&quot;https://drive.usercontent.google.com/download?id=&quot;</span>, </span>
<span id="cb2185-13"><a href="integral-population-models.html#cb2185-13" tabindex="-1"></a>    file_id, <span class="st">&quot;&amp;export=download&amp;confirm=t&quot;</span></span>
<span id="cb2185-14"><a href="integral-population-models.html#cb2185-14" tabindex="-1"></a>  ),</span>
<span id="cb2185-15"><a href="integral-population-models.html#cb2185-15" tabindex="-1"></a>  <span class="fu">write_disk</span>(<span class="st">&quot;data/silene_census.csv&quot;</span>, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2185-16"><a href="integral-population-models.html#cb2185-16" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## Response [https://drive.usercontent.google.com/download?id=1WCToCDAuSR_Wd2ScRy9yhwiDGijL47OX&amp;export=download&amp;confirm=t]
##   Date: 2026-02-22 15:08
##   Status: 200
##   Content-Type: application/octet-stream
##   Size: 5.03 kB
## &lt;ON DISK&gt;  /Users/sks379/Desktop/GitHubProjects/DataAnalysis/data/silene_census.csv</code></pre>
<div class="sourceCode" id="cb2187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2187-1"><a href="integral-population-models.html#cb2187-1" tabindex="-1"></a><span class="co"># Load the data</span></span>
<span id="cb2187-2"><a href="integral-population-models.html#cb2187-2" tabindex="-1"></a>silene <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/silene_census.csv&quot;</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb2187-3"><a href="integral-population-models.html#cb2187-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>...<span class="dv">1</span>)  <span class="co"># Remove row number column</span></span>
<span id="cb2187-4"><a href="integral-population-models.html#cb2187-4" tabindex="-1"></a></span>
<span id="cb2187-5"><a href="integral-population-models.html#cb2187-5" tabindex="-1"></a><span class="co"># Quick summary</span></span>
<span id="cb2187-6"><a href="integral-population-models.html#cb2187-6" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Dataset dimensions:&quot;</span>, <span class="fu">nrow</span>(silene), <span class="st">&quot;rows x&quot;</span>, <span class="fu">ncol</span>(silene), <span class="st">&quot;columns</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Dataset dimensions: 100 rows x 11 columns</code></pre>
<div class="sourceCode" id="cb2189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2189-1"><a href="integral-population-models.html#cb2189-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Number of unique individuals:&quot;</span>, <span class="fu">n_distinct</span>(silene<span class="sc">$</span>UniqueID), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Number of unique individuals: 100</code></pre>
<div class="sourceCode" id="cb2191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2191-1"><a href="integral-population-models.html#cb2191-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Weed control treatments:&quot;</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(silene<span class="sc">$</span>WeedControl), <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Weed control treatments: C, W</code></pre>
<div class="sourceCode" id="cb2193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2193-1"><a href="integral-population-models.html#cb2193-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Map units:&quot;</span>, <span class="fu">paste</span>(<span class="fu">sort</span>(<span class="fu">unique</span>(silene<span class="sc">$</span>MapUnit)), <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Map units: 1, 2, 3</code></pre>
<div class="sourceCode" id="cb2195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2195-1"><a href="integral-population-models.html#cb2195-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Proportion reproduced:&quot;</span>, <span class="fu">mean</span>(silene<span class="sc">$</span>Reproduced), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Proportion reproduced: 0.44</code></pre>
<div class="sourceCode" id="cb2197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2197-1"><a href="integral-population-models.html#cb2197-1" tabindex="-1"></a><span class="fu">glimpse</span>(silene)</span></code></pre></div>
<pre><code>## Rows: 100
## Columns: 11
## $ UniqueID                    &lt;chr&gt; &quot;Bobcat03_1&quot;, &quot;Bobcat03_2&quot;, &quot;Bobcat03_3&quot;, &quot;Bobcat03_4&quot;, &quot;Bobcat03_5&quot;, &quot;B…
## $ WeedControl                 &lt;chr&gt; &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C…
## $ MapUnit                     &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, …
## $ VegHeight                   &lt;dbl&gt; 5.4, 42.6, 8.3, 2.5, 31.0, 26.8, 39.1, 18.5, 14.4, 6.8, 18.5, 11.0, 13.8…
## $ VegReproHeight              &lt;dbl&gt; 5.4, 56.4, 8.3, 2.5, 31.0, 44.2, 53.6, 18.5, 14.4, 6.8, 18.5, 11.0, 13.8…
## $ StemDiameter                &lt;dbl&gt; 0.18, 0.30, 0.12, 0.06, 0.22, 0.19, 0.29, 0.17, 0.24, 0.15, 0.18, 0.12, …
## $ StemNum                     &lt;dbl&gt; 0, 6, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 6, …
## $ LengthLongestBranchwFoliage &lt;dbl&gt; 3.7, 1.0, 5.8, 1.5, 22.4, 18.0, 24.0, 16.4, 8.8, 4.3, 12.8, 5.1, 8.1, 12…
## $ InflorNum                   &lt;dbl&gt; 0, 6, 0, 0, 0, 4, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 9, 26,…
## $ Reproduced                  &lt;dbl&gt; 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, …
## $ TotalReproOutput            &lt;dbl&gt; 0, 21, 0, 0, 0, 26, 36, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64,…</code></pre>
</div>
<div id="understanding-state-variables-in-ipms" class="section level3 hasAnchor" number="82.2.5">
<h3><span class="header-section-number">82.2.5</span> Understanding State Variables in IPMs<a href="integral-population-models.html#understanding-state-variables-in-ipms" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In an Integral Projection Model (IPM), we need to choose a state variable that describes the “state” of each individual in the population. This variable must:</p>
<ul>
<li>Predict vital rates: Larger/older individuals typically have different survival, growth, and reproduction than smaller/younger ones</li>
<li>Be continuously measured: IPMs work with continuous size distributions (unlike matrix models with discrete stages)</li>
<li>Be measurable across time: We need to track how individuals change in this variable from year to year</li>
<li>Capture biological meaningful variation: The variable should reflect real differences in demographic performance</li>
</ul>
<p>Common state variables in plant IPMs:</p>
<ul>
<li>Stem diameter</li>
<li>Plant height</li>
<li>Basal area</li>
<li>Number of stems/ramets</li>
<li>Leaf area</li>
</ul>
<p>Let’s explore our data to identify the best state variable for <em>S. lanceolata</em>.</p>
</div>
<div id="explore-potential-state-variables" class="section level3 hasAnchor" number="82.2.6">
<h3><span class="header-section-number">82.2.6</span> Explore Potential State Variables<a href="integral-population-models.html#explore-potential-state-variables" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb2199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2199-1"><a href="integral-population-models.html#cb2199-1" tabindex="-1"></a><span class="co"># Summary of candidate state variables</span></span>
<span id="cb2199-2"><a href="integral-population-models.html#cb2199-2" tabindex="-1"></a>state_summary <span class="ot">&lt;-</span> silene <span class="sc">|&gt;</span></span>
<span id="cb2199-3"><a href="integral-population-models.html#cb2199-3" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb2199-4"><a href="integral-population-models.html#cb2199-4" tabindex="-1"></a>    <span class="at">n_plants =</span> <span class="fu">n</span>(),</span>
<span id="cb2199-5"><a href="integral-population-models.html#cb2199-5" tabindex="-1"></a>    </span>
<span id="cb2199-6"><a href="integral-population-models.html#cb2199-6" tabindex="-1"></a>    <span class="co"># Continuous size measurements</span></span>
<span id="cb2199-7"><a href="integral-population-models.html#cb2199-7" tabindex="-1"></a>    <span class="at">mean_stem_diam =</span> <span class="fu">mean</span>(StemDiameter, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-8"><a href="integral-population-models.html#cb2199-8" tabindex="-1"></a>    <span class="at">sd_stem_diam =</span> <span class="fu">sd</span>(StemDiameter, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-9"><a href="integral-population-models.html#cb2199-9" tabindex="-1"></a>    </span>
<span id="cb2199-10"><a href="integral-population-models.html#cb2199-10" tabindex="-1"></a>    <span class="at">mean_height_veg =</span> <span class="fu">mean</span>(VegHeight, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-11"><a href="integral-population-models.html#cb2199-11" tabindex="-1"></a>    <span class="at">sd_height_veg =</span> <span class="fu">sd</span>(VegHeight, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-12"><a href="integral-population-models.html#cb2199-12" tabindex="-1"></a>    </span>
<span id="cb2199-13"><a href="integral-population-models.html#cb2199-13" tabindex="-1"></a>    <span class="at">mean_height_repro =</span> <span class="fu">mean</span>(VegReproHeight, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-14"><a href="integral-population-models.html#cb2199-14" tabindex="-1"></a>    <span class="at">sd_height_repro =</span> <span class="fu">sd</span>(VegReproHeight, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-15"><a href="integral-population-models.html#cb2199-15" tabindex="-1"></a>    </span>
<span id="cb2199-16"><a href="integral-population-models.html#cb2199-16" tabindex="-1"></a>    <span class="at">mean_branch_length =</span> <span class="fu">mean</span>(LengthLongestBranchwFoliage, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-17"><a href="integral-population-models.html#cb2199-17" tabindex="-1"></a>    <span class="at">sd_branch_length =</span> <span class="fu">sd</span>(LengthLongestBranchwFoliage, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-18"><a href="integral-population-models.html#cb2199-18" tabindex="-1"></a>    </span>
<span id="cb2199-19"><a href="integral-population-models.html#cb2199-19" tabindex="-1"></a>    <span class="co"># Count variables</span></span>
<span id="cb2199-20"><a href="integral-population-models.html#cb2199-20" tabindex="-1"></a>    <span class="at">mean_stems =</span> <span class="fu">mean</span>(StemNum, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-21"><a href="integral-population-models.html#cb2199-21" tabindex="-1"></a>    <span class="at">sd_stems =</span> <span class="fu">sd</span>(StemNum, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-22"><a href="integral-population-models.html#cb2199-22" tabindex="-1"></a>    </span>
<span id="cb2199-23"><a href="integral-population-models.html#cb2199-23" tabindex="-1"></a>    <span class="co"># Reproduction</span></span>
<span id="cb2199-24"><a href="integral-population-models.html#cb2199-24" tabindex="-1"></a>    <span class="at">mean_inflor =</span> <span class="fu">mean</span>(InflorNum, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-25"><a href="integral-population-models.html#cb2199-25" tabindex="-1"></a>    <span class="at">sd_inflor =</span> <span class="fu">sd</span>(InflorNum, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-26"><a href="integral-population-models.html#cb2199-26" tabindex="-1"></a>    <span class="at">mean_repro =</span> <span class="fu">mean</span>(TotalReproOutput, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-27"><a href="integral-population-models.html#cb2199-27" tabindex="-1"></a>    <span class="at">sd_repro =</span> <span class="fu">sd</span>(TotalReproOutput, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2199-28"><a href="integral-population-models.html#cb2199-28" tabindex="-1"></a>    <span class="at">prop_reproduced =</span> <span class="fu">mean</span>(Reproduced, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2199-29"><a href="integral-population-models.html#cb2199-29" tabindex="-1"></a>  )</span>
<span id="cb2199-30"><a href="integral-population-models.html#cb2199-30" tabindex="-1"></a></span>
<span id="cb2199-31"><a href="integral-population-models.html#cb2199-31" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">t</span>(state_summary), <span class="at">col.names =</span> <span class="st">&quot;Value&quot;</span>, <span class="at">digits =</span> <span class="dv">2</span>,</span>
<span id="cb2199-32"><a href="integral-population-models.html#cb2199-32" tabindex="-1"></a>             <span class="at">caption =</span> <span class="st">&quot;Summary statistics for potential state variables&quot;</span>)</span></code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-6">Table 82.1: </span>Summary statistics for potential state variables</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">n_plants</td>
<td align="right">100.00</td>
</tr>
<tr class="even">
<td align="left">mean_stem_diam</td>
<td align="right">0.54</td>
</tr>
<tr class="odd">
<td align="left">sd_stem_diam</td>
<td align="right">0.68</td>
</tr>
<tr class="even">
<td align="left">mean_height_veg</td>
<td align="right">49.33</td>
</tr>
<tr class="odd">
<td align="left">sd_height_veg</td>
<td align="right">43.32</td>
</tr>
<tr class="even">
<td align="left">mean_height_repro</td>
<td align="right">56.00</td>
</tr>
<tr class="odd">
<td align="left">sd_height_repro</td>
<td align="right">49.77</td>
</tr>
<tr class="even">
<td align="left">mean_branch_length</td>
<td align="right">12.03</td>
</tr>
<tr class="odd">
<td align="left">sd_branch_length</td>
<td align="right">7.14</td>
</tr>
<tr class="even">
<td align="left">mean_stems</td>
<td align="right">1.29</td>
</tr>
<tr class="odd">
<td align="left">sd_stems</td>
<td align="right">1.80</td>
</tr>
<tr class="even">
<td align="left">mean_inflor</td>
<td align="right">3.98</td>
</tr>
<tr class="odd">
<td align="left">sd_inflor</td>
<td align="right">6.67</td>
</tr>
<tr class="even">
<td align="left">mean_repro</td>
<td align="right">17.86</td>
</tr>
<tr class="odd">
<td align="left">sd_repro</td>
<td align="right">24.89</td>
</tr>
<tr class="even">
<td align="left">prop_reproduced</td>
<td align="right">0.44</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb2200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2200-1"><a href="integral-population-models.html#cb2200-1" tabindex="-1"></a><span class="co"># Check for missing data</span></span>
<span id="cb2200-2"><a href="integral-population-models.html#cb2200-2" tabindex="-1"></a>missing_data <span class="ot">&lt;-</span> silene <span class="sc">|&gt;</span></span>
<span id="cb2200-3"><a href="integral-population-models.html#cb2200-3" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">is.na</span>(.)))) <span class="sc">|&gt;</span></span>
<span id="cb2200-4"><a href="integral-population-models.html#cb2200-4" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">everything</span>(), <span class="at">names_to =</span> <span class="st">&quot;Variable&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Missing&quot;</span>) <span class="sc">|&gt;</span></span>
<span id="cb2200-5"><a href="integral-population-models.html#cb2200-5" tabindex="-1"></a>  <span class="fu">filter</span>(Missing <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb2200-6"><a href="integral-population-models.html#cb2200-6" tabindex="-1"></a></span>
<span id="cb2200-7"><a href="integral-population-models.html#cb2200-7" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(missing_data) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb2200-8"><a href="integral-population-models.html#cb2200-8" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Missing data detected:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2200-9"><a href="integral-population-models.html#cb2200-9" tabindex="-1"></a>  <span class="fu">print</span>(knitr<span class="sc">::</span><span class="fu">kable</span>(missing_data))</span>
<span id="cb2200-10"><a href="integral-population-models.html#cb2200-10" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb2200-11"><a href="integral-population-models.html#cb2200-11" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">✓ No missing data in size measurements!</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2200-12"><a href="integral-population-models.html#cb2200-12" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## 
## ✓ No missing data in size measurements!</code></pre>
</div>
<div id="visualize-distribution-of-candidate-state-variables" class="section level3 hasAnchor" number="82.2.7">
<h3><span class="header-section-number">82.2.7</span> Visualize Distribution of Candidate State Variables<a href="integral-population-models.html#visualize-distribution-of-candidate-state-variables" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb2202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2202-1"><a href="integral-population-models.html#cb2202-1" tabindex="-1"></a><span class="co"># Reshape data for plotting</span></span>
<span id="cb2202-2"><a href="integral-population-models.html#cb2202-2" tabindex="-1"></a>state_vars_long <span class="ot">&lt;-</span> silene <span class="sc">|&gt;</span></span>
<span id="cb2202-3"><a href="integral-population-models.html#cb2202-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb2202-4"><a href="integral-population-models.html#cb2202-4" tabindex="-1"></a>    UniqueID, </span>
<span id="cb2202-5"><a href="integral-population-models.html#cb2202-5" tabindex="-1"></a>    StemDiameter,</span>
<span id="cb2202-6"><a href="integral-population-models.html#cb2202-6" tabindex="-1"></a>    VegHeight,</span>
<span id="cb2202-7"><a href="integral-population-models.html#cb2202-7" tabindex="-1"></a>    VegReproHeight,</span>
<span id="cb2202-8"><a href="integral-population-models.html#cb2202-8" tabindex="-1"></a>    LengthLongestBranchwFoliage,</span>
<span id="cb2202-9"><a href="integral-population-models.html#cb2202-9" tabindex="-1"></a>    StemNum</span>
<span id="cb2202-10"><a href="integral-population-models.html#cb2202-10" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2202-11"><a href="integral-population-models.html#cb2202-11" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb2202-12"><a href="integral-population-models.html#cb2202-12" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span>UniqueID,</span>
<span id="cb2202-13"><a href="integral-population-models.html#cb2202-13" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;variable&quot;</span>,</span>
<span id="cb2202-14"><a href="integral-population-models.html#cb2202-14" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span></span>
<span id="cb2202-15"><a href="integral-population-models.html#cb2202-15" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2202-16"><a href="integral-population-models.html#cb2202-16" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb2202-17"><a href="integral-population-models.html#cb2202-17" tabindex="-1"></a>    <span class="at">variable =</span> dplyr<span class="sc">::</span><span class="fu">recode</span>(</span>
<span id="cb2202-18"><a href="integral-population-models.html#cb2202-18" tabindex="-1"></a>      variable,</span>
<span id="cb2202-19"><a href="integral-population-models.html#cb2202-19" tabindex="-1"></a>      <span class="at">StemDiameter =</span> <span class="st">&quot;Stem Diameter (cm)&quot;</span>,</span>
<span id="cb2202-20"><a href="integral-population-models.html#cb2202-20" tabindex="-1"></a>      <span class="at">VegHeight =</span> <span class="st">&quot;Vegetative Height (cm)&quot;</span>,</span>
<span id="cb2202-21"><a href="integral-population-models.html#cb2202-21" tabindex="-1"></a>      <span class="at">VegReproHeight =</span> <span class="st">&quot;Height with Repro (cm)&quot;</span>,</span>
<span id="cb2202-22"><a href="integral-population-models.html#cb2202-22" tabindex="-1"></a>      <span class="at">LengthLongestBranchwFoliage =</span> <span class="st">&quot;Longest Branch w/ Foliage (cm)&quot;</span>,</span>
<span id="cb2202-23"><a href="integral-population-models.html#cb2202-23" tabindex="-1"></a>      <span class="at">StemNum =</span> <span class="st">&quot;Number of Stems&quot;</span></span>
<span id="cb2202-24"><a href="integral-population-models.html#cb2202-24" tabindex="-1"></a>    )</span>
<span id="cb2202-25"><a href="integral-population-models.html#cb2202-25" tabindex="-1"></a>  )</span>
<span id="cb2202-26"><a href="integral-population-models.html#cb2202-26" tabindex="-1"></a></span>
<span id="cb2202-27"><a href="integral-population-models.html#cb2202-27" tabindex="-1"></a><span class="co"># Plot distributions</span></span>
<span id="cb2202-28"><a href="integral-population-models.html#cb2202-28" tabindex="-1"></a><span class="fu">ggplot</span>(state_vars_long, <span class="fu">aes</span>(<span class="at">x =</span> value)) <span class="sc">+</span></span>
<span id="cb2202-29"><a href="integral-population-models.html#cb2202-29" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">15</span>, <span class="at">fill =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">color =</span> <span class="st">&quot;white&quot;</span>) <span class="sc">+</span></span>
<span id="cb2202-30"><a href="integral-population-models.html#cb2202-30" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>variable, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb2202-31"><a href="integral-population-models.html#cb2202-31" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb2202-32"><a href="integral-population-models.html#cb2202-32" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Distribution of Potential State Variables&quot;</span>,</span>
<span id="cb2202-33"><a href="integral-population-models.html#cb2202-33" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;Which variable best captures individual variation in S. lanceolata?&quot;</span>,</span>
<span id="cb2202-34"><a href="integral-population-models.html#cb2202-34" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Measurement value&quot;</span>,</span>
<span id="cb2202-35"><a href="integral-population-models.html#cb2202-35" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Number of individuals&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-7-1.png" alt="" width="960" style="display: block; margin: auto;" />
### Assess Correlations Between Measurements
State variables should ideally be correlated with other size metrics (indicating they capture overall plant size) but also provide unique information.</p>
<div class="sourceCode" id="cb2203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2203-1"><a href="integral-population-models.html#cb2203-1" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb2203-2"><a href="integral-population-models.html#cb2203-2" tabindex="-1"></a></span>
<span id="cb2203-3"><a href="integral-population-models.html#cb2203-3" tabindex="-1"></a><span class="co"># Calculate correlation matrix</span></span>
<span id="cb2203-4"><a href="integral-population-models.html#cb2203-4" tabindex="-1"></a>cor_data <span class="ot">&lt;-</span> silene <span class="sc">|&gt;</span></span>
<span id="cb2203-5"><a href="integral-population-models.html#cb2203-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(StemDiameter, </span>
<span id="cb2203-6"><a href="integral-population-models.html#cb2203-6" tabindex="-1"></a>         VegHeight,</span>
<span id="cb2203-7"><a href="integral-population-models.html#cb2203-7" tabindex="-1"></a>         VegReproHeight,</span>
<span id="cb2203-8"><a href="integral-population-models.html#cb2203-8" tabindex="-1"></a>         LengthLongestBranchwFoliage,</span>
<span id="cb2203-9"><a href="integral-population-models.html#cb2203-9" tabindex="-1"></a>         StemNum,</span>
<span id="cb2203-10"><a href="integral-population-models.html#cb2203-10" tabindex="-1"></a>         InflorNum,</span>
<span id="cb2203-11"><a href="integral-population-models.html#cb2203-11" tabindex="-1"></a>         TotalReproOutput) <span class="sc">|&gt;</span></span>
<span id="cb2203-12"><a href="integral-population-models.html#cb2203-12" tabindex="-1"></a>  <span class="fu">rename</span>(</span>
<span id="cb2203-13"><a href="integral-population-models.html#cb2203-13" tabindex="-1"></a>    <span class="st">`</span><span class="at">Stem Diam</span><span class="st">`</span> <span class="ot">=</span> StemDiameter,</span>
<span id="cb2203-14"><a href="integral-population-models.html#cb2203-14" tabindex="-1"></a>    <span class="st">`</span><span class="at">Height Veg</span><span class="st">`</span> <span class="ot">=</span> VegHeight,</span>
<span id="cb2203-15"><a href="integral-population-models.html#cb2203-15" tabindex="-1"></a>    <span class="st">`</span><span class="at">Height Repro</span><span class="st">`</span> <span class="ot">=</span> VegReproHeight,</span>
<span id="cb2203-16"><a href="integral-population-models.html#cb2203-16" tabindex="-1"></a>    <span class="st">`</span><span class="at">Branch Length</span><span class="st">`</span> <span class="ot">=</span> LengthLongestBranchwFoliage,</span>
<span id="cb2203-17"><a href="integral-population-models.html#cb2203-17" tabindex="-1"></a>    <span class="st">`</span><span class="at">N Stems</span><span class="st">`</span> <span class="ot">=</span> StemNum,</span>
<span id="cb2203-18"><a href="integral-population-models.html#cb2203-18" tabindex="-1"></a>    <span class="st">`</span><span class="at">N Inflor</span><span class="st">`</span> <span class="ot">=</span> InflorNum,</span>
<span id="cb2203-19"><a href="integral-population-models.html#cb2203-19" tabindex="-1"></a>    <span class="st">`</span><span class="at">Repro Output</span><span class="st">`</span> <span class="ot">=</span> TotalReproOutput</span>
<span id="cb2203-20"><a href="integral-population-models.html#cb2203-20" tabindex="-1"></a>  )</span>
<span id="cb2203-21"><a href="integral-population-models.html#cb2203-21" tabindex="-1"></a></span>
<span id="cb2203-22"><a href="integral-population-models.html#cb2203-22" tabindex="-1"></a>cor_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(cor_data, <span class="at">use =</span> <span class="st">&quot;complete.obs&quot;</span>)</span>
<span id="cb2203-23"><a href="integral-population-models.html#cb2203-23" tabindex="-1"></a></span>
<span id="cb2203-24"><a href="integral-population-models.html#cb2203-24" tabindex="-1"></a><span class="co"># Visualize correlations</span></span>
<span id="cb2203-25"><a href="integral-population-models.html#cb2203-25" tabindex="-1"></a><span class="fu">corrplot</span>(cor_matrix, <span class="at">method =</span> <span class="st">&quot;color&quot;</span>, <span class="at">type =</span> <span class="st">&quot;upper&quot;</span>,</span>
<span id="cb2203-26"><a href="integral-population-models.html#cb2203-26" tabindex="-1"></a>         <span class="at">addCoef.col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">number.cex =</span> <span class="fl">0.7</span>,</span>
<span id="cb2203-27"><a href="integral-population-models.html#cb2203-27" tabindex="-1"></a>         <span class="at">tl.col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">tl.srt =</span> <span class="dv">45</span>, <span class="at">tl.cex =</span> <span class="fl">0.8</span>,</span>
<span id="cb2203-28"><a href="integral-population-models.html#cb2203-28" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">&quot;Correlations Between Size and Reproductive Metrics&quot;</span>,</span>
<span id="cb2203-29"><a href="integral-population-models.html#cb2203-29" tabindex="-1"></a>         <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">0</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-8-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
<p>In many demographic studies, time constraints limit how long we have to identify appropriate state variables, establish a census protocol, and collect sufficient data to parameterize transition matrices. Because survival and growth require observations across multiple years, early stages of a study often rely on variables that can be measured and linked to demographic processes within a single field season. For this reason, I often prioritize selecting state variables that best explain variation in reproductive rates, since reproduction can typically be measured alongside size or trait data in the same year.</p>
<p>Below, we use model selection to identify the state variable that best predicts reproductive output. In my experience, the state variable that most strongly explains reproduction often also performs well as a predictor of other vital rates, such as survival and growth, once enough data have been collected to evaluate these relationships. This approach provides a practical and biologically informed starting point for demographic modeling when data are initially limited.</p>
</div>
<div id="selecting-our-state-variable" class="section level3 hasAnchor" number="82.2.8">
<h3><span class="header-section-number">82.2.8</span> Selecting Our State Variable<a href="integral-population-models.html#selecting-our-state-variable" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Recall that we are trying to identify state variables with the following characteristics:</p>
<ul>
<li>Continuously distributed: Not discrete categories, works well with IPM framework</li>
<li>Strong biological meaning: Represents accumulated woody growth and age</li>
<li>Practical: Easy to measure consistently across years</li>
<li>Predictive: Likely correlates with survival (established plants) and reproduction (larger plants)</li>
<li>Low measurement error: Less affected by seasonal variation</li>
</ul>
<p>So far, the majority of our variables meet these criteria, but let’s test predictability of vital rates! The ultimate decider!</p>
<div id="quantitative-model-comparison-with-aic" class="section level4 hasAnchor" number="82.2.8.1">
<h4><span class="header-section-number">82.2.8.1</span> Quantitative Model Comparison with AIC<a href="integral-population-models.html#quantitative-model-comparison-with-aic" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb2204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2204-1"><a href="integral-population-models.html#cb2204-1" tabindex="-1"></a><span class="co"># Fit Poisson GLMs for each candidate state variable</span></span>
<span id="cb2204-2"><a href="integral-population-models.html#cb2204-2" tabindex="-1"></a><span class="co"># (Poisson is appropriate for count data like reproductive output)</span></span>
<span id="cb2204-3"><a href="integral-population-models.html#cb2204-3" tabindex="-1"></a></span>
<span id="cb2204-4"><a href="integral-population-models.html#cb2204-4" tabindex="-1"></a>model_diameter <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalReproOutput <span class="sc">~</span> StemDiameter, </span>
<span id="cb2204-5"><a href="integral-population-models.html#cb2204-5" tabindex="-1"></a>                      <span class="at">family =</span> poisson, <span class="at">data =</span> silene)</span>
<span id="cb2204-6"><a href="integral-population-models.html#cb2204-6" tabindex="-1"></a></span>
<span id="cb2204-7"><a href="integral-population-models.html#cb2204-7" tabindex="-1"></a>model_height <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalReproOutput <span class="sc">~</span> VegHeight, </span>
<span id="cb2204-8"><a href="integral-population-models.html#cb2204-8" tabindex="-1"></a>                    <span class="at">family =</span> poisson, <span class="at">data =</span> silene)</span>
<span id="cb2204-9"><a href="integral-population-models.html#cb2204-9" tabindex="-1"></a></span>
<span id="cb2204-10"><a href="integral-population-models.html#cb2204-10" tabindex="-1"></a>model_height_repro <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalReproOutput <span class="sc">~</span> VegReproHeight, </span>
<span id="cb2204-11"><a href="integral-population-models.html#cb2204-11" tabindex="-1"></a>                          <span class="at">family =</span> poisson, <span class="at">data =</span> silene)</span>
<span id="cb2204-12"><a href="integral-population-models.html#cb2204-12" tabindex="-1"></a></span>
<span id="cb2204-13"><a href="integral-population-models.html#cb2204-13" tabindex="-1"></a>model_branch <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalReproOutput <span class="sc">~</span> LengthLongestBranchwFoliage, </span>
<span id="cb2204-14"><a href="integral-population-models.html#cb2204-14" tabindex="-1"></a>                    <span class="at">family =</span> poisson, <span class="at">data =</span> silene)</span>
<span id="cb2204-15"><a href="integral-population-models.html#cb2204-15" tabindex="-1"></a></span>
<span id="cb2204-16"><a href="integral-population-models.html#cb2204-16" tabindex="-1"></a>model_stems <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalReproOutput <span class="sc">~</span> StemNum, </span>
<span id="cb2204-17"><a href="integral-population-models.html#cb2204-17" tabindex="-1"></a>                   <span class="at">family =</span> poisson, <span class="at">data =</span> silene)</span>
<span id="cb2204-18"><a href="integral-population-models.html#cb2204-18" tabindex="-1"></a></span>
<span id="cb2204-19"><a href="integral-population-models.html#cb2204-19" tabindex="-1"></a><span class="co"># Create comprehensive comparison table</span></span>
<span id="cb2204-20"><a href="integral-population-models.html#cb2204-20" tabindex="-1"></a>aic_comparison <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb2204-21"><a href="integral-population-models.html#cb2204-21" tabindex="-1"></a>  <span class="at">State_Variable =</span> <span class="fu">c</span>(<span class="st">&quot;Stem Diameter&quot;</span>, </span>
<span id="cb2204-22"><a href="integral-population-models.html#cb2204-22" tabindex="-1"></a>                     <span class="st">&quot;Vegetative Height&quot;</span>, </span>
<span id="cb2204-23"><a href="integral-population-models.html#cb2204-23" tabindex="-1"></a>                     <span class="st">&quot;Height with Repro&quot;</span>,</span>
<span id="cb2204-24"><a href="integral-population-models.html#cb2204-24" tabindex="-1"></a>                     <span class="st">&quot;Branch Length&quot;</span>, </span>
<span id="cb2204-25"><a href="integral-population-models.html#cb2204-25" tabindex="-1"></a>                     <span class="st">&quot;Number of Stems&quot;</span>),</span>
<span id="cb2204-26"><a href="integral-population-models.html#cb2204-26" tabindex="-1"></a>  </span>
<span id="cb2204-27"><a href="integral-population-models.html#cb2204-27" tabindex="-1"></a>  <span class="at">AIC =</span> <span class="fu">c</span>(<span class="fu">AIC</span>(model_diameter), </span>
<span id="cb2204-28"><a href="integral-population-models.html#cb2204-28" tabindex="-1"></a>          <span class="fu">AIC</span>(model_height), </span>
<span id="cb2204-29"><a href="integral-population-models.html#cb2204-29" tabindex="-1"></a>          <span class="fu">AIC</span>(model_height_repro),</span>
<span id="cb2204-30"><a href="integral-population-models.html#cb2204-30" tabindex="-1"></a>          <span class="fu">AIC</span>(model_branch), </span>
<span id="cb2204-31"><a href="integral-population-models.html#cb2204-31" tabindex="-1"></a>          <span class="fu">AIC</span>(model_stems)),</span>
<span id="cb2204-32"><a href="integral-population-models.html#cb2204-32" tabindex="-1"></a>  </span>
<span id="cb2204-33"><a href="integral-population-models.html#cb2204-33" tabindex="-1"></a>  <span class="at">df =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>)  <span class="co"># All have 2 parameters (intercept + slope)</span></span>
<span id="cb2204-34"><a href="integral-population-models.html#cb2204-34" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb2204-35"><a href="integral-population-models.html#cb2204-35" tabindex="-1"></a>  <span class="fu">arrange</span>(AIC) <span class="sc">|&gt;</span></span>
<span id="cb2204-36"><a href="integral-population-models.html#cb2204-36" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2204-37"><a href="integral-population-models.html#cb2204-37" tabindex="-1"></a>    <span class="at">Delta_AIC =</span> AIC <span class="sc">-</span> <span class="fu">min</span>(AIC),</span>
<span id="cb2204-38"><a href="integral-population-models.html#cb2204-38" tabindex="-1"></a>    <span class="at">Relative_Likelihood =</span> <span class="fu">exp</span>(<span class="sc">-</span><span class="fl">0.5</span> <span class="sc">*</span> Delta_AIC),</span>
<span id="cb2204-39"><a href="integral-population-models.html#cb2204-39" tabindex="-1"></a>    <span class="at">AIC_Weight =</span> Relative_Likelihood <span class="sc">/</span> <span class="fu">sum</span>(Relative_Likelihood),</span>
<span id="cb2204-40"><a href="integral-population-models.html#cb2204-40" tabindex="-1"></a>    <span class="at">Interpretation =</span> <span class="fu">case_when</span>(</span>
<span id="cb2204-41"><a href="integral-population-models.html#cb2204-41" tabindex="-1"></a>      Delta_AIC <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">&quot;Best model&quot;</span>,</span>
<span id="cb2204-42"><a href="integral-population-models.html#cb2204-42" tabindex="-1"></a>      Delta_AIC <span class="sc">&lt;</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Substantial support&quot;</span>,</span>
<span id="cb2204-43"><a href="integral-population-models.html#cb2204-43" tabindex="-1"></a>      Delta_AIC <span class="sc">&lt;</span> <span class="dv">7</span> <span class="sc">~</span> <span class="st">&quot;Considerably less support&quot;</span>,</span>
<span id="cb2204-44"><a href="integral-population-models.html#cb2204-44" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Essentially no support&quot;</span></span>
<span id="cb2204-45"><a href="integral-population-models.html#cb2204-45" tabindex="-1"></a>    )</span>
<span id="cb2204-46"><a href="integral-population-models.html#cb2204-46" tabindex="-1"></a>  )</span>
<span id="cb2204-47"><a href="integral-population-models.html#cb2204-47" tabindex="-1"></a></span>
<span id="cb2204-48"><a href="integral-population-models.html#cb2204-48" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(aic_comparison, <span class="at">digits =</span> <span class="dv">3</span>,</span>
<span id="cb2204-49"><a href="integral-population-models.html#cb2204-49" tabindex="-1"></a>             <span class="at">caption =</span> <span class="st">&quot;AIC comparison: Which size metric best predicts fecundity?&quot;</span>)</span></code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-9">Table 82.2: </span>AIC comparison: Which size metric best predicts fecundity?</caption>
<colgroup>
<col width="19%" />
<col width="9%" />
<col width="3%" />
<col width="10%" />
<col width="21%" />
<col width="11%" />
<col width="24%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">State_Variable</th>
<th align="right">AIC</th>
<th align="right">df</th>
<th align="right">Delta_AIC</th>
<th align="right">Relative_Likelihood</th>
<th align="right">AIC_Weight</th>
<th align="left">Interpretation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Height with Repro</td>
<td align="right">1588.305</td>
<td align="right">2</td>
<td align="right">0.000</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">Best model</td>
</tr>
<tr class="even">
<td align="left">Vegetative Height</td>
<td align="right">1876.079</td>
<td align="right">2</td>
<td align="right">287.774</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Essentially no support</td>
</tr>
<tr class="odd">
<td align="left">Number of Stems</td>
<td align="right">2193.372</td>
<td align="right">2</td>
<td align="right">605.067</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Essentially no support</td>
</tr>
<tr class="even">
<td align="left">Stem Diameter</td>
<td align="right">3057.851</td>
<td align="right">2</td>
<td align="right">1469.546</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Essentially no support</td>
</tr>
<tr class="odd">
<td align="left">Branch Length</td>
<td align="right">3401.991</td>
<td align="right">2</td>
<td align="right">1813.686</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Essentially no support</td>
</tr>
</tbody>
</table>
<p>Now, let’s visualize the top relationship!</p>
<div class="sourceCode" id="cb2205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2205-1"><a href="integral-population-models.html#cb2205-1" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2205-2"><a href="integral-population-models.html#cb2205-2" tabindex="-1"></a></span>
<span id="cb2205-3"><a href="integral-population-models.html#cb2205-3" tabindex="-1"></a><span class="co"># Get predictions from the best model for smooth curve</span></span>
<span id="cb2205-4"><a href="integral-population-models.html#cb2205-4" tabindex="-1"></a>pred_data <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb2205-5"><a href="integral-population-models.html#cb2205-5" tabindex="-1"></a>  <span class="at">VegReproHeight =</span> <span class="fu">seq</span>(<span class="fu">min</span>(silene<span class="sc">$</span>VegReproHeight, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2205-6"><a href="integral-population-models.html#cb2205-6" tabindex="-1"></a>                       <span class="fu">max</span>(silene<span class="sc">$</span>VegReproHeight, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2205-7"><a href="integral-population-models.html#cb2205-7" tabindex="-1"></a>                       <span class="at">length.out =</span> <span class="dv">100</span>)</span>
<span id="cb2205-8"><a href="integral-population-models.html#cb2205-8" tabindex="-1"></a>)</span>
<span id="cb2205-9"><a href="integral-population-models.html#cb2205-9" tabindex="-1"></a></span>
<span id="cb2205-10"><a href="integral-population-models.html#cb2205-10" tabindex="-1"></a>pred_data<span class="sc">$</span>predicted <span class="ot">&lt;-</span> <span class="fu">predict</span>(model_height_repro, </span>
<span id="cb2205-11"><a href="integral-population-models.html#cb2205-11" tabindex="-1"></a>                               <span class="at">newdata =</span> pred_data, </span>
<span id="cb2205-12"><a href="integral-population-models.html#cb2205-12" tabindex="-1"></a>                               <span class="at">type =</span> <span class="st">&quot;response&quot;</span>)</span>
<span id="cb2205-13"><a href="integral-population-models.html#cb2205-13" tabindex="-1"></a></span>
<span id="cb2205-14"><a href="integral-population-models.html#cb2205-14" tabindex="-1"></a><span class="co"># Main scatter plot with model fit</span></span>
<span id="cb2205-15"><a href="integral-population-models.html#cb2205-15" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(silene, <span class="fu">aes</span>(<span class="at">x =</span> VegReproHeight, <span class="at">y =</span> TotalReproOutput)) <span class="sc">+</span></span>
<span id="cb2205-16"><a href="integral-population-models.html#cb2205-16" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> <span class="fu">factor</span>(Reproduced)), </span>
<span id="cb2205-17"><a href="integral-population-models.html#cb2205-17" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb2205-18"><a href="integral-population-models.html#cb2205-18" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> pred_data, </span>
<span id="cb2205-19"><a href="integral-population-models.html#cb2205-19" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> VegReproHeight, <span class="at">y =</span> predicted),</span>
<span id="cb2205-20"><a href="integral-population-models.html#cb2205-20" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">&quot;coral&quot;</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb2205-21"><a href="integral-population-models.html#cb2205-21" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;0&quot;</span> <span class="ot">=</span> <span class="st">&quot;gray60&quot;</span>, <span class="st">&quot;1&quot;</span> <span class="ot">=</span> <span class="st">&quot;steelblue&quot;</span>),</span>
<span id="cb2205-22"><a href="integral-population-models.html#cb2205-22" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;No reproduction&quot;</span>, <span class="st">&quot;Reproduced&quot;</span>),</span>
<span id="cb2205-23"><a href="integral-population-models.html#cb2205-23" tabindex="-1"></a>                     <span class="at">name =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb2205-24"><a href="integral-population-models.html#cb2205-24" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb2205-25"><a href="integral-population-models.html#cb2205-25" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Best Predictor: Height with Reproductive Structures&quot;</span>,</span>
<span id="cb2205-26"><a href="integral-population-models.html#cb2205-26" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">&quot;AIC = &quot;</span>, <span class="fu">round</span>(<span class="fu">AIC</span>(model_height_repro), <span class="dv">2</span>), </span>
<span id="cb2205-27"><a href="integral-population-models.html#cb2205-27" tabindex="-1"></a>                        <span class="st">&quot; | AIC Weight = &quot;</span>, </span>
<span id="cb2205-28"><a href="integral-population-models.html#cb2205-28" tabindex="-1"></a>                        <span class="fu">round</span>(aic_comparison<span class="sc">$</span>AIC_Weight[aic_comparison<span class="sc">$</span>State_Variable <span class="sc">==</span> <span class="st">&quot;Height with Repro&quot;</span>], <span class="dv">3</span>)),</span>
<span id="cb2205-29"><a href="integral-population-models.html#cb2205-29" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Height with reproductive structures (cm)&quot;</span>,</span>
<span id="cb2205-30"><a href="integral-population-models.html#cb2205-30" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Total reproductive output (flowers + seeds)&quot;</span>) <span class="sc">+</span></span>
<span id="cb2205-31"><a href="integral-population-models.html#cb2205-31" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;top&quot;</span>)</span>
<span id="cb2205-32"><a href="integral-population-models.html#cb2205-32" tabindex="-1"></a></span>
<span id="cb2205-33"><a href="integral-population-models.html#cb2205-33" tabindex="-1"></a><span class="co"># Residual plot to check model fit</span></span>
<span id="cb2205-34"><a href="integral-population-models.html#cb2205-34" tabindex="-1"></a>silene_resid <span class="ot">&lt;-</span> silene <span class="sc">|&gt;</span></span>
<span id="cb2205-35"><a href="integral-population-models.html#cb2205-35" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2205-36"><a href="integral-population-models.html#cb2205-36" tabindex="-1"></a>    <span class="at">fitted =</span> <span class="fu">predict</span>(model_height_repro, <span class="at">type =</span> <span class="st">&quot;response&quot;</span>),</span>
<span id="cb2205-37"><a href="integral-population-models.html#cb2205-37" tabindex="-1"></a>    <span class="at">residuals =</span> TotalReproOutput <span class="sc">-</span> fitted</span>
<span id="cb2205-38"><a href="integral-population-models.html#cb2205-38" tabindex="-1"></a>  )</span>
<span id="cb2205-39"><a href="integral-population-models.html#cb2205-39" tabindex="-1"></a></span>
<span id="cb2205-40"><a href="integral-population-models.html#cb2205-40" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(silene_resid, <span class="fu">aes</span>(<span class="at">x =</span> fitted, <span class="at">y =</span> residuals)) <span class="sc">+</span></span>
<span id="cb2205-41"><a href="integral-population-models.html#cb2205-41" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">color =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="sc">+</span></span>
<span id="cb2205-42"><a href="integral-population-models.html#cb2205-42" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;coral&quot;</span>) <span class="sc">+</span></span>
<span id="cb2205-43"><a href="integral-population-models.html#cb2205-43" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">color =</span> <span class="st">&quot;gray30&quot;</span>) <span class="sc">+</span></span>
<span id="cb2205-44"><a href="integral-population-models.html#cb2205-44" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb2205-45"><a href="integral-population-models.html#cb2205-45" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Model Diagnostics: Residual Plot&quot;</span>,</span>
<span id="cb2205-46"><a href="integral-population-models.html#cb2205-46" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Fitted values&quot;</span>,</span>
<span id="cb2205-47"><a href="integral-population-models.html#cb2205-47" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Residuals&quot;</span>)</span>
<span id="cb2205-48"><a href="integral-population-models.html#cb2205-48" tabindex="-1"></a></span>
<span id="cb2205-49"><a href="integral-population-models.html#cb2205-49" tabindex="-1"></a>p1 <span class="sc">/</span> p2 <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-10-1.png" alt="" width="960" style="display: block; margin: auto;" />
<strong>Top Panel: The Size-Fecundity Relationship</strong></p>
<p>Let’s check out the figure and think about the implications:</p>
<ol style="list-style-type: decimal">
<li><strong>The Zero-Inflation Problem</strong>: Notice the large cluster of gray points at reproductive output = 0. These are plants that didn’t reproduce at all (56% of individuals!). This creates a challenge for Poisson models, which don’t naturally handle “excess zeros.”</li>
<li><strong>Clear Size Threshold</strong>: Among reproducing plants (blue), there’s a strong positive relationship - taller plants with larger reproductive structures produce more seeds. But notice that ALL small plants (&lt;~40 cm) failed to reproduce. This suggests a <strong>reproductive size threshold</strong>.</li>
<li><strong>Increasing Variance</strong>: As height increases, the spread of blue points gets wider. Small reproductive plants show relatively consistent output (~10-30 seeds), while large plants vary dramatically (20-100 seeds). This <strong>heteroscedasticity</strong> (unequal variance) can be an issue for simple linear models.</li>
<li><strong>The Perfect Fit “Problem”</strong>: The coral line fits the reproducing plants beautifully, BUT this creates a conceptual issue for IPMs: <code>VegReproHeight</code> includes the reproductive stalks themselves! This is somewhat circular - we’re using reproductive structures to predict reproduction.</li>
</ol>
<p><strong>Bottom Panel: Model Diagnostics (Critical!)</strong></p>
<p>Let’s check out the figure and think about the implications:</p>
<ol start="5" style="list-style-type: decimal">
<li><strong>Non-Random Residual Pattern</strong>: The residuals should be randomly scattered around zero (the dashed orange line). Instead, notice:
<ul>
<li><strong>Left side</strong> (fitted values 0-25): Residuals are tightly clustered near zero (all those non-reproductive plants)</li>
<li><strong>Middle</strong> (fitted values 25-50): Residuals are nicely random ✓</li>
<li><strong>Right side</strong> (fitted values &gt;75): The smooth line curves downward, indicating the model <strong>underpredicts</strong> reproductive output for the largest plants</li>
</ul></li>
<li><strong>The Outlier</strong>: See that point at the far right with residual ≈ -75? That’s a very large plant that produced MUCH less than expected. What happened? (Disease? Herbivory? Late-season measurement?)</li>
</ol>
<p><strong>What This Means for Model Selection:</strong></p>
<p>Let’s think this through:</p>
<ul>
<li><strong>Statistically</strong>: VegReproHeight has the best AIC, BUT the residual pattern suggests the Poisson model isn’t perfect.</li>
<li><strong>Biologically</strong>: This relationship makes sense - plants invest in tall reproductive stalks to display flowers and disperse seeds.</li>
</ul>
<p>A classic (and very common) way to handle “zero inflation” in plant fecundity when building IPMs is to use a two‐part (hurdle) model. In this approach, we first model whether an individual reproduces at all (i.e., produces any flowers/fruits/seeds) using a binomial model. Then, conditional on reproduction occurring, we model the amount of reproductive output (e.g., number of fruits or seeds) using a count model such as a Poisson (or often a negative binomial if counts are overdispersed). This is often more biologically realistic because it separates two distinct processes: (1) the decision/ability to reproduce and (2) how much is produced once reproduction happens.</p>
<p>A zero-inflated Poisson (ZIP) or zero-inflated negative binomial (ZINB) also addresses excess zeros, but it assumes the zeros come from two sources:</p>
<pre><code>1.  Structural zeros: individuals in a “certain zero” state (they cannot reproduce / are not eligible)
2.  Sampling zeros: individuals who are “eligible,” but still produce zero by chance under the count distribution</code></pre>
<p>Thus, in the hurdle model, all zeros are handled by the binomial part; the count model is fit to positive counts only. In the zero-inflated model, zeros can come from either the inflation process or the count process. The hurdle (binomial + conditional count) approach aligns well with plant reproductive biology because:</p>
<ul>
<li>Reproduction often reflects a threshold-like process (size/condition must be sufficient to flower/fruit).</li>
<li>Once reproducing, output typically scales with size and resources.</li>
<li>It yields two interpretable functions for the IPM:</li>
<li>probability of reproduction as a function of size</li>
<li>expected fecundity given reproduction as a function of size</li>
<li>and, bonus, it’s also often simpler to implement and communicate in an IPM context</li>
</ul>
<p>Let’s check this out:</p>
<div class="sourceCode" id="cb2207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2207-1"><a href="integral-population-models.html#cb2207-1" tabindex="-1"></a><span class="co"># Fit logistic regression for each candidate state variable</span></span>
<span id="cb2207-2"><a href="integral-population-models.html#cb2207-2" tabindex="-1"></a><span class="co"># Response: Reproduced (0 = no, 1 = yes)</span></span>
<span id="cb2207-3"><a href="integral-population-models.html#cb2207-3" tabindex="-1"></a></span>
<span id="cb2207-4"><a href="integral-population-models.html#cb2207-4" tabindex="-1"></a>logit_diameter <span class="ot">&lt;-</span> <span class="fu">glm</span>(Reproduced <span class="sc">~</span> VegReproHeight, </span>
<span id="cb2207-5"><a href="integral-population-models.html#cb2207-5" tabindex="-1"></a>                      <span class="at">family =</span> binomial, <span class="at">data =</span> silene)</span>
<span id="cb2207-6"><a href="integral-population-models.html#cb2207-6" tabindex="-1"></a></span>
<span id="cb2207-7"><a href="integral-population-models.html#cb2207-7" tabindex="-1"></a>logit_height <span class="ot">&lt;-</span> <span class="fu">glm</span>(Reproduced <span class="sc">~</span> VegHeight, </span>
<span id="cb2207-8"><a href="integral-population-models.html#cb2207-8" tabindex="-1"></a>                    <span class="at">family =</span> binomial, <span class="at">data =</span> silene)</span>
<span id="cb2207-9"><a href="integral-population-models.html#cb2207-9" tabindex="-1"></a></span>
<span id="cb2207-10"><a href="integral-population-models.html#cb2207-10" tabindex="-1"></a>logit_height_repro <span class="ot">&lt;-</span> <span class="fu">glm</span>(Reproduced <span class="sc">~</span> VegReproHeight, </span>
<span id="cb2207-11"><a href="integral-population-models.html#cb2207-11" tabindex="-1"></a>                          <span class="at">family =</span> binomial, <span class="at">data =</span> silene)</span>
<span id="cb2207-12"><a href="integral-population-models.html#cb2207-12" tabindex="-1"></a></span>
<span id="cb2207-13"><a href="integral-population-models.html#cb2207-13" tabindex="-1"></a>logit_branch <span class="ot">&lt;-</span> <span class="fu">glm</span>(Reproduced <span class="sc">~</span> LengthLongestBranchwFoliage, </span>
<span id="cb2207-14"><a href="integral-population-models.html#cb2207-14" tabindex="-1"></a>                    <span class="at">family =</span> binomial, <span class="at">data =</span> silene)</span>
<span id="cb2207-15"><a href="integral-population-models.html#cb2207-15" tabindex="-1"></a></span>
<span id="cb2207-16"><a href="integral-population-models.html#cb2207-16" tabindex="-1"></a>logit_stems <span class="ot">&lt;-</span> <span class="fu">glm</span>(Reproduced <span class="sc">~</span> StemNum, </span>
<span id="cb2207-17"><a href="integral-population-models.html#cb2207-17" tabindex="-1"></a>                   <span class="at">family =</span> binomial, <span class="at">data =</span> silene)</span>
<span id="cb2207-18"><a href="integral-population-models.html#cb2207-18" tabindex="-1"></a></span>
<span id="cb2207-19"><a href="integral-population-models.html#cb2207-19" tabindex="-1"></a><span class="co"># Compare models with AIC</span></span>
<span id="cb2207-20"><a href="integral-population-models.html#cb2207-20" tabindex="-1"></a>aic_reproduction <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb2207-21"><a href="integral-population-models.html#cb2207-21" tabindex="-1"></a>  <span class="at">State_Variable =</span> <span class="fu">c</span>(<span class="st">&quot;Stem Diameter&quot;</span>, <span class="st">&quot;Vegetative Height&quot;</span>, </span>
<span id="cb2207-22"><a href="integral-population-models.html#cb2207-22" tabindex="-1"></a>                     <span class="st">&quot;Height with Repro&quot;</span>, <span class="st">&quot;Branch Length&quot;</span>, <span class="st">&quot;Number of Stems&quot;</span>),</span>
<span id="cb2207-23"><a href="integral-population-models.html#cb2207-23" tabindex="-1"></a>  <span class="at">AIC =</span> <span class="fu">c</span>(<span class="fu">AIC</span>(logit_diameter), <span class="fu">AIC</span>(logit_height), </span>
<span id="cb2207-24"><a href="integral-population-models.html#cb2207-24" tabindex="-1"></a>          <span class="fu">AIC</span>(logit_height_repro), <span class="fu">AIC</span>(logit_branch), <span class="fu">AIC</span>(logit_stems))</span>
<span id="cb2207-25"><a href="integral-population-models.html#cb2207-25" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb2207-26"><a href="integral-population-models.html#cb2207-26" tabindex="-1"></a>  <span class="fu">arrange</span>(AIC) <span class="sc">|&gt;</span></span>
<span id="cb2207-27"><a href="integral-population-models.html#cb2207-27" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2207-28"><a href="integral-population-models.html#cb2207-28" tabindex="-1"></a>    <span class="at">Delta_AIC =</span> AIC <span class="sc">-</span> <span class="fu">min</span>(AIC),</span>
<span id="cb2207-29"><a href="integral-population-models.html#cb2207-29" tabindex="-1"></a>    <span class="at">AIC_Weight =</span> <span class="fu">exp</span>(<span class="sc">-</span><span class="fl">0.5</span> <span class="sc">*</span> Delta_AIC) <span class="sc">/</span> <span class="fu">sum</span>(<span class="fu">exp</span>(<span class="sc">-</span><span class="fl">0.5</span> <span class="sc">*</span> Delta_AIC))</span>
<span id="cb2207-30"><a href="integral-population-models.html#cb2207-30" tabindex="-1"></a>  )</span>
<span id="cb2207-31"><a href="integral-population-models.html#cb2207-31" tabindex="-1"></a></span>
<span id="cb2207-32"><a href="integral-population-models.html#cb2207-32" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(aic_reproduction, <span class="at">digits =</span> <span class="dv">3</span>,</span>
<span id="cb2207-33"><a href="integral-population-models.html#cb2207-33" tabindex="-1"></a>             <span class="at">caption =</span> <span class="st">&quot;Step 1: Which size metric best predicts probability of reproduction?&quot;</span>)</span></code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-11">Table 82.3: </span>Step 1: Which size metric best predicts probability of reproduction?</caption>
<thead>
<tr class="header">
<th align="left">State_Variable</th>
<th align="right">AIC</th>
<th align="right">Delta_AIC</th>
<th align="right">AIC_Weight</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Stem Diameter</td>
<td align="right">38.661</td>
<td align="right">0.000</td>
<td align="right">0.497</td>
</tr>
<tr class="even">
<td align="left">Height with Repro</td>
<td align="right">38.661</td>
<td align="right">0.000</td>
<td align="right">0.497</td>
</tr>
<tr class="odd">
<td align="left">Number of Stems</td>
<td align="right">48.159</td>
<td align="right">9.498</td>
<td align="right">0.004</td>
</tr>
<tr class="even">
<td align="left">Vegetative Height</td>
<td align="right">50.583</td>
<td align="right">11.923</td>
<td align="right">0.001</td>
</tr>
<tr class="odd">
<td align="left">Branch Length</td>
<td align="right">129.301</td>
<td align="right">90.640</td>
<td align="right">0.000</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb2208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2208-1"><a href="integral-population-models.html#cb2208-1" tabindex="-1"></a><span class="co"># Filter to plants that reproduced</span></span>
<span id="cb2208-2"><a href="integral-population-models.html#cb2208-2" tabindex="-1"></a>silene_reproductive <span class="ot">&lt;-</span> silene <span class="sc">|&gt;</span></span>
<span id="cb2208-3"><a href="integral-population-models.html#cb2208-3" tabindex="-1"></a>  <span class="fu">filter</span>(Reproduced <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb2208-4"><a href="integral-population-models.html#cb2208-4" tabindex="-1"></a></span>
<span id="cb2208-5"><a href="integral-population-models.html#cb2208-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Reproductive plants:&quot;</span>, <span class="fu">nrow</span>(silene_reproductive), </span>
<span id="cb2208-6"><a href="integral-population-models.html#cb2208-6" tabindex="-1"></a>    <span class="st">&quot;out of&quot;</span>, <span class="fu">nrow</span>(silene), </span>
<span id="cb2208-7"><a href="integral-population-models.html#cb2208-7" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="st">&quot;(&quot;</span>, <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> <span class="fu">nrow</span>(silene_reproductive)<span class="sc">/</span><span class="fu">nrow</span>(silene), <span class="dv">1</span>), <span class="st">&quot;%)</span><span class="sc">\n</span><span class="st">&quot;</span>))</span></code></pre></div>
<pre><code>## Reproductive plants: 44 out of 100 (44%)</code></pre>
<div class="sourceCode" id="cb2210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2210-1"><a href="integral-population-models.html#cb2210-1" tabindex="-1"></a><span class="co"># Fit Poisson regression for seed production (reproductive plants only)</span></span>
<span id="cb2210-2"><a href="integral-population-models.html#cb2210-2" tabindex="-1"></a>pois_diameter <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalReproOutput <span class="sc">~</span> VegReproHeight, </span>
<span id="cb2210-3"><a href="integral-population-models.html#cb2210-3" tabindex="-1"></a>                     <span class="at">family =</span> poisson, <span class="at">data =</span> silene_reproductive)</span>
<span id="cb2210-4"><a href="integral-population-models.html#cb2210-4" tabindex="-1"></a></span>
<span id="cb2210-5"><a href="integral-population-models.html#cb2210-5" tabindex="-1"></a>pois_height <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalReproOutput <span class="sc">~</span> VegHeight, </span>
<span id="cb2210-6"><a href="integral-population-models.html#cb2210-6" tabindex="-1"></a>                   <span class="at">family =</span> poisson, <span class="at">data =</span> silene_reproductive)</span>
<span id="cb2210-7"><a href="integral-population-models.html#cb2210-7" tabindex="-1"></a></span>
<span id="cb2210-8"><a href="integral-population-models.html#cb2210-8" tabindex="-1"></a>pois_height_repro <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalReproOutput <span class="sc">~</span> VegReproHeight, </span>
<span id="cb2210-9"><a href="integral-population-models.html#cb2210-9" tabindex="-1"></a>                         <span class="at">family =</span> poisson, <span class="at">data =</span> silene_reproductive)</span>
<span id="cb2210-10"><a href="integral-population-models.html#cb2210-10" tabindex="-1"></a></span>
<span id="cb2210-11"><a href="integral-population-models.html#cb2210-11" tabindex="-1"></a>pois_branch <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalReproOutput <span class="sc">~</span> LengthLongestBranchwFoliage, </span>
<span id="cb2210-12"><a href="integral-population-models.html#cb2210-12" tabindex="-1"></a>                   <span class="at">family =</span> poisson, <span class="at">data =</span> silene_reproductive)</span>
<span id="cb2210-13"><a href="integral-population-models.html#cb2210-13" tabindex="-1"></a></span>
<span id="cb2210-14"><a href="integral-population-models.html#cb2210-14" tabindex="-1"></a>pois_stems <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalReproOutput <span class="sc">~</span> StemNum, </span>
<span id="cb2210-15"><a href="integral-population-models.html#cb2210-15" tabindex="-1"></a>                  <span class="at">family =</span> poisson, <span class="at">data =</span> silene_reproductive)</span>
<span id="cb2210-16"><a href="integral-population-models.html#cb2210-16" tabindex="-1"></a></span>
<span id="cb2210-17"><a href="integral-population-models.html#cb2210-17" tabindex="-1"></a><span class="co"># Compare models</span></span>
<span id="cb2210-18"><a href="integral-population-models.html#cb2210-18" tabindex="-1"></a>aic_seed_production <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb2210-19"><a href="integral-population-models.html#cb2210-19" tabindex="-1"></a>  <span class="at">State_Variable =</span> <span class="fu">c</span>(<span class="st">&quot;Stem Diameter&quot;</span>, <span class="st">&quot;Vegetative Height&quot;</span>, </span>
<span id="cb2210-20"><a href="integral-population-models.html#cb2210-20" tabindex="-1"></a>                     <span class="st">&quot;Height with Repro&quot;</span>, <span class="st">&quot;Branch Length&quot;</span>, <span class="st">&quot;Number of Stems&quot;</span>),</span>
<span id="cb2210-21"><a href="integral-population-models.html#cb2210-21" tabindex="-1"></a>  <span class="at">AIC =</span> <span class="fu">c</span>(<span class="fu">AIC</span>(pois_diameter), <span class="fu">AIC</span>(pois_height), </span>
<span id="cb2210-22"><a href="integral-population-models.html#cb2210-22" tabindex="-1"></a>          <span class="fu">AIC</span>(pois_height_repro), <span class="fu">AIC</span>(pois_branch), <span class="fu">AIC</span>(pois_stems))</span>
<span id="cb2210-23"><a href="integral-population-models.html#cb2210-23" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb2210-24"><a href="integral-population-models.html#cb2210-24" tabindex="-1"></a>  <span class="fu">arrange</span>(AIC) <span class="sc">|&gt;</span></span>
<span id="cb2210-25"><a href="integral-population-models.html#cb2210-25" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2210-26"><a href="integral-population-models.html#cb2210-26" tabindex="-1"></a>    <span class="at">Delta_AIC =</span> AIC <span class="sc">-</span> <span class="fu">min</span>(AIC),</span>
<span id="cb2210-27"><a href="integral-population-models.html#cb2210-27" tabindex="-1"></a>    <span class="at">AIC_Weight =</span> <span class="fu">exp</span>(<span class="sc">-</span><span class="fl">0.5</span> <span class="sc">*</span> Delta_AIC) <span class="sc">/</span> <span class="fu">sum</span>(<span class="fu">exp</span>(<span class="sc">-</span><span class="fl">0.5</span> <span class="sc">*</span> Delta_AIC))</span>
<span id="cb2210-28"><a href="integral-population-models.html#cb2210-28" tabindex="-1"></a>  )</span>
<span id="cb2210-29"><a href="integral-population-models.html#cb2210-29" tabindex="-1"></a></span>
<span id="cb2210-30"><a href="integral-population-models.html#cb2210-30" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(aic_seed_production, <span class="at">digits =</span> <span class="dv">3</span>,</span>
<span id="cb2210-31"><a href="integral-population-models.html#cb2210-31" tabindex="-1"></a>             <span class="at">caption =</span> <span class="st">&quot;Step 2: Which size metric best predicts seed count among reproducers?&quot;</span>)</span></code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-11">Table 82.3: </span>Step 2: Which size metric best predicts seed count among reproducers?</caption>
<thead>
<tr class="header">
<th align="left">State_Variable</th>
<th align="right">AIC</th>
<th align="right">Delta_AIC</th>
<th align="right">AIC_Weight</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Number of Stems</td>
<td align="right">726.427</td>
<td align="right">0.000</td>
<td align="right">0.813</td>
</tr>
<tr class="even">
<td align="left">Stem Diameter</td>
<td align="right">730.757</td>
<td align="right">4.331</td>
<td align="right">0.093</td>
</tr>
<tr class="odd">
<td align="left">Height with Repro</td>
<td align="right">730.757</td>
<td align="right">4.331</td>
<td align="right">0.093</td>
</tr>
<tr class="even">
<td align="left">Vegetative Height</td>
<td align="right">747.756</td>
<td align="right">21.329</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">Branch Length</td>
<td align="right">798.683</td>
<td align="right">72.256</td>
<td align="right">0.000</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb2211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2211-1"><a href="integral-population-models.html#cb2211-1" tabindex="-1"></a><span class="co"># Combine the two tables</span></span>
<span id="cb2211-2"><a href="integral-population-models.html#cb2211-2" tabindex="-1"></a>combined_aic <span class="ot">&lt;-</span> aic_reproduction <span class="sc">|&gt;</span></span>
<span id="cb2211-3"><a href="integral-population-models.html#cb2211-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb2211-4"><a href="integral-population-models.html#cb2211-4" tabindex="-1"></a>    <span class="at">AIC_Reproduction =</span> AIC,</span>
<span id="cb2211-5"><a href="integral-population-models.html#cb2211-5" tabindex="-1"></a>    <span class="at">Weight_Reproduction =</span> AIC_Weight</span>
<span id="cb2211-6"><a href="integral-population-models.html#cb2211-6" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2211-7"><a href="integral-population-models.html#cb2211-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb2211-8"><a href="integral-population-models.html#cb2211-8" tabindex="-1"></a>    State_Variable,</span>
<span id="cb2211-9"><a href="integral-population-models.html#cb2211-9" tabindex="-1"></a>    AIC_Reproduction,</span>
<span id="cb2211-10"><a href="integral-population-models.html#cb2211-10" tabindex="-1"></a>    Weight_Reproduction</span>
<span id="cb2211-11"><a href="integral-population-models.html#cb2211-11" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2211-12"><a href="integral-population-models.html#cb2211-12" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(</span>
<span id="cb2211-13"><a href="integral-population-models.html#cb2211-13" tabindex="-1"></a>    aic_seed_production <span class="sc">|&gt;</span></span>
<span id="cb2211-14"><a href="integral-population-models.html#cb2211-14" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb2211-15"><a href="integral-population-models.html#cb2211-15" tabindex="-1"></a>        <span class="at">AIC_SeedCount =</span> AIC,</span>
<span id="cb2211-16"><a href="integral-population-models.html#cb2211-16" tabindex="-1"></a>        <span class="at">Weight_SeedCount =</span> AIC_Weight</span>
<span id="cb2211-17"><a href="integral-population-models.html#cb2211-17" tabindex="-1"></a>      ) <span class="sc">|&gt;</span></span>
<span id="cb2211-18"><a href="integral-population-models.html#cb2211-18" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb2211-19"><a href="integral-population-models.html#cb2211-19" tabindex="-1"></a>        State_Variable,</span>
<span id="cb2211-20"><a href="integral-population-models.html#cb2211-20" tabindex="-1"></a>        AIC_SeedCount,</span>
<span id="cb2211-21"><a href="integral-population-models.html#cb2211-21" tabindex="-1"></a>        Weight_SeedCount</span>
<span id="cb2211-22"><a href="integral-population-models.html#cb2211-22" tabindex="-1"></a>      ),</span>
<span id="cb2211-23"><a href="integral-population-models.html#cb2211-23" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">&quot;State_Variable&quot;</span></span>
<span id="cb2211-24"><a href="integral-population-models.html#cb2211-24" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2211-25"><a href="integral-population-models.html#cb2211-25" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb2211-26"><a href="integral-population-models.html#cb2211-26" tabindex="-1"></a>    <span class="at">Combined_Weight =</span> Weight_Reproduction <span class="sc">*</span> Weight_SeedCount,</span>
<span id="cb2211-27"><a href="integral-population-models.html#cb2211-27" tabindex="-1"></a>    <span class="at">Rank_Reproduction =</span> <span class="fu">rank</span>(AIC_Reproduction),</span>
<span id="cb2211-28"><a href="integral-population-models.html#cb2211-28" tabindex="-1"></a>    <span class="at">Rank_SeedCount =</span> <span class="fu">rank</span>(AIC_SeedCount),</span>
<span id="cb2211-29"><a href="integral-population-models.html#cb2211-29" tabindex="-1"></a>    <span class="at">Average_Rank =</span> (Rank_Reproduction <span class="sc">+</span> Rank_SeedCount) <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb2211-30"><a href="integral-population-models.html#cb2211-30" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2211-31"><a href="integral-population-models.html#cb2211-31" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(Average_Rank)</span>
<span id="cb2211-32"><a href="integral-population-models.html#cb2211-32" tabindex="-1"></a></span>
<span id="cb2211-33"><a href="integral-population-models.html#cb2211-33" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(combined_aic, <span class="at">digits =</span> <span class="dv">3</span>,</span>
<span id="cb2211-34"><a href="integral-population-models.html#cb2211-34" tabindex="-1"></a>             <span class="at">caption =</span> <span class="st">&quot;Combined performance: Which variable is best at BOTH tasks?&quot;</span>)</span></code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-11">Table 82.3: </span>Combined performance: Which variable is best at BOTH tasks?</caption>
<colgroup>
<col width="12%" />
<col width="11%" />
<col width="13%" />
<col width="9%" />
<col width="11%" />
<col width="10%" />
<col width="12%" />
<col width="10%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">State_Variable</th>
<th align="right">AIC_Reproduction</th>
<th align="right">Weight_Reproduction</th>
<th align="right">AIC_SeedCount</th>
<th align="right">Weight_SeedCount</th>
<th align="right">Combined_Weight</th>
<th align="right">Rank_Reproduction</th>
<th align="right">Rank_SeedCount</th>
<th align="right">Average_Rank</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Stem Diameter</td>
<td align="right">38.661</td>
<td align="right">0.497</td>
<td align="right">730.757</td>
<td align="right">0.093</td>
<td align="right">0.046</td>
<td align="right">1.5</td>
<td align="right">2.5</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">Height with Repro</td>
<td align="right">38.661</td>
<td align="right">0.497</td>
<td align="right">730.757</td>
<td align="right">0.093</td>
<td align="right">0.046</td>
<td align="right">1.5</td>
<td align="right">2.5</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Number of Stems</td>
<td align="right">48.159</td>
<td align="right">0.004</td>
<td align="right">726.427</td>
<td align="right">0.813</td>
<td align="right">0.004</td>
<td align="right">3.0</td>
<td align="right">1.0</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">Vegetative Height</td>
<td align="right">50.583</td>
<td align="right">0.001</td>
<td align="right">747.756</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">4.0</td>
<td align="right">4.0</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">Branch Length</td>
<td align="right">129.301</td>
<td align="right">0.000</td>
<td align="right">798.683</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">5.0</td>
<td align="right">5.0</td>
<td align="right">5</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb2212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2212-1"><a href="integral-population-models.html#cb2212-1" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2212-2"><a href="integral-population-models.html#cb2212-2" tabindex="-1"></a></span>
<span id="cb2212-3"><a href="integral-population-models.html#cb2212-3" tabindex="-1"></a><span class="co"># Use the best model (let&#39;s assume it&#39;s stem diameter for example)</span></span>
<span id="cb2212-4"><a href="integral-population-models.html#cb2212-4" tabindex="-1"></a><span class="co"># Create prediction data</span></span>
<span id="cb2212-5"><a href="integral-population-models.html#cb2212-5" tabindex="-1"></a>pred_data <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb2212-6"><a href="integral-population-models.html#cb2212-6" tabindex="-1"></a>  <span class="at">VegReproHeight =</span> <span class="fu">seq</span>(<span class="fu">min</span>(silene<span class="sc">$</span>VegReproHeight, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2212-7"><a href="integral-population-models.html#cb2212-7" tabindex="-1"></a>                     <span class="fu">max</span>(silene<span class="sc">$</span>VegReproHeight, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2212-8"><a href="integral-population-models.html#cb2212-8" tabindex="-1"></a>                     <span class="at">length.out =</span> <span class="dv">100</span>)</span>
<span id="cb2212-9"><a href="integral-population-models.html#cb2212-9" tabindex="-1"></a>)</span>
<span id="cb2212-10"><a href="integral-population-models.html#cb2212-10" tabindex="-1"></a></span>
<span id="cb2212-11"><a href="integral-population-models.html#cb2212-11" tabindex="-1"></a><span class="co"># Predict probability of reproduction</span></span>
<span id="cb2212-12"><a href="integral-population-models.html#cb2212-12" tabindex="-1"></a>pred_data<span class="sc">$</span>prob_reproduce <span class="ot">&lt;-</span> <span class="fu">predict</span>(logit_diameter, </span>
<span id="cb2212-13"><a href="integral-population-models.html#cb2212-13" tabindex="-1"></a>                                    <span class="at">newdata =</span> pred_data, </span>
<span id="cb2212-14"><a href="integral-population-models.html#cb2212-14" tabindex="-1"></a>                                    <span class="at">type =</span> <span class="st">&quot;response&quot;</span>)</span>
<span id="cb2212-15"><a href="integral-population-models.html#cb2212-15" tabindex="-1"></a></span>
<span id="cb2212-16"><a href="integral-population-models.html#cb2212-16" tabindex="-1"></a><span class="co"># Predict seed count (on log scale, then back-transform)</span></span>
<span id="cb2212-17"><a href="integral-population-models.html#cb2212-17" tabindex="-1"></a>pred_data<span class="sc">$</span>expected_seeds <span class="ot">&lt;-</span> <span class="fu">predict</span>(pois_diameter, </span>
<span id="cb2212-18"><a href="integral-population-models.html#cb2212-18" tabindex="-1"></a>                                   <span class="at">newdata =</span> pred_data, </span>
<span id="cb2212-19"><a href="integral-population-models.html#cb2212-19" tabindex="-1"></a>                                   <span class="at">type =</span> <span class="st">&quot;response&quot;</span>)</span>
<span id="cb2212-20"><a href="integral-population-models.html#cb2212-20" tabindex="-1"></a></span>
<span id="cb2212-21"><a href="integral-population-models.html#cb2212-21" tabindex="-1"></a><span class="co"># Combined expected output</span></span>
<span id="cb2212-22"><a href="integral-population-models.html#cb2212-22" tabindex="-1"></a>pred_data<span class="sc">$</span>expected_total <span class="ot">&lt;-</span> pred_data<span class="sc">$</span>prob_reproduce <span class="sc">*</span> pred_data<span class="sc">$</span>expected_seeds</span>
<span id="cb2212-23"><a href="integral-population-models.html#cb2212-23" tabindex="-1"></a></span>
<span id="cb2212-24"><a href="integral-population-models.html#cb2212-24" tabindex="-1"></a><span class="co"># Panel 1: Probability of Reproduction</span></span>
<span id="cb2212-25"><a href="integral-population-models.html#cb2212-25" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(silene, <span class="fu">aes</span>(<span class="at">x =</span> VegReproHeight, <span class="at">y =</span> Reproduced)) <span class="sc">+</span></span>
<span id="cb2212-26"><a href="integral-population-models.html#cb2212-26" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">size =</span> <span class="dv">2</span>, </span>
<span id="cb2212-27"><a href="integral-population-models.html#cb2212-27" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">height =</span> <span class="fl">0.02</span>, <span class="at">width =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb2212-28"><a href="integral-population-models.html#cb2212-28" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> pred_data, </span>
<span id="cb2212-29"><a href="integral-population-models.html#cb2212-29" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> VegReproHeight, <span class="at">y =</span> prob_reproduce),</span>
<span id="cb2212-30"><a href="integral-population-models.html#cb2212-30" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">&quot;coral&quot;</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb2212-31"><a href="integral-population-models.html#cb2212-31" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb2212-32"><a href="integral-population-models.html#cb2212-32" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Step 1: Probability of Reproducing&quot;</span>,</span>
<span id="cb2212-33"><a href="integral-population-models.html#cb2212-33" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;Logistic regression on all plants&quot;</span>,</span>
<span id="cb2212-34"><a href="integral-population-models.html#cb2212-34" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Stem diameter (cm)&quot;</span>,</span>
<span id="cb2212-35"><a href="integral-population-models.html#cb2212-35" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;P(Reproduction)&quot;</span>) <span class="sc">+</span></span>
<span id="cb2212-36"><a href="integral-population-models.html#cb2212-36" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.05</span>, <span class="fl">1.05</span>)</span>
<span id="cb2212-37"><a href="integral-population-models.html#cb2212-37" tabindex="-1"></a></span>
<span id="cb2212-38"><a href="integral-population-models.html#cb2212-38" tabindex="-1"></a><span class="co"># Panel 2: Seeds IF reproducing</span></span>
<span id="cb2212-39"><a href="integral-population-models.html#cb2212-39" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(silene_reproductive, <span class="fu">aes</span>(<span class="at">x =</span> VegReproHeight, <span class="at">y =</span> TotalReproOutput)) <span class="sc">+</span></span>
<span id="cb2212-40"><a href="integral-population-models.html#cb2212-40" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">color =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="sc">+</span></span>
<span id="cb2212-41"><a href="integral-population-models.html#cb2212-41" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> pred_data, </span>
<span id="cb2212-42"><a href="integral-population-models.html#cb2212-42" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> VegReproHeight, <span class="at">y =</span> expected_seeds),</span>
<span id="cb2212-43"><a href="integral-population-models.html#cb2212-43" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">&quot;coral&quot;</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb2212-44"><a href="integral-population-models.html#cb2212-44" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb2212-45"><a href="integral-population-models.html#cb2212-45" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Step 2: Seeds IF Reproducing&quot;</span>,</span>
<span id="cb2212-46"><a href="integral-population-models.html#cb2212-46" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;Poisson regression on reproductive plants only&quot;</span>,</span>
<span id="cb2212-47"><a href="integral-population-models.html#cb2212-47" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Stem diameter (cm)&quot;</span>,</span>
<span id="cb2212-48"><a href="integral-population-models.html#cb2212-48" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Seed count (given reproduction)&quot;</span>)</span>
<span id="cb2212-49"><a href="integral-population-models.html#cb2212-49" tabindex="-1"></a></span>
<span id="cb2212-50"><a href="integral-population-models.html#cb2212-50" tabindex="-1"></a><span class="co"># Panel 3: Combined expected reproduction</span></span>
<span id="cb2212-51"><a href="integral-population-models.html#cb2212-51" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(silene, <span class="fu">aes</span>(<span class="at">x =</span> VegReproHeight, <span class="at">y =</span> TotalReproOutput)) <span class="sc">+</span></span>
<span id="cb2212-52"><a href="integral-population-models.html#cb2212-52" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> <span class="fu">factor</span>(Reproduced)), <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb2212-53"><a href="integral-population-models.html#cb2212-53" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> pred_data, </span>
<span id="cb2212-54"><a href="integral-population-models.html#cb2212-54" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> VegReproHeight, <span class="at">y =</span> expected_total),</span>
<span id="cb2212-55"><a href="integral-population-models.html#cb2212-55" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">&quot;coral&quot;</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb2212-56"><a href="integral-population-models.html#cb2212-56" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;0&quot;</span> <span class="ot">=</span> <span class="st">&quot;gray60&quot;</span>, <span class="st">&quot;1&quot;</span> <span class="ot">=</span> <span class="st">&quot;steelblue&quot;</span>),</span>
<span id="cb2212-57"><a href="integral-population-models.html#cb2212-57" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Did not reproduce&quot;</span>, <span class="st">&quot;Reproduced&quot;</span>),</span>
<span id="cb2212-58"><a href="integral-population-models.html#cb2212-58" tabindex="-1"></a>                     <span class="at">name =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb2212-59"><a href="integral-population-models.html#cb2212-59" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb2212-60"><a href="integral-population-models.html#cb2212-60" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Combined: Expected Reproduction&quot;</span>,</span>
<span id="cb2212-61"><a href="integral-population-models.html#cb2212-61" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;E[seeds] = P(reproduce) × E[seeds | reproduce]&quot;</span>,</span>
<span id="cb2212-62"><a href="integral-population-models.html#cb2212-62" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Stem diameter (cm)&quot;</span>,</span>
<span id="cb2212-63"><a href="integral-population-models.html#cb2212-63" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Expected reproductive output&quot;</span>) <span class="sc">+</span></span>
<span id="cb2212-64"><a href="integral-population-models.html#cb2212-64" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;top&quot;</span>)</span>
<span id="cb2212-65"><a href="integral-population-models.html#cb2212-65" tabindex="-1"></a></span>
<span id="cb2212-66"><a href="integral-population-models.html#cb2212-66" tabindex="-1"></a>p1 <span class="sc">+</span> p2 <span class="sc">+</span> p3</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-11-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="why-use-only-one-state-variable-the-multi-dimensional-trade-off" class="section level4 hasAnchor" number="82.2.8.2">
<h4><span class="header-section-number">82.2.8.2</span> Why Use Only One State Variable? The Multi-Dimensional Trade-off<a href="integral-population-models.html#why-use-only-one-state-variable-the-multi-dimensional-trade-off" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>State variables are rarely perfect predictors of vital rates. As a result, it is tempting to assume that including additional state variables will always improve demographic models. In practice, however, adding multiple state variables introduces substantial challenges. First, it increases the amount of field data required, often extending monitoring timelines and complicating census protocols. Multi-variable models can also be less intuitive for land managers, making it harder to interpret key drivers of population dynamics or implement streamlined monitoring programs.</p>
<p>Second—and more importantly—adding state variables dramatically increases model complexity through what is known as the dimensionality problem, which we describe in the following section.</p>
</div>
</div>
<div id="the-dimensionality-problem" class="section level3 hasAnchor" number="82.2.9">
<h3><span class="header-section-number">82.2.9</span> The dimensionality problem<a href="integral-population-models.html#the-dimensionality-problem" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>An Integral Projection Model (IPM) works by tracking how individuals move through a
<em>continuous distribution of sizes</em> over time. Mathematically, this involves solving
an integral equation:</p>
<p><span class="math display">\[
n(z&#39;, t+1) = \int \left[ s(z)\, g(z&#39;, z) + f(z&#39;, z) \right] n(z, t)\, dz
\]</span></p>
<p>where:</p>
<ul>
<li><span class="math inline">\(z\)</span> is the state variable (e.g., size) at time <span class="math inline">\(t\)</span></li>
<li><span class="math inline">\(z&#39;\)</span> is the state variable at time <span class="math inline">\(t+1\)</span></li>
<li><span class="math inline">\(n(z, t)\)</span> is the population density of individuals of size <span class="math inline">\(z\)</span> at time <span class="math inline">\(t\)</span></li>
<li><span class="math inline">\(s(z)\)</span> is the probability of survival for individuals of size <span class="math inline">\(z\)</span></li>
<li><span class="math inline">\(g(z&#39;, z)\)</span> describes growth from size <span class="math inline">\(z\)</span> to size <span class="math inline">\(z&#39;\)</span></li>
<li><span class="math inline">\(f(z&#39;, z)\)</span> describes reproductive contributions from individuals of size <span class="math inline">\(z\)</span> to size <span class="math inline">\(z&#39;\)</span></li>
</ul>
<p>The integral is evaluated over all possible values of <span class="math inline">\(z\)</span>, meaning that contributions
from individuals of every size at time <span class="math inline">\(t\)</span> are summed to determine the population
distribution at time <span class="math inline">\(t+1\)</span>.</p>
<p>With ONE state variable (e.g., stem diameter):</p>
<ul>
<li>We discretize the size range into ~100-200 cells (bins)</li>
<li>This creates a 100×100 projection matrix (manageable!)</li>
<li>Computational time: seconds</li>
</ul>
<p>With TWO state variables (e.g., diameter AND height):</p>
<ul>
<li>We need a 2-dimensional grid: diameter × height</li>
<li>With 100 bins each → 100×100 = 10,000 cells</li>
<li>Projection matrix becomes 10,000×10,000</li>
<li>Computational time: minutes to hours</li>
<li><strong>Data requirements: Need MANY more individuals to fill that grid</strong></li>
</ul>
<p>With THREE state variables (diameter, height, number of stems):</p>
<ul>
<li>100×100×100 = 1,000,000 cells</li>
<li>Matrix size: 1,000,000×1,000,000</li>
<li>Computational time: potentially days</li>
<li><strong>Data requirements: Tens of thousands of observations</strong> Most cells would be EMPTY</li>
</ul>
<p>This is called the “curse of dimensionality” - complexity grows exponentially with each added dimension. Especially since IPMs are often used to model rare plants characterized by low population sizes, the data requirements imposed by the inclusion of even a second state variable, renders this approach intractable. On rare occasions, it might be worth it to include to an additional state variable, if the two variables two variables capture different aspects of performance and they are uncorrelated and if you have HUGE datasets.</p>
<p>Onward! With total height selected as our state variable, we’re ready to build the IPM vital rate functions!</p>
</div>
</div>
<div id="packages-and-approaches-to-creating-ipms" class="section level2 hasAnchor" number="82.3">
<h2><span class="header-section-number">82.3</span> Packages and approaches to creating IPMs<a href="integral-population-models.html#packages-and-approaches-to-creating-ipms" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>There are multiple ways to build and analyze Integral Projection Models (IPMs) in R.
All approaches are based on the same underlying demographic theory, but they differ
in how much structure is imposed, how much code must be written by the user, and
how transparent the resulting model is.</p>
<p>Here, we briefly introduce three commonly used approaches: fully automated IPM
frameworks, semi-structured IPM tools, and hand-coded IPMs. In the sections that
follow, we use these approaches to illustrate model fitting and selection using
the same biological questions.</p>
<div id="ipmr-a-modern-flexible-ipm-framework" class="section level3 hasAnchor" number="82.3.1">
<h3><span class="header-section-number">82.3.1</span> ipmr: a modern, flexible IPM framework<a href="integral-population-models.html#ipmr-a-modern-flexible-ipm-framework" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <strong>ipmr</strong> package provides a modern, modular framework for building IPMs in R.
Rather than fitting the IPM as a single object, ipmr encourages users to specify
each vital-rate submodel (survival, growth, reproduction) explicitly and then
combine them into an IPM kernel.</p>
<p>This approach emphasizes transparency and flexibility. Model components can be
easily modified, compared, and extended, making ipmr particularly well suited for
model selection, uncertainty analysis, and advanced demographic metrics such as
sensitivities, elasticities, and transient dynamics.</p>
<p>Because ipmr requires users to think carefully about each model component, it has
a steeper learning curve than more automated approaches, but it provides the most
control and clarity for research-grade analyses.</p>
</div>
<div id="ipmpack-a-more-automated-ipm-workflow" class="section level3 hasAnchor" number="82.3.2">
<h3><span class="header-section-number">82.3.2</span> IPMpack: a more automated IPM workflow<a href="integral-population-models.html#ipmpack-a-more-automated-ipm-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <strong>IPMpack</strong> package offers a more automated approach to building IPMs, handling
many steps of kernel construction and numerical integration internally. Users
specify vital-rate models and size bounds, and the package generates the IPM and
associated outputs.</p>
<p>This approach can be efficient and relatively easy to implement for standard
one-dimensional IPMs. However, the abstraction can make it harder to see how
individual model components contribute to population dynamics, and flexibility
is more limited when extending models or exploring alternative formulations.</p>
</div>
<div id="hand-coded-ipms-maximum-transparency" class="section level3 hasAnchor" number="82.3.3">
<h3><span class="header-section-number">82.3.3</span> Hand-coded IPMs: maximum transparency<a href="integral-population-models.html#hand-coded-ipms-maximum-transparency" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A third approach is to build IPMs entirely by hand using base R code. In this
workflow, vital-rate models are fit using standard statistical tools (e.g., <code>glm</code>,
<code>lm</code>, <code>gam</code>), and the IPM kernel is constructed explicitly using numerical
integration and matrix operations.</p>
<p>Hand-coded IPMs provide complete transparency and conceptual clarity, making them
valuable for understanding how IPMs work at a fundamental level. However, they
require substantially more code, are more prone to implementation errors, and can
be difficult to maintain or extend.</p>
<p>Now let’s create the linear models that make up the IPM kernel! Through, we can work through the process using each approach!</p>
</div>
</div>
<div id="building-an-ipm-structuring-linear-models-variable-transformations-and-dealing-with-outliers" class="section level2 hasAnchor" number="82.4">
<h2><span class="header-section-number">82.4</span> Building an IPM: Structuring linear models, variable transformations, and dealing with outliers<a href="integral-population-models.html#building-an-ipm-structuring-linear-models-variable-transformations-and-dealing-with-outliers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this section, we build the statistical models that form the core of an Integral
Projection Model (IPM). Regardless of whether an IPM is implemented using a package
such as <code>ipmr</code>, <code>IPMpack</code>, or coded by hand, the underlying structure is the same:
IPMs are constructed from a set of regression models that describe how individual
state variables influence survival, growth, and reproduction.</p>
<p>Before assembling an IPM kernel, it is therefore essential to think carefully about
(1) which vital rates will be modeled, (2) how state variables enter those models,
and (3) how issues such as zero inflation, outliers, and missing values are handled.</p>
<p>First, let’s go ahead and pull in our transition dataset. A transition dataset consists of two years of matched data, in which the fate of each individual from one year to the next is documented. Notice the structure of the dataset! This dataframe tracks the fate of all individuals in the population from time 1 to time 2. The ‘id’ column corresponds with the tag number for each plant. The ‘size’ column contains sizes (in this case heights) of individuals in time 1, while ‘sizeNext’ indicates the size at time 2. In the ‘surv’ column, the fate of the individual at time 2 is indicated by either a 0 (died) or 1 (survived). For individuals that died, we place an ‘NA’ in the sizeNext column, since that individual didn’t exist that year!</p>
<p>Reproduction is always the most complex part of demographic modeling for plant species. Often, a reproductive cycle includes cryptic stages (seeds!) that are hard to track, may include vegetative and sexual reproductive components, and may involve several processes (like flower production and seed production) that may be important to examine to understand <em>why</em> population growth rates vary!</p>
<p>For Silene lanceolata, we don’t know enough about the seed bank to include this in our models. Seed dynamics are difficult to track across multiple sites and years. For this reason, we have built the models such that seeds produced in year 1 has the potential to germinate in year 2 or perish. Luckily, ignoring the seed bank for species with high rates of seed production has negligible affects on population growth. A new seedling won’t have a size, survival or reproductive values in year 1 (NAs are added to this column), but will have a measurement in the ‘sizeNext’ column - using this pattern in the data is <em>how</em> we tell R to pull data on seedlings, so it is important to have this information correctly entered in the dataframe.</p>
<p>We’ve broken reproduction into two components, fec1 and fec2. If an individual reproduced, we mark a ‘1’ in fec1, if the individual didn’t reproduce, then we add a ‘0’ to the fec1 column. For those individuals that did reproduce, the fec2 column contains the number of seeds produced. Again notice that if an individual didn’t reproduce, we place an ‘NA’ in the fec2 column. Finally, we’ve documented other observations for each individual that we think might explain population growth rates, like whether a plant has been browsed by deer or is in competition with other vegetation.</p>
<p>IPMs are only as good as the linear models that comprise them! As a critical first step in analyses, we have to take a look at the relationship between our state variable and growth, reproduction and survival. We want to be sure to address any issues with normality and unequal variance. First, examine variance; if variation depends on the state variable, include this in the model (we’ll do this in a minute). It is fairly common in plants to see, for instance, that growth rates are more variable in larger individuals.</p>
<p>For the ipmr and handcoding, column names could essentially be anything, though it is good practice to name columns in a way that is intuitive for your analyses. The IPMpack syntax is stricter, so we will default to using those naming conventions. Let’s pull in our data and look at its structure.</p>
<div class="sourceCode" id="cb2213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2213-1"><a href="integral-population-models.html#cb2213-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2213-2"><a href="integral-population-models.html#cb2213-2" tabindex="-1"></a><span class="fu">library</span>(httr)</span>
<span id="cb2213-3"><a href="integral-population-models.html#cb2213-3" tabindex="-1"></a></span>
<span id="cb2213-4"><a href="integral-population-models.html#cb2213-4" tabindex="-1"></a><span class="co"># Download transition data</span></span>
<span id="cb2213-5"><a href="integral-population-models.html#cb2213-5" tabindex="-1"></a>file_id <span class="ot">&lt;-</span> <span class="st">&quot;1V0qWq0SlA9rWzyraZaOFdWDVDKQyiu53&quot;</span></span>
<span id="cb2213-6"><a href="integral-population-models.html#cb2213-6" tabindex="-1"></a></span>
<span id="cb2213-7"><a href="integral-population-models.html#cb2213-7" tabindex="-1"></a><span class="fu">GET</span>(</span>
<span id="cb2213-8"><a href="integral-population-models.html#cb2213-8" tabindex="-1"></a>  <span class="at">url =</span> <span class="fu">paste0</span>(<span class="st">&quot;https://drive.usercontent.google.com/download?id=&quot;</span>, </span>
<span id="cb2213-9"><a href="integral-population-models.html#cb2213-9" tabindex="-1"></a>               file_id, <span class="st">&quot;&amp;export=download&amp;confirm=t&quot;</span>),</span>
<span id="cb2213-10"><a href="integral-population-models.html#cb2213-10" tabindex="-1"></a>  <span class="fu">write_disk</span>(<span class="st">&quot;data/silene_transitions.csv&quot;</span>, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2213-11"><a href="integral-population-models.html#cb2213-11" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## Response [https://drive.usercontent.google.com/download?id=1V0qWq0SlA9rWzyraZaOFdWDVDKQyiu53&amp;export=download&amp;confirm=t]
##   Date: 2026-02-22 15:08
##   Status: 200
##   Content-Type: application/octet-stream
##   Size: 18.2 kB
## &lt;ON DISK&gt;  /Users/sks379/Desktop/GitHubProjects/DataAnalysis/data/silene_transitions.csv</code></pre>
<div class="sourceCode" id="cb2215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2215-1"><a href="integral-population-models.html#cb2215-1" tabindex="-1"></a><span class="co"># Load the data</span></span>
<span id="cb2215-2"><a href="integral-population-models.html#cb2215-2" tabindex="-1"></a>transitions <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/silene_transitions.csv&quot;</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2215-3"><a href="integral-population-models.html#cb2215-3" tabindex="-1"></a></span>
<span id="cb2215-4"><a href="integral-population-models.html#cb2215-4" tabindex="-1"></a><span class="co"># Initial exploration</span></span>
<span id="cb2215-5"><a href="integral-population-models.html#cb2215-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Dataset dimensions:&quot;</span>, <span class="fu">nrow</span>(transitions), <span class="st">&quot;rows x&quot;</span>, <span class="fu">ncol</span>(transitions), <span class="st">&quot;columns</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Dataset dimensions: 392 rows x 14 columns</code></pre>
<div class="sourceCode" id="cb2217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2217-1"><a href="integral-population-models.html#cb2217-1" tabindex="-1"></a><span class="fu">glimpse</span>(transitions)</span></code></pre></div>
<pre><code>## Rows: 392
## Columns: 14
## $ id                            &lt;dbl&gt; 7, 34, 2, 46, 31, 23, 36, 32, 1, 41, 6, 23, 35, 22, 32, 26, 4, 1, 40, …
## $ size                          &lt;dbl&gt; 110.5, 121.6, 56.4, 97.5, 90.0, 144.5, 103.4, 90.2, 76.2, 106.8, 94.8,…
## $ surv                          &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0…
## $ sizeNext                      &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, 26.7, NA, NA, NA, NA, NA, NA, NA, NA, …
## $ fec1                          &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0…
## $ fec2                          &lt;dbl&gt; 3600, 10080, 11340, 12150, 13500, 15120, 15750, 18000, 29700, 31500, 7…
## $ Exclude                       &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0…
## $ Site                          &lt;chr&gt; &quot;KalawamounaE&quot;, &quot;Bobcat03&quot;, &quot;Bobcat03&quot;, &quot;KalawamounaE&quot;, &quot;KalawamounaE&quot;…
## $ `Herbivory (Y/N)_2024`        &lt;chr&gt; NA, NA, NA, &quot;N&quot;, NA, NA, NA, NA, &quot;N&quot;, &quot;N&quot;, NA, NA, NA, NA, NA, NA, &quot;N&quot;…
## $ `Herbivory Agent_2024`        &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…
## $ `Disease (Y/N)_2024`          &lt;chr&gt; &quot;N&quot;, &quot;N&quot;, NA, &quot;N&quot;, &quot;N&quot;, &quot;N&quot;, &quot;N&quot;, &quot;N&quot;, &quot;N&quot;, &quot;N&quot;, &quot;N&quot;, NA, &quot;N&quot;, &quot;N&quot;, &quot;N…
## $ `Description of Disease_2024` &lt;lgl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…
## $ NotesYear1                    &lt;chr&gt; NA, NA, NA, &quot;most leaves are gone&quot;, NA, NA, NA, NA, NA, NA, NA, NA, NA…
## $ NotesYear2                    &lt;chr&gt; &quot;NO&quot;, &quot;NO&quot;, NA, &quot;NO&quot;, &quot;NO&quot;, &quot;NO&quot;, &quot;NO&quot;, &quot;NO&quot;, &quot;NO&quot;, &quot;NO&quot;, &quot;NO&quot;, NA, &quot;N…</code></pre>
<div class="sourceCode" id="cb2219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2219-1"><a href="integral-population-models.html#cb2219-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== Column names ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === Column names ===</code></pre>
<div class="sourceCode" id="cb2221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2221-1"><a href="integral-population-models.html#cb2221-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">names</span>(transitions))</span></code></pre></div>
<pre><code>##  [1] &quot;id&quot;                          &quot;size&quot;                        &quot;surv&quot;                       
##  [4] &quot;sizeNext&quot;                    &quot;fec1&quot;                        &quot;fec2&quot;                       
##  [7] &quot;Exclude&quot;                     &quot;Site&quot;                        &quot;Herbivory (Y/N)_2024&quot;       
## [10] &quot;Herbivory Agent_2024&quot;        &quot;Disease (Y/N)_2024&quot;          &quot;Description of Disease_2024&quot;
## [13] &quot;NotesYear1&quot;                  &quot;NotesYear2&quot;</code></pre>
<div class="sourceCode" id="cb2223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2223-1"><a href="integral-population-models.html#cb2223-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== Summary statistics ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === Summary statistics ===</code></pre>
<div class="sourceCode" id="cb2225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2225-1"><a href="integral-population-models.html#cb2225-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">summary</span>(transitions))</span></code></pre></div>
<pre><code>##        id              size             surv          sizeNext           fec1             fec2        
##  Min.   :  1.00   Min.   :  2.50   Min.   :0.000   Min.   : 0.700   Min.   :0.0000   Min.   :     90  
##  1st Qu.: 39.75   1st Qu.: 11.00   1st Qu.:0.000   1st Qu.: 3.000   1st Qu.:0.0000   1st Qu.:  10800  
##  Median :111.50   Median : 32.30   Median :1.000   Median : 4.300   Median :0.0000   Median :  30240  
##  Mean   :125.44   Mean   : 55.29   Mean   :0.596   Mean   : 6.167   Mean   :0.4343   Mean   : 130305  
##  3rd Qu.:190.25   3rd Qu.:103.30   3rd Qu.:1.000   3rd Qu.: 6.525   3rd Qu.:1.0000   3rd Qu.: 127665  
##  Max.   :479.00   Max.   :160.20   Max.   :1.000   Max.   :44.000   Max.   :1.0000   Max.   :1086120  
##                   NA&#39;s   :293      NA&#39;s   :293     NA&#39;s   :40       NA&#39;s   :293      NA&#39;s   :349      
##     Exclude            Site           Herbivory (Y/N)_2024 Herbivory Agent_2024 Disease (Y/N)_2024
##  Min.   :0.00000   Length:392         Length:392           Length:392           Length:392        
##  1st Qu.:0.00000   Class :character   Class :character     Class :character     Class :character  
##  Median :0.00000   Mode  :character   Mode  :character     Mode  :character     Mode  :character  
##  Mean   :0.04337                                                                                  
##  3rd Qu.:0.00000                                                                                  
##  Max.   :1.00000                                                                                  
##                                                                                                   
##  Description of Disease_2024  NotesYear1         NotesYear2       
##  Mode:logical                Length:392         Length:392        
##  NA&#39;s:392                    Class :character   Class :character  
##                              Mode  :character   Mode  :character  
##                                                                   
##                                                                   
##                                                                   
## </code></pre>
<div class="sourceCode" id="cb2227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2227-1"><a href="integral-population-models.html#cb2227-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== First few rows ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === First few rows ===</code></pre>
<div class="sourceCode" id="cb2229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2229-1"><a href="integral-population-models.html#cb2229-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(transitions, <span class="dv">10</span>))</span></code></pre></div>
<pre><code>## # A tibble: 10 × 14
##       id  size  surv sizeNext  fec1  fec2 Exclude Site         `Herbivory (Y/N)_2024` `Herbivory Agent_2024`
##    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;        &lt;chr&gt;                  &lt;chr&gt;                 
##  1     7 110.      0     NA       1  3600       0 KalawamounaE &lt;NA&gt;                   &lt;NA&gt;                  
##  2    34 122.      0     NA       1 10080       0 Bobcat03     &lt;NA&gt;                   &lt;NA&gt;                  
##  3     2  56.4     0     NA       1 11340       0 Bobcat03     &lt;NA&gt;                   &lt;NA&gt;                  
##  4    46  97.5     0     NA       1 12150       0 KalawamounaE N                      &lt;NA&gt;                  
##  5    31  90       0     NA       1 13500       0 KalawamounaE &lt;NA&gt;                   &lt;NA&gt;                  
##  6    23 144.      0     NA       1 15120       0 KalawamounaE &lt;NA&gt;                   &lt;NA&gt;                  
##  7    36 103.      0     NA       1 15750       0 KalawamounaE &lt;NA&gt;                   &lt;NA&gt;                  
##  8    32  90.2     0     NA       1 18000       0 KalawamounaE &lt;NA&gt;                   &lt;NA&gt;                  
##  9     1  76.2     1     26.7     1 29700       0 KalawamounaE N                      &lt;NA&gt;                  
## 10    41 107.      0     NA       1 31500       0 KalawamounaE N                      &lt;NA&gt;                  
## # ℹ 4 more variables: `Disease (Y/N)_2024` &lt;chr&gt;, `Description of Disease_2024` &lt;lgl&gt;, NotesYear1 &lt;chr&gt;,
## #   NotesYear2 &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb2231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2231-1"><a href="integral-population-models.html#cb2231-1" tabindex="-1"></a><span class="co"># I always include an Exclude column to allow folks entering or collecting data to identify a priori really weird stuff to exclude!</span></span>
<span id="cb2231-2"><a href="integral-population-models.html#cb2231-2" tabindex="-1"></a>final_data <span class="ot">&lt;-</span> transitions <span class="sc">|&gt;</span></span>
<span id="cb2231-3"><a href="integral-population-models.html#cb2231-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Exclude <span class="sc">==</span> <span class="dv">0</span>)</span></code></pre></div>
<p>Key features:</p>
<ul>
<li>survival (surv) coded as a binary, 0 = dead, 1 = alive</li>
<li>size in year one size, continuous variable</li>
<li>sizeNext, year two size, continuous variable</li>
<li>fec1 = reproduced, yes or no, coded as 0 for no, and 1 for yes</li>
<li>fec2 two number of seeds produced, estimated from floral counts, rounded to the nearest whole number</li>
<li>includes any important information that could explain what happened to the plants (where they eaten by deer? where were they located?)</li>
<li>includes an ID number</li>
<li>new seedlings indicated by not having a size year 1 (coded as NA), survival information (NA), reproduction (NA), and a sizeNext entry</li>
<li>NA indicate missing data, and 0s are true zero</li>
<li>A column labeled ‘Exclude’ used to mark observations with unusual circumstances (e.g., severe damage, measurement error, or other anomalies) that would otherwise distort model fitting. Rather than deleting these observations permanently, retaining them in the dataset and filtering them explicitly allows decisions about inclusion to remain transparent and reproducible.</li>
</ul>
<p>Using this dataset, we will build four core components of the IPM:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Survival model</strong><br />
Probability of surviving from time <span class="math inline">\(t\)</span> to <span class="math inline">\(t+1\)</span> as a function of size.</p></li>
<li><p><strong>Growth model</strong><br />
Expected size at time <span class="math inline">\(t+1\)</span> given size at time <span class="math inline">\(t\)</span>, conditional on survival.</p></li>
<li><p><strong>Probability of reproduction</strong><br />
Probability that an individual reproduces as a function of size.</p></li>
<li><p><strong>Reproductive output</strong><br />
Number of offspring produced, conditional on reproduction occurring.</p></li>
</ol>
<div id="variable-transformations-and-scale" class="section level3 hasAnchor" number="82.4.1">
<h3><span class="header-section-number">82.4.1</span> Variable transformations and scale<a href="integral-population-models.html#variable-transformations-and-scale" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Demographic rates often exhibit nonlinear relationships with size and strong
right-skew, particularly for growth and fecundity. As a result, transformations are
commonly applied to improve model fit and biological realism.</p>
<p>In this example, we will explore transformations such as:</p>
<ul>
<li>log or square-root transformations of size</li>
<li>log transformations of reproductive output</li>
<li>centering and scaling of predictors for numerical stability</li>
</ul>
<p>Rather than assuming a single transformation <em>a priori</em>, we will use model selection
to evaluate competing formulations and identify those that best explain variation
in each vital rate.</p>
</div>
<div id="dealing-with-zeros-missing-values-and-outliers" class="section level3 hasAnchor" number="82.4.2">
<h3><span class="header-section-number">82.4.2</span> Dealing with zeros, missing values, and outliers<a href="integral-population-models.html#dealing-with-zeros-missing-values-and-outliers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This dataset illustrates several common challenges in demographic data:</p>
<ul>
<li>Survival and reproduction include many zeros</li>
<li>Growth is undefined for individuals that did not survive</li>
<li>Reproductive output spans several orders of magnitude</li>
<li>Some individuals are flagged for exclusion</li>
</ul>
<p>We address these issues explicitly by:</p>
<ul>
<li>Modeling survival and reproduction as binomial processes</li>
<li>Conditioning growth models on survival</li>
<li>Using two-stage (hurdle) models for fecundity</li>
<li>Excluding flagged individuals prior to model fitting</li>
</ul>
<p>These steps ensure that each statistical model aligns with the biological process
it represents.</p>
</div>
<div id="parallel-ipm-workflows" class="section level3 hasAnchor" number="82.4.3">
<h3><span class="header-section-number">82.4.3</span> Parallel IPM workflows<a href="integral-population-models.html#parallel-ipm-workflows" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Importantly, the statistical models described above are identical across IPM
implementations. What differs among approaches is how these models are incorporated
into a population projection framework.</p>
<p>In the sections that follow, we use the same fitted vital-rate models to construct
IPMs using three approaches:</p>
<ol style="list-style-type: decimal">
<li><p><strong><code>ipmr</code></strong> — a modular, transparent framework that explicitly links each model
component to the IPM kernel.</p></li>
<li><p><strong><code>IPMpack</code></strong> — a more automated approach that streamlines kernel construction.</p></li>
<li><p><strong>Hand-coded IPMs</strong> — a fully explicit implementation that reveals the underlying
mechanics of numerical integration and matrix approximation.</p></li>
</ol>
<p>By working through each approach in parallel, we emphasize that differences among
packages reflect differences in software design, not differences in demographic
theory.</p>
<p>We begin by fitting and comparing candidate models for each vital rate, starting
with growth. At each step, we evaluate model fit, interpret parameter estimates,
and discuss how these choices influence the resulting IPM.</p>
</div>
<div id="growth-transitions" class="section level3 hasAnchor" number="82.4.4">
<h3><span class="header-section-number">82.4.4</span> Growth transitions<a href="integral-population-models.html#growth-transitions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>I typically start by modeling growth. Why? Growth is typically modeled as a continuous relationship between size at time <span class="math inline">\(t\)</span> and
size at time <span class="math inline">\(t+1\)</span>. Extreme deviations from this relationship—such as implausibly large
shrinks or jumps—can strongly influence parameter estimates and propagate unrealistic
behavior throughout the IPM kernel. In contrast, extreme reproductive values often reflect genuine biological processes
and should be treated with caution rather than automatically excluded.</p>
<p>Demographic data often contain extreme values, but not all extremes should be treated
as outliers. In particular, reproductive output can exhibit large variation that is
biologically real (e.g., mast events, episodic flowering), whereas extreme values in
growth are more likely to reflect measurement error or data recording issues.</p>
<p>For this reason, we focus first on identifying and handling outliers in the <strong>growth
model</strong>, rather than applying blanket outlier removal across all vital-rate models.</p>
<p>We begin by fitting a preliminary growth model using all available, non-excluded data. Remember, that we will only work with individuals that survived (otherwise there is no growth!)</p>
<div class="sourceCode" id="cb2232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2232-1"><a href="integral-population-models.html#cb2232-1" tabindex="-1"></a>growth_data <span class="ot">&lt;-</span> final_data <span class="sc">|&gt;</span></span>
<span id="cb2232-2"><a href="integral-population-models.html#cb2232-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(surv <span class="sc">==</span> <span class="dv">1</span>, <span class="sc">!</span><span class="fu">is.na</span>(sizeNext))</span>
<span id="cb2232-3"><a href="integral-population-models.html#cb2232-3" tabindex="-1"></a></span>
<span id="cb2232-4"><a href="integral-population-models.html#cb2232-4" tabindex="-1"></a>m_growth_lin <span class="ot">&lt;-</span> <span class="fu">lm</span>(sizeNext <span class="sc">~</span> size,</span>
<span id="cb2232-5"><a href="integral-population-models.html#cb2232-5" tabindex="-1"></a>                   <span class="at">data =</span> growth_data)</span>
<span id="cb2232-6"><a href="integral-population-models.html#cb2232-6" tabindex="-1"></a></span>
<span id="cb2232-7"><a href="integral-population-models.html#cb2232-7" tabindex="-1"></a>m_growth_log <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="fu">log</span>(sizeNext) <span class="sc">~</span> <span class="fu">log</span>(size),</span>
<span id="cb2232-8"><a href="integral-population-models.html#cb2232-8" tabindex="-1"></a>                   <span class="at">data =</span> growth_data)</span>
<span id="cb2232-9"><a href="integral-population-models.html#cb2232-9" tabindex="-1"></a></span>
<span id="cb2232-10"><a href="integral-population-models.html#cb2232-10" tabindex="-1"></a><span class="fu">AIC</span>(m_growth_lin, m_growth_log)</span></code></pre></div>
<pre><code>##              df      AIC
## m_growth_lin  3 430.8275
## m_growth_log  3 122.3578</code></pre>
<p>Great! The lower AIC value wins and we see that we need to log transform the data. Now lets look at model fit (linear vs polynomial).</p>
<div class="sourceCode" id="cb2234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2234-1"><a href="integral-population-models.html#cb2234-1" tabindex="-1"></a>m_growth_log_lin <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="fu">log</span>(sizeNext) <span class="sc">~</span> <span class="fu">log</span>(size),</span>
<span id="cb2234-2"><a href="integral-population-models.html#cb2234-2" tabindex="-1"></a>                       <span class="at">data =</span> growth_data)</span>
<span id="cb2234-3"><a href="integral-population-models.html#cb2234-3" tabindex="-1"></a></span>
<span id="cb2234-4"><a href="integral-population-models.html#cb2234-4" tabindex="-1"></a>m_growth_log_poly <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="fu">log</span>(sizeNext) <span class="sc">~</span> <span class="fu">poly</span>(<span class="fu">log</span>(size), <span class="dv">2</span>, <span class="at">raw =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2234-5"><a href="integral-population-models.html#cb2234-5" tabindex="-1"></a>                        <span class="at">data =</span> growth_data)</span>
<span id="cb2234-6"><a href="integral-population-models.html#cb2234-6" tabindex="-1"></a></span>
<span id="cb2234-7"><a href="integral-population-models.html#cb2234-7" tabindex="-1"></a><span class="fu">AIC</span>(m_growth_log_lin, m_growth_log_poly)</span></code></pre></div>
<pre><code>##                   df      AIC
## m_growth_log_lin   3 122.3578
## m_growth_log_poly  4 120.1388</code></pre>
<p>The polynomial model is best! But wait, before we proceed - let’s check out the fit. This is critical for two reasons. First, we want to identify and remove any outliers. Remember, these outliers are more likely to be ‘true’ outliers (mis-ided individuals, measurement error). Secondly, sometimes higher order models produce biological unrealistic fits, especially when sample size is low (i.e., odd peaks and valleys for survival). When this occurs, it is typically better to revert to the simpler model! Let’s take a look at both fits.</p>
<div class="sourceCode" id="cb2236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2236-1"><a href="integral-population-models.html#cb2236-1" tabindex="-1"></a>newdat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2236-2"><a href="integral-population-models.html#cb2236-2" tabindex="-1"></a>  <span class="at">size =</span> <span class="fu">seq</span>(<span class="fu">min</span>(growth_data<span class="sc">$</span>size),</span>
<span id="cb2236-3"><a href="integral-population-models.html#cb2236-3" tabindex="-1"></a>             <span class="fu">max</span>(growth_data<span class="sc">$</span>size),</span>
<span id="cb2236-4"><a href="integral-population-models.html#cb2236-4" tabindex="-1"></a>             <span class="at">length.out =</span> <span class="dv">200</span>)</span>
<span id="cb2236-5"><a href="integral-population-models.html#cb2236-5" tabindex="-1"></a>)</span>
<span id="cb2236-6"><a href="integral-population-models.html#cb2236-6" tabindex="-1"></a></span>
<span id="cb2236-7"><a href="integral-population-models.html#cb2236-7" tabindex="-1"></a>newdat<span class="sc">$</span>lin  <span class="ot">&lt;-</span> <span class="fu">exp</span>(<span class="fu">predict</span>(m_growth_log_lin,  newdat))</span>
<span id="cb2236-8"><a href="integral-population-models.html#cb2236-8" tabindex="-1"></a>newdat<span class="sc">$</span>poly <span class="ot">&lt;-</span> <span class="fu">exp</span>(<span class="fu">predict</span>(m_growth_log_poly, newdat))</span>
<span id="cb2236-9"><a href="integral-population-models.html#cb2236-9" tabindex="-1"></a></span>
<span id="cb2236-10"><a href="integral-population-models.html#cb2236-10" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2236-11"><a href="integral-population-models.html#cb2236-11" tabindex="-1"></a></span>
<span id="cb2236-12"><a href="integral-population-models.html#cb2236-12" tabindex="-1"></a><span class="fu">ggplot</span>(growth_data, <span class="fu">aes</span>(size, sizeNext)) <span class="sc">+</span></span>
<span id="cb2236-13"><a href="integral-population-models.html#cb2236-13" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb2236-14"><a href="integral-population-models.html#cb2236-14" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> newdat, <span class="fu">aes</span>(size, lin,  <span class="at">color =</span> <span class="st">&quot;Linear&quot;</span>), <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb2236-15"><a href="integral-population-models.html#cb2236-15" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> newdat, <span class="fu">aes</span>(size, poly, <span class="at">color =</span> <span class="st">&quot;Quadratic&quot;</span>), <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb2236-16"><a href="integral-population-models.html#cb2236-16" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb2236-17"><a href="integral-population-models.html#cb2236-17" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb2236-18"><a href="integral-population-models.html#cb2236-18" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb2236-19"><a href="integral-population-models.html#cb2236-19" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Size at time t&quot;</span>,</span>
<span id="cb2236-20"><a href="integral-population-models.html#cb2236-20" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Size at time t+1&quot;</span>,</span>
<span id="cb2236-21"><a href="integral-population-models.html#cb2236-21" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Model&quot;</span></span>
<span id="cb2236-22"><a href="integral-population-models.html#cb2236-22" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2236-23"><a href="integral-population-models.html#cb2236-23" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-15-1.png" alt="" width="960" style="display: block; margin: auto;" />
Interesting! The polynomial fit indicates that growth decreases at larger sizes, and indication of aging or some biophysical limit on growth at larger sizes. This seems biologically defensible and from a modeling standpoint (lower AIC) defensible. Let’s keep the polynomial fit!</p>
<div id="outlier-removal" class="section level4 hasAnchor" number="82.4.4.1">
<h4><span class="header-section-number">82.4.4.1</span> Outlier removal<a href="integral-population-models.html#outlier-removal" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We identify growth outliers using standardized residuals from the fitted growth
model, flagging extreme deviations for biological inspection rather than automatic
removal. Only observations that are clearly inconsistent with plausible growth are
excluded, and the model is refit to ensure stable kernel behavior.</p>
<div class="sourceCode" id="cb2237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2237-1"><a href="integral-population-models.html#cb2237-1" tabindex="-1"></a>m_growth_quad <span class="ot">&lt;-</span> <span class="fu">lm</span>(</span>
<span id="cb2237-2"><a href="integral-population-models.html#cb2237-2" tabindex="-1"></a>  <span class="fu">log</span>(sizeNext) <span class="sc">~</span> <span class="fu">poly</span>(<span class="fu">log</span>(size), <span class="dv">2</span>, <span class="at">raw =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2237-3"><a href="integral-population-models.html#cb2237-3" tabindex="-1"></a>  <span class="at">data =</span> growth_data</span>
<span id="cb2237-4"><a href="integral-population-models.html#cb2237-4" tabindex="-1"></a>)</span>
<span id="cb2237-5"><a href="integral-population-models.html#cb2237-5" tabindex="-1"></a></span>
<span id="cb2237-6"><a href="integral-population-models.html#cb2237-6" tabindex="-1"></a><span class="co"># Extract residuals</span></span>
<span id="cb2237-7"><a href="integral-population-models.html#cb2237-7" tabindex="-1"></a>growth_data <span class="ot">&lt;-</span> growth_data <span class="sc">|&gt;</span></span>
<span id="cb2237-8"><a href="integral-population-models.html#cb2237-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb2237-9"><a href="integral-population-models.html#cb2237-9" tabindex="-1"></a>    <span class="at">fitted   =</span> <span class="fu">fitted</span>(m_growth_quad),</span>
<span id="cb2237-10"><a href="integral-population-models.html#cb2237-10" tabindex="-1"></a>    <span class="at">resid    =</span> <span class="fu">residuals</span>(m_growth_quad),</span>
<span id="cb2237-11"><a href="integral-population-models.html#cb2237-11" tabindex="-1"></a>    <span class="at">std_res  =</span> <span class="fu">rstandard</span>(m_growth_quad)</span>
<span id="cb2237-12"><a href="integral-population-models.html#cb2237-12" tabindex="-1"></a>  )</span>
<span id="cb2237-13"><a href="integral-population-models.html#cb2237-13" tabindex="-1"></a></span>
<span id="cb2237-14"><a href="integral-population-models.html#cb2237-14" tabindex="-1"></a><span class="co"># Visual residuals</span></span>
<span id="cb2237-15"><a href="integral-population-models.html#cb2237-15" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2237-16"><a href="integral-population-models.html#cb2237-16" tabindex="-1"></a></span>
<span id="cb2237-17"><a href="integral-population-models.html#cb2237-17" tabindex="-1"></a><span class="fu">ggplot</span>(growth_data, <span class="fu">aes</span>(<span class="fu">log</span>(size), std_res)) <span class="sc">+</span></span>
<span id="cb2237-18"><a href="integral-population-models.html#cb2237-18" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb2237-19"><a href="integral-population-models.html#cb2237-19" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">3</span>),</span>
<span id="cb2237-20"><a href="integral-population-models.html#cb2237-20" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb2237-21"><a href="integral-population-models.html#cb2237-21" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb2237-22"><a href="integral-population-models.html#cb2237-22" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb2237-23"><a href="integral-population-models.html#cb2237-23" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;log(size at time t)&quot;</span>,</span>
<span id="cb2237-24"><a href="integral-population-models.html#cb2237-24" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Standardized residual&quot;</span></span>
<span id="cb2237-25"><a href="integral-population-models.html#cb2237-25" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2237-26"><a href="integral-population-models.html#cb2237-26" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-16-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb2238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2238-1"><a href="integral-population-models.html#cb2238-1" tabindex="-1"></a><span class="co"># Apply ±3 is a flagging threshold, not an automatic deletion rule</span></span>
<span id="cb2238-2"><a href="integral-population-models.html#cb2238-2" tabindex="-1"></a>growth_data <span class="ot">&lt;-</span> growth_data <span class="sc">|&gt;</span></span>
<span id="cb2238-3"><a href="integral-population-models.html#cb2238-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb2238-4"><a href="integral-population-models.html#cb2238-4" tabindex="-1"></a>    <span class="at">flag_outlier =</span> <span class="fu">abs</span>(std_res) <span class="sc">&gt;</span> <span class="dv">3</span></span>
<span id="cb2238-5"><a href="integral-population-models.html#cb2238-5" tabindex="-1"></a>  )</span></code></pre></div>
<p>Here, we don’t see any significant outliers! Awesome! Let take a different look:</p>
<div class="sourceCode" id="cb2239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2239-1"><a href="integral-population-models.html#cb2239-1" tabindex="-1"></a><span class="co"># Evaluating data for IPM construction using log(size)</span></span>
<span id="cb2239-2"><a href="integral-population-models.html#cb2239-2" tabindex="-1"></a>growth_log <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="fu">log</span>(sizeNext) <span class="sc">~</span> <span class="fu">log</span>(size),</span>
<span id="cb2239-3"><a href="integral-population-models.html#cb2239-3" tabindex="-1"></a>                 <span class="at">na.action =</span> na.omit,</span>
<span id="cb2239-4"><a href="integral-population-models.html#cb2239-4" tabindex="-1"></a>                 <span class="at">data =</span> final_data)</span>
<span id="cb2239-5"><a href="integral-population-models.html#cb2239-5" tabindex="-1"></a></span>
<span id="cb2239-6"><a href="integral-population-models.html#cb2239-6" tabindex="-1"></a>resid_log <span class="ot">&lt;-</span> <span class="fu">residuals</span>(growth_log)</span>
<span id="cb2239-7"><a href="integral-population-models.html#cb2239-7" tabindex="-1"></a></span>
<span id="cb2239-8"><a href="integral-population-models.html#cb2239-8" tabindex="-1"></a><span class="co"># Normality test</span></span>
<span id="cb2239-9"><a href="integral-population-models.html#cb2239-9" tabindex="-1"></a><span class="fu">shapiro.test</span>(resid_log)</span></code></pre></div>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  resid_log
## W = 0.97379, p-value = 0.2316</code></pre>
<div class="sourceCode" id="cb2241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2241-1"><a href="integral-population-models.html#cb2241-1" tabindex="-1"></a><span class="co"># Residuals vs fitted</span></span>
<span id="cb2241-2"><a href="integral-population-models.html#cb2241-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">fitted</span>(growth_log), resid_log,</span>
<span id="cb2241-3"><a href="integral-population-models.html#cb2241-3" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Fitted values (log sizeNext)&quot;</span>,</span>
<span id="cb2241-4"><a href="integral-population-models.html#cb2241-4" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;Residuals&quot;</span>,</span>
<span id="cb2241-5"><a href="integral-population-models.html#cb2241-5" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;Residuals vs Fitted (log–log model)&quot;</span>)</span>
<span id="cb2241-6"><a href="integral-population-models.html#cb2241-6" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-17-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb2242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2242-1"><a href="integral-population-models.html#cb2242-1" tabindex="-1"></a>id.n <span class="ot">&lt;-</span> <span class="fu">nrow</span>(final_data)</span>
<span id="cb2242-2"><a href="integral-population-models.html#cb2242-2" tabindex="-1"></a></span>
<span id="cb2242-3"><a href="integral-population-models.html#cb2242-3" tabindex="-1"></a><span class="fu">qqPlot</span>(growth_log,</span>
<span id="cb2242-4"><a href="integral-population-models.html#cb2242-4" tabindex="-1"></a>       <span class="at">distribution =</span> <span class="st">&quot;norm&quot;</span>,</span>
<span id="cb2242-5"><a href="integral-population-models.html#cb2242-5" tabindex="-1"></a>       <span class="at">id.method =</span> <span class="st">&quot;y&quot;</span>,</span>
<span id="cb2242-6"><a href="integral-population-models.html#cb2242-6" tabindex="-1"></a>       <span class="at">id.cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb2242-7"><a href="integral-population-models.html#cb2242-7" tabindex="-1"></a>       <span class="at">id.n =</span> id.n,</span>
<span id="cb2242-8"><a href="integral-population-models.html#cb2242-8" tabindex="-1"></a>       <span class="at">id.col =</span> <span class="st">&quot;blue&quot;</span>,</span>
<span id="cb2242-9"><a href="integral-population-models.html#cb2242-9" tabindex="-1"></a>       <span class="at">id.location =</span> <span class="st">&quot;ab&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-17-2.png" alt="" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## [1] 72 81</code></pre>
<p>This figure is showing a scatter plot of residuals on the y axis and fitted values (estimated responses) on the x axis. We use these scatterplots to detect non-linearity, unequal error variances, and outliers. The residuals should “bounce randomly” around the 0 line, indicating that the assumption that the relationship is linear is reasonable. You also show see that the residuals roughly form a “horizontal band” around the 0 line, indicating that the variances of the error terms are equal. Finally, we looking for a dataset in which no one residual “stands out” from the basic random pattern of residuals. Let’s look at these residuals another way and use label the outliers so that we can remove them if necessary!</p>
<p>A quantile-quantile plot, often abbreviated as Q-Q plot, is a graphical tool used to assess whether a dataset follows a particular theoretical distribution, such as the normal distribution. It compares the quantiles of the observed data against the quantiles of the expected theoretical distribution.</p>
<p>Here’s how to interpret a Q-Q plot: The <em>x-axis</em> of the Q-Q plot represents the theoretical quantiles from a specified distribution (e.g., the normal distribution). The <em>y-axis</em> represents the quantiles of the observed data. If the points on the Q-Q plot fall approximately along a straight line, it suggests that the data follows the theoretical distribution. Deviations from the straight line indicate departures from the assumed distribution. If points are above the line, it suggests that the observed values are higher than expected for that quantile. If points are below the line, it suggests that the observed values are lower than expected for that quantile. The ends of the Q-Q plot are often of particular interest. Deviations in the tails can indicate differences in tail behavior.</p>
<p>These data look fine! In cases where this is deviation from normality, it is good practice to try to determine whether it is legitimate to remove outliers from any statistical analysis. In demographic studies, there are so many plants and many reasonable possibilities for why we may observe a strange transition from one year to the next.</p>
<p>For example:</p>
<ul>
<li>In the first year, someone measured the plant, but it had already been browsed by a deer and so was extra short, and then appears to grow like wild in the following year</li>
<li>Someone took data on the wrong space on the datasheet</li>
<li>Someone couldn’t find a mama plant, who actually died, so accidentally measured a baby that had germinated in a nearby spot</li>
</ul>
<p>Given this, unlike with standard statistical analyses, we tend to be less stringent about when we remove outliers. Let’s take a look at residuals.</p>
<div class="sourceCode" id="cb2244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2244-1"><a href="integral-population-models.html#cb2244-1" tabindex="-1"></a><span class="co"># row_index &lt;- 59</span></span>
<span id="cb2244-2"><a href="integral-population-models.html#cb2244-2" tabindex="-1"></a><span class="co"># value &lt;- final_data[row_index,]</span></span>
<span id="cb2244-3"><a href="integral-population-models.html#cb2244-3" tabindex="-1"></a><span class="co"># print(value)</span></span>
<span id="cb2244-4"><a href="integral-population-models.html#cb2244-4" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb2244-5"><a href="integral-population-models.html#cb2244-5" tabindex="-1"></a><span class="co"># #or you can generate the id and use that to subset (just looking at the id number from above)</span></span>
<span id="cb2244-6"><a href="integral-population-models.html#cb2244-6" tabindex="-1"></a><span class="co"># target_id &lt;- 17</span></span>
<span id="cb2244-7"><a href="integral-population-models.html#cb2244-7" tabindex="-1"></a><span class="co"># row_data_index &lt;- final_data[final_data$ID == target_id, ]</span></span>
<span id="cb2244-8"><a href="integral-population-models.html#cb2244-8" tabindex="-1"></a><span class="co"># print(row_data_index)</span></span>
<span id="cb2244-9"><a href="integral-population-models.html#cb2244-9" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb2244-10"><a href="integral-population-models.html#cb2244-10" tabindex="-1"></a><span class="co"># #check out the other outlier</span></span>
<span id="cb2244-11"><a href="integral-population-models.html#cb2244-11" tabindex="-1"></a><span class="co"># row_index &lt;- 74</span></span>
<span id="cb2244-12"><a href="integral-population-models.html#cb2244-12" tabindex="-1"></a><span class="co"># value &lt;- final_data[row_index,]</span></span>
<span id="cb2244-13"><a href="integral-population-models.html#cb2244-13" tabindex="-1"></a><span class="co"># print(value) </span></span></code></pre></div>
<p>We could decide to remove outliers! Example below:</p>
<div class="sourceCode" id="cb2245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2245-1"><a href="integral-population-models.html#cb2245-1" tabindex="-1"></a><span class="co"># # Remove identified outliers</span></span>
<span id="cb2245-2"><a href="integral-population-models.html#cb2245-2" tabindex="-1"></a><span class="co"># final_data &lt;- final_data[-c(44, 74), ] </span></span>
<span id="cb2245-3"><a href="integral-population-models.html#cb2245-3" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb2245-4"><a href="integral-population-models.html#cb2245-4" tabindex="-1"></a><span class="co"># # Fit growth model on the log scale</span></span>
<span id="cb2245-5"><a href="integral-population-models.html#cb2245-5" tabindex="-1"></a><span class="co"># growth_log &lt;- lm(log(sizeNext) ~ log(size),</span></span>
<span id="cb2245-6"><a href="integral-population-models.html#cb2245-6" tabindex="-1"></a><span class="co">#                  na.action = na.omit,</span></span>
<span id="cb2245-7"><a href="integral-population-models.html#cb2245-7" tabindex="-1"></a><span class="co">#                  data = final_data)</span></span>
<span id="cb2245-8"><a href="integral-population-models.html#cb2245-8" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb2245-9"><a href="integral-population-models.html#cb2245-9" tabindex="-1"></a><span class="co"># # Extract residuals</span></span>
<span id="cb2245-10"><a href="integral-population-models.html#cb2245-10" tabindex="-1"></a><span class="co"># resid_log &lt;- residuals(growth_log)</span></span>
<span id="cb2245-11"><a href="integral-population-models.html#cb2245-11" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb2245-12"><a href="integral-population-models.html#cb2245-12" tabindex="-1"></a><span class="co"># # Test normality of residuals (on log scale)</span></span>
<span id="cb2245-13"><a href="integral-population-models.html#cb2245-13" tabindex="-1"></a><span class="co"># shapiro.test(resid_log)</span></span>
<span id="cb2245-14"><a href="integral-population-models.html#cb2245-14" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb2245-15"><a href="integral-population-models.html#cb2245-15" tabindex="-1"></a><span class="co"># # Set up plotting window</span></span>
<span id="cb2245-16"><a href="integral-population-models.html#cb2245-16" tabindex="-1"></a><span class="co"># par(mfrow = c(1, 2), mar = c(4, 4, 2, 1))</span></span>
<span id="cb2245-17"><a href="integral-population-models.html#cb2245-17" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb2245-18"><a href="integral-population-models.html#cb2245-18" tabindex="-1"></a><span class="co"># # Scatter plot of log(size) vs log(sizeNext)</span></span>
<span id="cb2245-19"><a href="integral-population-models.html#cb2245-19" tabindex="-1"></a><span class="co"># plot(log(sizeNext) ~ log(size),</span></span>
<span id="cb2245-20"><a href="integral-population-models.html#cb2245-20" tabindex="-1"></a><span class="co">#      data = final_data,</span></span>
<span id="cb2245-21"><a href="integral-population-models.html#cb2245-21" tabindex="-1"></a><span class="co">#      xlab = &quot;log(Size at time t)&quot;,</span></span>
<span id="cb2245-22"><a href="integral-population-models.html#cb2245-22" tabindex="-1"></a><span class="co">#      ylab = &quot;log(Size at time t+1)&quot;,</span></span>
<span id="cb2245-23"><a href="integral-population-models.html#cb2245-23" tabindex="-1"></a><span class="co">#      main = &quot;Growth relationship (log–log)&quot;)</span></span>
<span id="cb2245-24"><a href="integral-population-models.html#cb2245-24" tabindex="-1"></a><span class="co"># abline(growth_log, col = &quot;red&quot;, lwd = 2)</span></span>
<span id="cb2245-25"><a href="integral-population-models.html#cb2245-25" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb2245-26"><a href="integral-population-models.html#cb2245-26" tabindex="-1"></a><span class="co"># # Q-Q plot of residuals</span></span>
<span id="cb2245-27"><a href="integral-population-models.html#cb2245-27" tabindex="-1"></a><span class="co"># id.n &lt;- nrow(final_data)</span></span>
<span id="cb2245-28"><a href="integral-population-models.html#cb2245-28" tabindex="-1"></a><span class="co"># qqPlot(growth_log,</span></span>
<span id="cb2245-29"><a href="integral-population-models.html#cb2245-29" tabindex="-1"></a><span class="co">#        distribution = &quot;norm&quot;,</span></span>
<span id="cb2245-30"><a href="integral-population-models.html#cb2245-30" tabindex="-1"></a><span class="co">#        id.method = &quot;y&quot;,</span></span>
<span id="cb2245-31"><a href="integral-population-models.html#cb2245-31" tabindex="-1"></a><span class="co">#        id.cex = 0.6,</span></span>
<span id="cb2245-32"><a href="integral-population-models.html#cb2245-32" tabindex="-1"></a><span class="co">#        id.n = id.n,</span></span>
<span id="cb2245-33"><a href="integral-population-models.html#cb2245-33" tabindex="-1"></a><span class="co">#        id.col = &quot;blue&quot;,</span></span>
<span id="cb2245-34"><a href="integral-population-models.html#cb2245-34" tabindex="-1"></a><span class="co">#        id.location = &quot;ab&quot;)</span></span></code></pre></div>
<p>Remember, if you do remove outliers, you must refit the model!</p>
<div class="sourceCode" id="cb2246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2246-1"><a href="integral-population-models.html#cb2246-1" tabindex="-1"></a>growth_clean <span class="ot">&lt;-</span> growth_data <span class="sc">|&gt;</span></span>
<span id="cb2246-2"><a href="integral-population-models.html#cb2246-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>flag_outlier)</span>
<span id="cb2246-3"><a href="integral-population-models.html#cb2246-3" tabindex="-1"></a></span>
<span id="cb2246-4"><a href="integral-population-models.html#cb2246-4" tabindex="-1"></a><span class="co"># Refit model without outliers</span></span>
<span id="cb2246-5"><a href="integral-population-models.html#cb2246-5" tabindex="-1"></a>m_growth_quad_clean <span class="ot">&lt;-</span> <span class="fu">lm</span>(</span>
<span id="cb2246-6"><a href="integral-population-models.html#cb2246-6" tabindex="-1"></a>  <span class="fu">log</span>(sizeNext) <span class="sc">~</span> <span class="fu">poly</span>(<span class="fu">log</span>(size), <span class="dv">2</span>, <span class="at">raw =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2246-7"><a href="integral-population-models.html#cb2246-7" tabindex="-1"></a>  <span class="at">data =</span> growth_clean</span>
<span id="cb2246-8"><a href="integral-population-models.html#cb2246-8" tabindex="-1"></a>)</span>
<span id="cb2246-9"><a href="integral-population-models.html#cb2246-9" tabindex="-1"></a></span>
<span id="cb2246-10"><a href="integral-population-models.html#cb2246-10" tabindex="-1"></a><span class="co"># Compare models in order to determine whether the removal of the outlier changed fit.</span></span>
<span id="cb2246-11"><a href="integral-population-models.html#cb2246-11" tabindex="-1"></a><span class="fu">summary</span>(m_growth_quad)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = log(sizeNext) ~ poly(log(size), 2, raw = TRUE), 
##     data = growth_data)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -1.25728 -0.44403 -0.01676  0.38419  1.38985 
## 
## Coefficients:
##                                 Estimate Std. Error t value Pr(&gt;|t|)   
## (Intercept)                     -1.95684    1.09712  -1.784  0.07990 . 
## poly(log(size), 2, raw = TRUE)1  1.96402    0.67401   2.914  0.00512 **
## poly(log(size), 2, raw = TRUE)2 -0.19582    0.09611  -2.037  0.04634 * 
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.6424 on 56 degrees of freedom
## Multiple R-squared:  0.5209, Adjusted R-squared:  0.5038 
## F-statistic: 30.44 on 2 and 56 DF,  p-value: 1.128e-09</code></pre>
<div class="sourceCode" id="cb2248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2248-1"><a href="integral-population-models.html#cb2248-1" tabindex="-1"></a><span class="fu">summary</span>(m_growth_quad_clean)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = log(sizeNext) ~ poly(log(size), 2, raw = TRUE), 
##     data = growth_clean)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -1.25728 -0.44403 -0.01676  0.38419  1.38985 
## 
## Coefficients:
##                                 Estimate Std. Error t value Pr(&gt;|t|)   
## (Intercept)                     -1.95684    1.09712  -1.784  0.07990 . 
## poly(log(size), 2, raw = TRUE)1  1.96402    0.67401   2.914  0.00512 **
## poly(log(size), 2, raw = TRUE)2 -0.19582    0.09611  -2.037  0.04634 * 
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.6424 on 56 degrees of freedom
## Multiple R-squared:  0.5209, Adjusted R-squared:  0.5038 
## F-statistic: 30.44 on 2 and 56 DF,  p-value: 1.128e-09</code></pre>
<p>Note that if you do remove outliers you must repeat the model fitting process and update the final model that can be included in the kernel!</p>
</div>
<div id="probability-of-reproduction" class="section level4 hasAnchor" number="82.4.4.2">
<h4><span class="header-section-number">82.4.4.2</span> Probability of reproduction<a href="integral-population-models.html#probability-of-reproduction" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Next, we repeat this process for all individuals. I generally do not perform outlier removal for the rest of the analyses, since we want to capture true variation in vital rates. Thus, the oddballs can be important for understanding the system, but only if they are real outliers, rather than created by human error.</p>
<p>We model whether an individual reproduced (<code>fec1</code>) as a function of size:</p>
<p><span class="math display">\[
\text{logit}(p_{\text{repr}}) = \beta_0 + \beta_1 z
\]</span></p>
<p>where <span class="math inline">\(z\)</span> is the state variable (size at time <span class="math inline">\(t\)</span>).</p>
<p>First, let’s determine whether we should use a log transformation. We will apply this consistently across all models. Reproduction is often log transformed, since reproduction is expected to increase multiplicatively with size.</p>
<div class="sourceCode" id="cb2250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2250-1"><a href="integral-population-models.html#cb2250-1" tabindex="-1"></a>d_repr <span class="ot">&lt;-</span> final_data[final_data<span class="sc">$</span>Exclude <span class="sc">==</span> <span class="dv">0</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb2250-2"><a href="integral-population-models.html#cb2250-2" tabindex="-1"></a>d_repr <span class="ot">&lt;-</span> d_repr[<span class="sc">!</span><span class="fu">is.na</span>(d_repr<span class="sc">$</span>fec1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(d_repr<span class="sc">$</span>size) <span class="sc">&amp;</span> d_repr<span class="sc">$</span>size <span class="sc">&gt;</span> <span class="dv">0</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb2250-3"><a href="integral-population-models.html#cb2250-3" tabindex="-1"></a></span>
<span id="cb2250-4"><a href="integral-population-models.html#cb2250-4" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">is.factor</span>(d_repr<span class="sc">$</span>fec1) <span class="sc">||</span> <span class="fu">is.character</span>(d_repr<span class="sc">$</span>fec1)) {</span>
<span id="cb2250-5"><a href="integral-population-models.html#cb2250-5" tabindex="-1"></a>  d_repr<span class="sc">$</span>fec1 <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">as.factor</span>(d_repr<span class="sc">$</span>fec1)) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb2250-6"><a href="integral-population-models.html#cb2250-6" tabindex="-1"></a>}</span>
<span id="cb2250-7"><a href="integral-population-models.html#cb2250-7" tabindex="-1"></a></span>
<span id="cb2250-8"><a href="integral-population-models.html#cb2250-8" tabindex="-1"></a>m_repr     <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">glm</span>(fec1 <span class="sc">~</span> size,      <span class="at">data =</span> d_repr, <span class="at">family =</span> stats<span class="sc">::</span><span class="fu">binomial</span>())</span>
<span id="cb2250-9"><a href="integral-population-models.html#cb2250-9" tabindex="-1"></a>m_repr_log <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">glm</span>(fec1 <span class="sc">~</span> <span class="fu">log</span>(size), <span class="at">data =</span> d_repr, <span class="at">family =</span> stats<span class="sc">::</span><span class="fu">binomial</span>())</span>
<span id="cb2250-10"><a href="integral-population-models.html#cb2250-10" tabindex="-1"></a></span>
<span id="cb2250-11"><a href="integral-population-models.html#cb2250-11" tabindex="-1"></a>stats<span class="sc">::</span><span class="fu">AIC</span>(m_repr, m_repr_log)</span></code></pre></div>
<pre><code>##            df      AIC
## m_repr      2 35.03888
## m_repr_log  2 30.57976</code></pre>
<p>And yes, the log scale has the lower AIC value, so we choose log transformed data! Now let’s compare model fits (i.e., linear vs polynomial models).</p>
<div class="sourceCode" id="cb2252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2252-1"><a href="integral-population-models.html#cb2252-1" tabindex="-1"></a><span class="co"># Compare linear vs quadratic on log(size) for reproduction probability</span></span>
<span id="cb2252-2"><a href="integral-population-models.html#cb2252-2" tabindex="-1"></a></span>
<span id="cb2252-3"><a href="integral-population-models.html#cb2252-3" tabindex="-1"></a><span class="co"># Start with included rows only (keep this line if you use Exclude)</span></span>
<span id="cb2252-4"><a href="integral-population-models.html#cb2252-4" tabindex="-1"></a>d_repr <span class="ot">&lt;-</span> final_data[final_data<span class="sc">$</span>Exclude <span class="sc">==</span> <span class="dv">0</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb2252-5"><a href="integral-population-models.html#cb2252-5" tabindex="-1"></a></span>
<span id="cb2252-6"><a href="integral-population-models.html#cb2252-6" tabindex="-1"></a><span class="co"># Keep complete cases for variables used + ensure size is positive for log()</span></span>
<span id="cb2252-7"><a href="integral-population-models.html#cb2252-7" tabindex="-1"></a>d_repr <span class="ot">&lt;-</span> d_repr[<span class="sc">!</span><span class="fu">is.na</span>(d_repr<span class="sc">$</span>fec1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(d_repr<span class="sc">$</span>size) <span class="sc">&amp;</span> d_repr<span class="sc">$</span>size <span class="sc">&gt;</span> <span class="dv">0</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb2252-8"><a href="integral-population-models.html#cb2252-8" tabindex="-1"></a></span>
<span id="cb2252-9"><a href="integral-population-models.html#cb2252-9" tabindex="-1"></a><span class="co"># Ensure fec1 is 0/1 numeric</span></span>
<span id="cb2252-10"><a href="integral-population-models.html#cb2252-10" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">is.factor</span>(d_repr<span class="sc">$</span>fec1) <span class="sc">||</span> <span class="fu">is.character</span>(d_repr<span class="sc">$</span>fec1)) {</span>
<span id="cb2252-11"><a href="integral-population-models.html#cb2252-11" tabindex="-1"></a>  d_repr<span class="sc">$</span>fec1 <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">as.factor</span>(d_repr<span class="sc">$</span>fec1)) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb2252-12"><a href="integral-population-models.html#cb2252-12" tabindex="-1"></a>}</span>
<span id="cb2252-13"><a href="integral-population-models.html#cb2252-13" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(d_repr<span class="sc">$</span>fec1 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)))</span>
<span id="cb2252-14"><a href="integral-population-models.html#cb2252-14" tabindex="-1"></a></span>
<span id="cb2252-15"><a href="integral-population-models.html#cb2252-15" tabindex="-1"></a>m_repr_log_lin <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">glm</span>(</span>
<span id="cb2252-16"><a href="integral-population-models.html#cb2252-16" tabindex="-1"></a>  fec1 <span class="sc">~</span> <span class="fu">log</span>(size),</span>
<span id="cb2252-17"><a href="integral-population-models.html#cb2252-17" tabindex="-1"></a>  <span class="at">data   =</span> d_repr,</span>
<span id="cb2252-18"><a href="integral-population-models.html#cb2252-18" tabindex="-1"></a>  <span class="at">family =</span> stats<span class="sc">::</span><span class="fu">binomial</span>()</span>
<span id="cb2252-19"><a href="integral-population-models.html#cb2252-19" tabindex="-1"></a>)</span>
<span id="cb2252-20"><a href="integral-population-models.html#cb2252-20" tabindex="-1"></a></span>
<span id="cb2252-21"><a href="integral-population-models.html#cb2252-21" tabindex="-1"></a>m_repr_log_poly <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">glm</span>(</span>
<span id="cb2252-22"><a href="integral-population-models.html#cb2252-22" tabindex="-1"></a>  fec1 <span class="sc">~</span> <span class="fu">poly</span>(<span class="fu">log</span>(size), <span class="dv">2</span>, <span class="at">raw =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2252-23"><a href="integral-population-models.html#cb2252-23" tabindex="-1"></a>  <span class="at">data   =</span> d_repr,</span>
<span id="cb2252-24"><a href="integral-population-models.html#cb2252-24" tabindex="-1"></a>  <span class="at">family =</span> stats<span class="sc">::</span><span class="fu">binomial</span>()</span>
<span id="cb2252-25"><a href="integral-population-models.html#cb2252-25" tabindex="-1"></a>)</span>
<span id="cb2252-26"><a href="integral-population-models.html#cb2252-26" tabindex="-1"></a></span>
<span id="cb2252-27"><a href="integral-population-models.html#cb2252-27" tabindex="-1"></a>stats<span class="sc">::</span><span class="fu">AIC</span>(m_repr_log_lin, m_repr_log_poly)</span></code></pre></div>
<pre><code>##                 df      AIC
## m_repr_log_lin   2 30.57976
## m_repr_log_poly  3 31.11613</code></pre>
<p>Wonderful - the linear model wins! Next, we need to visualize the state variable-vital rate relationship.</p>
<div class="sourceCode" id="cb2254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2254-1"><a href="integral-population-models.html#cb2254-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2254-2"><a href="integral-population-models.html#cb2254-2" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb2254-3"><a href="integral-population-models.html#cb2254-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2254-4"><a href="integral-population-models.html#cb2254-4" tabindex="-1"></a></span>
<span id="cb2254-5"><a href="integral-population-models.html#cb2254-5" tabindex="-1"></a>newdat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2254-6"><a href="integral-population-models.html#cb2254-6" tabindex="-1"></a>  <span class="at">size =</span> <span class="fu">seq</span>(<span class="fu">min</span>(final_data<span class="sc">$</span>size, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2254-7"><a href="integral-population-models.html#cb2254-7" tabindex="-1"></a>             <span class="fu">max</span>(final_data<span class="sc">$</span>size, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2254-8"><a href="integral-population-models.html#cb2254-8" tabindex="-1"></a>             <span class="at">length.out =</span> <span class="dv">200</span>)</span>
<span id="cb2254-9"><a href="integral-population-models.html#cb2254-9" tabindex="-1"></a>)</span>
<span id="cb2254-10"><a href="integral-population-models.html#cb2254-10" tabindex="-1"></a></span>
<span id="cb2254-11"><a href="integral-population-models.html#cb2254-11" tabindex="-1"></a>pred_df <span class="ot">&lt;-</span> newdat <span class="sc">|&gt;</span></span>
<span id="cb2254-12"><a href="integral-population-models.html#cb2254-12" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2254-13"><a href="integral-population-models.html#cb2254-13" tabindex="-1"></a>    <span class="st">`</span><span class="at">Linear (log size)</span><span class="st">`</span>     <span class="ot">=</span> <span class="fu">predict</span>(m_repr_log_lin,  newdat, <span class="at">type =</span> <span class="st">&quot;response&quot;</span>),</span>
<span id="cb2254-14"><a href="integral-population-models.html#cb2254-14" tabindex="-1"></a>    <span class="st">`</span><span class="at">Quadratic (log size)</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">predict</span>(m_repr_log_poly, newdat, <span class="at">type =</span> <span class="st">&quot;response&quot;</span>)</span>
<span id="cb2254-15"><a href="integral-population-models.html#cb2254-15" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2254-16"><a href="integral-population-models.html#cb2254-16" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb2254-17"><a href="integral-population-models.html#cb2254-17" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span>size,</span>
<span id="cb2254-18"><a href="integral-population-models.html#cb2254-18" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;Model&quot;</span>,</span>
<span id="cb2254-19"><a href="integral-population-models.html#cb2254-19" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;Prediction&quot;</span></span>
<span id="cb2254-20"><a href="integral-population-models.html#cb2254-20" tabindex="-1"></a>  )</span>
<span id="cb2254-21"><a href="integral-population-models.html#cb2254-21" tabindex="-1"></a></span>
<span id="cb2254-22"><a href="integral-population-models.html#cb2254-22" tabindex="-1"></a><span class="fu">ggplot</span>(final_data, <span class="fu">aes</span>(size, fec1)) <span class="sc">+</span></span>
<span id="cb2254-23"><a href="integral-population-models.html#cb2254-23" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">height =</span> <span class="fl">0.05</span>, <span class="at">width =</span> <span class="dv">0</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb2254-24"><a href="integral-population-models.html#cb2254-24" tabindex="-1"></a>  <span class="fu">geom_line</span>(</span>
<span id="cb2254-25"><a href="integral-population-models.html#cb2254-25" tabindex="-1"></a>    <span class="at">data =</span> pred_df,</span>
<span id="cb2254-26"><a href="integral-population-models.html#cb2254-26" tabindex="-1"></a>    <span class="fu">aes</span>(size, Prediction, <span class="at">color =</span> Model),</span>
<span id="cb2254-27"><a href="integral-population-models.html#cb2254-27" tabindex="-1"></a>    <span class="at">linewidth =</span> <span class="dv">1</span></span>
<span id="cb2254-28"><a href="integral-population-models.html#cb2254-28" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2254-29"><a href="integral-population-models.html#cb2254-29" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb2254-30"><a href="integral-population-models.html#cb2254-30" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Probability of reproduction&quot;</span>,</span>
<span id="cb2254-31"><a href="integral-population-models.html#cb2254-31" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Model fit&quot;</span></span>
<span id="cb2254-32"><a href="integral-population-models.html#cb2254-32" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2254-33"><a href="integral-population-models.html#cb2254-33" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-21-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="seed-production" class="section level4 hasAnchor" number="82.4.4.3">
<h4><span class="header-section-number">82.4.4.3</span> Seed production<a href="integral-population-models.html#seed-production" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Seed production (<code>fec2</code>) is only observed for individuals that reproduced. We therefore
model fecundity conditional on reproduction (<code>fec1 == 1</code>). Because seed counts are
highly right-skewed and often overdispersed, we use a negative binomial model.</p>
<p>As above, we first evaluate whether a log transformation of size is appropriate, then
compare linear versus polynomial functional forms.</p>
<div class="sourceCode" id="cb2255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2255-1"><a href="integral-population-models.html#cb2255-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2255-2"><a href="integral-population-models.html#cb2255-2" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb2255-3"><a href="integral-population-models.html#cb2255-3" tabindex="-1"></a></span>
<span id="cb2255-4"><a href="integral-population-models.html#cb2255-4" tabindex="-1"></a>fec_data <span class="ot">&lt;-</span> final_data <span class="sc">%&gt;%</span></span>
<span id="cb2255-5"><a href="integral-population-models.html#cb2255-5" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb2255-6"><a href="integral-population-models.html#cb2255-6" tabindex="-1"></a>    fec1 <span class="sc">==</span> <span class="dv">1</span>,</span>
<span id="cb2255-7"><a href="integral-population-models.html#cb2255-7" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(fec2),</span>
<span id="cb2255-8"><a href="integral-population-models.html#cb2255-8" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(size)</span>
<span id="cb2255-9"><a href="integral-population-models.html#cb2255-9" tabindex="-1"></a>  )</span>
<span id="cb2255-10"><a href="integral-population-models.html#cb2255-10" tabindex="-1"></a></span>
<span id="cb2255-11"><a href="integral-population-models.html#cb2255-11" tabindex="-1"></a><span class="co"># if you use log(size), require positive size</span></span>
<span id="cb2255-12"><a href="integral-population-models.html#cb2255-12" tabindex="-1"></a>fec_data_log <span class="ot">&lt;-</span> fec_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(size <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb2255-13"><a href="integral-population-models.html#cb2255-13" tabindex="-1"></a></span>
<span id="cb2255-14"><a href="integral-population-models.html#cb2255-14" tabindex="-1"></a><span class="co"># (Recommended) use the SAME dataset for both models:</span></span>
<span id="cb2255-15"><a href="integral-population-models.html#cb2255-15" tabindex="-1"></a><span class="co"># if any size &lt;= 0 exists, drop them for both models</span></span>
<span id="cb2255-16"><a href="integral-population-models.html#cb2255-16" tabindex="-1"></a>fec_data_use <span class="ot">&lt;-</span> fec_data_log</span>
<span id="cb2255-17"><a href="integral-population-models.html#cb2255-17" tabindex="-1"></a></span>
<span id="cb2255-18"><a href="integral-population-models.html#cb2255-18" tabindex="-1"></a>m_fec_lin <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">glm.nb</span>(</span>
<span id="cb2255-19"><a href="integral-population-models.html#cb2255-19" tabindex="-1"></a>  fec2 <span class="sc">~</span> size,</span>
<span id="cb2255-20"><a href="integral-population-models.html#cb2255-20" tabindex="-1"></a>  <span class="at">data =</span> fec_data_use</span>
<span id="cb2255-21"><a href="integral-population-models.html#cb2255-21" tabindex="-1"></a>)</span>
<span id="cb2255-22"><a href="integral-population-models.html#cb2255-22" tabindex="-1"></a></span>
<span id="cb2255-23"><a href="integral-population-models.html#cb2255-23" tabindex="-1"></a>m_fec_log <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">glm.nb</span>(</span>
<span id="cb2255-24"><a href="integral-population-models.html#cb2255-24" tabindex="-1"></a>  fec2 <span class="sc">~</span> <span class="fu">log</span>(size),</span>
<span id="cb2255-25"><a href="integral-population-models.html#cb2255-25" tabindex="-1"></a>  <span class="at">data =</span> fec_data_use</span>
<span id="cb2255-26"><a href="integral-population-models.html#cb2255-26" tabindex="-1"></a>)</span>
<span id="cb2255-27"><a href="integral-population-models.html#cb2255-27" tabindex="-1"></a></span>
<span id="cb2255-28"><a href="integral-population-models.html#cb2255-28" tabindex="-1"></a><span class="fu">AIC</span>(m_fec_lin, m_fec_log)</span></code></pre></div>
<pre><code>##           df      AIC
## m_fec_lin  3 972.3240
## m_fec_log  3 967.9926</code></pre>
<p>Log transformation wins again! As with reproduction probability, fecundity is often expected to increase
multiplicatively with size.</p>
<div class="sourceCode" id="cb2257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2257-1"><a href="integral-population-models.html#cb2257-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2257-2"><a href="integral-population-models.html#cb2257-2" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb2257-3"><a href="integral-population-models.html#cb2257-3" tabindex="-1"></a></span>
<span id="cb2257-4"><a href="integral-population-models.html#cb2257-4" tabindex="-1"></a><span class="co"># Ensure the exact same rows are used in both models</span></span>
<span id="cb2257-5"><a href="integral-population-models.html#cb2257-5" tabindex="-1"></a>fec_data2 <span class="ot">&lt;-</span> fec_data <span class="sc">%&gt;%</span></span>
<span id="cb2257-6"><a href="integral-population-models.html#cb2257-6" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb2257-7"><a href="integral-population-models.html#cb2257-7" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(fec2),</span>
<span id="cb2257-8"><a href="integral-population-models.html#cb2257-8" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(size),</span>
<span id="cb2257-9"><a href="integral-population-models.html#cb2257-9" tabindex="-1"></a>    <span class="fu">is.finite</span>(size),</span>
<span id="cb2257-10"><a href="integral-population-models.html#cb2257-10" tabindex="-1"></a>    size <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb2257-11"><a href="integral-population-models.html#cb2257-11" tabindex="-1"></a>  )</span>
<span id="cb2257-12"><a href="integral-population-models.html#cb2257-12" tabindex="-1"></a></span>
<span id="cb2257-13"><a href="integral-population-models.html#cb2257-13" tabindex="-1"></a><span class="co"># Optional sanity check</span></span>
<span id="cb2257-14"><a href="integral-population-models.html#cb2257-14" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">nrow</span>(fec_data2) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb2257-15"><a href="integral-population-models.html#cb2257-15" tabindex="-1"></a></span>
<span id="cb2257-16"><a href="integral-population-models.html#cb2257-16" tabindex="-1"></a>m_fec_log_lin <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">glm.nb</span>(</span>
<span id="cb2257-17"><a href="integral-population-models.html#cb2257-17" tabindex="-1"></a>  fec2 <span class="sc">~</span> <span class="fu">log</span>(size),</span>
<span id="cb2257-18"><a href="integral-population-models.html#cb2257-18" tabindex="-1"></a>  <span class="at">data =</span> fec_data2</span>
<span id="cb2257-19"><a href="integral-population-models.html#cb2257-19" tabindex="-1"></a>)</span>
<span id="cb2257-20"><a href="integral-population-models.html#cb2257-20" tabindex="-1"></a></span>
<span id="cb2257-21"><a href="integral-population-models.html#cb2257-21" tabindex="-1"></a>m_fec_log_poly <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">glm.nb</span>(</span>
<span id="cb2257-22"><a href="integral-population-models.html#cb2257-22" tabindex="-1"></a>  fec2 <span class="sc">~</span> <span class="fu">poly</span>(<span class="fu">log</span>(size), <span class="dv">2</span>, <span class="at">raw =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2257-23"><a href="integral-population-models.html#cb2257-23" tabindex="-1"></a>  <span class="at">data =</span> fec_data2</span>
<span id="cb2257-24"><a href="integral-population-models.html#cb2257-24" tabindex="-1"></a>)</span>
<span id="cb2257-25"><a href="integral-population-models.html#cb2257-25" tabindex="-1"></a></span>
<span id="cb2257-26"><a href="integral-population-models.html#cb2257-26" tabindex="-1"></a><span class="fu">AIC</span>(m_fec_log_lin, m_fec_log_poly)</span></code></pre></div>
<pre><code>##                df      AIC
## m_fec_log_lin   3 967.9926
## m_fec_log_poly  4 963.6462</code></pre>
<p>Polynomial model wins! But note, we have a warning that the model didn’t converge! This is annoying, but not uncommon for rare plants. This typically means that we should choose the simpler model, despite AIC findings. Let’s see whether the simpler model converges.</p>
<div class="sourceCode" id="cb2259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2259-1"><a href="integral-population-models.html#cb2259-1" tabindex="-1"></a><span class="fu">summary</span>(m_fec_log_lin)</span></code></pre></div>
<pre><code>## 
## Call:
## MASS::glm.nb(formula = fec2 ~ log(size), data = fec_data2, init.theta = 0.4985063643, 
##     link = log)
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)  -4.8531     2.9095  -1.668   0.0953 .  
## log(size)     3.5580     0.6291   5.656 1.55e-08 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for Negative Binomial(0.4985) family taken to be 1)
## 
##     Null deviance: 60.115  on 38  degrees of freedom
## Residual deviance: 49.562  on 37  degrees of freedom
## AIC: 967.99
## 
## Number of Fisher Scoring iterations: 1
## 
## 
##               Theta:  0.4985 
##           Std. Err.:  0.0932 
## 
##  2 x log-likelihood:  -961.9930</code></pre>
<p>O.K. The simple model converged! We will likely go with that model. Let’s still consider whether the preferred model is biologically feasible. Since convergence errors often occur due to low sample size, it is usually safer to go with the simpler model! Let’s visualize both models.</p>
<div class="sourceCode" id="cb2261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2261-1"><a href="integral-population-models.html#cb2261-1" tabindex="-1"></a>newdat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2261-2"><a href="integral-population-models.html#cb2261-2" tabindex="-1"></a>  <span class="at">size =</span> <span class="fu">seq</span>(<span class="fu">min</span>(fec_data<span class="sc">$</span>size, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2261-3"><a href="integral-population-models.html#cb2261-3" tabindex="-1"></a>             <span class="fu">max</span>(fec_data<span class="sc">$</span>size, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2261-4"><a href="integral-population-models.html#cb2261-4" tabindex="-1"></a>             <span class="at">length.out =</span> <span class="dv">200</span>)</span>
<span id="cb2261-5"><a href="integral-population-models.html#cb2261-5" tabindex="-1"></a>)</span>
<span id="cb2261-6"><a href="integral-population-models.html#cb2261-6" tabindex="-1"></a></span>
<span id="cb2261-7"><a href="integral-population-models.html#cb2261-7" tabindex="-1"></a>pred_df <span class="ot">&lt;-</span> newdat <span class="sc">|&gt;</span></span>
<span id="cb2261-8"><a href="integral-population-models.html#cb2261-8" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2261-9"><a href="integral-population-models.html#cb2261-9" tabindex="-1"></a>    <span class="st">`</span><span class="at">Linear (log size)</span><span class="st">`</span>    <span class="ot">=</span> <span class="fu">predict</span>(m_fec_log_lin,  newdat, <span class="at">type =</span> <span class="st">&quot;response&quot;</span>),</span>
<span id="cb2261-10"><a href="integral-population-models.html#cb2261-10" tabindex="-1"></a>    <span class="st">`</span><span class="at">Quadratic (log size)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">predict</span>(m_fec_log_poly, newdat, <span class="at">type =</span> <span class="st">&quot;response&quot;</span>)</span>
<span id="cb2261-11"><a href="integral-population-models.html#cb2261-11" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2261-12"><a href="integral-population-models.html#cb2261-12" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb2261-13"><a href="integral-population-models.html#cb2261-13" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span>size,</span>
<span id="cb2261-14"><a href="integral-population-models.html#cb2261-14" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;Model&quot;</span>,</span>
<span id="cb2261-15"><a href="integral-population-models.html#cb2261-15" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;Prediction&quot;</span></span>
<span id="cb2261-16"><a href="integral-population-models.html#cb2261-16" tabindex="-1"></a>  )</span>
<span id="cb2261-17"><a href="integral-population-models.html#cb2261-17" tabindex="-1"></a></span>
<span id="cb2261-18"><a href="integral-population-models.html#cb2261-18" tabindex="-1"></a><span class="fu">ggplot</span>(fec_data, <span class="fu">aes</span>(size, fec2)) <span class="sc">+</span></span>
<span id="cb2261-19"><a href="integral-population-models.html#cb2261-19" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb2261-20"><a href="integral-population-models.html#cb2261-20" tabindex="-1"></a>  <span class="fu">geom_line</span>(</span>
<span id="cb2261-21"><a href="integral-population-models.html#cb2261-21" tabindex="-1"></a>    <span class="at">data =</span> pred_df,</span>
<span id="cb2261-22"><a href="integral-population-models.html#cb2261-22" tabindex="-1"></a>    <span class="fu">aes</span>(size, Prediction, <span class="at">color =</span> Model),</span>
<span id="cb2261-23"><a href="integral-population-models.html#cb2261-23" tabindex="-1"></a>    <span class="at">linewidth =</span> <span class="dv">1</span></span>
<span id="cb2261-24"><a href="integral-population-models.html#cb2261-24" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2261-25"><a href="integral-population-models.html#cb2261-25" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb2261-26"><a href="integral-population-models.html#cb2261-26" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb2261-27"><a href="integral-population-models.html#cb2261-27" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Seed production&quot;</span>,</span>
<span id="cb2261-28"><a href="integral-population-models.html#cb2261-28" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Model fit&quot;</span></span>
<span id="cb2261-29"><a href="integral-population-models.html#cb2261-29" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2261-30"><a href="integral-population-models.html#cb2261-30" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-24-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb2262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2262-1"><a href="integral-population-models.html#cb2262-1" tabindex="-1"></a><span class="co"># SEED PRODUCTION MODEL (Poisson GLM) ----</span></span>
<span id="cb2262-2"><a href="integral-population-models.html#cb2262-2" tabindex="-1"></a><span class="co"># Check for OVERDISPERSION </span></span>
<span id="cb2262-3"><a href="integral-population-models.html#cb2262-3" tabindex="-1"></a><span class="co"># check_overdispersion &lt;- function(model) {</span></span>
<span id="cb2262-4"><a href="integral-population-models.html#cb2262-4" tabindex="-1"></a><span class="co">#   resid_dev &lt;- sum(residuals(model, type = &quot;deviance&quot;)^2)</span></span>
<span id="cb2262-5"><a href="integral-population-models.html#cb2262-5" tabindex="-1"></a><span class="co">#   ratio &lt;- resid_dev / model$df.residual</span></span>
<span id="cb2262-6"><a href="integral-population-models.html#cb2262-6" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb2262-7"><a href="integral-population-models.html#cb2262-7" tabindex="-1"></a><span class="co">#   cat(&quot;\n=== Overdispersion Check (Poisson Model) ===\n&quot;)</span></span>
<span id="cb2262-8"><a href="integral-population-models.html#cb2262-8" tabindex="-1"></a><span class="co">#   cat(&quot;Residual deviance:&quot;, round(resid_dev, 2), &quot;\n&quot;)</span></span>
<span id="cb2262-9"><a href="integral-population-models.html#cb2262-9" tabindex="-1"></a><span class="co">#   cat(&quot;Degrees of freedom:&quot;, model$df.residual, &quot;\n&quot;)</span></span>
<span id="cb2262-10"><a href="integral-population-models.html#cb2262-10" tabindex="-1"></a><span class="co">#   cat(&quot;Dispersion ratio:&quot;, round(ratio, 3), &quot;\n&quot;)</span></span>
<span id="cb2262-11"><a href="integral-population-models.html#cb2262-11" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb2262-12"><a href="integral-population-models.html#cb2262-12" tabindex="-1"></a><span class="co">#   if(ratio &gt; 2) {</span></span>
<span id="cb2262-13"><a href="integral-population-models.html#cb2262-13" tabindex="-1"></a><span class="co">#     cat(&quot;⚠️  OVERDISPERSED - consider using negative binomial or quasipoisson\n&quot;)</span></span>
<span id="cb2262-14"><a href="integral-population-models.html#cb2262-14" tabindex="-1"></a><span class="co">#   } else if(ratio &lt; 0.5) {</span></span>
<span id="cb2262-15"><a href="integral-population-models.html#cb2262-15" tabindex="-1"></a><span class="co">#     cat(&quot;⚠️  UNDERDISPERSED - check for zero-inflation or model structure\n&quot;)</span></span>
<span id="cb2262-16"><a href="integral-population-models.html#cb2262-16" tabindex="-1"></a><span class="co">#   } else {</span></span>
<span id="cb2262-17"><a href="integral-population-models.html#cb2262-17" tabindex="-1"></a><span class="co">#     cat(&quot;✓ Dispersion looks reasonable for Poisson\n&quot;)</span></span>
<span id="cb2262-18"><a href="integral-population-models.html#cb2262-18" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb2262-19"><a href="integral-population-models.html#cb2262-19" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb2262-20"><a href="integral-population-models.html#cb2262-20" tabindex="-1"></a><span class="co">#   return(ratio)</span></span>
<span id="cb2262-21"><a href="integral-population-models.html#cb2262-21" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb2262-22"><a href="integral-population-models.html#cb2262-22" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb2262-23"><a href="integral-population-models.html#cb2262-23" tabindex="-1"></a><span class="co"># dispersion_ratio &lt;- check_overdispersion(seed_mod)</span></span>
<span id="cb2262-24"><a href="integral-population-models.html#cb2262-24" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb2262-25"><a href="integral-population-models.html#cb2262-25" tabindex="-1"></a><span class="co"># # If overdispersed, consider refitting:</span></span>
<span id="cb2262-26"><a href="integral-population-models.html#cb2262-26" tabindex="-1"></a><span class="co"># if(dispersion_ratio &gt; 2) {</span></span>
<span id="cb2262-27"><a href="integral-population-models.html#cb2262-27" tabindex="-1"></a><span class="co">#   cat(&quot;\n📝 Consider refitting as:\n&quot;)</span></span>
<span id="cb2262-28"><a href="integral-population-models.html#cb2262-28" tabindex="-1"></a><span class="co">#   cat(&quot;   library(MASS)\n&quot;)</span></span>
<span id="cb2262-29"><a href="integral-population-models.html#cb2262-29" tabindex="-1"></a><span class="co">#   cat(&quot;   seed_mod_nb &lt;- glm.nb(fec2 ~ size_log + I(size_log^2), data = final_data)\n&quot;)</span></span>
<span id="cb2262-30"><a href="integral-population-models.html#cb2262-30" tabindex="-1"></a><span class="co"># }</span></span></code></pre></div>
<p>O.K. we have a judgement call, we’ve seen ‘aging’ in other metrics, so this could be reasonable, but the apparent downturn looks like it is driven by a small number of large individuals, and with the lack of model convergence, any estimate of “aging effects” would be highly uncertain. This could go either way and we could compare the difference in outcomes!</p>
</div>
<div id="survival" class="section level4 hasAnchor" number="82.4.4.4">
<h4><span class="header-section-number">82.4.4.4</span> Survival<a href="integral-population-models.html#survival" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Survival is modeled as a binary response and is therefore fit using a binomial model.
We do not perform outlier removal for survival, as extreme values (alive vs. dead)
are intrinsic to the process.</p>
<div class="sourceCode" id="cb2263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2263-1"><a href="integral-population-models.html#cb2263-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2263-2"><a href="integral-population-models.html#cb2263-2" tabindex="-1"></a></span>
<span id="cb2263-3"><a href="integral-population-models.html#cb2263-3" tabindex="-1"></a><span class="co"># Use identical rows for both models (critical for AIC comparisons)</span></span>
<span id="cb2263-4"><a href="integral-population-models.html#cb2263-4" tabindex="-1"></a>surv_data <span class="ot">&lt;-</span> final_data <span class="sc">%&gt;%</span></span>
<span id="cb2263-5"><a href="integral-population-models.html#cb2263-5" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb2263-6"><a href="integral-population-models.html#cb2263-6" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(surv),</span>
<span id="cb2263-7"><a href="integral-population-models.html#cb2263-7" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(size),</span>
<span id="cb2263-8"><a href="integral-population-models.html#cb2263-8" tabindex="-1"></a>    <span class="fu">is.finite</span>(size),</span>
<span id="cb2263-9"><a href="integral-population-models.html#cb2263-9" tabindex="-1"></a>    size <span class="sc">&gt;</span> <span class="dv">0</span>   <span class="co"># needed for log(size)</span></span>
<span id="cb2263-10"><a href="integral-population-models.html#cb2263-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb2263-11"><a href="integral-population-models.html#cb2263-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2263-12"><a href="integral-population-models.html#cb2263-12" tabindex="-1"></a>    <span class="at">surv =</span> <span class="fu">as.integer</span>(surv)  <span class="co"># optional: ensure 0/1</span></span>
<span id="cb2263-13"><a href="integral-population-models.html#cb2263-13" tabindex="-1"></a>  )</span>
<span id="cb2263-14"><a href="integral-population-models.html#cb2263-14" tabindex="-1"></a></span>
<span id="cb2263-15"><a href="integral-population-models.html#cb2263-15" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">nrow</span>(surv_data) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb2263-16"><a href="integral-population-models.html#cb2263-16" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(surv_data<span class="sc">$</span>surv <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)))</span>
<span id="cb2263-17"><a href="integral-population-models.html#cb2263-17" tabindex="-1"></a></span>
<span id="cb2263-18"><a href="integral-population-models.html#cb2263-18" tabindex="-1"></a>m_surv_lin <span class="ot">&lt;-</span> <span class="fu">glm</span>(</span>
<span id="cb2263-19"><a href="integral-population-models.html#cb2263-19" tabindex="-1"></a>  surv <span class="sc">~</span> size,</span>
<span id="cb2263-20"><a href="integral-population-models.html#cb2263-20" tabindex="-1"></a>  <span class="at">data =</span> surv_data,</span>
<span id="cb2263-21"><a href="integral-population-models.html#cb2263-21" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">binomial</span>()</span>
<span id="cb2263-22"><a href="integral-population-models.html#cb2263-22" tabindex="-1"></a>)</span>
<span id="cb2263-23"><a href="integral-population-models.html#cb2263-23" tabindex="-1"></a></span>
<span id="cb2263-24"><a href="integral-population-models.html#cb2263-24" tabindex="-1"></a>m_surv_log <span class="ot">&lt;-</span> <span class="fu">glm</span>(</span>
<span id="cb2263-25"><a href="integral-population-models.html#cb2263-25" tabindex="-1"></a>  surv <span class="sc">~</span> <span class="fu">log</span>(size),</span>
<span id="cb2263-26"><a href="integral-population-models.html#cb2263-26" tabindex="-1"></a>  <span class="at">data =</span> surv_data,</span>
<span id="cb2263-27"><a href="integral-population-models.html#cb2263-27" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">binomial</span>()</span>
<span id="cb2263-28"><a href="integral-population-models.html#cb2263-28" tabindex="-1"></a>)</span>
<span id="cb2263-29"><a href="integral-population-models.html#cb2263-29" tabindex="-1"></a></span>
<span id="cb2263-30"><a href="integral-population-models.html#cb2263-30" tabindex="-1"></a><span class="fu">AIC</span>(m_surv_lin, m_surv_log)</span></code></pre></div>
<pre><code>##            df      AIC
## m_surv_lin  2 93.24766
## m_surv_log  2 93.78090</code></pre>
<p>These are so close! Given the other relationships, I’m just going to go with the log scale. Notably, you can include both scales for different variables in an IPM.</p>
<p>In an Integral Projection Model (IPM), all vital rates are linked to the same underlying
<strong>state variable</strong> (e.g., size), but they do <strong>not</strong> need to share the same functional
form. This means that it is perfectly acceptable for one vital rate to be modeled with
a linear relationship, while another uses a polynomial or nonlinear form.</p>
<p>Each vital rate reflects a different biological process, and these processes often
scale with size in different ways. For example, growth may decelerate at large sizes,
while survival changes more gradually, and reproduction increases multiplicatively
with size. Allowing each vital rate to adopt the functional form that best reflects its
biology improves realism and model performance.</p>
<p>Importantly, all vital-rate models must be well supported by the data and behave
sensibly when extrapolated across the full range of sizes included in the IPM. More
complex models (e.g., polynomial fits) are retained only when they converge reliably
and do not introduce implausible behavior at the boundaries of the size distribution.</p>
<p>In practice, this means that a single IPM may include:</p>
<ul>
<li>a quadratic growth model,</li>
<li>a linear (log-scale) reproduction model, and</li>
<li>a linear survival model,</li>
</ul>
<p>all operating together within the same projection kernel. The consistency of the IPM
comes from the shared state variable, not from enforcing identical mathematical forms
across vital rates.</p>
<p>All that said, I tend to find that the same transformation works across all equations and I find it more parsimonous to include the same transformation, especially when data are sparse.</p>
<div class="sourceCode" id="cb2265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2265-1"><a href="integral-population-models.html#cb2265-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2265-2"><a href="integral-population-models.html#cb2265-2" tabindex="-1"></a></span>
<span id="cb2265-3"><a href="integral-population-models.html#cb2265-3" tabindex="-1"></a><span class="co"># one shared, &quot;safe for log()&quot; dataset for BOTH models</span></span>
<span id="cb2265-4"><a href="integral-population-models.html#cb2265-4" tabindex="-1"></a>surv_data <span class="ot">&lt;-</span> final_data <span class="sc">%&gt;%</span></span>
<span id="cb2265-5"><a href="integral-population-models.html#cb2265-5" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb2265-6"><a href="integral-population-models.html#cb2265-6" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(surv),</span>
<span id="cb2265-7"><a href="integral-population-models.html#cb2265-7" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(size),</span>
<span id="cb2265-8"><a href="integral-population-models.html#cb2265-8" tabindex="-1"></a>    <span class="fu">is.finite</span>(size),</span>
<span id="cb2265-9"><a href="integral-population-models.html#cb2265-9" tabindex="-1"></a>    size <span class="sc">&gt;</span> <span class="dv">0</span>          <span class="co"># required for log(size)</span></span>
<span id="cb2265-10"><a href="integral-population-models.html#cb2265-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb2265-11"><a href="integral-population-models.html#cb2265-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2265-12"><a href="integral-population-models.html#cb2265-12" tabindex="-1"></a>    <span class="at">surv =</span> <span class="fu">as.integer</span>(surv)  <span class="co"># ensure 0/1 (optional but nice)</span></span>
<span id="cb2265-13"><a href="integral-population-models.html#cb2265-13" tabindex="-1"></a>  )</span>
<span id="cb2265-14"><a href="integral-population-models.html#cb2265-14" tabindex="-1"></a></span>
<span id="cb2265-15"><a href="integral-population-models.html#cb2265-15" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">nrow</span>(surv_data) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb2265-16"><a href="integral-population-models.html#cb2265-16" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(surv_data<span class="sc">$</span>surv <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)))</span>
<span id="cb2265-17"><a href="integral-population-models.html#cb2265-17" tabindex="-1"></a></span>
<span id="cb2265-18"><a href="integral-population-models.html#cb2265-18" tabindex="-1"></a>m_surv_log_lin <span class="ot">&lt;-</span> <span class="fu">glm</span>(</span>
<span id="cb2265-19"><a href="integral-population-models.html#cb2265-19" tabindex="-1"></a>  surv <span class="sc">~</span> <span class="fu">log</span>(size),</span>
<span id="cb2265-20"><a href="integral-population-models.html#cb2265-20" tabindex="-1"></a>  <span class="at">data   =</span> surv_data,</span>
<span id="cb2265-21"><a href="integral-population-models.html#cb2265-21" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">binomial</span>()</span>
<span id="cb2265-22"><a href="integral-population-models.html#cb2265-22" tabindex="-1"></a>)</span>
<span id="cb2265-23"><a href="integral-population-models.html#cb2265-23" tabindex="-1"></a></span>
<span id="cb2265-24"><a href="integral-population-models.html#cb2265-24" tabindex="-1"></a>m_surv_log_poly <span class="ot">&lt;-</span> <span class="fu">glm</span>(</span>
<span id="cb2265-25"><a href="integral-population-models.html#cb2265-25" tabindex="-1"></a>  surv <span class="sc">~</span> <span class="fu">poly</span>(<span class="fu">log</span>(size), <span class="dv">2</span>, <span class="at">raw =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2265-26"><a href="integral-population-models.html#cb2265-26" tabindex="-1"></a>  <span class="at">data   =</span> surv_data,</span>
<span id="cb2265-27"><a href="integral-population-models.html#cb2265-27" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">binomial</span>()</span>
<span id="cb2265-28"><a href="integral-population-models.html#cb2265-28" tabindex="-1"></a>)</span>
<span id="cb2265-29"><a href="integral-population-models.html#cb2265-29" tabindex="-1"></a></span>
<span id="cb2265-30"><a href="integral-population-models.html#cb2265-30" tabindex="-1"></a><span class="fu">AIC</span>(m_surv_log_lin, m_surv_log_poly)</span></code></pre></div>
<pre><code>##                 df      AIC
## m_surv_log_lin   2 93.78090
## m_surv_log_poly  3 94.31272</code></pre>
<p>Let’s visualize!</p>
<div class="sourceCode" id="cb2267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2267-1"><a href="integral-population-models.html#cb2267-1" tabindex="-1"></a>newdat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2267-2"><a href="integral-population-models.html#cb2267-2" tabindex="-1"></a>  <span class="at">size =</span> <span class="fu">seq</span>(<span class="fu">min</span>(final_data<span class="sc">$</span>size, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2267-3"><a href="integral-population-models.html#cb2267-3" tabindex="-1"></a>             <span class="fu">max</span>(final_data<span class="sc">$</span>size, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2267-4"><a href="integral-population-models.html#cb2267-4" tabindex="-1"></a>             <span class="at">length.out =</span> <span class="dv">200</span>)</span>
<span id="cb2267-5"><a href="integral-population-models.html#cb2267-5" tabindex="-1"></a>)</span>
<span id="cb2267-6"><a href="integral-population-models.html#cb2267-6" tabindex="-1"></a></span>
<span id="cb2267-7"><a href="integral-population-models.html#cb2267-7" tabindex="-1"></a>pred_df <span class="ot">&lt;-</span> newdat <span class="sc">|&gt;</span></span>
<span id="cb2267-8"><a href="integral-population-models.html#cb2267-8" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2267-9"><a href="integral-population-models.html#cb2267-9" tabindex="-1"></a>    <span class="st">`</span><span class="at">Linear (log size)</span><span class="st">`</span>    <span class="ot">=</span> <span class="fu">predict</span>(m_surv_log_lin,  newdat, <span class="at">type =</span> <span class="st">&quot;response&quot;</span>),</span>
<span id="cb2267-10"><a href="integral-population-models.html#cb2267-10" tabindex="-1"></a>    <span class="st">`</span><span class="at">Quadratic (log size)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">predict</span>(m_surv_log_poly, newdat, <span class="at">type =</span> <span class="st">&quot;response&quot;</span>)</span>
<span id="cb2267-11"><a href="integral-population-models.html#cb2267-11" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2267-12"><a href="integral-population-models.html#cb2267-12" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb2267-13"><a href="integral-population-models.html#cb2267-13" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span>size,</span>
<span id="cb2267-14"><a href="integral-population-models.html#cb2267-14" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;Model&quot;</span>,</span>
<span id="cb2267-15"><a href="integral-population-models.html#cb2267-15" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;Prediction&quot;</span></span>
<span id="cb2267-16"><a href="integral-population-models.html#cb2267-16" tabindex="-1"></a>  )</span>
<span id="cb2267-17"><a href="integral-population-models.html#cb2267-17" tabindex="-1"></a></span>
<span id="cb2267-18"><a href="integral-population-models.html#cb2267-18" tabindex="-1"></a><span class="fu">ggplot</span>(final_data, <span class="fu">aes</span>(size, surv)) <span class="sc">+</span></span>
<span id="cb2267-19"><a href="integral-population-models.html#cb2267-19" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">height =</span> <span class="fl">0.05</span>, <span class="at">width =</span> <span class="dv">0</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb2267-20"><a href="integral-population-models.html#cb2267-20" tabindex="-1"></a>  <span class="fu">geom_line</span>(</span>
<span id="cb2267-21"><a href="integral-population-models.html#cb2267-21" tabindex="-1"></a>    <span class="at">data =</span> pred_df,</span>
<span id="cb2267-22"><a href="integral-population-models.html#cb2267-22" tabindex="-1"></a>    <span class="fu">aes</span>(size, Prediction, <span class="at">color =</span> Model),</span>
<span id="cb2267-23"><a href="integral-population-models.html#cb2267-23" tabindex="-1"></a>    <span class="at">linewidth =</span> <span class="dv">1</span></span>
<span id="cb2267-24"><a href="integral-population-models.html#cb2267-24" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2267-25"><a href="integral-population-models.html#cb2267-25" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb2267-26"><a href="integral-population-models.html#cb2267-26" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Survival probability&quot;</span>,</span>
<span id="cb2267-27"><a href="integral-population-models.html#cb2267-27" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Model fit&quot;</span></span>
<span id="cb2267-28"><a href="integral-population-models.html#cb2267-28" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2267-29"><a href="integral-population-models.html#cb2267-29" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-27-1.png" alt="" width="960" style="display: block; margin: auto;" />
Oh yeah. Confirmed. That polynomial looks bonkers!</p>
<p>One thing to note in terms of packages. The IPMpack package has fairly slick built in equations for these initial model comparisons. For this reason, I sometimes mix-and-match packages, using the IPM pack for initial visualization and other packages or hand-coding for the actual modeling! The approach is up to you, though sometimes coding just gives you more flexibility!</p>
<p>At this point, we have assembled all of the vital-rate models needed to build a
population projection model: (1) growth, (2) survival, (3) probability of reproduction,
and (4) reproductive output. Together, these components describe how individuals move
through the population from one time step to the next.</p>
<p>In principle, additional demographic processes could be modeled if sufficient data
were available. For example, reproduction could be decomposed into multiple steps,
such as (1) the probability of flowering, (2) the probability that a flower sets seed,
and (3) reproductive output conditional on successful flowering and seed set. Each of
these processes may respond differently to environmental conditions or individual
state.</p>
<p>In practice, however, demographic data are extremely time- and labor-intensive to
collect, and most studies are limited to the transitions described above. As a result,
population models typically combine multiple biological processes into a smaller set
of composite vital rates.</p>
<p>That said, additional model complexity can be justified when driven by specific
hypotheses. For example, if a study focuses on pollination limitation or herbivore
effects on flowering, explicitly modeling intermediate reproductive steps may be
necessary and informative. The key is to match model complexity to both the available
data and the biological questions being asked.</p>
</div>
</div>
</div>
<div id="constructing-a-transition-matrix" class="section level2 hasAnchor" number="82.5">
<h2><span class="header-section-number">82.5</span> Constructing a transition matrix<a href="integral-population-models.html#constructing-a-transition-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To build a population projection model, we must translate our fitted vital-rate
models into a structure that projects the population forward in time. In matrix
population models, this structure is a <strong>transition matrix</strong>, where each element
describes the contribution of individuals in one state to another state in the next
time step.</p>
<p>In stage-structured models, the transition matrix is constructed directly from
estimated transition probabilities among discrete stages. In contrast, IPMs begin
with continuous relationships between size and vital rates. To construct a transition
matrix from these continuous functions, we discretize the size distribution into a
finite number of size classes. This discretized matrix serves as a numerical
approximation of the continuous IPM kernel.</p>
<p>Thus, even though IPMs are conceptually continuous, they are implemented computationally
as large matrices that closely approximate continuous dynamics.</p>
<p>In an IPM, population dynamics are governed by a continuous projection equation:
<span class="math display">\[
n(z&#39;, t+1) = \int K(z&#39;, z)\, n(z, t)\, dz
\]</span></p>
<p>where:
- <span class="math inline">\(z\)</span> is individual size at time <span class="math inline">\(t\)</span>,
- <span class="math inline">\(z&#39;\)</span> is size at time <span class="math inline">\(t+1\)</span>,
- <span class="math inline">\(n(z,t)\)</span> is the size distribution at time <span class="math inline">\(t\)</span>,
- <span class="math inline">\(K(z&#39;,z)\)</span> is the <strong>kernel</strong> describing transitions from size <span class="math inline">\(z\)</span> to <span class="math inline">\(z&#39;\)</span>.</p>
<p>The kernel is typically decomposed into survival–growth (<span class="math inline">\(P\)</span>) and reproduction (<span class="math inline">\(F\)</span>)
components:</p>
<p><span class="math display">\[
K(z&#39;, z) = P(z&#39;, z) + F(z&#39;, z)
\]</span>
When implemented numerically, the continuous kernel is approximated by a large matrix
<span class="math inline">\(\mathbf{K}\)</span>, where each element represents transitions between discretized size
classes.</p>
</div>
<div id="building-an-ipm-boundary-points-mesh-points-and-step-size" class="section level2 hasAnchor" number="82.6">
<h2><span class="header-section-number">82.6</span> Building an IPM: Boundary points, mesh points, and step size<a href="integral-population-models.html#building-an-ipm-boundary-points-mesh-points-and-step-size" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Constructing an IPM requires defining the size domain over which the population is
modeled and how finely that domain is discretized. This involves three key choices:
boundary points, mesh points, and step size.</p>
<p><strong>Boundary points</strong> define the minimum and maximum sizes included in the model.
These limits should encompass all biologically plausible sizes observed in the data,
often with a small buffer to avoid edge effects.</p>
<p><strong>Mesh points</strong> are the discrete size values used to approximate the continuous size
distribution. The number of mesh points determines the resolution of the IPM: more
mesh points yield a closer approximation to the continuous kernel but increase
computational cost.</p>
<p><strong>Step size</strong> is the distance between adjacent mesh points and is determined by the
boundary range divided by the number of mesh points. The step size plays a critical
role in numerical accuracy, as it scales the contribution of individuals across size
classes.</p>
<p>Together, these choices determine how well the discretized matrix approximates the
underlying continuous population dynamics. In practice, mesh sizes of 50–200 points
often provide a good balance between accuracy and computational efficiency, though
convergence should always be checked.</p>
</div>
<div id="model-evaluation-eviction" class="section level2 hasAnchor" number="82.7">
<h2><span class="header-section-number">82.7</span> Model Evaluation: Eviction<a href="integral-population-models.html#model-evaluation-eviction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>One challenge in Integral Projection Models (IPMs) arises when predicted growth pushes
individuals outside the defined size boundaries of the model. For example, a growth
model may predict that some individuals shrink below the minimum size or grow beyond
the maximum size. This issue is referred to as <strong>eviction</strong>.</p>
<p>If not handled properly, eviction can lead to loss or artificial accumulation of
individuals at the boundaries, distorting population dynamics and biasing estimates
of population growth rate (<span class="math inline">\(\lambda\)</span>).</p>
<div id="why-eviction-matters" class="section level3 hasAnchor" number="82.7.1">
<h3><span class="header-section-number">82.7.1</span> Why Eviction Matters<a href="integral-population-models.html#why-eviction-matters" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Consider a large plant with size <span class="math inline">\(z = 150 \,\text{cm}^2\)</span> and a growth model that
predicts next year’s size as:</p>
<ul>
<li>mean: <span class="math inline">\(\mu(z) = 155 \,\text{cm}^2\)</span></li>
<li>standard deviation: <span class="math inline">\(\sigma = 10 \,\text{cm}^2\)</span></li>
</ul>
<p>If the maximum size boundary is <span class="math inline">\(U = 176 \,\text{cm}^2\)</span>, some probability mass from
this individual’s growth distribution extends beyond the modeled size range. Without
correction, this probability mass is effectively lost, implicitly assuming those
individuals die. This assumption is often biologically unrealistic.</p>
</div>
<div id="eviction-correction-strategies" class="section level3 hasAnchor" number="82.7.2">
<h3><span class="header-section-number">82.7.2</span> Eviction Correction Strategies<a href="integral-population-models.html#eviction-correction-strategies" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Several strategies exist to address eviction, each corresponding to different
biological assumptions.</p>
<div id="truncation-renormalization" class="section level4 hasAnchor" number="82.7.2.1">
<h4><span class="header-section-number">82.7.2.1</span> 1. Truncation (Renormalization)<a href="integral-population-models.html#truncation-renormalization" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Method:</strong></p>
<p>Redistribute probability mass that would fall outside the boundaries back into the
valid size range by renormalizing the growth distribution.</p>
<p><strong>Mathematical implementation:</strong></p>
<p><span class="math display">\[
g_{\text{truncated}}(z&#39;, z) =
\frac{g_{\text{uncorrected}}(z&#39;, z)}
{\int_L^U g_{\text{uncorrected}}(z&#39;, z)\, dz&#39;}
\]</span>
where the denominator is:</p>
<p><span class="math display">\[
\int_L^U g_{\text{uncorrected}}(z&#39;, z)\, dz&#39;
=
\Phi\left(\frac{U - \mu(z)}{\sigma}\right)
-
\Phi\left(\frac{L - \mu(z)}{\sigma}\right)
\]</span>
and <span class="math inline">\(\Phi(\cdot)\)</span> is the cumulative distribution function of the normal distribution.</p>
<p>Note two different kinds of truncation:</p>
<p><strong>Simple truncation</strong> divides by (p_upper - p_lower) for EVERY transition:</p>
<ul>
<li>Pro: Conceptually simpler</li>
<li>Con: Applies unnecessary operations when eviction risk is ~0%</li>
<li>Con: Can introduce small numerical artifacts across the entire matrix</li>
</ul>
<p><strong>Conditional truncation</strong> only corrects when eviction is meaningful (&gt;1e-10 probability):</p>
<ul>
<li>Pro: More numerically stable</li>
<li>Pro: Matches industry-standard implementations (ipmr)</li>
<li>Pro: Computationally more accurate</li>
<li>Pro: Only modifies transitions that actually need correction</li>
</ul>
<p><strong>Biological interpretation:</strong></p>
<p>Individuals do not actually leave the population; instead, extreme sizes are “folded
back” into the observable range. This approach assumes that the boundaries reflect
measurement or detection limits rather than true biological limits.</p>
<p><strong>When appropriate:</strong></p>
<ul>
<li>Size represents natural constraints<br />
</li>
<li>Boundaries are conservative relative to the true biological range<br />
</li>
<li>Extreme sizes are rare in the data</li>
</ul>
<p>For plants demography, I typically use truncation!</p>
</div>
<div id="reflection" class="section level4 hasAnchor" number="82.7.2.2">
<h4><span class="header-section-number">82.7.2.2</span> 2. Reflection<a href="integral-population-models.html#reflection" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Method:</strong></p>
<p>Reflect probability mass back into the domain, creating symmetric probability density
at the boundaries.</p>
<p><strong>Mathematical implementation:</strong></p>
<p><span class="math display">\[
g_{\text{reflected}}(z&#39;, z) =
g(z&#39;, z) +
g(2L - z&#39;, z) +
g(2U - z&#39;, z)
\]</span></p>
<p><strong>Biological interpretation:</strong></p>
<p>Minimum and maximum sizes act as hard biological barriers, such as physical or
architectural constraints.</p>
<p><strong>When appropriate:</strong></p>
<ul>
<li>Clear biological maxima exist (e.g., determinate growth)<br />
</li>
<li>Minimum size represents a functional threshold</li>
</ul>
</div>
<div id="redistribution-to-boundary-classes" class="section level4 hasAnchor" number="82.7.2.3">
<h4><span class="header-section-number">82.7.2.3</span> 3. Redistribution to Boundary Classes<a href="integral-population-models.html#redistribution-to-boundary-classes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Method:</strong></p>
<p>Accumulate all evicted probability mass into the boundary size classes.</p>
<p><strong>Biological interpretation:</strong></p>
<p>Individuals reach a size asymptote and remain at that size indefinitely.</p>
<p><strong>When appropriate:</strong></p>
<ul>
<li>Species with determinate growth<br />
</li>
<li>Maximum size represents a stable life stage</li>
</ul>
</div>
<div id="no-correction-accept-loss" class="section level4 hasAnchor" number="82.7.2.4">
<h4><span class="header-section-number">82.7.2.4</span> 4. No Correction (Accept Loss)<a href="integral-population-models.html#no-correction-accept-loss" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Method:</strong></p>
<p>Allow probability mass to escape the modeled domain, treating eviction as mortality.</p>
<p><strong>Biological interpretation:</strong></p>
<p>Individuals that grow beyond <span class="math inline">\(U\)</span> or shrink below <span class="math inline">\(L\)</span> are assumed to die.</p>
<p><strong>When appropriate:</strong></p>
<ul>
<li>Rarely appropriate in practice<br />
</li>
<li>May be acceptable if eviction is negligible (e.g., &lt;1% of total probability mass)</li>
</ul>
</div>
</div>
<div id="summary-1" class="section level3 hasAnchor" number="82.7.3">
<h3><span class="header-section-number">82.7.3</span> Summary<a href="integral-population-models.html#summary-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Evaluating and correcting for eviction is a critical step in IPM construction.
Excessive eviction often signals issues with boundary selection, growth model
specification, or extrapolation beyond the data. The chosen correction method should
align with both the biology of the species and the interpretation of the size variable.</p>
</div>
</div>
<div id="estimating-population-dynamics-parameters" class="section level2 hasAnchor" number="82.8">
<h2><span class="header-section-number">82.8</span> Estimating population dynamics parameters<a href="integral-population-models.html#estimating-population-dynamics-parameters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Once the transition matrix has been constructed, we can calculate key population
dynamics parameters. The most important of these is the long-term population growth
rate, <span class="math inline">\(\lambda\)</span>, which describes whether the population is expected to grow, decline,
or remain stable over time.</p>
<p>Additional quantities of interest include the stable size distribution, which
describes the relative frequency of individuals across sizes at equilibrium, and the
reproductive value, which measures the relative contribution of individuals of
different sizes to future population growth.</p>
<p>Beyond asymptotic behavior, IPMs also allow us to quantify short-term or <strong>transient
dynamics</strong>, such as population amplification, attenuation, and inertia. These metrics
describe how populations respond to disturbances before reaching long-term equilibrium
and are often particularly relevant for conservation and management.</p>
</div>
<div id="calculating-confidence-intervals" class="section level2 hasAnchor" number="82.9">
<h2><span class="header-section-number">82.9</span> Calculating confidence intervals<a href="integral-population-models.html#calculating-confidence-intervals" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Demographic models are built from estimated vital-rate relationships, each of which
contains uncertainty. To understand how this uncertainty propagates into population
dynamics, we must quantify uncertainty around derived parameters such as <span class="math inline">\(\lambda\)</span>.</p>
<p>A common approach is to use resampling methods, such as bootstrapping, in which vital
rate models are repeatedly refit to resampled data. Each refitted model is used to
reconstruct the IPM and calculate population metrics, generating empirical
distributions for parameters of interest.</p>
<p>Confidence intervals derived from these distributions provide insight into the
robustness of model predictions and help distinguish meaningful biological patterns
from sampling variability. Incorporating uncertainty is particularly important when
IPMs are used to inform conservation decisions or compare population viability across
sites or treatments.</p>
</div>
<div id="putting-this-all-together-in-code-form" class="section level2 hasAnchor" number="82.10">
<h2><span class="header-section-number">82.10</span> Putting this all together in code form!<a href="integral-population-models.html#putting-this-all-together-in-code-form" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For simplicity, let’s go ahead and code this using the ipmr program! Note: From this point forward, the IPM is constructed using log(size) as the state variable. All kernels, mesh points, eviction correction, and demographic metrics are defined on the log scale unless explicitly stated otherwise. Raw units (cm) are used only for interpretation and visualization.</p>
<div class="sourceCode" id="cb2268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2268-1"><a href="integral-population-models.html#cb2268-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2268-2"><a href="integral-population-models.html#cb2268-2" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb2268-3"><a href="integral-population-models.html#cb2268-3" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb2268-4"><a href="integral-population-models.html#cb2268-4" tabindex="-1"></a><span class="fu">library</span>(ipmr)</span>
<span id="cb2268-5"><a href="integral-population-models.html#cb2268-5" tabindex="-1"></a></span>
<span id="cb2268-6"><a href="integral-population-models.html#cb2268-6" tabindex="-1"></a>species_name <span class="ot">&lt;-</span> <span class="st">&quot;SILLAN&quot;</span></span>
<span id="cb2268-7"><a href="integral-population-models.html#cb2268-7" tabindex="-1"></a>trans_years  <span class="ot">&lt;-</span> <span class="st">&quot;20232024&quot;</span></span>
<span id="cb2268-8"><a href="integral-population-models.html#cb2268-8" tabindex="-1"></a></span>
<span id="cb2268-9"><a href="integral-population-models.html#cb2268-9" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2268-10"><a href="integral-population-models.html#cb2268-10" tabindex="-1"></a><span class="co"># 0) Transform + keep raw and log</span></span>
<span id="cb2268-11"><a href="integral-population-models.html#cb2268-11" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2268-12"><a href="integral-population-models.html#cb2268-12" tabindex="-1"></a>final_data <span class="ot">&lt;-</span> final_data <span class="sc">%&gt;%</span></span>
<span id="cb2268-13"><a href="integral-population-models.html#cb2268-13" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2268-14"><a href="integral-population-models.html#cb2268-14" tabindex="-1"></a>    <span class="at">size_raw     =</span> size,</span>
<span id="cb2268-15"><a href="integral-population-models.html#cb2268-15" tabindex="-1"></a>    <span class="at">sizeNext_raw =</span> sizeNext,</span>
<span id="cb2268-16"><a href="integral-population-models.html#cb2268-16" tabindex="-1"></a>    <span class="at">size_log     =</span> <span class="fu">if_else</span>(<span class="sc">!</span><span class="fu">is.na</span>(size)     <span class="sc">&amp;</span> <span class="fu">is.finite</span>(size)     <span class="sc">&amp;</span> size     <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="fu">log</span>(size),     <span class="cn">NA_real_</span>),</span>
<span id="cb2268-17"><a href="integral-population-models.html#cb2268-17" tabindex="-1"></a>    <span class="at">sizeNext_log =</span> <span class="fu">if_else</span>(<span class="sc">!</span><span class="fu">is.na</span>(sizeNext) <span class="sc">&amp;</span> <span class="fu">is.finite</span>(sizeNext) <span class="sc">&amp;</span> sizeNext <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="fu">log</span>(sizeNext), <span class="cn">NA_real_</span>)</span>
<span id="cb2268-18"><a href="integral-population-models.html#cb2268-18" tabindex="-1"></a>  )</span>
<span id="cb2268-19"><a href="integral-population-models.html#cb2268-19" tabindex="-1"></a></span>
<span id="cb2268-20"><a href="integral-population-models.html#cb2268-20" tabindex="-1"></a><span class="co"># If you are using Exclude, make it explicit here (optional)</span></span>
<span id="cb2268-21"><a href="integral-population-models.html#cb2268-21" tabindex="-1"></a><span class="cf">if</span> (<span class="st">&quot;Exclude&quot;</span> <span class="sc">%in%</span> <span class="fu">names</span>(final_data)) {</span>
<span id="cb2268-22"><a href="integral-population-models.html#cb2268-22" tabindex="-1"></a>  final_data <span class="ot">&lt;-</span> final_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Exclude <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb2268-23"><a href="integral-population-models.html#cb2268-23" tabindex="-1"></a>}</span>
<span id="cb2268-24"><a href="integral-population-models.html#cb2268-24" tabindex="-1"></a></span>
<span id="cb2268-25"><a href="integral-population-models.html#cb2268-25" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2268-26"><a href="integral-population-models.html#cb2268-26" tabindex="-1"></a><span class="co"># 1) Build model-specific datasets (THIS fixes NA/AIC issues)</span></span>
<span id="cb2268-27"><a href="integral-population-models.html#cb2268-27" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2268-28"><a href="integral-population-models.html#cb2268-28" tabindex="-1"></a></span>
<span id="cb2268-29"><a href="integral-population-models.html#cb2268-29" tabindex="-1"></a><span class="co"># Growth: needs size_log + sizeNext_log</span></span>
<span id="cb2268-30"><a href="integral-population-models.html#cb2268-30" tabindex="-1"></a>grow_data <span class="ot">&lt;-</span> final_data <span class="sc">%&gt;%</span></span>
<span id="cb2268-31"><a href="integral-population-models.html#cb2268-31" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(size_log), <span class="sc">!</span><span class="fu">is.na</span>(sizeNext_log))</span>
<span id="cb2268-32"><a href="integral-population-models.html#cb2268-32" tabindex="-1"></a></span>
<span id="cb2268-33"><a href="integral-population-models.html#cb2268-33" tabindex="-1"></a><span class="co"># Survival: needs size_log + surv</span></span>
<span id="cb2268-34"><a href="integral-population-models.html#cb2268-34" tabindex="-1"></a>surv_data <span class="ot">&lt;-</span> final_data <span class="sc">%&gt;%</span></span>
<span id="cb2268-35"><a href="integral-population-models.html#cb2268-35" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(size_log), <span class="sc">!</span><span class="fu">is.na</span>(surv)) <span class="sc">%&gt;%</span></span>
<span id="cb2268-36"><a href="integral-population-models.html#cb2268-36" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">surv =</span> <span class="fu">as.integer</span>(surv))</span>
<span id="cb2268-37"><a href="integral-population-models.html#cb2268-37" tabindex="-1"></a></span>
<span id="cb2268-38"><a href="integral-population-models.html#cb2268-38" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(surv_data<span class="sc">$</span>surv <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)))</span>
<span id="cb2268-39"><a href="integral-population-models.html#cb2268-39" tabindex="-1"></a></span>
<span id="cb2268-40"><a href="integral-population-models.html#cb2268-40" tabindex="-1"></a><span class="co"># Reproduction: needs size_log + fec1 (0/1)</span></span>
<span id="cb2268-41"><a href="integral-population-models.html#cb2268-41" tabindex="-1"></a>repr_data <span class="ot">&lt;-</span> final_data <span class="sc">%&gt;%</span></span>
<span id="cb2268-42"><a href="integral-population-models.html#cb2268-42" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(size_log), <span class="sc">!</span><span class="fu">is.na</span>(fec1)) <span class="sc">%&gt;%</span></span>
<span id="cb2268-43"><a href="integral-population-models.html#cb2268-43" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fec1 =</span> <span class="fu">as.integer</span>(fec1))</span>
<span id="cb2268-44"><a href="integral-population-models.html#cb2268-44" tabindex="-1"></a></span>
<span id="cb2268-45"><a href="integral-population-models.html#cb2268-45" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(repr_data<span class="sc">$</span>fec1 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)))</span>
<span id="cb2268-46"><a href="integral-population-models.html#cb2268-46" tabindex="-1"></a></span>
<span id="cb2268-47"><a href="integral-population-models.html#cb2268-47" tabindex="-1"></a><span class="co"># Fecundity: CONDITIONAL on reproducing (fec1 == 1)</span></span>
<span id="cb2268-48"><a href="integral-population-models.html#cb2268-48" tabindex="-1"></a>fec_data <span class="ot">&lt;-</span> final_data <span class="sc">%&gt;%</span></span>
<span id="cb2268-49"><a href="integral-population-models.html#cb2268-49" tabindex="-1"></a>  <span class="fu">filter</span>(fec1 <span class="sc">==</span> <span class="dv">1</span>, <span class="sc">!</span><span class="fu">is.na</span>(size_log), <span class="sc">!</span><span class="fu">is.na</span>(fec2))</span>
<span id="cb2268-50"><a href="integral-population-models.html#cb2268-50" tabindex="-1"></a></span>
<span id="cb2268-51"><a href="integral-population-models.html#cb2268-51" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2268-52"><a href="integral-population-models.html#cb2268-52" tabindex="-1"></a><span class="co"># 2) Fit vital rate models on consistent, clean data</span></span>
<span id="cb2268-53"><a href="integral-population-models.html#cb2268-53" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2268-54"><a href="integral-population-models.html#cb2268-54" tabindex="-1"></a></span>
<span id="cb2268-55"><a href="integral-population-models.html#cb2268-55" tabindex="-1"></a>grow_mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(sizeNext_log <span class="sc">~</span> size_log <span class="sc">+</span> <span class="fu">I</span>(size_log<span class="sc">^</span><span class="dv">2</span>), <span class="at">data =</span> grow_data)</span>
<span id="cb2268-56"><a href="integral-population-models.html#cb2268-56" tabindex="-1"></a>grow_sd  <span class="ot">&lt;-</span> <span class="fu">sd</span>(<span class="fu">resid</span>(grow_mod))</span>
<span id="cb2268-57"><a href="integral-population-models.html#cb2268-57" tabindex="-1"></a></span>
<span id="cb2268-58"><a href="integral-population-models.html#cb2268-58" tabindex="-1"></a>surv_mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(surv <span class="sc">~</span> size_log, <span class="at">data =</span> surv_data, <span class="at">family =</span> <span class="fu">binomial</span>())</span>
<span id="cb2268-59"><a href="integral-population-models.html#cb2268-59" tabindex="-1"></a></span>
<span id="cb2268-60"><a href="integral-population-models.html#cb2268-60" tabindex="-1"></a>repr_mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(fec1 <span class="sc">~</span> size_log <span class="sc">+</span> <span class="fu">I</span>(size_log<span class="sc">^</span><span class="dv">2</span>), <span class="at">data =</span> repr_data, <span class="at">family =</span> <span class="fu">binomial</span>())</span>
<span id="cb2268-61"><a href="integral-population-models.html#cb2268-61" tabindex="-1"></a></span>
<span id="cb2268-62"><a href="integral-population-models.html#cb2268-62" tabindex="-1"></a><span class="co"># fecundity model: expected fec2 GIVEN reproduction</span></span>
<span id="cb2268-63"><a href="integral-population-models.html#cb2268-63" tabindex="-1"></a>seed_mod <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">glm.nb</span>(fec2 <span class="sc">~</span> size_log <span class="sc">+</span> <span class="fu">I</span>(size_log<span class="sc">^</span><span class="dv">2</span>), <span class="at">data =</span> fec_data)</span>
<span id="cb2268-64"><a href="integral-population-models.html#cb2268-64" tabindex="-1"></a><span class="co"># (If you really want Poisson, swap to glm(..., family = poisson()), but NB is safer.)</span></span>
<span id="cb2268-65"><a href="integral-population-models.html#cb2268-65" tabindex="-1"></a></span>
<span id="cb2268-66"><a href="integral-population-models.html#cb2268-66" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2268-67"><a href="integral-population-models.html#cb2268-67" tabindex="-1"></a><span class="co"># 3) Recruitment (strict recruits) + establishment prob</span></span>
<span id="cb2268-68"><a href="integral-population-models.html#cb2268-68" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2268-69"><a href="integral-population-models.html#cb2268-69" tabindex="-1"></a>robust_recruit_logical <span class="ot">&lt;-</span> <span class="fu">is.na</span>(final_data<span class="sc">$</span>size_log) <span class="sc">&amp;</span></span>
<span id="cb2268-70"><a href="integral-population-models.html#cb2268-70" tabindex="-1"></a>  <span class="fu">is.na</span>(final_data<span class="sc">$</span>surv) <span class="sc">&amp;</span></span>
<span id="cb2268-71"><a href="integral-population-models.html#cb2268-71" tabindex="-1"></a>  <span class="fu">is.na</span>(final_data<span class="sc">$</span>fec1) <span class="sc">&amp;</span></span>
<span id="cb2268-72"><a href="integral-population-models.html#cb2268-72" tabindex="-1"></a>  <span class="sc">!</span><span class="fu">is.na</span>(final_data<span class="sc">$</span>sizeNext_log)</span>
<span id="cb2268-73"><a href="integral-population-models.html#cb2268-73" tabindex="-1"></a></span>
<span id="cb2268-74"><a href="integral-population-models.html#cb2268-74" tabindex="-1"></a>recr_data <span class="ot">&lt;-</span> final_data[robust_recruit_logical, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb2268-75"><a href="integral-population-models.html#cb2268-75" tabindex="-1"></a></span>
<span id="cb2268-76"><a href="integral-population-models.html#cb2268-76" tabindex="-1"></a>recr_mu <span class="ot">&lt;-</span> <span class="fu">mean</span>(recr_data<span class="sc">$</span>sizeNext_log, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2268-77"><a href="integral-population-models.html#cb2268-77" tabindex="-1"></a>recr_sd <span class="ot">&lt;-</span> <span class="fu">sd</span>(recr_data<span class="sc">$</span>sizeNext_log, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2268-78"><a href="integral-population-models.html#cb2268-78" tabindex="-1"></a>recr_n  <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.finite</span>(recr_data<span class="sc">$</span>sizeNext_log))</span>
<span id="cb2268-79"><a href="integral-population-models.html#cb2268-79" tabindex="-1"></a></span>
<span id="cb2268-80"><a href="integral-population-models.html#cb2268-80" tabindex="-1"></a><span class="co"># seeds produced by repro individuals (matches seed_mod being conditional)</span></span>
<span id="cb2268-81"><a href="integral-population-models.html#cb2268-81" tabindex="-1"></a>seed_n  <span class="ot">&lt;-</span> <span class="fu">sum</span>(fec_data<span class="sc">$</span>fec2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2268-82"><a href="integral-population-models.html#cb2268-82" tabindex="-1"></a></span>
<span id="cb2268-83"><a href="integral-population-models.html#cb2268-83" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.finite</span>(seed_n) <span class="sc">||</span> seed_n <span class="sc">&lt;=</span> <span class="dv">0</span>) <span class="fu">stop</span>(<span class="st">&quot;seed_n is 0 or non-finite; cannot build fecundity kernel&quot;</span>)</span>
<span id="cb2268-84"><a href="integral-population-models.html#cb2268-84" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.finite</span>(recr_n) <span class="sc">||</span> recr_n <span class="sc">&lt;=</span> <span class="dv">0</span>) <span class="fu">stop</span>(<span class="st">&quot;recr_n is 0 or non-finite; cannot build recruitment kernel&quot;</span>)</span>
<span id="cb2268-85"><a href="integral-population-models.html#cb2268-85" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.finite</span>(recr_mu) <span class="sc">||</span> <span class="sc">!</span><span class="fu">is.finite</span>(recr_sd) <span class="sc">||</span> recr_sd <span class="sc">&lt;=</span> <span class="dv">0</span>) <span class="fu">stop</span>(<span class="st">&quot;Recruit size distribution invalid (mu/sd)&quot;</span>)</span>
<span id="cb2268-86"><a href="integral-population-models.html#cb2268-86" tabindex="-1"></a></span>
<span id="cb2268-87"><a href="integral-population-models.html#cb2268-87" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== Recruitment Validation ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === Recruitment Validation ===</code></pre>
<div class="sourceCode" id="cb2270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2270-1"><a href="integral-population-models.html#cb2270-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Recruits:&quot;</span>, recr_n, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Recruits: 293</code></pre>
<div class="sourceCode" id="cb2272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2272-1"><a href="integral-population-models.html#cb2272-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Total seeds produced (fec1==1 only):&quot;</span>, seed_n, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Total seeds produced (fec1==1 only): 5317830</code></pre>
<div class="sourceCode" id="cb2274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2274-1"><a href="integral-population-models.html#cb2274-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Establishment probability:&quot;</span>, <span class="fu">round</span>(recr_n <span class="sc">/</span> seed_n, <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Establishment probability: 1e-04</code></pre>
<div class="sourceCode" id="cb2276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2276-1"><a href="integral-population-models.html#cb2276-1" tabindex="-1"></a><span class="co"># Minimum repro size on log scale (guarded)</span></span>
<span id="cb2276-2"><a href="integral-population-models.html#cb2276-2" tabindex="-1"></a>min_repro_size_log <span class="ot">&lt;-</span> <span class="fu">suppressWarnings</span>(<span class="fu">min</span>(repr_data<span class="sc">$</span>size_log[repr_data<span class="sc">$</span>fec1 <span class="sc">==</span> <span class="dv">1</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb2276-3"><a href="integral-population-models.html#cb2276-3" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.finite</span>(min_repro_size_log)) {</span>
<span id="cb2276-4"><a href="integral-population-models.html#cb2276-4" tabindex="-1"></a>  min_repro_size_log <span class="ot">&lt;-</span> <span class="cn">Inf</span></span>
<span id="cb2276-5"><a href="integral-population-models.html#cb2276-5" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;⚠️ No reproductive individuals (fec1==1). Setting min_repro_size_log = Inf (F kernel will be ~0).&quot;</span>)</span>
<span id="cb2276-6"><a href="integral-population-models.html#cb2276-6" tabindex="-1"></a>}</span>
<span id="cb2276-7"><a href="integral-population-models.html#cb2276-7" tabindex="-1"></a></span>
<span id="cb2276-8"><a href="integral-population-models.html#cb2276-8" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2276-9"><a href="integral-population-models.html#cb2276-9" tabindex="-1"></a><span class="co"># 4) Domain on LOG scale (pad by range, not multiply)</span></span>
<span id="cb2276-10"><a href="integral-population-models.html#cb2276-10" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2276-11"><a href="integral-population-models.html#cb2276-11" tabindex="-1"></a>all_sizes_log <span class="ot">&lt;-</span> <span class="fu">c</span>(final_data<span class="sc">$</span>size_log, final_data<span class="sc">$</span>sizeNext_log)</span>
<span id="cb2276-12"><a href="integral-population-models.html#cb2276-12" tabindex="-1"></a>q <span class="ot">&lt;-</span> <span class="fu">quantile</span>(all_sizes_log, <span class="at">probs =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.99</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2276-13"><a href="integral-population-models.html#cb2276-13" tabindex="-1"></a></span>
<span id="cb2276-14"><a href="integral-population-models.html#cb2276-14" tabindex="-1"></a>rng <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(q[<span class="dv">2</span>] <span class="sc">-</span> q[<span class="dv">1</span>])</span>
<span id="cb2276-15"><a href="integral-population-models.html#cb2276-15" tabindex="-1"></a>pad <span class="ot">&lt;-</span> <span class="fl">0.05</span> <span class="sc">*</span> rng</span>
<span id="cb2276-16"><a href="integral-population-models.html#cb2276-16" tabindex="-1"></a></span>
<span id="cb2276-17"><a href="integral-population-models.html#cb2276-17" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(q[<span class="dv">1</span>] <span class="sc">-</span> pad)</span>
<span id="cb2276-18"><a href="integral-population-models.html#cb2276-18" tabindex="-1"></a>U <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(q[<span class="dv">2</span>] <span class="sc">+</span> pad)</span>
<span id="cb2276-19"><a href="integral-population-models.html#cb2276-19" tabindex="-1"></a></span>
<span id="cb2276-20"><a href="integral-population-models.html#cb2276-20" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.finite</span>(L) <span class="sc">||</span> <span class="sc">!</span><span class="fu">is.finite</span>(U) <span class="sc">||</span> L <span class="sc">&gt;=</span> U) <span class="fu">stop</span>(<span class="st">&quot;Invalid domain bounds: check sizes and quantiles&quot;</span>)</span>
<span id="cb2276-21"><a href="integral-population-models.html#cb2276-21" tabindex="-1"></a></span>
<span id="cb2276-22"><a href="integral-population-models.html#cb2276-22" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== Domain Boundaries (LOG SCALE) ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === Domain Boundaries (LOG SCALE) ===</code></pre>
<div class="sourceCode" id="cb2278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2278-1"><a href="integral-population-models.html#cb2278-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Log scale: [&quot;</span>, <span class="fu">round</span>(L, <span class="dv">3</span>), <span class="st">&quot;,&quot;</span>, <span class="fu">round</span>(U, <span class="dv">3</span>), <span class="st">&quot;]</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Log scale: [ -0.248 , 5.204 ]</code></pre>
<div class="sourceCode" id="cb2280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2280-1"><a href="integral-population-models.html#cb2280-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Raw scale: [&quot;</span>, <span class="fu">round</span>(<span class="fu">exp</span>(L), <span class="dv">3</span>), <span class="st">&quot;,&quot;</span>, <span class="fu">round</span>(<span class="fu">exp</span>(U), <span class="dv">3</span>), <span class="st">&quot;]</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Raw scale: [ 0.781 , 181.972 ]</code></pre>
<div class="sourceCode" id="cb2282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2282-1"><a href="integral-population-models.html#cb2282-1" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2282-2"><a href="integral-population-models.html#cb2282-2" tabindex="-1"></a><span class="co"># 5) Pack params for ipmr kernels</span></span>
<span id="cb2282-3"><a href="integral-population-models.html#cb2282-3" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2282-4"><a href="integral-population-models.html#cb2282-4" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb2282-5"><a href="integral-population-models.html#cb2282-5" tabindex="-1"></a>  <span class="at">recr_mu            =</span> recr_mu,</span>
<span id="cb2282-6"><a href="integral-population-models.html#cb2282-6" tabindex="-1"></a>  <span class="at">recr_sd            =</span> recr_sd,</span>
<span id="cb2282-7"><a href="integral-population-models.html#cb2282-7" tabindex="-1"></a>  <span class="at">grow_sd            =</span> grow_sd,</span>
<span id="cb2282-8"><a href="integral-population-models.html#cb2282-8" tabindex="-1"></a>  <span class="at">surv_mod           =</span> surv_mod,</span>
<span id="cb2282-9"><a href="integral-population-models.html#cb2282-9" tabindex="-1"></a>  <span class="at">grow_mod           =</span> grow_mod,</span>
<span id="cb2282-10"><a href="integral-population-models.html#cb2282-10" tabindex="-1"></a>  <span class="at">repr_mod           =</span> repr_mod,</span>
<span id="cb2282-11"><a href="integral-population-models.html#cb2282-11" tabindex="-1"></a>  <span class="at">seed_mod           =</span> seed_mod,</span>
<span id="cb2282-12"><a href="integral-population-models.html#cb2282-12" tabindex="-1"></a>  <span class="at">recr_n             =</span> recr_n,</span>
<span id="cb2282-13"><a href="integral-population-models.html#cb2282-13" tabindex="-1"></a>  <span class="at">seed_n             =</span> seed_n,</span>
<span id="cb2282-14"><a href="integral-population-models.html#cb2282-14" tabindex="-1"></a>  <span class="at">min_repro_size_log =</span> min_repro_size_log</span>
<span id="cb2282-15"><a href="integral-population-models.html#cb2282-15" tabindex="-1"></a>)</span>
<span id="cb2282-16"><a href="integral-population-models.html#cb2282-16" tabindex="-1"></a></span>
<span id="cb2282-17"><a href="integral-population-models.html#cb2282-17" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2282-18"><a href="integral-population-models.html#cb2282-18" tabindex="-1"></a><span class="co"># 6) Build IPM (log state variable everywhere)</span></span>
<span id="cb2282-19"><a href="integral-population-models.html#cb2282-19" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2282-20"><a href="integral-population-models.html#cb2282-20" tabindex="-1"></a>obs_ipm <span class="ot">&lt;-</span> <span class="fu">init_ipm</span>(</span>
<span id="cb2282-21"><a href="integral-population-models.html#cb2282-21" tabindex="-1"></a>  <span class="at">sim_gen   =</span> <span class="st">&quot;simple&quot;</span>,</span>
<span id="cb2282-22"><a href="integral-population-models.html#cb2282-22" tabindex="-1"></a>  <span class="at">di_dd     =</span> <span class="st">&quot;di&quot;</span>,</span>
<span id="cb2282-23"><a href="integral-population-models.html#cb2282-23" tabindex="-1"></a>  <span class="at">det_stoch =</span> <span class="st">&quot;det&quot;</span></span>
<span id="cb2282-24"><a href="integral-population-models.html#cb2282-24" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb2282-25"><a href="integral-population-models.html#cb2282-25" tabindex="-1"></a></span>
<span id="cb2282-26"><a href="integral-population-models.html#cb2282-26" tabindex="-1"></a>  <span class="fu">define_kernel</span>(</span>
<span id="cb2282-27"><a href="integral-population-models.html#cb2282-27" tabindex="-1"></a>    <span class="at">name    =</span> <span class="st">&quot;P&quot;</span>,</span>
<span id="cb2282-28"><a href="integral-population-models.html#cb2282-28" tabindex="-1"></a>    <span class="at">family  =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb2282-29"><a href="integral-population-models.html#cb2282-29" tabindex="-1"></a>    <span class="at">formula =</span> s <span class="sc">*</span> g,</span>
<span id="cb2282-30"><a href="integral-population-models.html#cb2282-30" tabindex="-1"></a></span>
<span id="cb2282-31"><a href="integral-population-models.html#cb2282-31" tabindex="-1"></a>    <span class="at">s =</span> <span class="fu">predict</span>(surv_mod, <span class="at">newdata =</span> <span class="fu">data.frame</span>(<span class="at">size_log =</span> sa_1), <span class="at">type =</span> <span class="st">&quot;response&quot;</span>),</span>
<span id="cb2282-32"><a href="integral-population-models.html#cb2282-32" tabindex="-1"></a></span>
<span id="cb2282-33"><a href="integral-population-models.html#cb2282-33" tabindex="-1"></a>    <span class="at">g_mu =</span> <span class="fu">predict</span>(grow_mod, <span class="at">newdata =</span> <span class="fu">data.frame</span>(<span class="at">size_log =</span> sa_1), <span class="at">type =</span> <span class="st">&quot;response&quot;</span>),</span>
<span id="cb2282-34"><a href="integral-population-models.html#cb2282-34" tabindex="-1"></a>    <span class="at">g    =</span> <span class="fu">dnorm</span>(sa_2, g_mu, grow_sd),</span>
<span id="cb2282-35"><a href="integral-population-models.html#cb2282-35" tabindex="-1"></a></span>
<span id="cb2282-36"><a href="integral-population-models.html#cb2282-36" tabindex="-1"></a>    <span class="at">states        =</span> <span class="fu">list</span>(<span class="st">&quot;sa&quot;</span>),</span>
<span id="cb2282-37"><a href="integral-population-models.html#cb2282-37" tabindex="-1"></a>    <span class="at">data_list     =</span> params,</span>
<span id="cb2282-38"><a href="integral-population-models.html#cb2282-38" tabindex="-1"></a>    <span class="at">uses_par_sets =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2282-39"><a href="integral-population-models.html#cb2282-39" tabindex="-1"></a></span>
<span id="cb2282-40"><a href="integral-population-models.html#cb2282-40" tabindex="-1"></a>    <span class="at">evict_cor =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2282-41"><a href="integral-population-models.html#cb2282-41" tabindex="-1"></a>    <span class="at">evict_fun =</span> <span class="fu">truncated_distributions</span>(<span class="at">fun =</span> <span class="st">&quot;norm&quot;</span>, <span class="at">target =</span> <span class="st">&quot;g&quot;</span>)</span>
<span id="cb2282-42"><a href="integral-population-models.html#cb2282-42" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb2282-43"><a href="integral-population-models.html#cb2282-43" tabindex="-1"></a></span>
<span id="cb2282-44"><a href="integral-population-models.html#cb2282-44" tabindex="-1"></a>  <span class="fu">define_kernel</span>(</span>
<span id="cb2282-45"><a href="integral-population-models.html#cb2282-45" tabindex="-1"></a>    <span class="at">name    =</span> <span class="st">&quot;F&quot;</span>,</span>
<span id="cb2282-46"><a href="integral-population-models.html#cb2282-46" tabindex="-1"></a>    <span class="at">family  =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb2282-47"><a href="integral-population-models.html#cb2282-47" tabindex="-1"></a>    <span class="at">formula =</span> r_p <span class="sc">*</span> r_s <span class="sc">*</span> r_r <span class="sc">*</span> r_d,</span>
<span id="cb2282-48"><a href="integral-population-models.html#cb2282-48" tabindex="-1"></a></span>
<span id="cb2282-49"><a href="integral-population-models.html#cb2282-49" tabindex="-1"></a>    <span class="at">r_p =</span> <span class="fu">ifelse</span>(</span>
<span id="cb2282-50"><a href="integral-population-models.html#cb2282-50" tabindex="-1"></a>      sa_1 <span class="sc">&lt;</span> min_repro_size_log,</span>
<span id="cb2282-51"><a href="integral-population-models.html#cb2282-51" tabindex="-1"></a>      <span class="dv">0</span>,</span>
<span id="cb2282-52"><a href="integral-population-models.html#cb2282-52" tabindex="-1"></a>      <span class="fu">predict</span>(repr_mod, <span class="at">newdata =</span> <span class="fu">data.frame</span>(<span class="at">size_log =</span> sa_1), <span class="at">type =</span> <span class="st">&quot;response&quot;</span>)</span>
<span id="cb2282-53"><a href="integral-population-models.html#cb2282-53" tabindex="-1"></a>    ),</span>
<span id="cb2282-54"><a href="integral-population-models.html#cb2282-54" tabindex="-1"></a></span>
<span id="cb2282-55"><a href="integral-population-models.html#cb2282-55" tabindex="-1"></a>    <span class="co"># expected fec2 GIVEN reproduction</span></span>
<span id="cb2282-56"><a href="integral-population-models.html#cb2282-56" tabindex="-1"></a>    <span class="at">r_s =</span> <span class="fu">predict</span>(seed_mod, <span class="at">newdata =</span> <span class="fu">data.frame</span>(<span class="at">size_log =</span> sa_1), <span class="at">type =</span> <span class="st">&quot;response&quot;</span>),</span>
<span id="cb2282-57"><a href="integral-population-models.html#cb2282-57" tabindex="-1"></a></span>
<span id="cb2282-58"><a href="integral-population-models.html#cb2282-58" tabindex="-1"></a>    <span class="at">r_r =</span> recr_n <span class="sc">/</span> seed_n,</span>
<span id="cb2282-59"><a href="integral-population-models.html#cb2282-59" tabindex="-1"></a></span>
<span id="cb2282-60"><a href="integral-population-models.html#cb2282-60" tabindex="-1"></a>    <span class="at">r_d =</span> <span class="fu">dnorm</span>(sa_2, recr_mu, recr_sd),</span>
<span id="cb2282-61"><a href="integral-population-models.html#cb2282-61" tabindex="-1"></a></span>
<span id="cb2282-62"><a href="integral-population-models.html#cb2282-62" tabindex="-1"></a>    <span class="at">states        =</span> <span class="fu">list</span>(<span class="st">&quot;sa&quot;</span>),</span>
<span id="cb2282-63"><a href="integral-population-models.html#cb2282-63" tabindex="-1"></a>    <span class="at">data_list     =</span> params,</span>
<span id="cb2282-64"><a href="integral-population-models.html#cb2282-64" tabindex="-1"></a>    <span class="at">uses_par_sets =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2282-65"><a href="integral-population-models.html#cb2282-65" tabindex="-1"></a></span>
<span id="cb2282-66"><a href="integral-population-models.html#cb2282-66" tabindex="-1"></a>    <span class="at">evict_cor =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2282-67"><a href="integral-population-models.html#cb2282-67" tabindex="-1"></a>    <span class="at">evict_fun =</span> <span class="fu">truncated_distributions</span>(<span class="at">fun =</span> <span class="st">&quot;norm&quot;</span>, <span class="at">target =</span> <span class="st">&quot;r_d&quot;</span>)</span>
<span id="cb2282-68"><a href="integral-population-models.html#cb2282-68" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb2282-69"><a href="integral-population-models.html#cb2282-69" tabindex="-1"></a></span>
<span id="cb2282-70"><a href="integral-population-models.html#cb2282-70" tabindex="-1"></a>  <span class="fu">define_impl</span>(</span>
<span id="cb2282-71"><a href="integral-population-models.html#cb2282-71" tabindex="-1"></a>    <span class="fu">make_impl_args_list</span>(</span>
<span id="cb2282-72"><a href="integral-population-models.html#cb2282-72" tabindex="-1"></a>      <span class="at">kernel_names =</span> <span class="fu">c</span>(<span class="st">&quot;P&quot;</span>, <span class="st">&quot;F&quot;</span>),</span>
<span id="cb2282-73"><a href="integral-population-models.html#cb2282-73" tabindex="-1"></a>      <span class="at">int_rule     =</span> <span class="fu">rep</span>(<span class="st">&quot;midpoint&quot;</span>, <span class="dv">2</span>),</span>
<span id="cb2282-74"><a href="integral-population-models.html#cb2282-74" tabindex="-1"></a>      <span class="at">state_start  =</span> <span class="fu">rep</span>(<span class="st">&quot;sa&quot;</span>, <span class="dv">2</span>),</span>
<span id="cb2282-75"><a href="integral-population-models.html#cb2282-75" tabindex="-1"></a>      <span class="at">state_end    =</span> <span class="fu">rep</span>(<span class="st">&quot;sa&quot;</span>, <span class="dv">2</span>)</span>
<span id="cb2282-76"><a href="integral-population-models.html#cb2282-76" tabindex="-1"></a>    )</span>
<span id="cb2282-77"><a href="integral-population-models.html#cb2282-77" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb2282-78"><a href="integral-population-models.html#cb2282-78" tabindex="-1"></a></span>
<span id="cb2282-79"><a href="integral-population-models.html#cb2282-79" tabindex="-1"></a>  <span class="fu">define_domains</span>(<span class="at">sa =</span> <span class="fu">c</span>(L, U, <span class="dv">100</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2282-80"><a href="integral-population-models.html#cb2282-80" tabindex="-1"></a>  <span class="fu">define_pop_state</span>(<span class="at">n_sa =</span> <span class="fu">runif</span>(<span class="dv">100</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2282-81"><a href="integral-population-models.html#cb2282-81" tabindex="-1"></a></span>
<span id="cb2282-82"><a href="integral-population-models.html#cb2282-82" tabindex="-1"></a>  <span class="fu">make_ipm</span>(<span class="at">iterate =</span> <span class="cn">TRUE</span>, <span class="at">iterations =</span> <span class="dv">100</span>)</span>
<span id="cb2282-83"><a href="integral-population-models.html#cb2282-83" tabindex="-1"></a></span>
<span id="cb2282-84"><a href="integral-population-models.html#cb2282-84" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2282-85"><a href="integral-population-models.html#cb2282-85" tabindex="-1"></a><span class="co"># 7) Extract kernels + lambda</span></span>
<span id="cb2282-86"><a href="integral-population-models.html#cb2282-86" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2282-87"><a href="integral-population-models.html#cb2282-87" tabindex="-1"></a>P <span class="ot">&lt;-</span> obs_ipm<span class="sc">$</span>sub_kernels<span class="sc">$</span>P</span>
<span id="cb2282-88"><a href="integral-population-models.html#cb2282-88" tabindex="-1"></a>F <span class="ot">&lt;-</span> obs_ipm<span class="sc">$</span>sub_kernels<span class="sc">$</span>F</span>
<span id="cb2282-89"><a href="integral-population-models.html#cb2282-89" tabindex="-1"></a>K_matrix <span class="ot">&lt;-</span> P <span class="sc">+</span> F</span>
<span id="cb2282-90"><a href="integral-population-models.html#cb2282-90" tabindex="-1"></a></span>
<span id="cb2282-91"><a href="integral-population-models.html#cb2282-91" tabindex="-1"></a>eigs <span class="ot">&lt;-</span> <span class="fu">eigen</span>(K_matrix)</span>
<span id="cb2282-92"><a href="integral-population-models.html#cb2282-92" tabindex="-1"></a>lambda_raw <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">Re</span>(eigs<span class="sc">$</span>values))</span>
<span id="cb2282-93"><a href="integral-population-models.html#cb2282-93" tabindex="-1"></a></span>
<span id="cb2282-94"><a href="integral-population-models.html#cb2282-94" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== IPM Results ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === IPM Results ===</code></pre>
<div class="sourceCode" id="cb2284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2284-1"><a href="integral-population-models.html#cb2284-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Population growth rate (λ):&quot;</span>, <span class="fu">round</span>(lambda_raw, <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Population growth rate (λ): 0.9694</code></pre>
<div class="sourceCode" id="cb2286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2286-1"><a href="integral-population-models.html#cb2286-1" tabindex="-1"></a>stable_dist_raw <span class="ot">&lt;-</span> <span class="fu">Re</span>(eigs<span class="sc">$</span>vectors[, <span class="fu">which.max</span>(<span class="fu">Re</span>(eigs<span class="sc">$</span>values))])</span>
<span id="cb2286-2"><a href="integral-population-models.html#cb2286-2" tabindex="-1"></a>stable_dist_raw <span class="ot">&lt;-</span> stable_dist_raw <span class="sc">/</span> <span class="fu">sum</span>(stable_dist_raw)</span>
<span id="cb2286-3"><a href="integral-population-models.html#cb2286-3" tabindex="-1"></a></span>
<span id="cb2286-4"><a href="integral-population-models.html#cb2286-4" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2286-5"><a href="integral-population-models.html#cb2286-5" tabindex="-1"></a><span class="co"># 8) Save RDS</span></span>
<span id="cb2286-6"><a href="integral-population-models.html#cb2286-6" tabindex="-1"></a><span class="co">#========================================================</span></span>
<span id="cb2286-7"><a href="integral-population-models.html#cb2286-7" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;IPM_outputs&quot;</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2286-8"><a href="integral-population-models.html#cb2286-8" tabindex="-1"></a>species_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;IPM_outputs&quot;</span>, <span class="fu">paste0</span>(species_name, <span class="st">&quot;_&quot;</span>, trans_years))</span>
<span id="cb2286-9"><a href="integral-population-models.html#cb2286-9" tabindex="-1"></a><span class="fu">dir.create</span>(species_dir, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2286-10"><a href="integral-population-models.html#cb2286-10" tabindex="-1"></a></span>
<span id="cb2286-11"><a href="integral-population-models.html#cb2286-11" tabindex="-1"></a>mesh_points <span class="ot">&lt;-</span> obs_ipm<span class="sc">$</span>domains<span class="sc">$</span>sa</span>
<span id="cb2286-12"><a href="integral-population-models.html#cb2286-12" tabindex="-1"></a>mesh_width  <span class="ot">&lt;-</span> <span class="fu">diff</span>(mesh_points)[<span class="dv">1</span>]</span>
<span id="cb2286-13"><a href="integral-population-models.html#cb2286-13" tabindex="-1"></a></span>
<span id="cb2286-14"><a href="integral-population-models.html#cb2286-14" tabindex="-1"></a>ipm_rds <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb2286-15"><a href="integral-population-models.html#cb2286-15" tabindex="-1"></a>  <span class="at">metadata =</span> <span class="fu">list</span>(</span>
<span id="cb2286-16"><a href="integral-population-models.html#cb2286-16" tabindex="-1"></a>    <span class="at">species        =</span> species_name,</span>
<span id="cb2286-17"><a href="integral-population-models.html#cb2286-17" tabindex="-1"></a>    <span class="at">transition     =</span> trans_years,</span>
<span id="cb2286-18"><a href="integral-population-models.html#cb2286-18" tabindex="-1"></a>    <span class="at">date_created   =</span> <span class="fu">Sys.Date</span>(),</span>
<span id="cb2286-19"><a href="integral-population-models.html#cb2286-19" tabindex="-1"></a>    <span class="at">state_variable =</span> <span class="st">&quot;size_log&quot;</span>,</span>
<span id="cb2286-20"><a href="integral-population-models.html#cb2286-20" tabindex="-1"></a>    <span class="at">state_scale    =</span> <span class="st">&quot;log&quot;</span>,</span>
<span id="cb2286-21"><a href="integral-population-models.html#cb2286-21" tabindex="-1"></a>    <span class="at">mesh_points_n  =</span> <span class="fu">length</span>(mesh_points),</span>
<span id="cb2286-22"><a href="integral-population-models.html#cb2286-22" tabindex="-1"></a>    <span class="at">mesh_bounds    =</span> <span class="fu">c</span>(<span class="at">L =</span> L, <span class="at">U =</span> U),</span>
<span id="cb2286-23"><a href="integral-population-models.html#cb2286-23" tabindex="-1"></a>    <span class="at">mesh_width     =</span> mesh_width</span>
<span id="cb2286-24"><a href="integral-population-models.html#cb2286-24" tabindex="-1"></a>  ),</span>
<span id="cb2286-25"><a href="integral-population-models.html#cb2286-25" tabindex="-1"></a>  <span class="at">kernels =</span> <span class="fu">list</span>(<span class="at">K =</span> K_matrix, <span class="at">P =</span> P, <span class="at">F =</span> F),</span>
<span id="cb2286-26"><a href="integral-population-models.html#cb2286-26" tabindex="-1"></a>  <span class="at">size    =</span> <span class="fu">list</span>(<span class="at">mesh_points_log =</span> mesh_points, <span class="at">stable_dist =</span> stable_dist_raw),</span>
<span id="cb2286-27"><a href="integral-population-models.html#cb2286-27" tabindex="-1"></a>  <span class="at">population_metrics =</span> <span class="fu">list</span>(<span class="at">lambda =</span> lambda_raw),</span>
<span id="cb2286-28"><a href="integral-population-models.html#cb2286-28" tabindex="-1"></a>  <span class="at">models  =</span> <span class="fu">list</span>(<span class="at">growth =</span> grow_mod, <span class="at">survival =</span> surv_mod, <span class="at">reproduction =</span> repr_mod, <span class="at">fecundity =</span> seed_mod),</span>
<span id="cb2286-29"><a href="integral-population-models.html#cb2286-29" tabindex="-1"></a>  <span class="at">recruitment =</span> <span class="fu">list</span>(</span>
<span id="cb2286-30"><a href="integral-population-models.html#cb2286-30" tabindex="-1"></a>    <span class="at">recr_mu_log        =</span> recr_mu,</span>
<span id="cb2286-31"><a href="integral-population-models.html#cb2286-31" tabindex="-1"></a>    <span class="at">recr_sd_log        =</span> recr_sd,</span>
<span id="cb2286-32"><a href="integral-population-models.html#cb2286-32" tabindex="-1"></a>    <span class="at">recr_n             =</span> recr_n,</span>
<span id="cb2286-33"><a href="integral-population-models.html#cb2286-33" tabindex="-1"></a>    <span class="at">seed_n             =</span> seed_n,</span>
<span id="cb2286-34"><a href="integral-population-models.html#cb2286-34" tabindex="-1"></a>    <span class="at">establishment_prob =</span> recr_n <span class="sc">/</span> seed_n</span>
<span id="cb2286-35"><a href="integral-population-models.html#cb2286-35" tabindex="-1"></a>  ),</span>
<span id="cb2286-36"><a href="integral-population-models.html#cb2286-36" tabindex="-1"></a>  <span class="at">params =</span> params,</span>
<span id="cb2286-37"><a href="integral-population-models.html#cb2286-37" tabindex="-1"></a>  <span class="at">data   =</span> final_data</span>
<span id="cb2286-38"><a href="integral-population-models.html#cb2286-38" tabindex="-1"></a>)</span>
<span id="cb2286-39"><a href="integral-population-models.html#cb2286-39" tabindex="-1"></a></span>
<span id="cb2286-40"><a href="integral-population-models.html#cb2286-40" tabindex="-1"></a><span class="fu">saveRDS</span>(ipm_rds, <span class="at">file =</span> <span class="fu">file.path</span>(species_dir, <span class="st">&quot;IPM_full_object.rds&quot;</span>))</span>
<span id="cb2286-41"><a href="integral-population-models.html#cb2286-41" tabindex="-1"></a></span>
<span id="cb2286-42"><a href="integral-population-models.html#cb2286-42" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Saved full IPM object to:&quot;</span>, <span class="fu">file.path</span>(species_dir, <span class="st">&quot;IPM_full_object.rds&quot;</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Saved full IPM object to: IPM_outputs/SILLAN_20232024/IPM_full_object.rds</code></pre>
<p>Now, let’s bootstrap for confidence intervals!</p>
<div class="sourceCode" id="cb2288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2288-1"><a href="integral-population-models.html#cb2288-1" tabindex="-1"></a><span class="co">#===============================================</span></span>
<span id="cb2288-2"><a href="integral-population-models.html#cb2288-2" tabindex="-1"></a><span class="co"># BOOTSTRAP ANALYSIS - UNCERTAINTY IN LAMBDA</span></span>
<span id="cb2288-3"><a href="integral-population-models.html#cb2288-3" tabindex="-1"></a><span class="co"># (INLINE ONLY, FIXED MESH)</span></span>
<span id="cb2288-4"><a href="integral-population-models.html#cb2288-4" tabindex="-1"></a><span class="co">#===============================================</span></span>
<span id="cb2288-5"><a href="integral-population-models.html#cb2288-5" tabindex="-1"></a><span class="fu">library</span>(ipmr)</span>
<span id="cb2288-6"><a href="integral-population-models.html#cb2288-6" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2288-7"><a href="integral-population-models.html#cb2288-7" tabindex="-1"></a></span>
<span id="cb2288-8"><a href="integral-population-models.html#cb2288-8" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== Starting Bootstrap Analysis (fixed mesh) ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === Starting Bootstrap Analysis (fixed mesh) ===</code></pre>
<div class="sourceCode" id="cb2290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2290-1"><a href="integral-population-models.html#cb2290-1" tabindex="-1"></a>safe_glm <span class="ot">&lt;-</span> <span class="cf">function</span>(formula, data, family) {</span>
<span id="cb2290-2"><a href="integral-population-models.html#cb2290-2" tabindex="-1"></a>  <span class="fu">tryCatch</span>(<span class="fu">glm</span>(formula, <span class="at">data =</span> data, <span class="at">family =</span> family),</span>
<span id="cb2290-3"><a href="integral-population-models.html#cb2290-3" tabindex="-1"></a>           <span class="at">error   =</span> <span class="cf">function</span>(e) <span class="cn">NULL</span>,</span>
<span id="cb2290-4"><a href="integral-population-models.html#cb2290-4" tabindex="-1"></a>           <span class="at">warning =</span> <span class="cf">function</span>(w) <span class="cn">NULL</span>)</span>
<span id="cb2290-5"><a href="integral-population-models.html#cb2290-5" tabindex="-1"></a>}</span>
<span id="cb2290-6"><a href="integral-population-models.html#cb2290-6" tabindex="-1"></a>safe_lm <span class="ot">&lt;-</span> <span class="cf">function</span>(formula, data) {</span>
<span id="cb2290-7"><a href="integral-population-models.html#cb2290-7" tabindex="-1"></a>  <span class="fu">tryCatch</span>(<span class="fu">lm</span>(formula, <span class="at">data =</span> data),</span>
<span id="cb2290-8"><a href="integral-population-models.html#cb2290-8" tabindex="-1"></a>           <span class="at">error   =</span> <span class="cf">function</span>(e) <span class="cn">NULL</span>,</span>
<span id="cb2290-9"><a href="integral-population-models.html#cb2290-9" tabindex="-1"></a>           <span class="at">warning =</span> <span class="cf">function</span>(w) <span class="cn">NULL</span>)</span>
<span id="cb2290-10"><a href="integral-population-models.html#cb2290-10" tabindex="-1"></a>}</span>
<span id="cb2290-11"><a href="integral-population-models.html#cb2290-11" tabindex="-1"></a></span>
<span id="cb2290-12"><a href="integral-population-models.html#cb2290-12" tabindex="-1"></a>get_stable_vec <span class="ot">&lt;-</span> <span class="cf">function</span>(ipm, <span class="at">state_var =</span> <span class="st">&quot;sa&quot;</span>) {</span>
<span id="cb2290-13"><a href="integral-population-models.html#cb2290-13" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(ipmr<span class="sc">::</span><span class="fu">right_ev</span>(ipm), <span class="at">error =</span> <span class="cf">function</span>(e) <span class="cn">NULL</span>)</span>
<span id="cb2290-14"><a href="integral-population-models.html#cb2290-14" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(out)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb2290-15"><a href="integral-population-models.html#cb2290-15" tabindex="-1"></a></span>
<span id="cb2290-16"><a href="integral-population-models.html#cb2290-16" tabindex="-1"></a>  sv <span class="ot">&lt;-</span> out[[<span class="fu">paste0</span>(state_var, <span class="st">&quot;_w&quot;</span>)]]  <span class="co"># e.g., &quot;sa_w&quot;</span></span>
<span id="cb2290-17"><a href="integral-population-models.html#cb2290-17" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(sv)) sv <span class="ot">&lt;-</span> out[[<span class="dv">1</span>]]</span>
<span id="cb2290-18"><a href="integral-population-models.html#cb2290-18" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(sv)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb2290-19"><a href="integral-population-models.html#cb2290-19" tabindex="-1"></a></span>
<span id="cb2290-20"><a href="integral-population-models.html#cb2290-20" tabindex="-1"></a>  sv <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(sv)</span>
<span id="cb2290-21"><a href="integral-population-models.html#cb2290-21" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">any</span>(<span class="sc">!</span><span class="fu">is.finite</span>(sv)) <span class="sc">||</span> <span class="fu">sum</span>(sv) <span class="sc">&lt;=</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb2290-22"><a href="integral-population-models.html#cb2290-22" tabindex="-1"></a>  sv <span class="sc">/</span> <span class="fu">sum</span>(sv)</span>
<span id="cb2290-23"><a href="integral-population-models.html#cb2290-23" tabindex="-1"></a>}</span>
<span id="cb2290-24"><a href="integral-population-models.html#cb2290-24" tabindex="-1"></a></span>
<span id="cb2290-25"><a href="integral-population-models.html#cb2290-25" tabindex="-1"></a><span class="co">#-------------------------------</span></span>
<span id="cb2290-26"><a href="integral-population-models.html#cb2290-26" tabindex="-1"></a><span class="co"># Data splits</span></span>
<span id="cb2290-27"><a href="integral-population-models.html#cb2290-27" tabindex="-1"></a><span class="co">#-------------------------------</span></span>
<span id="cb2290-28"><a href="integral-population-models.html#cb2290-28" tabindex="-1"></a>adults <span class="ot">&lt;-</span> final_data[<span class="fu">is.finite</span>(final_data<span class="sc">$</span>size_log), ]</span>
<span id="cb2290-29"><a href="integral-population-models.html#cb2290-29" tabindex="-1"></a></span>
<span id="cb2290-30"><a href="integral-population-models.html#cb2290-30" tabindex="-1"></a>robust_recruit_logical <span class="ot">&lt;-</span> <span class="fu">is.na</span>(final_data<span class="sc">$</span>size_log) <span class="sc">&amp;</span></span>
<span id="cb2290-31"><a href="integral-population-models.html#cb2290-31" tabindex="-1"></a>  <span class="fu">is.na</span>(final_data<span class="sc">$</span>surv) <span class="sc">&amp;</span></span>
<span id="cb2290-32"><a href="integral-population-models.html#cb2290-32" tabindex="-1"></a>  <span class="fu">is.na</span>(final_data<span class="sc">$</span>fec1) <span class="sc">&amp;</span></span>
<span id="cb2290-33"><a href="integral-population-models.html#cb2290-33" tabindex="-1"></a>  <span class="fu">is.finite</span>(final_data<span class="sc">$</span>sizeNext_log)</span>
<span id="cb2290-34"><a href="integral-population-models.html#cb2290-34" tabindex="-1"></a></span>
<span id="cb2290-35"><a href="integral-population-models.html#cb2290-35" tabindex="-1"></a>recr_data <span class="ot">&lt;-</span> final_data[robust_recruit_logical, ]</span>
<span id="cb2290-36"><a href="integral-population-models.html#cb2290-36" tabindex="-1"></a></span>
<span id="cb2290-37"><a href="integral-population-models.html#cb2290-37" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Adults (bootstrapped):&quot;</span>, <span class="fu">nrow</span>(adults), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Adults (bootstrapped): 82</code></pre>
<div class="sourceCode" id="cb2292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2292-1"><a href="integral-population-models.html#cb2292-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Recruits (held constant):&quot;</span>, <span class="fu">nrow</span>(recr_data), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Recruits (held constant): 293</code></pre>
<div class="sourceCode" id="cb2294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2294-1"><a href="integral-population-models.html#cb2294-1" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(adults) <span class="sc">&lt;</span> <span class="dv">10</span>) <span class="fu">stop</span>(<span class="st">&quot;Too few adults to bootstrap reliably.&quot;</span>)</span>
<span id="cb2294-2"><a href="integral-population-models.html#cb2294-2" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(recr_data) <span class="sc">&lt;</span> <span class="dv">1</span>) <span class="fu">stop</span>(<span class="st">&quot;No recruits under strict definition; cannot hold recruitment constant.&quot;</span>)</span>
<span id="cb2294-3"><a href="integral-population-models.html#cb2294-3" tabindex="-1"></a></span>
<span id="cb2294-4"><a href="integral-population-models.html#cb2294-4" tabindex="-1"></a><span class="co">#-------------------------------</span></span>
<span id="cb2294-5"><a href="integral-population-models.html#cb2294-5" tabindex="-1"></a><span class="co"># FIXED MESH: use main_env grid, not domains spec</span></span>
<span id="cb2294-6"><a href="integral-population-models.html#cb2294-6" tabindex="-1"></a><span class="co">#-------------------------------</span></span>
<span id="cb2294-7"><a href="integral-population-models.html#cb2294-7" tabindex="-1"></a>state_var <span class="ot">&lt;-</span> <span class="st">&quot;sa&quot;</span></span>
<span id="cb2294-8"><a href="integral-population-models.html#cb2294-8" tabindex="-1"></a></span>
<span id="cb2294-9"><a href="integral-population-models.html#cb2294-9" tabindex="-1"></a>Pm <span class="ot">&lt;-</span> obs_ipm<span class="sc">$</span>sub_kernels<span class="sc">$</span>P</span>
<span id="cb2294-10"><a href="integral-population-models.html#cb2294-10" tabindex="-1"></a>n_mesh <span class="ot">&lt;-</span> <span class="fu">nrow</span>(Pm)</span>
<span id="cb2294-11"><a href="integral-population-models.html#cb2294-11" tabindex="-1"></a></span>
<span id="cb2294-12"><a href="integral-population-models.html#cb2294-12" tabindex="-1"></a>sa_1_full <span class="ot">&lt;-</span> <span class="fu">get</span>(<span class="st">&quot;sa_1&quot;</span>, <span class="at">envir =</span> obs_ipm<span class="sc">$</span>env_list<span class="sc">$</span>main_env)</span>
<span id="cb2294-13"><a href="integral-population-models.html#cb2294-13" tabindex="-1"></a>fixed_mesh <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(<span class="fu">as.numeric</span>(sa_1_full)))</span>
<span id="cb2294-14"><a href="integral-population-models.html#cb2294-14" tabindex="-1"></a></span>
<span id="cb2294-15"><a href="integral-population-models.html#cb2294-15" tabindex="-1"></a><span class="co"># sanity</span></span>
<span id="cb2294-16"><a href="integral-population-models.html#cb2294-16" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(fixed_mesh) <span class="sc">!=</span> n_mesh) {</span>
<span id="cb2294-17"><a href="integral-population-models.html#cb2294-17" tabindex="-1"></a>  <span class="co"># fallback: generate mesh from range if something is odd</span></span>
<span id="cb2294-18"><a href="integral-population-models.html#cb2294-18" tabindex="-1"></a>  fixed_mesh <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">min</span>(fixed_mesh, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2294-19"><a href="integral-population-models.html#cb2294-19" tabindex="-1"></a>                    <span class="fu">max</span>(fixed_mesh, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2294-20"><a href="integral-population-models.html#cb2294-20" tabindex="-1"></a>                    <span class="at">length.out =</span> n_mesh)</span>
<span id="cb2294-21"><a href="integral-population-models.html#cb2294-21" tabindex="-1"></a>}</span>
<span id="cb2294-22"><a href="integral-population-models.html#cb2294-22" tabindex="-1"></a></span>
<span id="cb2294-23"><a href="integral-population-models.html#cb2294-23" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Kernel dimension n_mesh:&quot;</span>, n_mesh, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Kernel dimension n_mesh: 100</code></pre>
<div class="sourceCode" id="cb2296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2296-1"><a href="integral-population-models.html#cb2296-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Fixed mesh points:&quot;</span>, <span class="fu">length</span>(fixed_mesh), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Fixed mesh points: 100</code></pre>
<div class="sourceCode" id="cb2298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2298-1"><a href="integral-population-models.html#cb2298-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Domain (log):&quot;</span>, <span class="fu">paste</span>(<span class="fu">round</span>(<span class="fu">range</span>(fixed_mesh), <span class="dv">3</span>), <span class="at">collapse =</span> <span class="st">&quot; to &quot;</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Domain (log): -0.221 to 5.177</code></pre>
<div class="sourceCode" id="cb2300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2300-1"><a href="integral-population-models.html#cb2300-1" tabindex="-1"></a><span class="co"># proto</span></span>
<span id="cb2300-2"><a href="integral-population-models.html#cb2300-2" tabindex="-1"></a>proto_template <span class="ot">&lt;-</span> obs_ipm<span class="sc">$</span>proto_ipm</span>
<span id="cb2300-3"><a href="integral-population-models.html#cb2300-3" tabindex="-1"></a></span>
<span id="cb2300-4"><a href="integral-population-models.html#cb2300-4" tabindex="-1"></a><span class="co"># full-data fallbacks</span></span>
<span id="cb2300-5"><a href="integral-population-models.html#cb2300-5" tabindex="-1"></a>surv_mod_full <span class="ot">&lt;-</span> surv_mod</span>
<span id="cb2300-6"><a href="integral-population-models.html#cb2300-6" tabindex="-1"></a>repr_mod_full <span class="ot">&lt;-</span> repr_mod</span>
<span id="cb2300-7"><a href="integral-population-models.html#cb2300-7" tabindex="-1"></a>seed_mod_full <span class="ot">&lt;-</span> seed_mod</span>
<span id="cb2300-8"><a href="integral-population-models.html#cb2300-8" tabindex="-1"></a></span>
<span id="cb2300-9"><a href="integral-population-models.html#cb2300-9" tabindex="-1"></a><span class="co"># recruitment params held constant</span></span>
<span id="cb2300-10"><a href="integral-population-models.html#cb2300-10" tabindex="-1"></a>recr_mu_full   <span class="ot">&lt;-</span> recr_mu</span>
<span id="cb2300-11"><a href="integral-population-models.html#cb2300-11" tabindex="-1"></a>recr_sd_full   <span class="ot">&lt;-</span> recr_sd</span>
<span id="cb2300-12"><a href="integral-population-models.html#cb2300-12" tabindex="-1"></a>recr_n_full    <span class="ot">&lt;-</span> recr_n</span>
<span id="cb2300-13"><a href="integral-population-models.html#cb2300-13" tabindex="-1"></a>min_repro_full <span class="ot">&lt;-</span> min_repro_size_log</span>
<span id="cb2300-14"><a href="integral-population-models.html#cb2300-14" tabindex="-1"></a></span>
<span id="cb2300-15"><a href="integral-population-models.html#cb2300-15" tabindex="-1"></a><span class="co"># IMPORTANT: hold seed_n constant to avoid resamples crashing establishment</span></span>
<span id="cb2300-16"><a href="integral-population-models.html#cb2300-16" tabindex="-1"></a>seed_n_full <span class="ot">&lt;-</span> <span class="fu">sum</span>(final_data<span class="sc">$</span>fec2[final_data<span class="sc">$</span>fec1 <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(final_data<span class="sc">$</span>fec2)], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2300-17"><a href="integral-population-models.html#cb2300-17" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.finite</span>(seed_n_full) <span class="sc">||</span> seed_n_full <span class="sc">&lt;=</span> <span class="dv">0</span>) {</span>
<span id="cb2300-18"><a href="integral-population-models.html#cb2300-18" tabindex="-1"></a>  <span class="fu">stop</span>(<span class="st">&quot;seed_n_full is 0/NA. Need at least one observed fec2 where fec1==1.&quot;</span>)</span>
<span id="cb2300-19"><a href="integral-population-models.html#cb2300-19" tabindex="-1"></a>}</span>
<span id="cb2300-20"><a href="integral-population-models.html#cb2300-20" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Fixed establishment r_r = recr_n/seed_n:&quot;</span>, <span class="fu">round</span>(recr_n_full<span class="sc">/</span>seed_n_full, <span class="dv">5</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Fixed establishment r_r = recr_n/seed_n: 6e-05</code></pre>
<div class="sourceCode" id="cb2302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2302-1"><a href="integral-population-models.html#cb2302-1" tabindex="-1"></a><span class="co">#-------------------------------</span></span>
<span id="cb2302-2"><a href="integral-population-models.html#cb2302-2" tabindex="-1"></a><span class="co"># Bootstrap controls</span></span>
<span id="cb2302-3"><a href="integral-population-models.html#cb2302-3" tabindex="-1"></a><span class="co">#-------------------------------</span></span>
<span id="cb2302-4"><a href="integral-population-models.html#cb2302-4" tabindex="-1"></a>n_boot    <span class="ot">&lt;-</span> <span class="dv">50</span><span class="dt">L</span></span>
<span id="cb2302-5"><a href="integral-population-models.html#cb2302-5" tabindex="-1"></a>max_tries <span class="ot">&lt;-</span> n_boot <span class="sc">*</span> <span class="dv">10</span><span class="dt">L</span></span>
<span id="cb2302-6"><a href="integral-population-models.html#cb2302-6" tabindex="-1"></a></span>
<span id="cb2302-7"><a href="integral-population-models.html#cb2302-7" tabindex="-1"></a>all_lambdas <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA_real_</span>, n_boot)</span>
<span id="cb2302-8"><a href="integral-population-models.html#cb2302-8" tabindex="-1"></a>valid_iter  <span class="ot">&lt;-</span> <span class="dv">0</span><span class="dt">L</span></span>
<span id="cb2302-9"><a href="integral-population-models.html#cb2302-9" tabindex="-1"></a>attempts    <span class="ot">&lt;-</span> <span class="dv">0</span><span class="dt">L</span></span>
<span id="cb2302-10"><a href="integral-population-models.html#cb2302-10" tabindex="-1"></a></span>
<span id="cb2302-11"><a href="integral-population-models.html#cb2302-11" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== Bootstrap Loop ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === Bootstrap Loop ===</code></pre>
<div class="sourceCode" id="cb2304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2304-1"><a href="integral-population-models.html#cb2304-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Target iterations:&quot;</span>, n_boot, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Target iterations: 50</code></pre>
<div class="sourceCode" id="cb2306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2306-1"><a href="integral-population-models.html#cb2306-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Maximum attempts:&quot;</span>, max_tries, <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Maximum attempts: 500</code></pre>
<div class="sourceCode" id="cb2308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2308-1"><a href="integral-population-models.html#cb2308-1" tabindex="-1"></a><span class="co">#-------------------------------</span></span>
<span id="cb2308-2"><a href="integral-population-models.html#cb2308-2" tabindex="-1"></a><span class="co"># Bootstrap loop</span></span>
<span id="cb2308-3"><a href="integral-population-models.html#cb2308-3" tabindex="-1"></a><span class="co">#-------------------------------</span></span>
<span id="cb2308-4"><a href="integral-population-models.html#cb2308-4" tabindex="-1"></a><span class="cf">while</span> (valid_iter <span class="sc">&lt;</span> n_boot <span class="sc">&amp;&amp;</span> attempts <span class="sc">&lt;</span> max_tries) {</span>
<span id="cb2308-5"><a href="integral-population-models.html#cb2308-5" tabindex="-1"></a></span>
<span id="cb2308-6"><a href="integral-population-models.html#cb2308-6" tabindex="-1"></a>  attempts <span class="ot">&lt;-</span> attempts <span class="sc">+</span> <span class="dv">1</span><span class="dt">L</span></span>
<span id="cb2308-7"><a href="integral-population-models.html#cb2308-7" tabindex="-1"></a>  <span class="cf">if</span> (attempts <span class="sc">%%</span> <span class="dv">25</span><span class="dt">L</span> <span class="sc">==</span> <span class="dv">0</span><span class="dt">L</span>) <span class="fu">cat</span>(<span class="st">&quot;Attempt&quot;</span>, attempts, <span class="st">&quot;| Valid:&quot;</span>, valid_iter, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2308-8"><a href="integral-population-models.html#cb2308-8" tabindex="-1"></a></span>
<span id="cb2308-9"><a href="integral-population-models.html#cb2308-9" tabindex="-1"></a>  boot_ind  <span class="ot">&lt;-</span> <span class="fu">sample.int</span>(<span class="fu">nrow</span>(adults), <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2308-10"><a href="integral-population-models.html#cb2308-10" tabindex="-1"></a>  boot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(adults[boot_ind, ], recr_data)</span>
<span id="cb2308-11"><a href="integral-population-models.html#cb2308-11" tabindex="-1"></a></span>
<span id="cb2308-12"><a href="integral-population-models.html#cb2308-12" tabindex="-1"></a>  <span class="co"># Growth (only rows with both)</span></span>
<span id="cb2308-13"><a href="integral-population-models.html#cb2308-13" tabindex="-1"></a>  gdat <span class="ot">&lt;-</span> boot_data[<span class="fu">is.finite</span>(boot_data<span class="sc">$</span>size_log) <span class="sc">&amp;</span> <span class="fu">is.finite</span>(boot_data<span class="sc">$</span>sizeNext_log), ]</span>
<span id="cb2308-14"><a href="integral-population-models.html#cb2308-14" tabindex="-1"></a>  grow_mod_boot <span class="ot">&lt;-</span> <span class="fu">safe_lm</span>(sizeNext_log <span class="sc">~</span> size_log <span class="sc">+</span> <span class="fu">I</span>(size_log<span class="sc">^</span><span class="dv">2</span>), <span class="at">data =</span> gdat)</span>
<span id="cb2308-15"><a href="integral-population-models.html#cb2308-15" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(grow_mod_boot)) <span class="cf">next</span></span>
<span id="cb2308-16"><a href="integral-population-models.html#cb2308-16" tabindex="-1"></a>  grow_sd_boot <span class="ot">&lt;-</span> <span class="fu">sd</span>(<span class="fu">resid</span>(grow_mod_boot), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2308-17"><a href="integral-population-models.html#cb2308-17" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.finite</span>(grow_sd_boot) <span class="sc">||</span> grow_sd_boot <span class="sc">&lt;=</span> <span class="dv">0</span>) <span class="cf">next</span></span>
<span id="cb2308-18"><a href="integral-population-models.html#cb2308-18" tabindex="-1"></a></span>
<span id="cb2308-19"><a href="integral-population-models.html#cb2308-19" tabindex="-1"></a>  <span class="co"># Survival</span></span>
<span id="cb2308-20"><a href="integral-population-models.html#cb2308-20" tabindex="-1"></a>  sdat <span class="ot">&lt;-</span> boot_data[<span class="fu">is.finite</span>(boot_data<span class="sc">$</span>size_log) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(boot_data<span class="sc">$</span>surv), ]</span>
<span id="cb2308-21"><a href="integral-population-models.html#cb2308-21" tabindex="-1"></a>  surv_mod_boot <span class="ot">&lt;-</span> <span class="fu">safe_glm</span>(surv <span class="sc">~</span> size_log, <span class="at">data =</span> sdat, <span class="at">family =</span> <span class="fu">binomial</span>())</span>
<span id="cb2308-22"><a href="integral-population-models.html#cb2308-22" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(surv_mod_boot)) surv_mod_boot <span class="ot">&lt;-</span> surv_mod_full</span>
<span id="cb2308-23"><a href="integral-population-models.html#cb2308-23" tabindex="-1"></a></span>
<span id="cb2308-24"><a href="integral-population-models.html#cb2308-24" tabindex="-1"></a>  <span class="co"># Reproduction</span></span>
<span id="cb2308-25"><a href="integral-population-models.html#cb2308-25" tabindex="-1"></a>  rdat <span class="ot">&lt;-</span> boot_data[<span class="fu">is.finite</span>(boot_data<span class="sc">$</span>size_log) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(boot_data<span class="sc">$</span>fec1), ]</span>
<span id="cb2308-26"><a href="integral-population-models.html#cb2308-26" tabindex="-1"></a>  repr_mod_boot <span class="ot">&lt;-</span> <span class="fu">safe_glm</span>(fec1 <span class="sc">~</span> size_log <span class="sc">+</span> <span class="fu">I</span>(size_log<span class="sc">^</span><span class="dv">2</span>), <span class="at">data =</span> rdat, <span class="at">family =</span> <span class="fu">binomial</span>())</span>
<span id="cb2308-27"><a href="integral-population-models.html#cb2308-27" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(repr_mod_boot)) repr_mod_boot <span class="ot">&lt;-</span> repr_mod_full</span>
<span id="cb2308-28"><a href="integral-population-models.html#cb2308-28" tabindex="-1"></a></span>
<span id="cb2308-29"><a href="integral-population-models.html#cb2308-29" tabindex="-1"></a>  <span class="co"># Seed production (fit only where fec2 observed AND fec1==1)</span></span>
<span id="cb2308-30"><a href="integral-population-models.html#cb2308-30" tabindex="-1"></a>  fdat <span class="ot">&lt;-</span> boot_data[<span class="fu">is.finite</span>(boot_data<span class="sc">$</span>size_log) <span class="sc">&amp;</span> boot_data<span class="sc">$</span>fec1 <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(boot_data<span class="sc">$</span>fec2), ]</span>
<span id="cb2308-31"><a href="integral-population-models.html#cb2308-31" tabindex="-1"></a>  seed_mod_boot <span class="ot">&lt;-</span> <span class="fu">safe_glm</span>(fec2 <span class="sc">~</span> size_log <span class="sc">+</span> <span class="fu">I</span>(size_log<span class="sc">^</span><span class="dv">2</span>), <span class="at">data =</span> fdat, <span class="at">family =</span> <span class="fu">poisson</span>())</span>
<span id="cb2308-32"><a href="integral-population-models.html#cb2308-32" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(seed_mod_boot)) seed_mod_boot <span class="ot">&lt;-</span> seed_mod_full</span>
<span id="cb2308-33"><a href="integral-population-models.html#cb2308-33" tabindex="-1"></a></span>
<span id="cb2308-34"><a href="integral-population-models.html#cb2308-34" tabindex="-1"></a>  params_boot <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb2308-35"><a href="integral-population-models.html#cb2308-35" tabindex="-1"></a>    <span class="at">recr_mu            =</span> recr_mu_full,</span>
<span id="cb2308-36"><a href="integral-population-models.html#cb2308-36" tabindex="-1"></a>    <span class="at">recr_sd            =</span> recr_sd_full,</span>
<span id="cb2308-37"><a href="integral-population-models.html#cb2308-37" tabindex="-1"></a>    <span class="at">grow_sd            =</span> grow_sd_boot,</span>
<span id="cb2308-38"><a href="integral-population-models.html#cb2308-38" tabindex="-1"></a>    <span class="at">surv_mod           =</span> surv_mod_boot,</span>
<span id="cb2308-39"><a href="integral-population-models.html#cb2308-39" tabindex="-1"></a>    <span class="at">grow_mod           =</span> grow_mod_boot,</span>
<span id="cb2308-40"><a href="integral-population-models.html#cb2308-40" tabindex="-1"></a>    <span class="at">repr_mod           =</span> repr_mod_boot,</span>
<span id="cb2308-41"><a href="integral-population-models.html#cb2308-41" tabindex="-1"></a>    <span class="at">seed_mod           =</span> seed_mod_boot,</span>
<span id="cb2308-42"><a href="integral-population-models.html#cb2308-42" tabindex="-1"></a>    <span class="at">recr_n             =</span> recr_n_full,</span>
<span id="cb2308-43"><a href="integral-population-models.html#cb2308-43" tabindex="-1"></a>    <span class="at">seed_n             =</span> seed_n_full,        <span class="co"># fixed</span></span>
<span id="cb2308-44"><a href="integral-population-models.html#cb2308-44" tabindex="-1"></a>    <span class="at">min_repro_size_log =</span> min_repro_full</span>
<span id="cb2308-45"><a href="integral-population-models.html#cb2308-45" tabindex="-1"></a>  )</span>
<span id="cb2308-46"><a href="integral-population-models.html#cb2308-46" tabindex="-1"></a></span>
<span id="cb2308-47"><a href="integral-population-models.html#cb2308-47" tabindex="-1"></a>  proto_boot <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">unserialize</span>(<span class="fu">serialize</span>(proto_template, <span class="cn">NULL</span>)),</span>
<span id="cb2308-48"><a href="integral-population-models.html#cb2308-48" tabindex="-1"></a>                        <span class="at">error =</span> <span class="cf">function</span>(e) proto_template)</span>
<span id="cb2308-49"><a href="integral-population-models.html#cb2308-49" tabindex="-1"></a></span>
<span id="cb2308-50"><a href="integral-population-models.html#cb2308-50" tabindex="-1"></a>  ipmr<span class="sc">::</span><span class="fu">parameters</span>(proto_boot) <span class="ot">&lt;-</span> params_boot</span>
<span id="cb2308-51"><a href="integral-population-models.html#cb2308-51" tabindex="-1"></a></span>
<span id="cb2308-52"><a href="integral-population-models.html#cb2308-52" tabindex="-1"></a>  boot_ipm <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(</span>
<span id="cb2308-53"><a href="integral-population-models.html#cb2308-53" tabindex="-1"></a>    <span class="fu">suppressMessages</span>(<span class="fu">suppressWarnings</span>(</span>
<span id="cb2308-54"><a href="integral-population-models.html#cb2308-54" tabindex="-1"></a>      <span class="fu">make_ipm</span>(proto_boot, <span class="at">iterate =</span> <span class="cn">TRUE</span>, <span class="at">iterations =</span> <span class="dv">100</span>)</span>
<span id="cb2308-55"><a href="integral-population-models.html#cb2308-55" tabindex="-1"></a>    )),</span>
<span id="cb2308-56"><a href="integral-population-models.html#cb2308-56" tabindex="-1"></a>    <span class="at">error =</span> <span class="cf">function</span>(e) <span class="cn">NULL</span></span>
<span id="cb2308-57"><a href="integral-population-models.html#cb2308-57" tabindex="-1"></a>  )</span>
<span id="cb2308-58"><a href="integral-population-models.html#cb2308-58" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(boot_ipm)) <span class="cf">next</span></span>
<span id="cb2308-59"><a href="integral-population-models.html#cb2308-59" tabindex="-1"></a></span>
<span id="cb2308-60"><a href="integral-population-models.html#cb2308-60" tabindex="-1"></a>  lam <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(ipmr<span class="sc">::</span><span class="fu">lambda</span>(boot_ipm), <span class="at">error =</span> <span class="cf">function</span>(e) <span class="cn">NA_real_</span>)</span>
<span id="cb2308-61"><a href="integral-population-models.html#cb2308-61" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.finite</span>(lam)) <span class="cf">next</span></span>
<span id="cb2308-62"><a href="integral-population-models.html#cb2308-62" tabindex="-1"></a></span>
<span id="cb2308-63"><a href="integral-population-models.html#cb2308-63" tabindex="-1"></a>  stable_vec <span class="ot">&lt;-</span> <span class="fu">get_stable_vec</span>(boot_ipm, <span class="at">state_var =</span> state_var)</span>
<span id="cb2308-64"><a href="integral-population-models.html#cb2308-64" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(stable_vec)) <span class="cf">next</span></span>
<span id="cb2308-65"><a href="integral-population-models.html#cb2308-65" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(stable_vec) <span class="sc">!=</span> n_mesh) <span class="cf">next</span>  <span class="co"># should now match!</span></span>
<span id="cb2308-66"><a href="integral-population-models.html#cb2308-66" tabindex="-1"></a></span>
<span id="cb2308-67"><a href="integral-population-models.html#cb2308-67" tabindex="-1"></a>  valid_iter <span class="ot">&lt;-</span> valid_iter <span class="sc">+</span> <span class="dv">1</span><span class="dt">L</span></span>
<span id="cb2308-68"><a href="integral-population-models.html#cb2308-68" tabindex="-1"></a>  all_lambdas[valid_iter] <span class="ot">&lt;-</span> lam</span>
<span id="cb2308-69"><a href="integral-population-models.html#cb2308-69" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Attempt 25 | Valid: 23 
## Attempt 50 | Valid: 47</code></pre>
<div class="sourceCode" id="cb2310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2310-1"><a href="integral-population-models.html#cb2310-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== Bootstrap Complete ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === Bootstrap Complete ===</code></pre>
<div class="sourceCode" id="cb2312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2312-1"><a href="integral-population-models.html#cb2312-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Valid iterations:&quot;</span>, valid_iter, <span class="st">&quot;/&quot;</span>, n_boot, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Valid iterations: 50 / 50</code></pre>
<div class="sourceCode" id="cb2314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2314-1"><a href="integral-population-models.html#cb2314-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Total attempts:&quot;</span>, attempts, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Total attempts: 53</code></pre>
<div class="sourceCode" id="cb2316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2316-1"><a href="integral-population-models.html#cb2316-1" tabindex="-1"></a><span class="cf">if</span> (valid_iter <span class="sc">==</span> <span class="dv">0</span><span class="dt">L</span>) {</span>
<span id="cb2316-2"><a href="integral-population-models.html#cb2316-2" tabindex="-1"></a>  <span class="fu">stop</span>(<span class="st">&quot;Bootstrap produced 0 valid iterations even after mesh fix. Next step: print make_ipm errors.&quot;</span>)</span>
<span id="cb2316-3"><a href="integral-population-models.html#cb2316-3" tabindex="-1"></a>}</span>
<span id="cb2316-4"><a href="integral-population-models.html#cb2316-4" tabindex="-1"></a></span>
<span id="cb2316-5"><a href="integral-population-models.html#cb2316-5" tabindex="-1"></a>all_lambdas <span class="ot">&lt;-</span> all_lambdas[<span class="fu">seq_len</span>(valid_iter)]</span>
<span id="cb2316-6"><a href="integral-population-models.html#cb2316-6" tabindex="-1"></a></span>
<span id="cb2316-7"><a href="integral-population-models.html#cb2316-7" tabindex="-1"></a>lambda_ci  <span class="ot">&lt;-</span> <span class="fu">quantile</span>(all_lambdas, <span class="at">probs =</span> <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.975</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2316-8"><a href="integral-population-models.html#cb2316-8" tabindex="-1"></a>lambda_obs <span class="ot">&lt;-</span> ipmr<span class="sc">::</span><span class="fu">lambda</span>(obs_ipm)</span>
<span id="cb2316-9"><a href="integral-population-models.html#cb2316-9" tabindex="-1"></a></span>
<span id="cb2316-10"><a href="integral-population-models.html#cb2316-10" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== Lambda Estimates ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === Lambda Estimates ===</code></pre>
<div class="sourceCode" id="cb2318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2318-1"><a href="integral-population-models.html#cb2318-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Observed λ:&quot;</span>, <span class="fu">round</span>(lambda_obs, <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Observed λ: 0.9694</code></pre>
<div class="sourceCode" id="cb2320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2320-1"><a href="integral-population-models.html#cb2320-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Bootstrap mean λ:&quot;</span>, <span class="fu">round</span>(<span class="fu">mean</span>(all_lambdas), <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Bootstrap mean λ: 0.9453</code></pre>
<div class="sourceCode" id="cb2322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2322-1"><a href="integral-population-models.html#cb2322-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;95% CI: [&quot;</span>, <span class="fu">round</span>(lambda_ci[<span class="dv">1</span>], <span class="dv">4</span>), <span class="st">&quot;,&quot;</span>, <span class="fu">round</span>(lambda_ci[<span class="dv">2</span>], <span class="dv">4</span>), <span class="st">&quot;]</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 95% CI: [ 0.8445 , 0.9946 ]</code></pre>
<div class="sourceCode" id="cb2324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2324-1"><a href="integral-population-models.html#cb2324-1" tabindex="-1"></a>lambda_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">lambda =</span> all_lambdas), <span class="fu">aes</span>(<span class="at">x =</span> lambda)) <span class="sc">+</span></span>
<span id="cb2324-2"><a href="integral-population-models.html#cb2324-2" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb2324-3"><a href="integral-population-models.html#cb2324-3" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> lambda_obs, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb2324-4"><a href="integral-population-models.html#cb2324-4" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> lambda_ci[<span class="dv">1</span>], <span class="at">linetype =</span> <span class="st">&quot;dotted&quot;</span>, <span class="at">linewidth =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb2324-5"><a href="integral-population-models.html#cb2324-5" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> lambda_ci[<span class="dv">2</span>], <span class="at">linetype =</span> <span class="st">&quot;dotted&quot;</span>, <span class="at">linewidth =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb2324-6"><a href="integral-population-models.html#cb2324-6" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb2324-7"><a href="integral-population-models.html#cb2324-7" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Bootstrap Distribution of λ&quot;</span>,</span>
<span id="cb2324-8"><a href="integral-population-models.html#cb2324-8" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">&quot;n = &quot;</span>, valid_iter, <span class="st">&quot; successful iterations&quot;</span>),</span>
<span id="cb2324-9"><a href="integral-population-models.html#cb2324-9" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Population growth rate (λ)&quot;</span>,</span>
<span id="cb2324-10"><a href="integral-population-models.html#cb2324-10" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Frequency&quot;</span></span>
<span id="cb2324-11"><a href="integral-population-models.html#cb2324-11" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2324-12"><a href="integral-population-models.html#cb2324-12" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>)</span>
<span id="cb2324-13"><a href="integral-population-models.html#cb2324-13" tabindex="-1"></a></span>
<span id="cb2324-14"><a href="integral-population-models.html#cb2324-14" tabindex="-1"></a><span class="fu">print</span>(lambda_plot)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-29-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
<p>Now, let’s look at these results! To better understand the life history strategy underlying the observed population growth rate, we calculated several additional demographic metrics. These included net reproductive rate (R₀; the expected lifetime offspring per individual), generation time (the average interval between generations), and size-specific patterns in reproduction and survival. Together, these metrics paint a fuller picture of how individuals progress through their life cycle and contribute to population dynamics.</p>
<div class="sourceCode" id="cb2325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2325-1"><a href="integral-population-models.html#cb2325-1" tabindex="-1"></a><span class="co">#==============================================================</span></span>
<span id="cb2325-2"><a href="integral-population-models.html#cb2325-2" tabindex="-1"></a><span class="co"># LIFE-HISTORY METRICS FROM A SIZE-STRUCTURED IPM</span></span>
<span id="cb2325-3"><a href="integral-population-models.html#cb2325-3" tabindex="-1"></a><span class="co">#==============================================================</span></span>
<span id="cb2325-4"><a href="integral-population-models.html#cb2325-4" tabindex="-1"></a><span class="fu">library</span>(ipmr)</span>
<span id="cb2325-5"><a href="integral-population-models.html#cb2325-5" tabindex="-1"></a></span>
<span id="cb2325-6"><a href="integral-population-models.html#cb2325-6" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== Computing Life-History Metrics ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === Computing Life-History Metrics ===</code></pre>
<div class="sourceCode" id="cb2327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2327-1"><a href="integral-population-models.html#cb2327-1" tabindex="-1"></a>state_var <span class="ot">&lt;-</span> <span class="st">&quot;sa&quot;</span></span>
<span id="cb2327-2"><a href="integral-population-models.html#cb2327-2" tabindex="-1"></a></span>
<span id="cb2327-3"><a href="integral-population-models.html#cb2327-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2327-4"><a href="integral-population-models.html#cb2327-4" tabindex="-1"></a><span class="co"># 1. Extract discretized kernels</span></span>
<span id="cb2327-5"><a href="integral-population-models.html#cb2327-5" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2327-6"><a href="integral-population-models.html#cb2327-6" tabindex="-1"></a>Pm <span class="ot">&lt;-</span> obs_ipm<span class="sc">$</span>sub_kernels<span class="sc">$</span>P</span>
<span id="cb2327-7"><a href="integral-population-models.html#cb2327-7" tabindex="-1"></a>Fm <span class="ot">&lt;-</span> obs_ipm<span class="sc">$</span>sub_kernels<span class="sc">$</span>F</span>
<span id="cb2327-8"><a href="integral-population-models.html#cb2327-8" tabindex="-1"></a>K  <span class="ot">&lt;-</span> Pm <span class="sc">+</span> Fm</span>
<span id="cb2327-9"><a href="integral-population-models.html#cb2327-9" tabindex="-1"></a></span>
<span id="cb2327-10"><a href="integral-population-models.html#cb2327-10" tabindex="-1"></a>n_mesh <span class="ot">&lt;-</span> <span class="fu">nrow</span>(Pm)</span>
<span id="cb2327-11"><a href="integral-population-models.html#cb2327-11" tabindex="-1"></a></span>
<span id="cb2327-12"><a href="integral-population-models.html#cb2327-12" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2327-13"><a href="integral-population-models.html#cb2327-13" tabindex="-1"></a><span class="co"># 2. Extract mesh (CORRECTED - get unique values)</span></span>
<span id="cb2327-14"><a href="integral-population-models.html#cb2327-14" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2327-15"><a href="integral-population-models.html#cb2327-15" tabindex="-1"></a></span>
<span id="cb2327-16"><a href="integral-population-models.html#cb2327-16" tabindex="-1"></a><span class="co"># sa_1 in main_env is the expanded grid (n_mesh × n_mesh values)</span></span>
<span id="cb2327-17"><a href="integral-population-models.html#cb2327-17" tabindex="-1"></a><span class="co"># We need the unique mesh points</span></span>
<span id="cb2327-18"><a href="integral-population-models.html#cb2327-18" tabindex="-1"></a>sa_1_full <span class="ot">&lt;-</span> <span class="fu">get</span>(<span class="st">&quot;sa_1&quot;</span>, <span class="at">envir =</span> obs_ipm<span class="sc">$</span>env_list<span class="sc">$</span>main_env)</span>
<span id="cb2327-19"><a href="integral-population-models.html#cb2327-19" tabindex="-1"></a>size_classes <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(<span class="fu">as.numeric</span>(sa_1_full)))</span>
<span id="cb2327-20"><a href="integral-population-models.html#cb2327-20" tabindex="-1"></a></span>
<span id="cb2327-21"><a href="integral-population-models.html#cb2327-21" tabindex="-1"></a><span class="co"># Verify length</span></span>
<span id="cb2327-22"><a href="integral-population-models.html#cb2327-22" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(size_classes) <span class="sc">!=</span> n_mesh) {</span>
<span id="cb2327-23"><a href="integral-population-models.html#cb2327-23" tabindex="-1"></a>  <span class="fu">stop</span>(<span class="st">&quot;Mesh length (&quot;</span>, <span class="fu">length</span>(size_classes), <span class="st">&quot;) != kernel dim (&quot;</span>, n_mesh, <span class="st">&quot;)&quot;</span>)</span>
<span id="cb2327-24"><a href="integral-population-models.html#cb2327-24" tabindex="-1"></a>}</span>
<span id="cb2327-25"><a href="integral-population-models.html#cb2327-25" tabindex="-1"></a></span>
<span id="cb2327-26"><a href="integral-population-models.html#cb2327-26" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Extracted mesh from main_env (unique values from sa_1)</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Extracted mesh from main_env (unique values from sa_1)</code></pre>
<div class="sourceCode" id="cb2329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2329-1"><a href="integral-population-models.html#cb2329-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Original sa_1 length:&quot;</span>, <span class="fu">length</span>(sa_1_full), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Original sa_1 length: 10000</code></pre>
<div class="sourceCode" id="cb2331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2331-1"><a href="integral-population-models.html#cb2331-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Unique mesh points:&quot;</span>, n_mesh, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Unique mesh points: 100</code></pre>
<div class="sourceCode" id="cb2333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2333-1"><a href="integral-population-models.html#cb2333-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Domain (log):&quot;</span>, <span class="fu">round</span>(<span class="fu">range</span>(size_classes), <span class="dv">3</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Domain (log): -0.221 5.177</code></pre>
<div class="sourceCode" id="cb2335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2335-1"><a href="integral-population-models.html#cb2335-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Domain (raw):&quot;</span>, <span class="fu">round</span>(<span class="fu">exp</span>(<span class="fu">range</span>(size_classes)), <span class="dv">2</span>), <span class="st">&quot;cm</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Domain (raw): 0.8 177.08 cm</code></pre>
<div class="sourceCode" id="cb2337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2337-1"><a href="integral-population-models.html#cb2337-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2337-2"><a href="integral-population-models.html#cb2337-2" tabindex="-1"></a><span class="co"># 3. Population growth rate (λ)</span></span>
<span id="cb2337-3"><a href="integral-population-models.html#cb2337-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2337-4"><a href="integral-population-models.html#cb2337-4" tabindex="-1"></a>lambda_obs <span class="ot">&lt;-</span> ipmr<span class="sc">::</span><span class="fu">lambda</span>(obs_ipm)</span>
<span id="cb2337-5"><a href="integral-population-models.html#cb2337-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">✓ Lambda:&quot;</span>, <span class="fu">round</span>(lambda_obs, <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## ✓ Lambda: 0.9694</code></pre>
<div class="sourceCode" id="cb2339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2339-1"><a href="integral-population-models.html#cb2339-1" tabindex="-1"></a><span class="cf">if</span>(lambda_obs <span class="sc">&gt;</span> <span class="dv">1</span>) {</span>
<span id="cb2339-2"><a href="integral-population-models.html#cb2339-2" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;  → Population GROWING at&quot;</span>, <span class="fu">round</span>((lambda_obs <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>), <span class="st">&quot;% per year</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2339-3"><a href="integral-population-models.html#cb2339-3" tabindex="-1"></a>} <span class="cf">else</span> <span class="cf">if</span>(lambda_obs <span class="sc">&lt;</span> <span class="dv">1</span>) {</span>
<span id="cb2339-4"><a href="integral-population-models.html#cb2339-4" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;  → Population DECLINING at&quot;</span>, <span class="fu">round</span>((<span class="dv">1</span> <span class="sc">-</span> lambda_obs) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>), <span class="st">&quot;% per year</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2339-5"><a href="integral-population-models.html#cb2339-5" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb2339-6"><a href="integral-population-models.html#cb2339-6" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;  → Population STABLE</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2339-7"><a href="integral-population-models.html#cb2339-7" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>##   → Population DECLINING at 3.06 % per year</code></pre>
<div class="sourceCode" id="cb2341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2341-1"><a href="integral-population-models.html#cb2341-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2341-2"><a href="integral-population-models.html#cb2341-2" tabindex="-1"></a><span class="co"># 4. Stable size distribution</span></span>
<span id="cb2341-3"><a href="integral-population-models.html#cb2341-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2341-4"><a href="integral-population-models.html#cb2341-4" tabindex="-1"></a>stable_output <span class="ot">&lt;-</span> ipmr<span class="sc">::</span><span class="fu">right_ev</span>(obs_ipm)</span>
<span id="cb2341-5"><a href="integral-population-models.html#cb2341-5" tabindex="-1"></a></span>
<span id="cb2341-6"><a href="integral-population-models.html#cb2341-6" tabindex="-1"></a>stable_vec <span class="ot">&lt;-</span> stable_output[[<span class="fu">paste0</span>(state_var, <span class="st">&quot;_w&quot;</span>)]]</span>
<span id="cb2341-7"><a href="integral-population-models.html#cb2341-7" tabindex="-1"></a>stable_vec <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(stable_vec)</span>
<span id="cb2341-8"><a href="integral-population-models.html#cb2341-8" tabindex="-1"></a></span>
<span id="cb2341-9"><a href="integral-population-models.html#cb2341-9" tabindex="-1"></a><span class="co"># Verify length</span></span>
<span id="cb2341-10"><a href="integral-population-models.html#cb2341-10" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(stable_vec) <span class="sc">!=</span> n_mesh) {</span>
<span id="cb2341-11"><a href="integral-population-models.html#cb2341-11" tabindex="-1"></a>  <span class="fu">stop</span>(<span class="st">&quot;Stable vector length (&quot;</span>, <span class="fu">length</span>(stable_vec), <span class="st">&quot;) != kernel dim (&quot;</span>, n_mesh, <span class="st">&quot;)&quot;</span>)</span>
<span id="cb2341-12"><a href="integral-population-models.html#cb2341-12" tabindex="-1"></a>}</span>
<span id="cb2341-13"><a href="integral-population-models.html#cb2341-13" tabindex="-1"></a></span>
<span id="cb2341-14"><a href="integral-population-models.html#cb2341-14" tabindex="-1"></a><span class="co"># Normalize (should already be 1)</span></span>
<span id="cb2341-15"><a href="integral-population-models.html#cb2341-15" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">isTRUE</span>(<span class="fu">all.equal</span>(<span class="fu">sum</span>(stable_vec), <span class="dv">1</span>))) {</span>
<span id="cb2341-16"><a href="integral-population-models.html#cb2341-16" tabindex="-1"></a>  stable_vec <span class="ot">&lt;-</span> stable_vec <span class="sc">/</span> <span class="fu">sum</span>(stable_vec)</span>
<span id="cb2341-17"><a href="integral-population-models.html#cb2341-17" tabindex="-1"></a>}</span>
<span id="cb2341-18"><a href="integral-population-models.html#cb2341-18" tabindex="-1"></a></span>
<span id="cb2341-19"><a href="integral-population-models.html#cb2341-19" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Stable distribution extracted (sum =&quot;</span>, <span class="fu">round</span>(<span class="fu">sum</span>(stable_vec), <span class="dv">6</span>), <span class="st">&quot;)</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Stable distribution extracted (sum = 1 )</code></pre>
<div class="sourceCode" id="cb2343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2343-1"><a href="integral-population-models.html#cb2343-1" tabindex="-1"></a><span class="co"># Mean size under stable distribution</span></span>
<span id="cb2343-2"><a href="integral-population-models.html#cb2343-2" tabindex="-1"></a>mean_size_stable_log <span class="ot">&lt;-</span> <span class="fu">sum</span>(size_classes <span class="sc">*</span> stable_vec)</span>
<span id="cb2343-3"><a href="integral-population-models.html#cb2343-3" tabindex="-1"></a>mean_size_stable_raw <span class="ot">&lt;-</span> <span class="fu">exp</span>(mean_size_stable_log)</span>
<span id="cb2343-4"><a href="integral-population-models.html#cb2343-4" tabindex="-1"></a></span>
<span id="cb2343-5"><a href="integral-population-models.html#cb2343-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Mean size (log):&quot;</span>, <span class="fu">round</span>(mean_size_stable_log, <span class="dv">3</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Mean size (log): -0.06</code></pre>
<div class="sourceCode" id="cb2345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2345-1"><a href="integral-population-models.html#cb2345-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Mean size (raw):&quot;</span>, <span class="fu">round</span>(mean_size_stable_raw, <span class="dv">2</span>), <span class="st">&quot;cm</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Mean size (raw): 0.94 cm</code></pre>
<div class="sourceCode" id="cb2347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2347-1"><a href="integral-population-models.html#cb2347-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2347-2"><a href="integral-population-models.html#cb2347-2" tabindex="-1"></a><span class="co"># 5. Reproductive value</span></span>
<span id="cb2347-3"><a href="integral-population-models.html#cb2347-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2347-4"><a href="integral-population-models.html#cb2347-4" tabindex="-1"></a>rv_output <span class="ot">&lt;-</span> ipmr<span class="sc">::</span><span class="fu">left_ev</span>(obs_ipm)</span>
<span id="cb2347-5"><a href="integral-population-models.html#cb2347-5" tabindex="-1"></a></span>
<span id="cb2347-6"><a href="integral-population-models.html#cb2347-6" tabindex="-1"></a><span class="co"># Try with suffix first, fallback to first element</span></span>
<span id="cb2347-7"><a href="integral-population-models.html#cb2347-7" tabindex="-1"></a>rv_vec <span class="ot">&lt;-</span> rv_output[[<span class="fu">paste0</span>(state_var, <span class="st">&quot;_w&quot;</span>)]]</span>
<span id="cb2347-8"><a href="integral-population-models.html#cb2347-8" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.null</span>(rv_vec)) rv_vec <span class="ot">&lt;-</span> rv_output[[<span class="dv">1</span>]]</span>
<span id="cb2347-9"><a href="integral-population-models.html#cb2347-9" tabindex="-1"></a></span>
<span id="cb2347-10"><a href="integral-population-models.html#cb2347-10" tabindex="-1"></a>rv_vec <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(rv_vec)</span>
<span id="cb2347-11"><a href="integral-population-models.html#cb2347-11" tabindex="-1"></a></span>
<span id="cb2347-12"><a href="integral-population-models.html#cb2347-12" tabindex="-1"></a><span class="co"># Verify length</span></span>
<span id="cb2347-13"><a href="integral-population-models.html#cb2347-13" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(rv_vec) <span class="sc">!=</span> n_mesh) {</span>
<span id="cb2347-14"><a href="integral-population-models.html#cb2347-14" tabindex="-1"></a>  <span class="fu">stop</span>(<span class="st">&quot;RV vector length (&quot;</span>, <span class="fu">length</span>(rv_vec), <span class="st">&quot;) != kernel dim (&quot;</span>, n_mesh, <span class="st">&quot;)&quot;</span>)</span>
<span id="cb2347-15"><a href="integral-population-models.html#cb2347-15" tabindex="-1"></a>}</span>
<span id="cb2347-16"><a href="integral-population-models.html#cb2347-16" tabindex="-1"></a></span>
<span id="cb2347-17"><a href="integral-population-models.html#cb2347-17" tabindex="-1"></a><span class="co"># Scale to max = 1</span></span>
<span id="cb2347-18"><a href="integral-population-models.html#cb2347-18" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">max</span>(rv_vec, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb2347-19"><a href="integral-population-models.html#cb2347-19" tabindex="-1"></a>  rv_vec <span class="ot">&lt;-</span> rv_vec <span class="sc">/</span> <span class="fu">max</span>(rv_vec, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2347-20"><a href="integral-population-models.html#cb2347-20" tabindex="-1"></a>}</span>
<span id="cb2347-21"><a href="integral-population-models.html#cb2347-21" tabindex="-1"></a></span>
<span id="cb2347-22"><a href="integral-population-models.html#cb2347-22" tabindex="-1"></a>max_rv <span class="ot">&lt;-</span> <span class="fu">max</span>(rv_vec, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2347-23"><a href="integral-population-models.html#cb2347-23" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Max reproductive value:&quot;</span>, <span class="fu">round</span>(max_rv, <span class="dv">3</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Max reproductive value: 1</code></pre>
<div class="sourceCode" id="cb2349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2349-1"><a href="integral-population-models.html#cb2349-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2349-2"><a href="integral-population-models.html#cb2349-2" tabindex="-1"></a><span class="co"># 6. Size-specific reproduction</span></span>
<span id="cb2349-3"><a href="integral-population-models.html#cb2349-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2349-4"><a href="integral-population-models.html#cb2349-4" tabindex="-1"></a>size_specific_repro <span class="ot">&lt;-</span> <span class="fu">colSums</span>(Fm)</span>
<span id="cb2349-5"><a href="integral-population-models.html#cb2349-5" tabindex="-1"></a></span>
<span id="cb2349-6"><a href="integral-population-models.html#cb2349-6" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">sum</span>(size_specific_repro) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb2349-7"><a href="integral-population-models.html#cb2349-7" tabindex="-1"></a>  mean_rep_obs_log <span class="ot">&lt;-</span> <span class="fu">sum</span>(size_classes <span class="sc">*</span> size_specific_repro) <span class="sc">/</span> <span class="fu">sum</span>(size_specific_repro)</span>
<span id="cb2349-8"><a href="integral-population-models.html#cb2349-8" tabindex="-1"></a>  mean_rep_obs_raw <span class="ot">&lt;-</span> <span class="fu">exp</span>(mean_rep_obs_log)</span>
<span id="cb2349-9"><a href="integral-population-models.html#cb2349-9" tabindex="-1"></a>  var_rep_obs_log  <span class="ot">&lt;-</span> <span class="fu">sum</span>((size_classes <span class="sc">-</span> mean_rep_obs_log)<span class="sc">^</span><span class="dv">2</span> <span class="sc">*</span> size_specific_repro) <span class="sc">/</span> </span>
<span id="cb2349-10"><a href="integral-population-models.html#cb2349-10" tabindex="-1"></a>                      <span class="fu">sum</span>(size_specific_repro)</span>
<span id="cb2349-11"><a href="integral-population-models.html#cb2349-11" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb2349-12"><a href="integral-population-models.html#cb2349-12" tabindex="-1"></a>  mean_rep_obs_log <span class="ot">&lt;-</span> <span class="cn">NA_real_</span></span>
<span id="cb2349-13"><a href="integral-population-models.html#cb2349-13" tabindex="-1"></a>  mean_rep_obs_raw <span class="ot">&lt;-</span> <span class="cn">NA_real_</span></span>
<span id="cb2349-14"><a href="integral-population-models.html#cb2349-14" tabindex="-1"></a>  var_rep_obs_log  <span class="ot">&lt;-</span> <span class="cn">NA_real_</span></span>
<span id="cb2349-15"><a href="integral-population-models.html#cb2349-15" tabindex="-1"></a>}</span>
<span id="cb2349-16"><a href="integral-population-models.html#cb2349-16" tabindex="-1"></a></span>
<span id="cb2349-17"><a href="integral-population-models.html#cb2349-17" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Mean reproductive size (log):&quot;</span>, <span class="fu">round</span>(mean_rep_obs_log, <span class="dv">3</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Mean reproductive size (log): 4.725</code></pre>
<div class="sourceCode" id="cb2351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2351-1"><a href="integral-population-models.html#cb2351-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Mean reproductive size (raw):&quot;</span>, <span class="fu">round</span>(mean_rep_obs_raw, <span class="dv">2</span>), <span class="st">&quot;cm</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Mean reproductive size (raw): 112.73 cm</code></pre>
<div class="sourceCode" id="cb2353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2353-1"><a href="integral-population-models.html#cb2353-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2353-2"><a href="integral-population-models.html#cb2353-2" tabindex="-1"></a><span class="co"># 7. Expected remaining lifetime by size</span></span>
<span id="cb2353-3"><a href="integral-population-models.html#cb2353-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2353-4"><a href="integral-population-models.html#cb2353-4" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== Computing Expected Remaining Lifetime ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === Computing Expected Remaining Lifetime ===</code></pre>
<div class="sourceCode" id="cb2355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2355-1"><a href="integral-population-models.html#cb2355-1" tabindex="-1"></a><span class="co"># Check P eigenvalue</span></span>
<span id="cb2355-2"><a href="integral-population-models.html#cb2355-2" tabindex="-1"></a>P_eigs <span class="ot">&lt;-</span> <span class="fu">eigen</span>(Pm)<span class="sc">$</span>values</span>
<span id="cb2355-3"><a href="integral-population-models.html#cb2355-3" tabindex="-1"></a>P_lambda <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">Re</span>(P_eigs))</span>
<span id="cb2355-4"><a href="integral-population-models.html#cb2355-4" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;P kernel eigenvalue:&quot;</span>, <span class="fu">round</span>(P_lambda, <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## P kernel eigenvalue: 0.9694</code></pre>
<div class="sourceCode" id="cb2357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2357-1"><a href="integral-population-models.html#cb2357-1" tabindex="-1"></a><span class="cf">if</span>(P_lambda <span class="sc">&gt;=</span> <span class="fl">0.99</span>) {</span>
<span id="cb2357-2"><a href="integral-population-models.html#cb2357-2" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;⚠️  High P eigenvalue (&gt;0.99) - expected lifetimes may be very long</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2357-3"><a href="integral-population-models.html#cb2357-3" tabindex="-1"></a>}</span>
<span id="cb2357-4"><a href="integral-population-models.html#cb2357-4" tabindex="-1"></a></span>
<span id="cb2357-5"><a href="integral-population-models.html#cb2357-5" tabindex="-1"></a>expected_remaining_life <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, n_mesh)</span>
<span id="cb2357-6"><a href="integral-population-models.html#cb2357-6" tabindex="-1"></a>v <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span>, n_mesh)</span>
<span id="cb2357-7"><a href="integral-population-models.html#cb2357-7" tabindex="-1"></a></span>
<span id="cb2357-8"><a href="integral-population-models.html#cb2357-8" tabindex="-1"></a>tol <span class="ot">&lt;-</span> <span class="fl">1e-8</span></span>
<span id="cb2357-9"><a href="integral-population-models.html#cb2357-9" tabindex="-1"></a>max_iter <span class="ot">&lt;-</span> <span class="dv">5000</span></span>
<span id="cb2357-10"><a href="integral-population-models.html#cb2357-10" tabindex="-1"></a>converged <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb2357-11"><a href="integral-population-models.html#cb2357-11" tabindex="-1"></a></span>
<span id="cb2357-12"><a href="integral-population-models.html#cb2357-12" tabindex="-1"></a><span class="cf">for</span>(iter <span class="cf">in</span> <span class="fu">seq_len</span>(max_iter)) {</span>
<span id="cb2357-13"><a href="integral-population-models.html#cb2357-13" tabindex="-1"></a>  expected_remaining_life <span class="ot">&lt;-</span> expected_remaining_life <span class="sc">+</span> v</span>
<span id="cb2357-14"><a href="integral-population-models.html#cb2357-14" tabindex="-1"></a>  v_new <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(Pm <span class="sc">%*%</span> v)</span>
<span id="cb2357-15"><a href="integral-population-models.html#cb2357-15" tabindex="-1"></a></span>
<span id="cb2357-16"><a href="integral-population-models.html#cb2357-16" tabindex="-1"></a>  diff <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">abs</span>(v_new <span class="sc">-</span> v), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2357-17"><a href="integral-population-models.html#cb2357-17" tabindex="-1"></a>  </span>
<span id="cb2357-18"><a href="integral-population-models.html#cb2357-18" tabindex="-1"></a>  <span class="cf">if</span>(diff <span class="sc">&lt;</span> tol) {</span>
<span id="cb2357-19"><a href="integral-population-models.html#cb2357-19" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;✓ Converged after&quot;</span>, iter, <span class="st">&quot;iterations</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2357-20"><a href="integral-population-models.html#cb2357-20" tabindex="-1"></a>    converged <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb2357-21"><a href="integral-population-models.html#cb2357-21" tabindex="-1"></a>    <span class="cf">break</span></span>
<span id="cb2357-22"><a href="integral-population-models.html#cb2357-22" tabindex="-1"></a>  }</span>
<span id="cb2357-23"><a href="integral-population-models.html#cb2357-23" tabindex="-1"></a></span>
<span id="cb2357-24"><a href="integral-population-models.html#cb2357-24" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">any</span>(<span class="sc">!</span><span class="fu">is.finite</span>(v_new)) <span class="sc">||</span> <span class="fu">max</span>(v_new) <span class="sc">&gt;</span> <span class="fl">1e6</span>) {</span>
<span id="cb2357-25"><a href="integral-population-models.html#cb2357-25" tabindex="-1"></a>    <span class="fu">warning</span>(<span class="st">&quot;⚠️  Lifetime calculation diverging at iteration &quot;</span>, iter)</span>
<span id="cb2357-26"><a href="integral-population-models.html#cb2357-26" tabindex="-1"></a>    <span class="cf">break</span></span>
<span id="cb2357-27"><a href="integral-population-models.html#cb2357-27" tabindex="-1"></a>  }</span>
<span id="cb2357-28"><a href="integral-population-models.html#cb2357-28" tabindex="-1"></a></span>
<span id="cb2357-29"><a href="integral-population-models.html#cb2357-29" tabindex="-1"></a>  v <span class="ot">&lt;-</span> v_new</span>
<span id="cb2357-30"><a href="integral-population-models.html#cb2357-30" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## ✓ Converged after 576 iterations</code></pre>
<div class="sourceCode" id="cb2359"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2359-1"><a href="integral-population-models.html#cb2359-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span>converged) {</span>
<span id="cb2359-2"><a href="integral-population-models.html#cb2359-2" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;⚠️  Did not fully converge after&quot;</span>, max_iter, <span class="st">&quot;iterations</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2359-3"><a href="integral-population-models.html#cb2359-3" tabindex="-1"></a>}</span>
<span id="cb2359-4"><a href="integral-population-models.html#cb2359-4" tabindex="-1"></a></span>
<span id="cb2359-5"><a href="integral-population-models.html#cb2359-5" tabindex="-1"></a>max_life <span class="ot">&lt;-</span> <span class="fu">max</span>(expected_remaining_life)</span>
<span id="cb2359-6"><a href="integral-population-models.html#cb2359-6" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Max expected lifetime:&quot;</span>, <span class="fu">round</span>(max_life, <span class="dv">2</span>), <span class="st">&quot;time steps</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Max expected lifetime: 544.83 time steps</code></pre>
<div class="sourceCode" id="cb2361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2361-1"><a href="integral-population-models.html#cb2361-1" tabindex="-1"></a><span class="cf">if</span>(max_life <span class="sc">&gt;</span> <span class="dv">100</span>) {</span>
<span id="cb2361-2"><a href="integral-population-models.html#cb2361-2" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;⚠️  Very high lifetime (&gt;100 years) - survival model may be overfitting</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2361-3"><a href="integral-population-models.html#cb2361-3" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## ⚠️  Very high lifetime (&gt;100 years) - survival model may be overfitting</code></pre>
<div class="sourceCode" id="cb2363"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2363-1"><a href="integral-population-models.html#cb2363-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2363-2"><a href="integral-population-models.html#cb2363-2" tabindex="-1"></a><span class="co"># 8. Median size (weighted by stable distribution)</span></span>
<span id="cb2363-3"><a href="integral-population-models.html#cb2363-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2363-4"><a href="integral-population-models.html#cb2363-4" tabindex="-1"></a>cumulative_dist <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(stable_vec)</span>
<span id="cb2363-5"><a href="integral-population-models.html#cb2363-5" tabindex="-1"></a>median_index <span class="ot">&lt;-</span> <span class="fu">which</span>(cumulative_dist <span class="sc">&gt;=</span> <span class="fl">0.5</span>)[<span class="dv">1</span>]</span>
<span id="cb2363-6"><a href="integral-population-models.html#cb2363-6" tabindex="-1"></a></span>
<span id="cb2363-7"><a href="integral-population-models.html#cb2363-7" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.na</span>(median_index)) {</span>
<span id="cb2363-8"><a href="integral-population-models.html#cb2363-8" tabindex="-1"></a>  median_index <span class="ot">&lt;-</span> <span class="fu">floor</span>(n_mesh <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb2363-9"><a href="integral-population-models.html#cb2363-9" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;⚠️  Could not find weighted median, using middle bin</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb2363-10"><a href="integral-population-models.html#cb2363-10" tabindex="-1"></a>}</span>
<span id="cb2363-11"><a href="integral-population-models.html#cb2363-11" tabindex="-1"></a></span>
<span id="cb2363-12"><a href="integral-population-models.html#cb2363-12" tabindex="-1"></a>median_size_log <span class="ot">&lt;-</span> size_classes[median_index]</span>
<span id="cb2363-13"><a href="integral-population-models.html#cb2363-13" tabindex="-1"></a>median_size_raw <span class="ot">&lt;-</span> <span class="fu">exp</span>(median_size_log)</span>
<span id="cb2363-14"><a href="integral-population-models.html#cb2363-14" tabindex="-1"></a>life_exp_median_obs <span class="ot">&lt;-</span> expected_remaining_life[median_index]</span>
<span id="cb2363-15"><a href="integral-population-models.html#cb2363-15" tabindex="-1"></a></span>
<span id="cb2363-16"><a href="integral-population-models.html#cb2363-16" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Median size (log):&quot;</span>, <span class="fu">round</span>(median_size_log, <span class="dv">3</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Median size (log): -0.112</code></pre>
<div class="sourceCode" id="cb2365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2365-1"><a href="integral-population-models.html#cb2365-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Median size (raw):&quot;</span>, <span class="fu">round</span>(median_size_raw, <span class="dv">2</span>), <span class="st">&quot;cm</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Median size (raw): 0.89 cm</code></pre>
<div class="sourceCode" id="cb2367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2367-1"><a href="integral-population-models.html#cb2367-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Expected lifetime at median:&quot;</span>, <span class="fu">round</span>(life_exp_median_obs, <span class="dv">2</span>), <span class="st">&quot;time steps</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Expected lifetime at median: 322.62 time steps</code></pre>
<div class="sourceCode" id="cb2369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2369-1"><a href="integral-population-models.html#cb2369-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2369-2"><a href="integral-population-models.html#cb2369-2" tabindex="-1"></a><span class="co"># 9. Summary table</span></span>
<span id="cb2369-3"><a href="integral-population-models.html#cb2369-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2369-4"><a href="integral-population-models.html#cb2369-4" tabindex="-1"></a>summary_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2369-5"><a href="integral-population-models.html#cb2369-5" tabindex="-1"></a>  <span class="at">Metric =</span> <span class="fu">c</span>(</span>
<span id="cb2369-6"><a href="integral-population-models.html#cb2369-6" tabindex="-1"></a>    <span class="st">&quot;Population growth rate (λ)&quot;</span>,</span>
<span id="cb2369-7"><a href="integral-population-models.html#cb2369-7" tabindex="-1"></a>    <span class="st">&quot;Mean size (stable, log)&quot;</span>,</span>
<span id="cb2369-8"><a href="integral-population-models.html#cb2369-8" tabindex="-1"></a>    <span class="st">&quot;Mean size (stable, cm)&quot;</span>,</span>
<span id="cb2369-9"><a href="integral-population-models.html#cb2369-9" tabindex="-1"></a>    <span class="st">&quot;Median size (stable, log)&quot;</span>,</span>
<span id="cb2369-10"><a href="integral-population-models.html#cb2369-10" tabindex="-1"></a>    <span class="st">&quot;Median size (stable, cm)&quot;</span>,</span>
<span id="cb2369-11"><a href="integral-population-models.html#cb2369-11" tabindex="-1"></a>    <span class="st">&quot;Mean reproductive size (log)&quot;</span>,</span>
<span id="cb2369-12"><a href="integral-population-models.html#cb2369-12" tabindex="-1"></a>    <span class="st">&quot;Mean reproductive size (cm)&quot;</span>,</span>
<span id="cb2369-13"><a href="integral-population-models.html#cb2369-13" tabindex="-1"></a>    <span class="st">&quot;Variance reproductive size (log)&quot;</span>,</span>
<span id="cb2369-14"><a href="integral-population-models.html#cb2369-14" tabindex="-1"></a>    <span class="st">&quot;Expected lifetime at median&quot;</span>,</span>
<span id="cb2369-15"><a href="integral-population-models.html#cb2369-15" tabindex="-1"></a>    <span class="st">&quot;Max reproductive value&quot;</span></span>
<span id="cb2369-16"><a href="integral-population-models.html#cb2369-16" tabindex="-1"></a>  ),</span>
<span id="cb2369-17"><a href="integral-population-models.html#cb2369-17" tabindex="-1"></a>  <span class="at">Estimate =</span> <span class="fu">round</span>(<span class="fu">c</span>(</span>
<span id="cb2369-18"><a href="integral-population-models.html#cb2369-18" tabindex="-1"></a>    lambda_obs,</span>
<span id="cb2369-19"><a href="integral-population-models.html#cb2369-19" tabindex="-1"></a>    mean_size_stable_log,</span>
<span id="cb2369-20"><a href="integral-population-models.html#cb2369-20" tabindex="-1"></a>    mean_size_stable_raw,</span>
<span id="cb2369-21"><a href="integral-population-models.html#cb2369-21" tabindex="-1"></a>    median_size_log,</span>
<span id="cb2369-22"><a href="integral-population-models.html#cb2369-22" tabindex="-1"></a>    median_size_raw,</span>
<span id="cb2369-23"><a href="integral-population-models.html#cb2369-23" tabindex="-1"></a>    mean_rep_obs_log,</span>
<span id="cb2369-24"><a href="integral-population-models.html#cb2369-24" tabindex="-1"></a>    mean_rep_obs_raw,</span>
<span id="cb2369-25"><a href="integral-population-models.html#cb2369-25" tabindex="-1"></a>    var_rep_obs_log,</span>
<span id="cb2369-26"><a href="integral-population-models.html#cb2369-26" tabindex="-1"></a>    life_exp_median_obs,</span>
<span id="cb2369-27"><a href="integral-population-models.html#cb2369-27" tabindex="-1"></a>    max_rv</span>
<span id="cb2369-28"><a href="integral-population-models.html#cb2369-28" tabindex="-1"></a>  ), <span class="dv">4</span>)</span>
<span id="cb2369-29"><a href="integral-population-models.html#cb2369-29" tabindex="-1"></a>)</span>
<span id="cb2369-30"><a href="integral-population-models.html#cb2369-30" tabindex="-1"></a></span>
<span id="cb2369-31"><a href="integral-population-models.html#cb2369-31" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">=== LIFE-HISTORY SUMMARY ===</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## === LIFE-HISTORY SUMMARY ===</code></pre>
<div class="sourceCode" id="cb2371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2371-1"><a href="integral-population-models.html#cb2371-1" tabindex="-1"></a><span class="fu">print</span>(summary_table, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>##                            Metric Estimate
##        Population growth rate (λ)   0.9694
##           Mean size (stable, log)  -0.0597
##            Mean size (stable, cm)   0.9421
##         Median size (stable, log)  -0.1115
##          Median size (stable, cm)   0.8945
##      Mean reproductive size (log)   4.7250
##       Mean reproductive size (cm) 112.7281
##  Variance reproductive size (log)   0.0790
##       Expected lifetime at median 322.6221
##            Max reproductive value   1.0000</code></pre>
<div class="sourceCode" id="cb2373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2373-1"><a href="integral-population-models.html#cb2373-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2373-2"><a href="integral-population-models.html#cb2373-2" tabindex="-1"></a><span class="co"># 10. Visualizations</span></span>
<span id="cb2373-3"><a href="integral-population-models.html#cb2373-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2373-4"><a href="integral-population-models.html#cb2373-4" tabindex="-1"></a></span>
<span id="cb2373-5"><a href="integral-population-models.html#cb2373-5" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">4.5</span>, <span class="fl">4.5</span>, <span class="dv">3</span>, <span class="dv">1</span>))</span>
<span id="cb2373-6"><a href="integral-population-models.html#cb2373-6" tabindex="-1"></a></span>
<span id="cb2373-7"><a href="integral-population-models.html#cb2373-7" tabindex="-1"></a><span class="co"># Plot 1: Expected lifetime vs log size</span></span>
<span id="cb2373-8"><a href="integral-population-models.html#cb2373-8" tabindex="-1"></a><span class="fu">plot</span>(size_classes, expected_remaining_life, </span>
<span id="cb2373-9"><a href="integral-population-models.html#cb2373-9" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">&quot;l&quot;</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="st">&quot;darkblue&quot;</span>,</span>
<span id="cb2373-10"><a href="integral-population-models.html#cb2373-10" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Size (log scale)&quot;</span>, </span>
<span id="cb2373-11"><a href="integral-population-models.html#cb2373-11" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;Expected remaining lifetime (years)&quot;</span>,</span>
<span id="cb2373-12"><a href="integral-population-models.html#cb2373-12" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;A. Expected Lifetime by Size (Log)&quot;</span>,</span>
<span id="cb2373-13"><a href="integral-population-models.html#cb2373-13" tabindex="-1"></a>     <span class="at">cex.lab =</span> <span class="fl">1.2</span>, <span class="at">cex.main =</span> <span class="fl">1.3</span>)</span>
<span id="cb2373-14"><a href="integral-population-models.html#cb2373-14" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> median_size_log, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb2373-15"><a href="integral-population-models.html#cb2373-15" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> mean_size_stable_log, <span class="at">col =</span> <span class="st">&quot;green&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb2373-16"><a href="integral-population-models.html#cb2373-16" tabindex="-1"></a><span class="fu">text</span>(median_size_log, <span class="fu">max</span>(expected_remaining_life) <span class="sc">*</span> <span class="fl">0.9</span>,</span>
<span id="cb2373-17"><a href="integral-population-models.html#cb2373-17" tabindex="-1"></a>     <span class="fu">paste0</span>(<span class="st">&quot;Median</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="fu">round</span>(life_exp_median_obs, <span class="dv">1</span>), <span class="st">&quot; yr&quot;</span>),</span>
<span id="cb2373-18"><a href="integral-population-models.html#cb2373-18" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">4</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">font =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">0.9</span>)</span>
<span id="cb2373-19"><a href="integral-population-models.html#cb2373-19" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, </span>
<span id="cb2373-20"><a href="integral-population-models.html#cb2373-20" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;Median&quot;</span>, <span class="st">&quot;Mean&quot;</span>),</span>
<span id="cb2373-21"><a href="integral-population-models.html#cb2373-21" tabindex="-1"></a>       <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>),</span>
<span id="cb2373-22"><a href="integral-population-models.html#cb2373-22" tabindex="-1"></a>       <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb2373-23"><a href="integral-population-models.html#cb2373-23" tabindex="-1"></a><span class="fu">grid</span>()</span>
<span id="cb2373-24"><a href="integral-population-models.html#cb2373-24" tabindex="-1"></a></span>
<span id="cb2373-25"><a href="integral-population-models.html#cb2373-25" tabindex="-1"></a><span class="co"># Plot 2: Expected lifetime vs raw size</span></span>
<span id="cb2373-26"><a href="integral-population-models.html#cb2373-26" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">exp</span>(size_classes), expected_remaining_life, </span>
<span id="cb2373-27"><a href="integral-population-models.html#cb2373-27" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">&quot;l&quot;</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="st">&quot;darkblue&quot;</span>,</span>
<span id="cb2373-28"><a href="integral-population-models.html#cb2373-28" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Size (cm)&quot;</span>, </span>
<span id="cb2373-29"><a href="integral-population-models.html#cb2373-29" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;Expected remaining lifetime (years)&quot;</span>,</span>
<span id="cb2373-30"><a href="integral-population-models.html#cb2373-30" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;B. Expected Lifetime by Size (Raw)&quot;</span>,</span>
<span id="cb2373-31"><a href="integral-population-models.html#cb2373-31" tabindex="-1"></a>     <span class="at">cex.lab =</span> <span class="fl">1.2</span>, <span class="at">cex.main =</span> <span class="fl">1.3</span>)</span>
<span id="cb2373-32"><a href="integral-population-models.html#cb2373-32" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> median_size_raw, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb2373-33"><a href="integral-population-models.html#cb2373-33" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> mean_size_stable_raw, <span class="at">col =</span> <span class="st">&quot;green&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb2373-34"><a href="integral-population-models.html#cb2373-34" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, </span>
<span id="cb2373-35"><a href="integral-population-models.html#cb2373-35" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;Median&quot;</span>, <span class="st">&quot;Mean&quot;</span>),</span>
<span id="cb2373-36"><a href="integral-population-models.html#cb2373-36" tabindex="-1"></a>       <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>),</span>
<span id="cb2373-37"><a href="integral-population-models.html#cb2373-37" tabindex="-1"></a>       <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb2373-38"><a href="integral-population-models.html#cb2373-38" tabindex="-1"></a><span class="fu">grid</span>()</span>
<span id="cb2373-39"><a href="integral-population-models.html#cb2373-39" tabindex="-1"></a></span>
<span id="cb2373-40"><a href="integral-population-models.html#cb2373-40" tabindex="-1"></a><span class="co"># Plot 3: Stable size distribution</span></span>
<span id="cb2373-41"><a href="integral-population-models.html#cb2373-41" tabindex="-1"></a><span class="fu">plot</span>(size_classes, stable_vec, </span>
<span id="cb2373-42"><a href="integral-population-models.html#cb2373-42" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">&quot;l&quot;</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="st">&quot;purple&quot;</span>,</span>
<span id="cb2373-43"><a href="integral-population-models.html#cb2373-43" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Size (log scale)&quot;</span>, </span>
<span id="cb2373-44"><a href="integral-population-models.html#cb2373-44" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;Proportion&quot;</span>,</span>
<span id="cb2373-45"><a href="integral-population-models.html#cb2373-45" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;C. Stable Size Distribution&quot;</span>,</span>
<span id="cb2373-46"><a href="integral-population-models.html#cb2373-46" tabindex="-1"></a>     <span class="at">cex.lab =</span> <span class="fl">1.2</span>, <span class="at">cex.main =</span> <span class="fl">1.3</span>)</span>
<span id="cb2373-47"><a href="integral-population-models.html#cb2373-47" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> median_size_log, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb2373-48"><a href="integral-population-models.html#cb2373-48" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> mean_size_stable_log, <span class="at">col =</span> <span class="st">&quot;green&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb2373-49"><a href="integral-population-models.html#cb2373-49" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, </span>
<span id="cb2373-50"><a href="integral-population-models.html#cb2373-50" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;Median&quot;</span>, <span class="st">&quot;Mean&quot;</span>),</span>
<span id="cb2373-51"><a href="integral-population-models.html#cb2373-51" tabindex="-1"></a>       <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>),</span>
<span id="cb2373-52"><a href="integral-population-models.html#cb2373-52" tabindex="-1"></a>       <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb2373-53"><a href="integral-population-models.html#cb2373-53" tabindex="-1"></a><span class="fu">grid</span>()</span>
<span id="cb2373-54"><a href="integral-population-models.html#cb2373-54" tabindex="-1"></a></span>
<span id="cb2373-55"><a href="integral-population-models.html#cb2373-55" tabindex="-1"></a><span class="co"># Plot 4: Reproductive value</span></span>
<span id="cb2373-56"><a href="integral-population-models.html#cb2373-56" tabindex="-1"></a><span class="fu">plot</span>(size_classes, rv_vec, </span>
<span id="cb2373-57"><a href="integral-population-models.html#cb2373-57" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">&quot;l&quot;</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="st">&quot;orange&quot;</span>,</span>
<span id="cb2373-58"><a href="integral-population-models.html#cb2373-58" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Size (log scale)&quot;</span>, </span>
<span id="cb2373-59"><a href="integral-population-models.html#cb2373-59" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;Reproductive value (scaled)&quot;</span>,</span>
<span id="cb2373-60"><a href="integral-population-models.html#cb2373-60" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;D. Reproductive Value by Size&quot;</span>,</span>
<span id="cb2373-61"><a href="integral-population-models.html#cb2373-61" tabindex="-1"></a>     <span class="at">cex.lab =</span> <span class="fl">1.2</span>, <span class="at">cex.main =</span> <span class="fl">1.3</span>)</span>
<span id="cb2373-62"><a href="integral-population-models.html#cb2373-62" tabindex="-1"></a><span class="fu">grid</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-30-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb2374"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2374-1"><a href="integral-population-models.html#cb2374-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode" id="cb2376"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2376-1"><a href="integral-population-models.html#cb2376-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb2376-2"><a href="integral-population-models.html#cb2376-2" tabindex="-1"></a></span>
<span id="cb2376-3"><a href="integral-population-models.html#cb2376-3" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">✓ Figure saved!</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## ✓ Figure saved!</code></pre>
<div class="sourceCode" id="cb2378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2378-1"><a href="integral-population-models.html#cb2378-1" tabindex="-1"></a>life_exp_detailed <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2378-2"><a href="integral-population-models.html#cb2378-2" tabindex="-1"></a>  <span class="at">size_log =</span> size_classes,</span>
<span id="cb2378-3"><a href="integral-population-models.html#cb2378-3" tabindex="-1"></a>  <span class="at">size_raw_cm =</span> <span class="fu">exp</span>(size_classes),</span>
<span id="cb2378-4"><a href="integral-population-models.html#cb2378-4" tabindex="-1"></a>  <span class="at">expected_remaining_lifetime =</span> expected_remaining_life,</span>
<span id="cb2378-5"><a href="integral-population-models.html#cb2378-5" tabindex="-1"></a>  <span class="at">stable_distribution =</span> stable_vec,</span>
<span id="cb2378-6"><a href="integral-population-models.html#cb2378-6" tabindex="-1"></a>  <span class="at">reproductive_value =</span> rv_vec</span>
<span id="cb2378-7"><a href="integral-population-models.html#cb2378-7" tabindex="-1"></a>)</span>
<span id="cb2378-8"><a href="integral-population-models.html#cb2378-8" tabindex="-1"></a></span>
<span id="cb2378-9"><a href="integral-population-models.html#cb2378-9" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ All results exported successfully!</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ All results exported successfully!</code></pre>
<p>Interesting! This figure is showing us size-dependent survival and its consequences for expected remaining lifetime in a size-structured Integral Projection Model (IPM). Each point shows the observed survival outcome (0 = died, 1 = survived) for an individual as a function of size at time t. Survival probability is high for small individuals but declines markedly at larger sizes, indicating that large individuals experience elevated mortality risk. This non-monotonic survival–size relationship directly shapes the expected remaining lifetime (ERL) curve derived from the IPM. Individuals at small sizes have long expected remaining lifetimes because they persist with high survival across multiple time steps, even if growth is slow. In contrast, individuals at larger sizes exhibit short expected remaining lifetimes, reflecting increased mortality once large size is reached. The dashed vertical line indicates the median size class, at which survival is approximately 50%, yielding an ERL near one time step. This pattern illustrates that, in size-structured populations, longevity is not necessarily associated with large size; instead, long expected lifetimes can arise from prolonged persistence at small sizes. The IPM faithfully translates this empirically observed survival–size relationship into emergent demographic properties, rather than imposing assumptions about monotonic size–survival relationships.</p>
<p>Now, let’s calculate transient dynamics! We have to use the original size distribution, and walk through an individual time step (rather than run to asymptotic population growth / stable size distribution).</p>
<div class="sourceCode" id="cb2380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2380-1"><a href="integral-population-models.html#cb2380-1" tabindex="-1"></a><span class="co">#==============================================================</span></span>
<span id="cb2380-2"><a href="integral-population-models.html#cb2380-2" tabindex="-1"></a><span class="co"># TRANSIENT DYNAMICS (LOG-SCALE STATE VARIABLE)</span></span>
<span id="cb2380-3"><a href="integral-population-models.html#cb2380-3" tabindex="-1"></a><span class="co">#==============================================================</span></span>
<span id="cb2380-4"><a href="integral-population-models.html#cb2380-4" tabindex="-1"></a><span class="fu">library</span>(ipmr)</span>
<span id="cb2380-5"><a href="integral-population-models.html#cb2380-5" tabindex="-1"></a></span>
<span id="cb2380-6"><a href="integral-population-models.html#cb2380-6" tabindex="-1"></a> <span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2380-7"><a href="integral-population-models.html#cb2380-7" tabindex="-1"></a><span class="co"># 1) Extract mesh + kernels (ROBUST to ipmr storage details)</span></span>
<span id="cb2380-8"><a href="integral-population-models.html#cb2380-8" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2380-9"><a href="integral-population-models.html#cb2380-9" tabindex="-1"></a></span>
<span id="cb2380-10"><a href="integral-population-models.html#cb2380-10" tabindex="-1"></a>P <span class="ot">&lt;-</span> obs_ipm<span class="sc">$</span>sub_kernels<span class="sc">$</span>P</span>
<span id="cb2380-11"><a href="integral-population-models.html#cb2380-11" tabindex="-1"></a>F <span class="ot">&lt;-</span> obs_ipm<span class="sc">$</span>sub_kernels<span class="sc">$</span>F</span>
<span id="cb2380-12"><a href="integral-population-models.html#cb2380-12" tabindex="-1"></a>K <span class="ot">&lt;-</span> P <span class="sc">+</span> F</span>
<span id="cb2380-13"><a href="integral-population-models.html#cb2380-13" tabindex="-1"></a></span>
<span id="cb2380-14"><a href="integral-population-models.html#cb2380-14" tabindex="-1"></a><span class="co"># The discretized matrix dimension is the truth</span></span>
<span id="cb2380-15"><a href="integral-population-models.html#cb2380-15" tabindex="-1"></a>n_mesh <span class="ot">&lt;-</span> <span class="fu">nrow</span>(K)</span>
<span id="cb2380-16"><a href="integral-population-models.html#cb2380-16" tabindex="-1"></a></span>
<span id="cb2380-17"><a href="integral-population-models.html#cb2380-17" tabindex="-1"></a><span class="co"># ---- Try to recover the mesh points in a way that matches K ----</span></span>
<span id="cb2380-18"><a href="integral-population-models.html#cb2380-18" tabindex="-1"></a><span class="co"># 1) Best case: row/col names are the mesh points</span></span>
<span id="cb2380-19"><a href="integral-population-models.html#cb2380-19" tabindex="-1"></a>size_classes <span class="ot">&lt;-</span> <span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(<span class="fu">rownames</span>(K)))</span>
<span id="cb2380-20"><a href="integral-population-models.html#cb2380-20" tabindex="-1"></a></span>
<span id="cb2380-21"><a href="integral-population-models.html#cb2380-21" tabindex="-1"></a><span class="co"># 2) If rownames aren&#39;t numeric, try colnames</span></span>
<span id="cb2380-22"><a href="integral-population-models.html#cb2380-22" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(size_classes) <span class="sc">!=</span> n_mesh <span class="sc">||</span> <span class="fu">anyNA</span>(size_classes)) {</span>
<span id="cb2380-23"><a href="integral-population-models.html#cb2380-23" tabindex="-1"></a>  size_classes <span class="ot">&lt;-</span> <span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(<span class="fu">colnames</span>(K)))</span>
<span id="cb2380-24"><a href="integral-population-models.html#cb2380-24" tabindex="-1"></a>}</span>
<span id="cb2380-25"><a href="integral-population-models.html#cb2380-25" tabindex="-1"></a></span>
<span id="cb2380-26"><a href="integral-population-models.html#cb2380-26" tabindex="-1"></a><span class="co"># 3) If still not valid, pull from ipmr environment (sa_1)</span></span>
<span id="cb2380-27"><a href="integral-population-models.html#cb2380-27" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(size_classes) <span class="sc">!=</span> n_mesh <span class="sc">||</span> <span class="fu">anyNA</span>(size_classes)) {</span>
<span id="cb2380-28"><a href="integral-population-models.html#cb2380-28" tabindex="-1"></a>  size_classes <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2380-29"><a href="integral-population-models.html#cb2380-29" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(obs_ipm<span class="sc">$</span>env_list<span class="sc">$</span>main_env) <span class="sc">&amp;&amp;</span></span>
<span id="cb2380-30"><a href="integral-population-models.html#cb2380-30" tabindex="-1"></a>      <span class="fu">exists</span>(<span class="st">&quot;sa_1&quot;</span>, <span class="at">envir =</span> obs_ipm<span class="sc">$</span>env_list<span class="sc">$</span>main_env, <span class="at">inherits =</span> <span class="cn">FALSE</span>)) {</span>
<span id="cb2380-31"><a href="integral-population-models.html#cb2380-31" tabindex="-1"></a>    sa_1_full <span class="ot">&lt;-</span> <span class="fu">get</span>(<span class="st">&quot;sa_1&quot;</span>, <span class="at">envir =</span> obs_ipm<span class="sc">$</span>env_list<span class="sc">$</span>main_env)</span>
<span id="cb2380-32"><a href="integral-population-models.html#cb2380-32" tabindex="-1"></a>    cand <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(<span class="fu">as.numeric</span>(sa_1_full)))</span>
<span id="cb2380-33"><a href="integral-population-models.html#cb2380-33" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(cand) <span class="sc">==</span> n_mesh) size_classes <span class="ot">&lt;-</span> cand</span>
<span id="cb2380-34"><a href="integral-population-models.html#cb2380-34" tabindex="-1"></a>  }</span>
<span id="cb2380-35"><a href="integral-population-models.html#cb2380-35" tabindex="-1"></a>}</span>
<span id="cb2380-36"><a href="integral-population-models.html#cb2380-36" tabindex="-1"></a></span>
<span id="cb2380-37"><a href="integral-population-models.html#cb2380-37" tabindex="-1"></a><span class="co"># 4) Last resort: reconstruct evenly spaced mesh from your domain bounds L, U</span></span>
<span id="cb2380-38"><a href="integral-population-models.html#cb2380-38" tabindex="-1"></a><span class="co">#    (works as long as you still have L and U in your workspace)</span></span>
<span id="cb2380-39"><a href="integral-population-models.html#cb2380-39" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">is.null</span>(size_classes) <span class="sc">||</span> <span class="fu">length</span>(size_classes) <span class="sc">!=</span> n_mesh <span class="sc">||</span> <span class="fu">anyNA</span>(size_classes)) {</span>
<span id="cb2380-40"><a href="integral-population-models.html#cb2380-40" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">exists</span>(<span class="st">&quot;L&quot;</span>, <span class="at">inherits =</span> <span class="cn">TRUE</span>) <span class="sc">||</span> <span class="sc">!</span><span class="fu">exists</span>(<span class="st">&quot;U&quot;</span>, <span class="at">inherits =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb2380-41"><a href="integral-population-models.html#cb2380-41" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Could not recover mesh points. Ensure L and U exist, or that K has rownames/colnames.&quot;</span>)</span>
<span id="cb2380-42"><a href="integral-population-models.html#cb2380-42" tabindex="-1"></a>  }</span>
<span id="cb2380-43"><a href="integral-population-models.html#cb2380-43" tabindex="-1"></a>  size_classes <span class="ot">&lt;-</span> <span class="fu">seq</span>(L, U, <span class="at">length.out =</span> n_mesh)</span>
<span id="cb2380-44"><a href="integral-population-models.html#cb2380-44" tabindex="-1"></a>}</span>
<span id="cb2380-45"><a href="integral-population-models.html#cb2380-45" tabindex="-1"></a></span>
<span id="cb2380-46"><a href="integral-population-models.html#cb2380-46" tabindex="-1"></a><span class="co"># Final sanity checks</span></span>
<span id="cb2380-47"><a href="integral-population-models.html#cb2380-47" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">nrow</span>(K) <span class="sc">==</span> n_mesh, <span class="fu">ncol</span>(K) <span class="sc">==</span> n_mesh, <span class="fu">length</span>(size_classes) <span class="sc">==</span> n_mesh)</span>
<span id="cb2380-48"><a href="integral-population-models.html#cb2380-48" tabindex="-1"></a></span>
<span id="cb2380-49"><a href="integral-population-models.html#cb2380-49" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Mesh points:&quot;</span>, n_mesh, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Mesh points: 100</code></pre>
<div class="sourceCode" id="cb2382"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2382-1"><a href="integral-population-models.html#cb2382-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Domain (log):&quot;</span>, <span class="fu">round</span>(<span class="fu">range</span>(size_classes), <span class="dv">3</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Domain (log): -0.221 5.177</code></pre>
<div class="sourceCode" id="cb2384"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2384-1"><a href="integral-population-models.html#cb2384-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2384-2"><a href="integral-population-models.html#cb2384-2" tabindex="-1"></a><span class="co"># 2) Initial population vector from observed sizes (LOG SCALE)</span></span>
<span id="cb2384-3"><a href="integral-population-models.html#cb2384-3" tabindex="-1"></a><span class="co">#    IMPORTANT: bins need to match the mesh discretization</span></span>
<span id="cb2384-4"><a href="integral-population-models.html#cb2384-4" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2384-5"><a href="integral-population-models.html#cb2384-5" tabindex="-1"></a>observed_sizes_log <span class="ot">&lt;-</span> final_data<span class="sc">$</span>size_log[<span class="fu">is.finite</span>(final_data<span class="sc">$</span>size_log)]</span>
<span id="cb2384-6"><a href="integral-population-models.html#cb2384-6" tabindex="-1"></a></span>
<span id="cb2384-7"><a href="integral-population-models.html#cb2384-7" tabindex="-1"></a><span class="co"># Build bin edges as midpoints between mesh points (plus endpoints)</span></span>
<span id="cb2384-8"><a href="integral-population-models.html#cb2384-8" tabindex="-1"></a><span class="co"># This aligns a histogram with the discretized state space</span></span>
<span id="cb2384-9"><a href="integral-population-models.html#cb2384-9" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb2384-10"><a href="integral-population-models.html#cb2384-10" tabindex="-1"></a>  size_classes[<span class="dv">1</span>] <span class="sc">-</span> <span class="fl">0.5</span> <span class="sc">*</span> (size_classes[<span class="dv">2</span>] <span class="sc">-</span> size_classes[<span class="dv">1</span>]),</span>
<span id="cb2384-11"><a href="integral-population-models.html#cb2384-11" tabindex="-1"></a>  (size_classes[<span class="sc">-</span><span class="dv">1</span>] <span class="sc">+</span> size_classes[<span class="sc">-</span>n_mesh]) <span class="sc">/</span> <span class="dv">2</span>,</span>
<span id="cb2384-12"><a href="integral-population-models.html#cb2384-12" tabindex="-1"></a>  size_classes[n_mesh] <span class="sc">+</span> <span class="fl">0.5</span> <span class="sc">*</span> (size_classes[n_mesh] <span class="sc">-</span> size_classes[n_mesh <span class="sc">-</span> <span class="dv">1</span>])</span>
<span id="cb2384-13"><a href="integral-population-models.html#cb2384-13" tabindex="-1"></a>)</span>
<span id="cb2384-14"><a href="integral-population-models.html#cb2384-14" tabindex="-1"></a></span>
<span id="cb2384-15"><a href="integral-population-models.html#cb2384-15" tabindex="-1"></a>hist_data <span class="ot">&lt;-</span> <span class="fu">hist</span>(observed_sizes_log, <span class="at">breaks =</span> edges, <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2384-16"><a href="integral-population-models.html#cb2384-16" tabindex="-1"></a>n0 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(hist_data<span class="sc">$</span>counts)</span>
<span id="cb2384-17"><a href="integral-population-models.html#cb2384-17" tabindex="-1"></a></span>
<span id="cb2384-18"><a href="integral-population-models.html#cb2384-18" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">sum</span>(n0) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">stop</span>(<span class="st">&quot;Initial vector n0 has all zeros (no finite size_log values).&quot;</span>)</span>
<span id="cb2384-19"><a href="integral-population-models.html#cb2384-19" tabindex="-1"></a></span>
<span id="cb2384-20"><a href="integral-population-models.html#cb2384-20" tabindex="-1"></a>n0 <span class="ot">&lt;-</span> n0 <span class="sc">/</span> <span class="fu">sum</span>(n0)  <span class="co"># normalize to proportions</span></span>
<span id="cb2384-21"><a href="integral-population-models.html#cb2384-21" tabindex="-1"></a></span>
<span id="cb2384-22"><a href="integral-population-models.html#cb2384-22" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Initial population vector:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Initial population vector:</code></pre>
<div class="sourceCode" id="cb2386"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2386-1"><a href="integral-population-models.html#cb2386-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Sum:&quot;</span>, <span class="fu">round</span>(<span class="fu">sum</span>(n0), <span class="dv">6</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Sum: 1</code></pre>
<div class="sourceCode" id="cb2388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2388-1"><a href="integral-population-models.html#cb2388-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Non-zero bins:&quot;</span>, <span class="fu">sum</span>(n0 <span class="sc">&gt;</span> <span class="dv">0</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Non-zero bins: 45</code></pre>
<div class="sourceCode" id="cb2390"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2390-1"><a href="integral-population-models.html#cb2390-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2390-2"><a href="integral-population-models.html#cb2390-2" tabindex="-1"></a><span class="co"># 3) Asymptotic properties (dominant eigenvalue/eigenvector)</span></span>
<span id="cb2390-3"><a href="integral-population-models.html#cb2390-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2390-4"><a href="integral-population-models.html#cb2390-4" tabindex="-1"></a>eig <span class="ot">&lt;-</span> <span class="fu">eigen</span>(K)</span>
<span id="cb2390-5"><a href="integral-population-models.html#cb2390-5" tabindex="-1"></a>lambda_asymptotic <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">Re</span>(eig<span class="sc">$</span>values))</span>
<span id="cb2390-6"><a href="integral-population-models.html#cb2390-6" tabindex="-1"></a>dom_idx <span class="ot">&lt;-</span> <span class="fu">which.max</span>(<span class="fu">Re</span>(eig<span class="sc">$</span>values))</span>
<span id="cb2390-7"><a href="integral-population-models.html#cb2390-7" tabindex="-1"></a></span>
<span id="cb2390-8"><a href="integral-population-models.html#cb2390-8" tabindex="-1"></a>w <span class="ot">&lt;-</span> <span class="fu">Re</span>(eig<span class="sc">$</span>vectors[, dom_idx])</span>
<span id="cb2390-9"><a href="integral-population-models.html#cb2390-9" tabindex="-1"></a>w[w <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb2390-10"><a href="integral-population-models.html#cb2390-10" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">sum</span>(w) <span class="sc">&lt;=</span> <span class="dv">0</span>) {</span>
<span id="cb2390-11"><a href="integral-population-models.html#cb2390-11" tabindex="-1"></a>  <span class="co"># fallback: absolute value then normalize</span></span>
<span id="cb2390-12"><a href="integral-population-models.html#cb2390-12" tabindex="-1"></a>  w <span class="ot">&lt;-</span> <span class="fu">abs</span>(<span class="fu">Re</span>(eig<span class="sc">$</span>vectors[, dom_idx]))</span>
<span id="cb2390-13"><a href="integral-population-models.html#cb2390-13" tabindex="-1"></a>}</span>
<span id="cb2390-14"><a href="integral-population-models.html#cb2390-14" tabindex="-1"></a>w <span class="ot">&lt;-</span> w <span class="sc">/</span> <span class="fu">sum</span>(w)</span>
<span id="cb2390-15"><a href="integral-population-models.html#cb2390-15" tabindex="-1"></a></span>
<span id="cb2390-16"><a href="integral-population-models.html#cb2390-16" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Asymptotic lambda:&quot;</span>, <span class="fu">round</span>(lambda_asymptotic, <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Asymptotic lambda: 0.9694</code></pre>
<div class="sourceCode" id="cb2392"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2392-1"><a href="integral-population-models.html#cb2392-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2392-2"><a href="integral-population-models.html#cb2392-2" tabindex="-1"></a><span class="co"># 4) One-step transient dynamics</span></span>
<span id="cb2392-3"><a href="integral-population-models.html#cb2392-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2392-4"><a href="integral-population-models.html#cb2392-4" tabindex="-1"></a>n1 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(K <span class="sc">%*%</span> n0)</span>
<span id="cb2392-5"><a href="integral-population-models.html#cb2392-5" tabindex="-1"></a>lambda_1 <span class="ot">&lt;-</span> <span class="fu">sum</span>(n1) <span class="sc">/</span> <span class="fu">sum</span>(n0)     <span class="co"># since sum(n0)=1, this is just sum(n1)</span></span>
<span id="cb2392-6"><a href="integral-population-models.html#cb2392-6" tabindex="-1"></a>n1_normalized <span class="ot">&lt;-</span> n1 <span class="sc">/</span> <span class="fu">sum</span>(n1)</span>
<span id="cb2392-7"><a href="integral-population-models.html#cb2392-7" tabindex="-1"></a></span>
<span id="cb2392-8"><a href="integral-population-models.html#cb2392-8" tabindex="-1"></a>deviation <span class="ot">&lt;-</span> n1_normalized <span class="sc">-</span> n0</span>
<span id="cb2392-9"><a href="integral-population-models.html#cb2392-9" tabindex="-1"></a></span>
<span id="cb2392-10"><a href="integral-population-models.html#cb2392-10" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">One-step dynamics:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## One-step dynamics:</code></pre>
<div class="sourceCode" id="cb2394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2394-1"><a href="integral-population-models.html#cb2394-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Lambda_1 (from observed n0):&quot;</span>, <span class="fu">round</span>(lambda_1, <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Lambda_1 (from observed n0): 4.2682</code></pre>
<div class="sourceCode" id="cb2396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2396-1"><a href="integral-population-models.html#cb2396-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Max deviation (|n1 - n0|):&quot;</span>, <span class="fu">round</span>(<span class="fu">max</span>(<span class="fu">abs</span>(deviation)), <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Max deviation (|n1 - n0|): 0.0975</code></pre>
<div class="sourceCode" id="cb2398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2398-1"><a href="integral-population-models.html#cb2398-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2398-2"><a href="integral-population-models.html#cb2398-2" tabindex="-1"></a><span class="co"># 5) Reactivity + inertia</span></span>
<span id="cb2398-3"><a href="integral-population-models.html#cb2398-3" tabindex="-1"></a><span class="co">#    - &quot;Observed one-step amplification&quot; = lambda_1</span></span>
<span id="cb2398-4"><a href="integral-population-models.html#cb2398-4" tabindex="-1"></a><span class="co">#    - &quot;Worst-case one-step amplification bound&quot; for total abundance:</span></span>
<span id="cb2398-5"><a href="integral-population-models.html#cb2398-5" tabindex="-1"></a><span class="co">#         max(colSums(K)) for nonnegative K under L1 norm</span></span>
<span id="cb2398-6"><a href="integral-population-models.html#cb2398-6" tabindex="-1"></a><span class="co">#    - Inertia: compare total N(t) to lambda^t scaling</span></span>
<span id="cb2398-7"><a href="integral-population-models.html#cb2398-7" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2398-8"><a href="integral-population-models.html#cb2398-8" tabindex="-1"></a><span class="co"># Worst-case one-step amplification (upper bound under L1)</span></span>
<span id="cb2398-9"><a href="integral-population-models.html#cb2398-9" tabindex="-1"></a>reactivity_bound <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">colSums</span>(K))</span>
<span id="cb2398-10"><a href="integral-population-models.html#cb2398-10" tabindex="-1"></a></span>
<span id="cb2398-11"><a href="integral-population-models.html#cb2398-11" tabindex="-1"></a><span class="co"># Inertia via repeated multiplication (avoid %^%)</span></span>
<span id="cb2398-12"><a href="integral-population-models.html#cb2398-12" tabindex="-1"></a>T <span class="ot">&lt;-</span> <span class="dv">50</span><span class="dt">L</span></span>
<span id="cb2398-13"><a href="integral-population-models.html#cb2398-13" tabindex="-1"></a>nt <span class="ot">&lt;-</span> n0</span>
<span id="cb2398-14"><a href="integral-population-models.html#cb2398-14" tabindex="-1"></a><span class="cf">for</span> (tt <span class="cf">in</span> <span class="fu">seq_len</span>(T)) nt <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(K <span class="sc">%*%</span> nt)</span>
<span id="cb2398-15"><a href="integral-population-models.html#cb2398-15" tabindex="-1"></a></span>
<span id="cb2398-16"><a href="integral-population-models.html#cb2398-16" tabindex="-1"></a>inertia <span class="ot">&lt;-</span> <span class="fu">sum</span>(nt) <span class="sc">/</span> (lambda_asymptotic<span class="sc">^</span>T <span class="sc">*</span> <span class="fu">sum</span>(n0))</span>
<span id="cb2398-17"><a href="integral-population-models.html#cb2398-17" tabindex="-1"></a></span>
<span id="cb2398-18"><a href="integral-population-models.html#cb2398-18" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Transient metrics:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Transient metrics:</code></pre>
<div class="sourceCode" id="cb2400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2400-1"><a href="integral-population-models.html#cb2400-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Reactivity (observed, 1-step):&quot;</span>, <span class="fu">round</span>(lambda_1, <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Reactivity (observed, 1-step): 4.2682</code></pre>
<div class="sourceCode" id="cb2402"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2402-1"><a href="integral-population-models.html#cb2402-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Reactivity (worst-case bound):&quot;</span>, <span class="fu">round</span>(reactivity_bound, <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>##   Reactivity (worst-case bound): 9.8296</code></pre>
<div class="sourceCode" id="cb2404"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2404-1"><a href="integral-population-models.html#cb2404-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;  Inertia (T = &quot;</span>, T, <span class="st">&quot;): &quot;</span>, <span class="fu">round</span>(inertia, <span class="dv">4</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<pre><code>##   Inertia (T = 50): 4.0442</code></pre>
<div class="sourceCode" id="cb2406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2406-1"><a href="integral-population-models.html#cb2406-1" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2406-2"><a href="integral-population-models.html#cb2406-2" tabindex="-1"></a><span class="co"># 6) Summary table + export</span></span>
<span id="cb2406-3"><a href="integral-population-models.html#cb2406-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2406-4"><a href="integral-population-models.html#cb2406-4" tabindex="-1"></a>transient_metrics <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2406-5"><a href="integral-population-models.html#cb2406-5" tabindex="-1"></a>  <span class="at">Transition          =</span> trans_years,</span>
<span id="cb2406-6"><a href="integral-population-models.html#cb2406-6" tabindex="-1"></a>  <span class="at">Lambda_1_observed   =</span> <span class="fu">round</span>(lambda_1, <span class="dv">4</span>),</span>
<span id="cb2406-7"><a href="integral-population-models.html#cb2406-7" tabindex="-1"></a>  <span class="at">Reactivity_bound    =</span> <span class="fu">round</span>(reactivity_bound, <span class="dv">4</span>),</span>
<span id="cb2406-8"><a href="integral-population-models.html#cb2406-8" tabindex="-1"></a>  <span class="at">Inertia_T50         =</span> <span class="fu">round</span>(inertia, <span class="dv">4</span>),</span>
<span id="cb2406-9"><a href="integral-population-models.html#cb2406-9" tabindex="-1"></a>  <span class="at">Lambda_asymptotic   =</span> <span class="fu">round</span>(lambda_asymptotic, <span class="dv">4</span>),</span>
<span id="cb2406-10"><a href="integral-population-models.html#cb2406-10" tabindex="-1"></a>  <span class="at">T_used              =</span> T</span>
<span id="cb2406-11"><a href="integral-population-models.html#cb2406-11" tabindex="-1"></a>)</span>
<span id="cb2406-12"><a href="integral-population-models.html#cb2406-12" tabindex="-1"></a></span>
<span id="cb2406-13"><a href="integral-population-models.html#cb2406-13" tabindex="-1"></a><span class="fu">print</span>(transient_metrics)</span></code></pre></div>
<pre><code>##   Transition Lambda_1_observed Reactivity_bound Inertia_T50 Lambda_asymptotic T_used
## 1   20232024            4.2682           9.8296      4.0442            0.9694     50</code></pre>
<div class="sourceCode" id="cb2408"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2408-1"><a href="integral-population-models.html#cb2408-1" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;IPM_outputs&quot;</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2408-2"><a href="integral-population-models.html#cb2408-2" tabindex="-1"></a></span>
<span id="cb2408-3"><a href="integral-population-models.html#cb2408-3" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2408-4"><a href="integral-population-models.html#cb2408-4" tabindex="-1"></a><span class="co"># 7) Visualization: deviation after one step</span></span>
<span id="cb2408-5"><a href="integral-population-models.html#cb2408-5" tabindex="-1"></a><span class="co">#--------------------------------------------------------------</span></span>
<span id="cb2408-6"><a href="integral-population-models.html#cb2408-6" tabindex="-1"></a>deviation_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2408-7"><a href="integral-population-models.html#cb2408-7" tabindex="-1"></a>  <span class="at">size_log     =</span> size_classes,</span>
<span id="cb2408-8"><a href="integral-population-models.html#cb2408-8" tabindex="-1"></a>  <span class="at">size_raw_cm  =</span> <span class="fu">exp</span>(size_classes),</span>
<span id="cb2408-9"><a href="integral-population-models.html#cb2408-9" tabindex="-1"></a>  <span class="at">deviation    =</span> deviation,</span>
<span id="cb2408-10"><a href="integral-population-models.html#cb2408-10" tabindex="-1"></a>  <span class="at">initial      =</span> n0,</span>
<span id="cb2408-11"><a href="integral-population-models.html#cb2408-11" tabindex="-1"></a>  <span class="at">final        =</span> n1_normalized</span>
<span id="cb2408-12"><a href="integral-population-models.html#cb2408-12" tabindex="-1"></a>)</span>
<span id="cb2408-13"><a href="integral-population-models.html#cb2408-13" tabindex="-1"></a></span>
<span id="cb2408-14"><a href="integral-population-models.html#cb2408-14" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2408-15"><a href="integral-population-models.html#cb2408-15" tabindex="-1"></a></span>
<span id="cb2408-16"><a href="integral-population-models.html#cb2408-16" tabindex="-1"></a>p_dev <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(deviation_df, <span class="fu">aes</span>(<span class="at">x =</span> size_log, <span class="at">y =</span> deviation, <span class="at">fill =</span> deviation <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb2408-17"><a href="integral-population-models.html#cb2408-17" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">linewidth =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb2408-18"><a href="integral-population-models.html#cb2408-18" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb2408-19"><a href="integral-population-models.html#cb2408-19" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;TRUE&quot;</span> <span class="ot">=</span> <span class="st">&quot;#3B82F6&quot;</span>, <span class="st">&quot;FALSE&quot;</span> <span class="ot">=</span> <span class="st">&quot;#EF4444&quot;</span>), <span class="at">guide =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb2408-20"><a href="integral-population-models.html#cb2408-20" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb2408-21"><a href="integral-population-models.html#cb2408-21" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Deviation from stable size distribution after one time step&quot;</span>,</span>
<span id="cb2408-22"><a href="integral-population-models.html#cb2408-22" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">&quot;Observed 1-step growth (λ1) = &quot;</span>, <span class="fu">round</span>(lambda_1, <span class="dv">3</span>)),</span>
<span id="cb2408-23"><a href="integral-population-models.html#cb2408-23" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Size class (log scale)&quot;</span>,</span>
<span id="cb2408-24"><a href="integral-population-models.html#cb2408-24" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Final − Initial proportion&quot;</span></span>
<span id="cb2408-25"><a href="integral-population-models.html#cb2408-25" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2408-26"><a href="integral-population-models.html#cb2408-26" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>)</span>
<span id="cb2408-27"><a href="integral-population-models.html#cb2408-27" tabindex="-1"></a></span>
<span id="cb2408-28"><a href="integral-population-models.html#cb2408-28" tabindex="-1"></a><span class="fu">print</span>(p_dev)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-31-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
<p>Sensitivity and vital rate calculations</p>
<div class="sourceCode" id="cb2409"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2409-1"><a href="integral-population-models.html#cb2409-1" tabindex="-1"></a><span class="co">#-------------------------------------------------</span></span>
<span id="cb2409-2"><a href="integral-population-models.html#cb2409-2" tabindex="-1"></a><span class="co"># Sensitivity and Elasticity of λ</span></span>
<span id="cb2409-3"><a href="integral-population-models.html#cb2409-3" tabindex="-1"></a><span class="co">#-------------------------------------------------</span></span>
<span id="cb2409-4"><a href="integral-population-models.html#cb2409-4" tabindex="-1"></a><span class="co"># Sensitivity: ∂λ / ∂K_ij</span></span>
<span id="cb2409-5"><a href="integral-population-models.html#cb2409-5" tabindex="-1"></a><span class="co"># Elasticity: proportional sensitivity = (K_ij / λ) * sensitivity</span></span>
<span id="cb2409-6"><a href="integral-population-models.html#cb2409-6" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb2409-7"><a href="integral-population-models.html#cb2409-7" tabindex="-1"></a><span class="co"># Row sums of elasticity:</span></span>
<span id="cb2409-8"><a href="integral-population-models.html#cb2409-8" tabindex="-1"></a><span class="co">#   Contribution of individuals of a given size at time t</span></span>
<span id="cb2409-9"><a href="integral-population-models.html#cb2409-9" tabindex="-1"></a><span class="co"># Column sums:</span></span>
<span id="cb2409-10"><a href="integral-population-models.html#cb2409-10" tabindex="-1"></a><span class="co">#   Contribution of individuals entering a given size at time t+1</span></span>
<span id="cb2409-11"><a href="integral-population-models.html#cb2409-11" tabindex="-1"></a><span class="co"># Row sums are most commonly interpreted and plotted.</span></span>
<span id="cb2409-12"><a href="integral-population-models.html#cb2409-12" tabindex="-1"></a><span class="co">#-------------------------------------------------</span></span>
<span id="cb2409-13"><a href="integral-population-models.html#cb2409-13" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb2409-14"><a href="integral-population-models.html#cb2409-14" tabindex="-1"></a><span class="fu">library</span>(ipmr)</span>
<span id="cb2409-15"><a href="integral-population-models.html#cb2409-15" tabindex="-1"></a></span>
<span id="cb2409-16"><a href="integral-population-models.html#cb2409-16" tabindex="-1"></a><span class="co"># Combine kernels</span></span>
<span id="cb2409-17"><a href="integral-population-models.html#cb2409-17" tabindex="-1"></a>K <span class="ot">&lt;-</span> obs_ipm<span class="sc">$</span>sub_kernels<span class="sc">$</span>P <span class="sc">+</span> obs_ipm<span class="sc">$</span>sub_kernels<span class="sc">$</span>F</span>
<span id="cb2409-18"><a href="integral-population-models.html#cb2409-18" tabindex="-1"></a>n_size <span class="ot">&lt;-</span> <span class="fu">nrow</span>(K)</span>
<span id="cb2409-19"><a href="integral-population-models.html#cb2409-19" tabindex="-1"></a></span>
<span id="cb2409-20"><a href="integral-population-models.html#cb2409-20" tabindex="-1"></a><span class="co"># Eigen decomposition</span></span>
<span id="cb2409-21"><a href="integral-population-models.html#cb2409-21" tabindex="-1"></a>eig <span class="ot">&lt;-</span> <span class="fu">eigen</span>(K)</span>
<span id="cb2409-22"><a href="integral-population-models.html#cb2409-22" tabindex="-1"></a>ord <span class="ot">&lt;-</span> <span class="fu">order</span>(<span class="fu">Re</span>(eig<span class="sc">$</span>values), <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2409-23"><a href="integral-population-models.html#cb2409-23" tabindex="-1"></a></span>
<span id="cb2409-24"><a href="integral-population-models.html#cb2409-24" tabindex="-1"></a>lambda_dom <span class="ot">&lt;-</span> <span class="fu">Re</span>(eig<span class="sc">$</span>values[ord[<span class="dv">1</span>]])</span>
<span id="cb2409-25"><a href="integral-population-models.html#cb2409-25" tabindex="-1"></a></span>
<span id="cb2409-26"><a href="integral-population-models.html#cb2409-26" tabindex="-1"></a><span class="co"># Right eigenvector (stable distribution)</span></span>
<span id="cb2409-27"><a href="integral-population-models.html#cb2409-27" tabindex="-1"></a>w <span class="ot">&lt;-</span> <span class="fu">Re</span>(eig<span class="sc">$</span>vectors[, ord[<span class="dv">1</span>]])</span>
<span id="cb2409-28"><a href="integral-population-models.html#cb2409-28" tabindex="-1"></a>w[w <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb2409-29"><a href="integral-population-models.html#cb2409-29" tabindex="-1"></a>w <span class="ot">&lt;-</span> w <span class="sc">/</span> <span class="fu">sum</span>(w)</span>
<span id="cb2409-30"><a href="integral-population-models.html#cb2409-30" tabindex="-1"></a></span>
<span id="cb2409-31"><a href="integral-population-models.html#cb2409-31" tabindex="-1"></a><span class="co"># Left eigenvector (reproductive value)</span></span>
<span id="cb2409-32"><a href="integral-population-models.html#cb2409-32" tabindex="-1"></a>eig_left <span class="ot">&lt;-</span> <span class="fu">eigen</span>(<span class="fu">t</span>(K))</span>
<span id="cb2409-33"><a href="integral-population-models.html#cb2409-33" tabindex="-1"></a>v <span class="ot">&lt;-</span> <span class="fu">Re</span>(eig_left<span class="sc">$</span>vectors[, ord[<span class="dv">1</span>]])</span>
<span id="cb2409-34"><a href="integral-population-models.html#cb2409-34" tabindex="-1"></a>v <span class="ot">&lt;-</span> v <span class="sc">/</span> <span class="fu">sum</span>(v <span class="sc">*</span> w)   <span class="co"># normalize so v · w = 1</span></span>
<span id="cb2409-35"><a href="integral-population-models.html#cb2409-35" tabindex="-1"></a></span>
<span id="cb2409-36"><a href="integral-population-models.html#cb2409-36" tabindex="-1"></a><span class="co"># Sensitivity and elasticity</span></span>
<span id="cb2409-37"><a href="integral-population-models.html#cb2409-37" tabindex="-1"></a>sensitivity <span class="ot">&lt;-</span> <span class="fu">outer</span>(v, w)</span>
<span id="cb2409-38"><a href="integral-population-models.html#cb2409-38" tabindex="-1"></a>elasticity  <span class="ot">&lt;-</span> (K <span class="sc">/</span> lambda_dom) <span class="sc">*</span> sensitivity</span>
<span id="cb2409-39"><a href="integral-population-models.html#cb2409-39" tabindex="-1"></a></span>
<span id="cb2409-40"><a href="integral-population-models.html#cb2409-40" tabindex="-1"></a><span class="co">#-------------------------------------------------</span></span>
<span id="cb2409-41"><a href="integral-population-models.html#cb2409-41" tabindex="-1"></a><span class="co"># Prepare plotting data (INDEX-ALIGNED)</span></span>
<span id="cb2409-42"><a href="integral-population-models.html#cb2409-42" tabindex="-1"></a><span class="co">#-------------------------------------------------</span></span>
<span id="cb2409-43"><a href="integral-population-models.html#cb2409-43" tabindex="-1"></a></span>
<span id="cb2409-44"><a href="integral-population-models.html#cb2409-44" tabindex="-1"></a><span class="co"># Use indices or safe mesh extraction</span></span>
<span id="cb2409-45"><a href="integral-population-models.html#cb2409-45" tabindex="-1"></a>size_log <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(obs_ipm<span class="sc">$</span>env_list<span class="sc">$</span>main_env<span class="sc">$</span>sa_1)[<span class="fu">seq_len</span>(n_size)]</span>
<span id="cb2409-46"><a href="integral-population-models.html#cb2409-46" tabindex="-1"></a>size_raw <span class="ot">&lt;-</span> <span class="fu">exp</span>(size_log)</span>
<span id="cb2409-47"><a href="integral-population-models.html#cb2409-47" tabindex="-1"></a></span>
<span id="cb2409-48"><a href="integral-population-models.html#cb2409-48" tabindex="-1"></a>sens_df <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(</span>
<span id="cb2409-49"><a href="integral-population-models.html#cb2409-49" tabindex="-1"></a>  <span class="at">from_size =</span> size_log,</span>
<span id="cb2409-50"><a href="integral-population-models.html#cb2409-50" tabindex="-1"></a>  <span class="at">to_size   =</span> size_log</span>
<span id="cb2409-51"><a href="integral-population-models.html#cb2409-51" tabindex="-1"></a>)</span>
<span id="cb2409-52"><a href="integral-population-models.html#cb2409-52" tabindex="-1"></a>sens_df<span class="sc">$</span>value <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(sensitivity)</span>
<span id="cb2409-53"><a href="integral-population-models.html#cb2409-53" tabindex="-1"></a></span>
<span id="cb2409-54"><a href="integral-population-models.html#cb2409-54" tabindex="-1"></a>elas_df <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(</span>
<span id="cb2409-55"><a href="integral-population-models.html#cb2409-55" tabindex="-1"></a>  <span class="at">from_size =</span> size_log,</span>
<span id="cb2409-56"><a href="integral-population-models.html#cb2409-56" tabindex="-1"></a>  <span class="at">to_size   =</span> size_log</span>
<span id="cb2409-57"><a href="integral-population-models.html#cb2409-57" tabindex="-1"></a>)</span>
<span id="cb2409-58"><a href="integral-population-models.html#cb2409-58" tabindex="-1"></a>elas_df<span class="sc">$</span>value <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(elasticity)</span>
<span id="cb2409-59"><a href="integral-population-models.html#cb2409-59" tabindex="-1"></a></span>
<span id="cb2409-60"><a href="integral-population-models.html#cb2409-60" tabindex="-1"></a>row_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2409-61"><a href="integral-population-models.html#cb2409-61" tabindex="-1"></a>  <span class="at">size_log =</span> size_log,</span>
<span id="cb2409-62"><a href="integral-population-models.html#cb2409-62" tabindex="-1"></a>  <span class="at">size_raw_cm =</span> size_raw,</span>
<span id="cb2409-63"><a href="integral-population-models.html#cb2409-63" tabindex="-1"></a>  <span class="at">elasticity_sum =</span> <span class="fu">rowSums</span>(elasticity)</span>
<span id="cb2409-64"><a href="integral-population-models.html#cb2409-64" tabindex="-1"></a>)</span>
<span id="cb2409-65"><a href="integral-population-models.html#cb2409-65" tabindex="-1"></a></span>
<span id="cb2409-66"><a href="integral-population-models.html#cb2409-66" tabindex="-1"></a><span class="co">#-------------------------------------------------</span></span>
<span id="cb2409-67"><a href="integral-population-models.html#cb2409-67" tabindex="-1"></a><span class="co"># Plots</span></span>
<span id="cb2409-68"><a href="integral-population-models.html#cb2409-68" tabindex="-1"></a><span class="co">#-------------------------------------------------</span></span>
<span id="cb2409-69"><a href="integral-population-models.html#cb2409-69" tabindex="-1"></a></span>
<span id="cb2409-70"><a href="integral-population-models.html#cb2409-70" tabindex="-1"></a>elas_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(elas_df, <span class="fu">aes</span>(from_size, to_size, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb2409-71"><a href="integral-population-models.html#cb2409-71" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb2409-72"><a href="integral-population-models.html#cb2409-72" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_c</span>(<span class="at">name =</span> <span class="st">&quot;Elasticity&quot;</span>) <span class="sc">+</span></span>
<span id="cb2409-73"><a href="integral-population-models.html#cb2409-73" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb2409-74"><a href="integral-population-models.html#cb2409-74" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Size at time t (log)&quot;</span>,</span>
<span id="cb2409-75"><a href="integral-population-models.html#cb2409-75" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Size at time t+1 (log)&quot;</span>,</span>
<span id="cb2409-76"><a href="integral-population-models.html#cb2409-76" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Elasticity matrix&quot;</span></span>
<span id="cb2409-77"><a href="integral-population-models.html#cb2409-77" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2409-78"><a href="integral-population-models.html#cb2409-78" tabindex="-1"></a>  <span class="fu">theme_pubr</span>()</span>
<span id="cb2409-79"><a href="integral-population-models.html#cb2409-79" tabindex="-1"></a></span>
<span id="cb2409-80"><a href="integral-population-models.html#cb2409-80" tabindex="-1"></a>row_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(row_df, <span class="fu">aes</span>(size_log, elasticity_sum)) <span class="sc">+</span></span>
<span id="cb2409-81"><a href="integral-population-models.html#cb2409-81" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="fl">1.1</span>) <span class="sc">+</span></span>
<span id="cb2409-82"><a href="integral-population-models.html#cb2409-82" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb2409-83"><a href="integral-population-models.html#cb2409-83" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Size at time t (log scale)&quot;</span>,</span>
<span id="cb2409-84"><a href="integral-population-models.html#cb2409-84" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Sum of elasticities&quot;</span>,</span>
<span id="cb2409-85"><a href="integral-population-models.html#cb2409-85" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Contribution of size classes to λ&quot;</span></span>
<span id="cb2409-86"><a href="integral-population-models.html#cb2409-86" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2409-87"><a href="integral-population-models.html#cb2409-87" tabindex="-1"></a>  <span class="fu">theme_pubr</span>()</span>
<span id="cb2409-88"><a href="integral-population-models.html#cb2409-88" tabindex="-1"></a></span>
<span id="cb2409-89"><a href="integral-population-models.html#cb2409-89" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(elas_plot, row_plot, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb2409-90"><a href="integral-population-models.html#cb2409-90" tabindex="-1"></a><span class="fu">print</span>(combined_plot)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-32-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="test-your-knowledge-2" class="section level2 hasAnchor" number="82.11">
<h2><span class="header-section-number">82.11</span> Test your knowledge<a href="integral-population-models.html#test-your-knowledge-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Conceptual Understanding:</p>
<pre><code>1.  What is an Integral Projection Model (IPM), and how does it differ from a matrix population model?</code></pre>
<p>(Hint: think about state variables and discretization.)
2. Why must IPM census timing align with biologically meaningful life-cycle events?
Give one example of how poor timing could bias demographic estimates.
3. What characteristics make a variable suitable as a state variable in an IPM?
List at least three and briefly explain why each matters.
4. Explain why using more than one state variable can quickly make IPMs impractical.
What is meant by the “curse of dimensionality”?</p>
<p>Vital Rates and Model Structure:</p>
<pre><code>5.  Why is reproduction often modeled using a two-stage (hurdle) approach in plant IPMs?</code></pre>
<p>What biological processes do each stage represent?
6. In this case study, why was it problematic to use “height with reproductive structures” as the state variable for fecundity?
What kind of modeling issue does this introduce?
7. Why is it generally inappropriate to remove outliers from survival models but reasonable to consider removing them from growth models?</p>
<p>Statistical Modeling Choices:</p>
<pre><code>8.  Why were log transformations applied to size in most vital-rate models?</code></pre>
<p>What biological assumption does this reflect?
9. A quadratic growth model had lower AIC than a linear model. Why is it still important to visually inspect the fitted relationship before choosing it?
10. What does overdispersion indicate in a Poisson model, and why might a negative binomial model be preferred for seed production?</p>
<p>IPM Construction and Interpretation:</p>
<pre><code>11. What do the $P$ and $F$ kernels represent biologically?</code></pre>
<p>Give one example of a process captured in each.
12. Why must the size domain boundaries (L and U) extend slightly beyond the observed data range?
What problem does this help prevent?
13. What is eviction in an IPM, and why can it bias estimates of population growth rate if not corrected?
14. Compare truncation and reflection as eviction-correction strategies.
Under what biological assumptions would truncation be preferred?</p>
<p>Population Metrics:</p>
<pre><code>15. Interpret the meaning of $\lambda &lt; 1$, $\lambda = 1$, and $\lambda &gt; 1$ in the context of conservation management.
16. What is the stable size distribution, and why does it not describe short-term population behavior?
17. What does reproductive value tell us about individuals of different sizes?</code></pre>
<p>How might managers use this information?</p>
<p>Transient Dynamics:</p>
<pre><code>18. Explain the difference between asymptotic population growth ($\lambda$) and one-step transient growth ($\lambda_1$).
19. What does reactivity measure, and why can it be important even when $\lambda &lt; 1$?
20. How is inertia interpreted biologically in a conservation context?</code></pre>
<p>What does high inertia imply about population response to disturbance?</p>
<p>Sensitivity and Elasticity:</p>
<pre><code>21. What is the difference between sensitivity and elasticity of $\lambda$?</code></pre>
<p>Why are elasticities often preferred for interpretation?
22. What does the row sum of the elasticity matrix represent biologically?
23. Why might a size class have high elasticity even if few individuals occupy that size class?</p>
<p>Uncertainty and Inference:</p>
<pre><code>24. Why is bootstrapping particularly important for IPMs built from small populations?
25. Why were recruitment parameters held constant during the bootstrap in this analysis?
26. How do uncertainty bands around vital-rate curves help prevent over-interpretation of demographic results?</code></pre>
<p>Synthesis / Short Answer:</p>
<pre><code>27. Based on the IPM results, would you prioritize protecting small individuals or large individuals for this species?</code></pre>
<p>Justify your answer using survival, reproduction, or elasticity results.
28. Give one example of a management action that could be evaluated using this IPM framework and describe which vital rate it would most likely affect.</p>
</div>
<div id="assignment-34" class="section level2 hasAnchor" number="82.12">
<h2><span class="header-section-number">82.12</span> Assignment<a href="integral-population-models.html#assignment-34" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using your data, build an ipm and calculate population dynamics!</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="population-sensation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="stochastic-population-dynamics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/sarasouther/DataAnalysis/edit/main/41-IntegralProjectionModels.Rmd",
    "text": "Edit this page"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
